<!DOCTYPE html>
<html>
<head><meta charset="utf-8" />

<title>NIH_LBD_FTD_SVs_for_sharing</title>

<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>



<style type="text/css">
    /*!
*
* Twitter Bootstrap
*
*/
/*!
 * Bootstrap v3.3.7 (http://getbootstrap.com)
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 */
/*! normalize.css v3.0.3 | MIT License | github.com/necolas/normalize.css */
html {
  font-family: sans-serif;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
}
body {
  margin: 0;
}
article,
aside,
details,
figcaption,
figure,
footer,
header,
hgroup,
main,
menu,
nav,
section,
summary {
  display: block;
}
audio,
canvas,
progress,
video {
  display: inline-block;
  vertical-align: baseline;
}
audio:not([controls]) {
  display: none;
  height: 0;
}
[hidden],
template {
  display: none;
}
a {
  background-color: transparent;
}
a:active,
a:hover {
  outline: 0;
}
abbr[title] {
  border-bottom: 1px dotted;
}
b,
strong {
  font-weight: bold;
}
dfn {
  font-style: italic;
}
h1 {
  font-size: 2em;
  margin: 0.67em 0;
}
mark {
  background: #ff0;
  color: #000;
}
small {
  font-size: 80%;
}
sub,
sup {
  font-size: 75%;
  line-height: 0;
  position: relative;
  vertical-align: baseline;
}
sup {
  top: -0.5em;
}
sub {
  bottom: -0.25em;
}
img {
  border: 0;
}
svg:not(:root) {
  overflow: hidden;
}
figure {
  margin: 1em 40px;
}
hr {
  box-sizing: content-box;
  height: 0;
}
pre {
  overflow: auto;
}
code,
kbd,
pre,
samp {
  font-family: monospace, monospace;
  font-size: 1em;
}
button,
input,
optgroup,
select,
textarea {
  color: inherit;
  font: inherit;
  margin: 0;
}
button {
  overflow: visible;
}
button,
select {
  text-transform: none;
}
button,
html input[type="button"],
input[type="reset"],
input[type="submit"] {
  -webkit-appearance: button;
  cursor: pointer;
}
button[disabled],
html input[disabled] {
  cursor: default;
}
button::-moz-focus-inner,
input::-moz-focus-inner {
  border: 0;
  padding: 0;
}
input {
  line-height: normal;
}
input[type="checkbox"],
input[type="radio"] {
  box-sizing: border-box;
  padding: 0;
}
input[type="number"]::-webkit-inner-spin-button,
input[type="number"]::-webkit-outer-spin-button {
  height: auto;
}
input[type="search"] {
  -webkit-appearance: textfield;
  box-sizing: content-box;
}
input[type="search"]::-webkit-search-cancel-button,
input[type="search"]::-webkit-search-decoration {
  -webkit-appearance: none;
}
fieldset {
  border: 1px solid #c0c0c0;
  margin: 0 2px;
  padding: 0.35em 0.625em 0.75em;
}
legend {
  border: 0;
  padding: 0;
}
textarea {
  overflow: auto;
}
optgroup {
  font-weight: bold;
}
table {
  border-collapse: collapse;
  border-spacing: 0;
}
td,
th {
  padding: 0;
}
/*! Source: https://github.com/h5bp/html5-boilerplate/blob/master/src/css/main.css */
@media print {
  *,
  *:before,
  *:after {
    background: transparent !important;
    box-shadow: none !important;
    text-shadow: none !important;
  }
  a,
  a:visited {
    text-decoration: underline;
  }
  a[href]:after {
    content: " (" attr(href) ")";
  }
  abbr[title]:after {
    content: " (" attr(title) ")";
  }
  a[href^="#"]:after,
  a[href^="javascript:"]:after {
    content: "";
  }
  pre,
  blockquote {
    border: 1px solid #999;
    page-break-inside: avoid;
  }
  thead {
    display: table-header-group;
  }
  tr,
  img {
    page-break-inside: avoid;
  }
  img {
    max-width: 100% !important;
  }
  p,
  h2,
  h3 {
    orphans: 3;
    widows: 3;
  }
  h2,
  h3 {
    page-break-after: avoid;
  }
  .navbar {
    display: none;
  }
  .btn > .caret,
  .dropup > .btn > .caret {
    border-top-color: #000 !important;
  }
  .label {
    border: 1px solid #000;
  }
  .table {
    border-collapse: collapse !important;
  }
  .table td,
  .table th {
    background-color: #fff !important;
  }
  .table-bordered th,
  .table-bordered td {
    border: 1px solid #ddd !important;
  }
}
@font-face {
  font-family: 'Glyphicons Halflings';
  src: url('../components/bootstrap/fonts/glyphicons-halflings-regular.eot');
  src: url('../components/bootstrap/fonts/glyphicons-halflings-regular.eot?#iefix') format('embedded-opentype'), url('../components/bootstrap/fonts/glyphicons-halflings-regular.woff2') format('woff2'), url('../components/bootstrap/fonts/glyphicons-halflings-regular.woff') format('woff'), url('../components/bootstrap/fonts/glyphicons-halflings-regular.ttf') format('truetype'), url('../components/bootstrap/fonts/glyphicons-halflings-regular.svg#glyphicons_halflingsregular') format('svg');
}
.glyphicon {
  position: relative;
  top: 1px;
  display: inline-block;
  font-family: 'Glyphicons Halflings';
  font-style: normal;
  font-weight: normal;
  line-height: 1;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
.glyphicon-asterisk:before {
  content: "\002a";
}
.glyphicon-plus:before {
  content: "\002b";
}
.glyphicon-euro:before,
.glyphicon-eur:before {
  content: "\20ac";
}
.glyphicon-minus:before {
  content: "\2212";
}
.glyphicon-cloud:before {
  content: "\2601";
}
.glyphicon-envelope:before {
  content: "\2709";
}
.glyphicon-pencil:before {
  content: "\270f";
}
.glyphicon-glass:before {
  content: "\e001";
}
.glyphicon-music:before {
  content: "\e002";
}
.glyphicon-search:before {
  content: "\e003";
}
.glyphicon-heart:before {
  content: "\e005";
}
.glyphicon-star:before {
  content: "\e006";
}
.glyphicon-star-empty:before {
  content: "\e007";
}
.glyphicon-user:before {
  content: "\e008";
}
.glyphicon-film:before {
  content: "\e009";
}
.glyphicon-th-large:before {
  content: "\e010";
}
.glyphicon-th:before {
  content: "\e011";
}
.glyphicon-th-list:before {
  content: "\e012";
}
.glyphicon-ok:before {
  content: "\e013";
}
.glyphicon-remove:before {
  content: "\e014";
}
.glyphicon-zoom-in:before {
  content: "\e015";
}
.glyphicon-zoom-out:before {
  content: "\e016";
}
.glyphicon-off:before {
  content: "\e017";
}
.glyphicon-signal:before {
  content: "\e018";
}
.glyphicon-cog:before {
  content: "\e019";
}
.glyphicon-trash:before {
  content: "\e020";
}
.glyphicon-home:before {
  content: "\e021";
}
.glyphicon-file:before {
  content: "\e022";
}
.glyphicon-time:before {
  content: "\e023";
}
.glyphicon-road:before {
  content: "\e024";
}
.glyphicon-download-alt:before {
  content: "\e025";
}
.glyphicon-download:before {
  content: "\e026";
}
.glyphicon-upload:before {
  content: "\e027";
}
.glyphicon-inbox:before {
  content: "\e028";
}
.glyphicon-play-circle:before {
  content: "\e029";
}
.glyphicon-repeat:before {
  content: "\e030";
}
.glyphicon-refresh:before {
  content: "\e031";
}
.glyphicon-list-alt:before {
  content: "\e032";
}
.glyphicon-lock:before {
  content: "\e033";
}
.glyphicon-flag:before {
  content: "\e034";
}
.glyphicon-headphones:before {
  content: "\e035";
}
.glyphicon-volume-off:before {
  content: "\e036";
}
.glyphicon-volume-down:before {
  content: "\e037";
}
.glyphicon-volume-up:before {
  content: "\e038";
}
.glyphicon-qrcode:before {
  content: "\e039";
}
.glyphicon-barcode:before {
  content: "\e040";
}
.glyphicon-tag:before {
  content: "\e041";
}
.glyphicon-tags:before {
  content: "\e042";
}
.glyphicon-book:before {
  content: "\e043";
}
.glyphicon-bookmark:before {
  content: "\e044";
}
.glyphicon-print:before {
  content: "\e045";
}
.glyphicon-camera:before {
  content: "\e046";
}
.glyphicon-font:before {
  content: "\e047";
}
.glyphicon-bold:before {
  content: "\e048";
}
.glyphicon-italic:before {
  content: "\e049";
}
.glyphicon-text-height:before {
  content: "\e050";
}
.glyphicon-text-width:before {
  content: "\e051";
}
.glyphicon-align-left:before {
  content: "\e052";
}
.glyphicon-align-center:before {
  content: "\e053";
}
.glyphicon-align-right:before {
  content: "\e054";
}
.glyphicon-align-justify:before {
  content: "\e055";
}
.glyphicon-list:before {
  content: "\e056";
}
.glyphicon-indent-left:before {
  content: "\e057";
}
.glyphicon-indent-right:before {
  content: "\e058";
}
.glyphicon-facetime-video:before {
  content: "\e059";
}
.glyphicon-picture:before {
  content: "\e060";
}
.glyphicon-map-marker:before {
  content: "\e062";
}
.glyphicon-adjust:before {
  content: "\e063";
}
.glyphicon-tint:before {
  content: "\e064";
}
.glyphicon-edit:before {
  content: "\e065";
}
.glyphicon-share:before {
  content: "\e066";
}
.glyphicon-check:before {
  content: "\e067";
}
.glyphicon-move:before {
  content: "\e068";
}
.glyphicon-step-backward:before {
  content: "\e069";
}
.glyphicon-fast-backward:before {
  content: "\e070";
}
.glyphicon-backward:before {
  content: "\e071";
}
.glyphicon-play:before {
  content: "\e072";
}
.glyphicon-pause:before {
  content: "\e073";
}
.glyphicon-stop:before {
  content: "\e074";
}
.glyphicon-forward:before {
  content: "\e075";
}
.glyphicon-fast-forward:before {
  content: "\e076";
}
.glyphicon-step-forward:before {
  content: "\e077";
}
.glyphicon-eject:before {
  content: "\e078";
}
.glyphicon-chevron-left:before {
  content: "\e079";
}
.glyphicon-chevron-right:before {
  content: "\e080";
}
.glyphicon-plus-sign:before {
  content: "\e081";
}
.glyphicon-minus-sign:before {
  content: "\e082";
}
.glyphicon-remove-sign:before {
  content: "\e083";
}
.glyphicon-ok-sign:before {
  content: "\e084";
}
.glyphicon-question-sign:before {
  content: "\e085";
}
.glyphicon-info-sign:before {
  content: "\e086";
}
.glyphicon-screenshot:before {
  content: "\e087";
}
.glyphicon-remove-circle:before {
  content: "\e088";
}
.glyphicon-ok-circle:before {
  content: "\e089";
}
.glyphicon-ban-circle:before {
  content: "\e090";
}
.glyphicon-arrow-left:before {
  content: "\e091";
}
.glyphicon-arrow-right:before {
  content: "\e092";
}
.glyphicon-arrow-up:before {
  content: "\e093";
}
.glyphicon-arrow-down:before {
  content: "\e094";
}
.glyphicon-share-alt:before {
  content: "\e095";
}
.glyphicon-resize-full:before {
  content: "\e096";
}
.glyphicon-resize-small:before {
  content: "\e097";
}
.glyphicon-exclamation-sign:before {
  content: "\e101";
}
.glyphicon-gift:before {
  content: "\e102";
}
.glyphicon-leaf:before {
  content: "\e103";
}
.glyphicon-fire:before {
  content: "\e104";
}
.glyphicon-eye-open:before {
  content: "\e105";
}
.glyphicon-eye-close:before {
  content: "\e106";
}
.glyphicon-warning-sign:before {
  content: "\e107";
}
.glyphicon-plane:before {
  content: "\e108";
}
.glyphicon-calendar:before {
  content: "\e109";
}
.glyphicon-random:before {
  content: "\e110";
}
.glyphicon-comment:before {
  content: "\e111";
}
.glyphicon-magnet:before {
  content: "\e112";
}
.glyphicon-chevron-up:before {
  content: "\e113";
}
.glyphicon-chevron-down:before {
  content: "\e114";
}
.glyphicon-retweet:before {
  content: "\e115";
}
.glyphicon-shopping-cart:before {
  content: "\e116";
}
.glyphicon-folder-close:before {
  content: "\e117";
}
.glyphicon-folder-open:before {
  content: "\e118";
}
.glyphicon-resize-vertical:before {
  content: "\e119";
}
.glyphicon-resize-horizontal:before {
  content: "\e120";
}
.glyphicon-hdd:before {
  content: "\e121";
}
.glyphicon-bullhorn:before {
  content: "\e122";
}
.glyphicon-bell:before {
  content: "\e123";
}
.glyphicon-certificate:before {
  content: "\e124";
}
.glyphicon-thumbs-up:before {
  content: "\e125";
}
.glyphicon-thumbs-down:before {
  content: "\e126";
}
.glyphicon-hand-right:before {
  content: "\e127";
}
.glyphicon-hand-left:before {
  content: "\e128";
}
.glyphicon-hand-up:before {
  content: "\e129";
}
.glyphicon-hand-down:before {
  content: "\e130";
}
.glyphicon-circle-arrow-right:before {
  content: "\e131";
}
.glyphicon-circle-arrow-left:before {
  content: "\e132";
}
.glyphicon-circle-arrow-up:before {
  content: "\e133";
}
.glyphicon-circle-arrow-down:before {
  content: "\e134";
}
.glyphicon-globe:before {
  content: "\e135";
}
.glyphicon-wrench:before {
  content: "\e136";
}
.glyphicon-tasks:before {
  content: "\e137";
}
.glyphicon-filter:before {
  content: "\e138";
}
.glyphicon-briefcase:before {
  content: "\e139";
}
.glyphicon-fullscreen:before {
  content: "\e140";
}
.glyphicon-dashboard:before {
  content: "\e141";
}
.glyphicon-paperclip:before {
  content: "\e142";
}
.glyphicon-heart-empty:before {
  content: "\e143";
}
.glyphicon-link:before {
  content: "\e144";
}
.glyphicon-phone:before {
  content: "\e145";
}
.glyphicon-pushpin:before {
  content: "\e146";
}
.glyphicon-usd:before {
  content: "\e148";
}
.glyphicon-gbp:before {
  content: "\e149";
}
.glyphicon-sort:before {
  content: "\e150";
}
.glyphicon-sort-by-alphabet:before {
  content: "\e151";
}
.glyphicon-sort-by-alphabet-alt:before {
  content: "\e152";
}
.glyphicon-sort-by-order:before {
  content: "\e153";
}
.glyphicon-sort-by-order-alt:before {
  content: "\e154";
}
.glyphicon-sort-by-attributes:before {
  content: "\e155";
}
.glyphicon-sort-by-attributes-alt:before {
  content: "\e156";
}
.glyphicon-unchecked:before {
  content: "\e157";
}
.glyphicon-expand:before {
  content: "\e158";
}
.glyphicon-collapse-down:before {
  content: "\e159";
}
.glyphicon-collapse-up:before {
  content: "\e160";
}
.glyphicon-log-in:before {
  content: "\e161";
}
.glyphicon-flash:before {
  content: "\e162";
}
.glyphicon-log-out:before {
  content: "\e163";
}
.glyphicon-new-window:before {
  content: "\e164";
}
.glyphicon-record:before {
  content: "\e165";
}
.glyphicon-save:before {
  content: "\e166";
}
.glyphicon-open:before {
  content: "\e167";
}
.glyphicon-saved:before {
  content: "\e168";
}
.glyphicon-import:before {
  content: "\e169";
}
.glyphicon-export:before {
  content: "\e170";
}
.glyphicon-send:before {
  content: "\e171";
}
.glyphicon-floppy-disk:before {
  content: "\e172";
}
.glyphicon-floppy-saved:before {
  content: "\e173";
}
.glyphicon-floppy-remove:before {
  content: "\e174";
}
.glyphicon-floppy-save:before {
  content: "\e175";
}
.glyphicon-floppy-open:before {
  content: "\e176";
}
.glyphicon-credit-card:before {
  content: "\e177";
}
.glyphicon-transfer:before {
  content: "\e178";
}
.glyphicon-cutlery:before {
  content: "\e179";
}
.glyphicon-header:before {
  content: "\e180";
}
.glyphicon-compressed:before {
  content: "\e181";
}
.glyphicon-earphone:before {
  content: "\e182";
}
.glyphicon-phone-alt:before {
  content: "\e183";
}
.glyphicon-tower:before {
  content: "\e184";
}
.glyphicon-stats:before {
  content: "\e185";
}
.glyphicon-sd-video:before {
  content: "\e186";
}
.glyphicon-hd-video:before {
  content: "\e187";
}
.glyphicon-subtitles:before {
  content: "\e188";
}
.glyphicon-sound-stereo:before {
  content: "\e189";
}
.glyphicon-sound-dolby:before {
  content: "\e190";
}
.glyphicon-sound-5-1:before {
  content: "\e191";
}
.glyphicon-sound-6-1:before {
  content: "\e192";
}
.glyphicon-sound-7-1:before {
  content: "\e193";
}
.glyphicon-copyright-mark:before {
  content: "\e194";
}
.glyphicon-registration-mark:before {
  content: "\e195";
}
.glyphicon-cloud-download:before {
  content: "\e197";
}
.glyphicon-cloud-upload:before {
  content: "\e198";
}
.glyphicon-tree-conifer:before {
  content: "\e199";
}
.glyphicon-tree-deciduous:before {
  content: "\e200";
}
.glyphicon-cd:before {
  content: "\e201";
}
.glyphicon-save-file:before {
  content: "\e202";
}
.glyphicon-open-file:before {
  content: "\e203";
}
.glyphicon-level-up:before {
  content: "\e204";
}
.glyphicon-copy:before {
  content: "\e205";
}
.glyphicon-paste:before {
  content: "\e206";
}
.glyphicon-alert:before {
  content: "\e209";
}
.glyphicon-equalizer:before {
  content: "\e210";
}
.glyphicon-king:before {
  content: "\e211";
}
.glyphicon-queen:before {
  content: "\e212";
}
.glyphicon-pawn:before {
  content: "\e213";
}
.glyphicon-bishop:before {
  content: "\e214";
}
.glyphicon-knight:before {
  content: "\e215";
}
.glyphicon-baby-formula:before {
  content: "\e216";
}
.glyphicon-tent:before {
  content: "\26fa";
}
.glyphicon-blackboard:before {
  content: "\e218";
}
.glyphicon-bed:before {
  content: "\e219";
}
.glyphicon-apple:before {
  content: "\f8ff";
}
.glyphicon-erase:before {
  content: "\e221";
}
.glyphicon-hourglass:before {
  content: "\231b";
}
.glyphicon-lamp:before {
  content: "\e223";
}
.glyphicon-duplicate:before {
  content: "\e224";
}
.glyphicon-piggy-bank:before {
  content: "\e225";
}
.glyphicon-scissors:before {
  content: "\e226";
}
.glyphicon-bitcoin:before {
  content: "\e227";
}
.glyphicon-btc:before {
  content: "\e227";
}
.glyphicon-xbt:before {
  content: "\e227";
}
.glyphicon-yen:before {
  content: "\00a5";
}
.glyphicon-jpy:before {
  content: "\00a5";
}
.glyphicon-ruble:before {
  content: "\20bd";
}
.glyphicon-rub:before {
  content: "\20bd";
}
.glyphicon-scale:before {
  content: "\e230";
}
.glyphicon-ice-lolly:before {
  content: "\e231";
}
.glyphicon-ice-lolly-tasted:before {
  content: "\e232";
}
.glyphicon-education:before {
  content: "\e233";
}
.glyphicon-option-horizontal:before {
  content: "\e234";
}
.glyphicon-option-vertical:before {
  content: "\e235";
}
.glyphicon-menu-hamburger:before {
  content: "\e236";
}
.glyphicon-modal-window:before {
  content: "\e237";
}
.glyphicon-oil:before {
  content: "\e238";
}
.glyphicon-grain:before {
  content: "\e239";
}
.glyphicon-sunglasses:before {
  content: "\e240";
}
.glyphicon-text-size:before {
  content: "\e241";
}
.glyphicon-text-color:before {
  content: "\e242";
}
.glyphicon-text-background:before {
  content: "\e243";
}
.glyphicon-object-align-top:before {
  content: "\e244";
}
.glyphicon-object-align-bottom:before {
  content: "\e245";
}
.glyphicon-object-align-horizontal:before {
  content: "\e246";
}
.glyphicon-object-align-left:before {
  content: "\e247";
}
.glyphicon-object-align-vertical:before {
  content: "\e248";
}
.glyphicon-object-align-right:before {
  content: "\e249";
}
.glyphicon-triangle-right:before {
  content: "\e250";
}
.glyphicon-triangle-left:before {
  content: "\e251";
}
.glyphicon-triangle-bottom:before {
  content: "\e252";
}
.glyphicon-triangle-top:before {
  content: "\e253";
}
.glyphicon-console:before {
  content: "\e254";
}
.glyphicon-superscript:before {
  content: "\e255";
}
.glyphicon-subscript:before {
  content: "\e256";
}
.glyphicon-menu-left:before {
  content: "\e257";
}
.glyphicon-menu-right:before {
  content: "\e258";
}
.glyphicon-menu-down:before {
  content: "\e259";
}
.glyphicon-menu-up:before {
  content: "\e260";
}
* {
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}
*:before,
*:after {
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}
html {
  font-size: 10px;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
}
body {
  font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
  font-size: 13px;
  line-height: 1.42857143;
  color: #000;
  background-color: #fff;
}
input,
button,
select,
textarea {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}
a {
  color: #337ab7;
  text-decoration: none;
}
a:hover,
a:focus {
  color: #23527c;
  text-decoration: underline;
}
a:focus {
  outline: 5px auto -webkit-focus-ring-color;
  outline-offset: -2px;
}
figure {
  margin: 0;
}
img {
  vertical-align: middle;
}
.img-responsive,
.thumbnail > img,
.thumbnail a > img,
.carousel-inner > .item > img,
.carousel-inner > .item > a > img {
  display: block;
  max-width: 100%;
  height: auto;
}
.img-rounded {
  border-radius: 3px;
}
.img-thumbnail {
  padding: 4px;
  line-height: 1.42857143;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 2px;
  -webkit-transition: all 0.2s ease-in-out;
  -o-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  display: inline-block;
  max-width: 100%;
  height: auto;
}
.img-circle {
  border-radius: 50%;
}
hr {
  margin-top: 18px;
  margin-bottom: 18px;
  border: 0;
  border-top: 1px solid #eeeeee;
}
.sr-only {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  padding: 0;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  border: 0;
}
.sr-only-focusable:active,
.sr-only-focusable:focus {
  position: static;
  width: auto;
  height: auto;
  margin: 0;
  overflow: visible;
  clip: auto;
}
[role="button"] {
  cursor: pointer;
}
h1,
h2,
h3,
h4,
h5,
h6,
.h1,
.h2,
.h3,
.h4,
.h5,
.h6 {
  font-family: inherit;
  font-weight: 500;
  line-height: 1.1;
  color: inherit;
}
h1 small,
h2 small,
h3 small,
h4 small,
h5 small,
h6 small,
.h1 small,
.h2 small,
.h3 small,
.h4 small,
.h5 small,
.h6 small,
h1 .small,
h2 .small,
h3 .small,
h4 .small,
h5 .small,
h6 .small,
.h1 .small,
.h2 .small,
.h3 .small,
.h4 .small,
.h5 .small,
.h6 .small {
  font-weight: normal;
  line-height: 1;
  color: #777777;
}
h1,
.h1,
h2,
.h2,
h3,
.h3 {
  margin-top: 18px;
  margin-bottom: 9px;
}
h1 small,
.h1 small,
h2 small,
.h2 small,
h3 small,
.h3 small,
h1 .small,
.h1 .small,
h2 .small,
.h2 .small,
h3 .small,
.h3 .small {
  font-size: 65%;
}
h4,
.h4,
h5,
.h5,
h6,
.h6 {
  margin-top: 9px;
  margin-bottom: 9px;
}
h4 small,
.h4 small,
h5 small,
.h5 small,
h6 small,
.h6 small,
h4 .small,
.h4 .small,
h5 .small,
.h5 .small,
h6 .small,
.h6 .small {
  font-size: 75%;
}
h1,
.h1 {
  font-size: 33px;
}
h2,
.h2 {
  font-size: 27px;
}
h3,
.h3 {
  font-size: 23px;
}
h4,
.h4 {
  font-size: 17px;
}
h5,
.h5 {
  font-size: 13px;
}
h6,
.h6 {
  font-size: 12px;
}
p {
  margin: 0 0 9px;
}
.lead {
  margin-bottom: 18px;
  font-size: 14px;
  font-weight: 300;
  line-height: 1.4;
}
@media (min-width: 768px) {
  .lead {
    font-size: 19.5px;
  }
}
small,
.small {
  font-size: 92%;
}
mark,
.mark {
  background-color: #fcf8e3;
  padding: .2em;
}
.text-left {
  text-align: left;
}
.text-right {
  text-align: right;
}
.text-center {
  text-align: center;
}
.text-justify {
  text-align: justify;
}
.text-nowrap {
  white-space: nowrap;
}
.text-lowercase {
  text-transform: lowercase;
}
.text-uppercase {
  text-transform: uppercase;
}
.text-capitalize {
  text-transform: capitalize;
}
.text-muted {
  color: #777777;
}
.text-primary {
  color: #337ab7;
}
a.text-primary:hover,
a.text-primary:focus {
  color: #286090;
}
.text-success {
  color: #3c763d;
}
a.text-success:hover,
a.text-success:focus {
  color: #2b542c;
}
.text-info {
  color: #31708f;
}
a.text-info:hover,
a.text-info:focus {
  color: #245269;
}
.text-warning {
  color: #8a6d3b;
}
a.text-warning:hover,
a.text-warning:focus {
  color: #66512c;
}
.text-danger {
  color: #a94442;
}
a.text-danger:hover,
a.text-danger:focus {
  color: #843534;
}
.bg-primary {
  color: #fff;
  background-color: #337ab7;
}
a.bg-primary:hover,
a.bg-primary:focus {
  background-color: #286090;
}
.bg-success {
  background-color: #dff0d8;
}
a.bg-success:hover,
a.bg-success:focus {
  background-color: #c1e2b3;
}
.bg-info {
  background-color: #d9edf7;
}
a.bg-info:hover,
a.bg-info:focus {
  background-color: #afd9ee;
}
.bg-warning {
  background-color: #fcf8e3;
}
a.bg-warning:hover,
a.bg-warning:focus {
  background-color: #f7ecb5;
}
.bg-danger {
  background-color: #f2dede;
}
a.bg-danger:hover,
a.bg-danger:focus {
  background-color: #e4b9b9;
}
.page-header {
  padding-bottom: 8px;
  margin: 36px 0 18px;
  border-bottom: 1px solid #eeeeee;
}
ul,
ol {
  margin-top: 0;
  margin-bottom: 9px;
}
ul ul,
ol ul,
ul ol,
ol ol {
  margin-bottom: 0;
}
.list-unstyled {
  padding-left: 0;
  list-style: none;
}
.list-inline {
  padding-left: 0;
  list-style: none;
  margin-left: -5px;
}
.list-inline > li {
  display: inline-block;
  padding-left: 5px;
  padding-right: 5px;
}
dl {
  margin-top: 0;
  margin-bottom: 18px;
}
dt,
dd {
  line-height: 1.42857143;
}
dt {
  font-weight: bold;
}
dd {
  margin-left: 0;
}
@media (min-width: 541px) {
  .dl-horizontal dt {
    float: left;
    width: 160px;
    clear: left;
    text-align: right;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
  .dl-horizontal dd {
    margin-left: 180px;
  }
}
abbr[title],
abbr[data-original-title] {
  cursor: help;
  border-bottom: 1px dotted #777777;
}
.initialism {
  font-size: 90%;
  text-transform: uppercase;
}
blockquote {
  padding: 9px 18px;
  margin: 0 0 18px;
  font-size: inherit;
  border-left: 5px solid #eeeeee;
}
blockquote p:last-child,
blockquote ul:last-child,
blockquote ol:last-child {
  margin-bottom: 0;
}
blockquote footer,
blockquote small,
blockquote .small {
  display: block;
  font-size: 80%;
  line-height: 1.42857143;
  color: #777777;
}
blockquote footer:before,
blockquote small:before,
blockquote .small:before {
  content: '\2014 \00A0';
}
.blockquote-reverse,
blockquote.pull-right {
  padding-right: 15px;
  padding-left: 0;
  border-right: 5px solid #eeeeee;
  border-left: 0;
  text-align: right;
}
.blockquote-reverse footer:before,
blockquote.pull-right footer:before,
.blockquote-reverse small:before,
blockquote.pull-right small:before,
.blockquote-reverse .small:before,
blockquote.pull-right .small:before {
  content: '';
}
.blockquote-reverse footer:after,
blockquote.pull-right footer:after,
.blockquote-reverse small:after,
blockquote.pull-right small:after,
.blockquote-reverse .small:after,
blockquote.pull-right .small:after {
  content: '\00A0 \2014';
}
address {
  margin-bottom: 18px;
  font-style: normal;
  line-height: 1.42857143;
}
code,
kbd,
pre,
samp {
  font-family: monospace;
}
code {
  padding: 2px 4px;
  font-size: 90%;
  color: #c7254e;
  background-color: #f9f2f4;
  border-radius: 2px;
}
kbd {
  padding: 2px 4px;
  font-size: 90%;
  color: #888;
  background-color: transparent;
  border-radius: 1px;
  box-shadow: inset 0 -1px 0 rgba(0, 0, 0, 0.25);
}
kbd kbd {
  padding: 0;
  font-size: 100%;
  font-weight: bold;
  box-shadow: none;
}
pre {
  display: block;
  padding: 8.5px;
  margin: 0 0 9px;
  font-size: 12px;
  line-height: 1.42857143;
  word-break: break-all;
  word-wrap: break-word;
  color: #333333;
  background-color: #f5f5f5;
  border: 1px solid #ccc;
  border-radius: 2px;
}
pre code {
  padding: 0;
  font-size: inherit;
  color: inherit;
  white-space: pre-wrap;
  background-color: transparent;
  border-radius: 0;
}
.pre-scrollable {
  max-height: 340px;
  overflow-y: scroll;
}
.container {
  margin-right: auto;
  margin-left: auto;
  padding-left: 0px;
  padding-right: 0px;
}
@media (min-width: 768px) {
  .container {
    width: 768px;
  }
}
@media (min-width: 992px) {
  .container {
    width: 940px;
  }
}
@media (min-width: 1200px) {
  .container {
    width: 1140px;
  }
}
.container-fluid {
  margin-right: auto;
  margin-left: auto;
  padding-left: 0px;
  padding-right: 0px;
}
.row {
  margin-left: 0px;
  margin-right: 0px;
}
.col-xs-1, .col-sm-1, .col-md-1, .col-lg-1, .col-xs-2, .col-sm-2, .col-md-2, .col-lg-2, .col-xs-3, .col-sm-3, .col-md-3, .col-lg-3, .col-xs-4, .col-sm-4, .col-md-4, .col-lg-4, .col-xs-5, .col-sm-5, .col-md-5, .col-lg-5, .col-xs-6, .col-sm-6, .col-md-6, .col-lg-6, .col-xs-7, .col-sm-7, .col-md-7, .col-lg-7, .col-xs-8, .col-sm-8, .col-md-8, .col-lg-8, .col-xs-9, .col-sm-9, .col-md-9, .col-lg-9, .col-xs-10, .col-sm-10, .col-md-10, .col-lg-10, .col-xs-11, .col-sm-11, .col-md-11, .col-lg-11, .col-xs-12, .col-sm-12, .col-md-12, .col-lg-12 {
  position: relative;
  min-height: 1px;
  padding-left: 0px;
  padding-right: 0px;
}
.col-xs-1, .col-xs-2, .col-xs-3, .col-xs-4, .col-xs-5, .col-xs-6, .col-xs-7, .col-xs-8, .col-xs-9, .col-xs-10, .col-xs-11, .col-xs-12 {
  float: left;
}
.col-xs-12 {
  width: 100%;
}
.col-xs-11 {
  width: 91.66666667%;
}
.col-xs-10 {
  width: 83.33333333%;
}
.col-xs-9 {
  width: 75%;
}
.col-xs-8 {
  width: 66.66666667%;
}
.col-xs-7 {
  width: 58.33333333%;
}
.col-xs-6 {
  width: 50%;
}
.col-xs-5 {
  width: 41.66666667%;
}
.col-xs-4 {
  width: 33.33333333%;
}
.col-xs-3 {
  width: 25%;
}
.col-xs-2 {
  width: 16.66666667%;
}
.col-xs-1 {
  width: 8.33333333%;
}
.col-xs-pull-12 {
  right: 100%;
}
.col-xs-pull-11 {
  right: 91.66666667%;
}
.col-xs-pull-10 {
  right: 83.33333333%;
}
.col-xs-pull-9 {
  right: 75%;
}
.col-xs-pull-8 {
  right: 66.66666667%;
}
.col-xs-pull-7 {
  right: 58.33333333%;
}
.col-xs-pull-6 {
  right: 50%;
}
.col-xs-pull-5 {
  right: 41.66666667%;
}
.col-xs-pull-4 {
  right: 33.33333333%;
}
.col-xs-pull-3 {
  right: 25%;
}
.col-xs-pull-2 {
  right: 16.66666667%;
}
.col-xs-pull-1 {
  right: 8.33333333%;
}
.col-xs-pull-0 {
  right: auto;
}
.col-xs-push-12 {
  left: 100%;
}
.col-xs-push-11 {
  left: 91.66666667%;
}
.col-xs-push-10 {
  left: 83.33333333%;
}
.col-xs-push-9 {
  left: 75%;
}
.col-xs-push-8 {
  left: 66.66666667%;
}
.col-xs-push-7 {
  left: 58.33333333%;
}
.col-xs-push-6 {
  left: 50%;
}
.col-xs-push-5 {
  left: 41.66666667%;
}
.col-xs-push-4 {
  left: 33.33333333%;
}
.col-xs-push-3 {
  left: 25%;
}
.col-xs-push-2 {
  left: 16.66666667%;
}
.col-xs-push-1 {
  left: 8.33333333%;
}
.col-xs-push-0 {
  left: auto;
}
.col-xs-offset-12 {
  margin-left: 100%;
}
.col-xs-offset-11 {
  margin-left: 91.66666667%;
}
.col-xs-offset-10 {
  margin-left: 83.33333333%;
}
.col-xs-offset-9 {
  margin-left: 75%;
}
.col-xs-offset-8 {
  margin-left: 66.66666667%;
}
.col-xs-offset-7 {
  margin-left: 58.33333333%;
}
.col-xs-offset-6 {
  margin-left: 50%;
}
.col-xs-offset-5 {
  margin-left: 41.66666667%;
}
.col-xs-offset-4 {
  margin-left: 33.33333333%;
}
.col-xs-offset-3 {
  margin-left: 25%;
}
.col-xs-offset-2 {
  margin-left: 16.66666667%;
}
.col-xs-offset-1 {
  margin-left: 8.33333333%;
}
.col-xs-offset-0 {
  margin-left: 0%;
}
@media (min-width: 768px) {
  .col-sm-1, .col-sm-2, .col-sm-3, .col-sm-4, .col-sm-5, .col-sm-6, .col-sm-7, .col-sm-8, .col-sm-9, .col-sm-10, .col-sm-11, .col-sm-12 {
    float: left;
  }
  .col-sm-12 {
    width: 100%;
  }
  .col-sm-11 {
    width: 91.66666667%;
  }
  .col-sm-10 {
    width: 83.33333333%;
  }
  .col-sm-9 {
    width: 75%;
  }
  .col-sm-8 {
    width: 66.66666667%;
  }
  .col-sm-7 {
    width: 58.33333333%;
  }
  .col-sm-6 {
    width: 50%;
  }
  .col-sm-5 {
    width: 41.66666667%;
  }
  .col-sm-4 {
    width: 33.33333333%;
  }
  .col-sm-3 {
    width: 25%;
  }
  .col-sm-2 {
    width: 16.66666667%;
  }
  .col-sm-1 {
    width: 8.33333333%;
  }
  .col-sm-pull-12 {
    right: 100%;
  }
  .col-sm-pull-11 {
    right: 91.66666667%;
  }
  .col-sm-pull-10 {
    right: 83.33333333%;
  }
  .col-sm-pull-9 {
    right: 75%;
  }
  .col-sm-pull-8 {
    right: 66.66666667%;
  }
  .col-sm-pull-7 {
    right: 58.33333333%;
  }
  .col-sm-pull-6 {
    right: 50%;
  }
  .col-sm-pull-5 {
    right: 41.66666667%;
  }
  .col-sm-pull-4 {
    right: 33.33333333%;
  }
  .col-sm-pull-3 {
    right: 25%;
  }
  .col-sm-pull-2 {
    right: 16.66666667%;
  }
  .col-sm-pull-1 {
    right: 8.33333333%;
  }
  .col-sm-pull-0 {
    right: auto;
  }
  .col-sm-push-12 {
    left: 100%;
  }
  .col-sm-push-11 {
    left: 91.66666667%;
  }
  .col-sm-push-10 {
    left: 83.33333333%;
  }
  .col-sm-push-9 {
    left: 75%;
  }
  .col-sm-push-8 {
    left: 66.66666667%;
  }
  .col-sm-push-7 {
    left: 58.33333333%;
  }
  .col-sm-push-6 {
    left: 50%;
  }
  .col-sm-push-5 {
    left: 41.66666667%;
  }
  .col-sm-push-4 {
    left: 33.33333333%;
  }
  .col-sm-push-3 {
    left: 25%;
  }
  .col-sm-push-2 {
    left: 16.66666667%;
  }
  .col-sm-push-1 {
    left: 8.33333333%;
  }
  .col-sm-push-0 {
    left: auto;
  }
  .col-sm-offset-12 {
    margin-left: 100%;
  }
  .col-sm-offset-11 {
    margin-left: 91.66666667%;
  }
  .col-sm-offset-10 {
    margin-left: 83.33333333%;
  }
  .col-sm-offset-9 {
    margin-left: 75%;
  }
  .col-sm-offset-8 {
    margin-left: 66.66666667%;
  }
  .col-sm-offset-7 {
    margin-left: 58.33333333%;
  }
  .col-sm-offset-6 {
    margin-left: 50%;
  }
  .col-sm-offset-5 {
    margin-left: 41.66666667%;
  }
  .col-sm-offset-4 {
    margin-left: 33.33333333%;
  }
  .col-sm-offset-3 {
    margin-left: 25%;
  }
  .col-sm-offset-2 {
    margin-left: 16.66666667%;
  }
  .col-sm-offset-1 {
    margin-left: 8.33333333%;
  }
  .col-sm-offset-0 {
    margin-left: 0%;
  }
}
@media (min-width: 992px) {
  .col-md-1, .col-md-2, .col-md-3, .col-md-4, .col-md-5, .col-md-6, .col-md-7, .col-md-8, .col-md-9, .col-md-10, .col-md-11, .col-md-12 {
    float: left;
  }
  .col-md-12 {
    width: 100%;
  }
  .col-md-11 {
    width: 91.66666667%;
  }
  .col-md-10 {
    width: 83.33333333%;
  }
  .col-md-9 {
    width: 75%;
  }
  .col-md-8 {
    width: 66.66666667%;
  }
  .col-md-7 {
    width: 58.33333333%;
  }
  .col-md-6 {
    width: 50%;
  }
  .col-md-5 {
    width: 41.66666667%;
  }
  .col-md-4 {
    width: 33.33333333%;
  }
  .col-md-3 {
    width: 25%;
  }
  .col-md-2 {
    width: 16.66666667%;
  }
  .col-md-1 {
    width: 8.33333333%;
  }
  .col-md-pull-12 {
    right: 100%;
  }
  .col-md-pull-11 {
    right: 91.66666667%;
  }
  .col-md-pull-10 {
    right: 83.33333333%;
  }
  .col-md-pull-9 {
    right: 75%;
  }
  .col-md-pull-8 {
    right: 66.66666667%;
  }
  .col-md-pull-7 {
    right: 58.33333333%;
  }
  .col-md-pull-6 {
    right: 50%;
  }
  .col-md-pull-5 {
    right: 41.66666667%;
  }
  .col-md-pull-4 {
    right: 33.33333333%;
  }
  .col-md-pull-3 {
    right: 25%;
  }
  .col-md-pull-2 {
    right: 16.66666667%;
  }
  .col-md-pull-1 {
    right: 8.33333333%;
  }
  .col-md-pull-0 {
    right: auto;
  }
  .col-md-push-12 {
    left: 100%;
  }
  .col-md-push-11 {
    left: 91.66666667%;
  }
  .col-md-push-10 {
    left: 83.33333333%;
  }
  .col-md-push-9 {
    left: 75%;
  }
  .col-md-push-8 {
    left: 66.66666667%;
  }
  .col-md-push-7 {
    left: 58.33333333%;
  }
  .col-md-push-6 {
    left: 50%;
  }
  .col-md-push-5 {
    left: 41.66666667%;
  }
  .col-md-push-4 {
    left: 33.33333333%;
  }
  .col-md-push-3 {
    left: 25%;
  }
  .col-md-push-2 {
    left: 16.66666667%;
  }
  .col-md-push-1 {
    left: 8.33333333%;
  }
  .col-md-push-0 {
    left: auto;
  }
  .col-md-offset-12 {
    margin-left: 100%;
  }
  .col-md-offset-11 {
    margin-left: 91.66666667%;
  }
  .col-md-offset-10 {
    margin-left: 83.33333333%;
  }
  .col-md-offset-9 {
    margin-left: 75%;
  }
  .col-md-offset-8 {
    margin-left: 66.66666667%;
  }
  .col-md-offset-7 {
    margin-left: 58.33333333%;
  }
  .col-md-offset-6 {
    margin-left: 50%;
  }
  .col-md-offset-5 {
    margin-left: 41.66666667%;
  }
  .col-md-offset-4 {
    margin-left: 33.33333333%;
  }
  .col-md-offset-3 {
    margin-left: 25%;
  }
  .col-md-offset-2 {
    margin-left: 16.66666667%;
  }
  .col-md-offset-1 {
    margin-left: 8.33333333%;
  }
  .col-md-offset-0 {
    margin-left: 0%;
  }
}
@media (min-width: 1200px) {
  .col-lg-1, .col-lg-2, .col-lg-3, .col-lg-4, .col-lg-5, .col-lg-6, .col-lg-7, .col-lg-8, .col-lg-9, .col-lg-10, .col-lg-11, .col-lg-12 {
    float: left;
  }
  .col-lg-12 {
    width: 100%;
  }
  .col-lg-11 {
    width: 91.66666667%;
  }
  .col-lg-10 {
    width: 83.33333333%;
  }
  .col-lg-9 {
    width: 75%;
  }
  .col-lg-8 {
    width: 66.66666667%;
  }
  .col-lg-7 {
    width: 58.33333333%;
  }
  .col-lg-6 {
    width: 50%;
  }
  .col-lg-5 {
    width: 41.66666667%;
  }
  .col-lg-4 {
    width: 33.33333333%;
  }
  .col-lg-3 {
    width: 25%;
  }
  .col-lg-2 {
    width: 16.66666667%;
  }
  .col-lg-1 {
    width: 8.33333333%;
  }
  .col-lg-pull-12 {
    right: 100%;
  }
  .col-lg-pull-11 {
    right: 91.66666667%;
  }
  .col-lg-pull-10 {
    right: 83.33333333%;
  }
  .col-lg-pull-9 {
    right: 75%;
  }
  .col-lg-pull-8 {
    right: 66.66666667%;
  }
  .col-lg-pull-7 {
    right: 58.33333333%;
  }
  .col-lg-pull-6 {
    right: 50%;
  }
  .col-lg-pull-5 {
    right: 41.66666667%;
  }
  .col-lg-pull-4 {
    right: 33.33333333%;
  }
  .col-lg-pull-3 {
    right: 25%;
  }
  .col-lg-pull-2 {
    right: 16.66666667%;
  }
  .col-lg-pull-1 {
    right: 8.33333333%;
  }
  .col-lg-pull-0 {
    right: auto;
  }
  .col-lg-push-12 {
    left: 100%;
  }
  .col-lg-push-11 {
    left: 91.66666667%;
  }
  .col-lg-push-10 {
    left: 83.33333333%;
  }
  .col-lg-push-9 {
    left: 75%;
  }
  .col-lg-push-8 {
    left: 66.66666667%;
  }
  .col-lg-push-7 {
    left: 58.33333333%;
  }
  .col-lg-push-6 {
    left: 50%;
  }
  .col-lg-push-5 {
    left: 41.66666667%;
  }
  .col-lg-push-4 {
    left: 33.33333333%;
  }
  .col-lg-push-3 {
    left: 25%;
  }
  .col-lg-push-2 {
    left: 16.66666667%;
  }
  .col-lg-push-1 {
    left: 8.33333333%;
  }
  .col-lg-push-0 {
    left: auto;
  }
  .col-lg-offset-12 {
    margin-left: 100%;
  }
  .col-lg-offset-11 {
    margin-left: 91.66666667%;
  }
  .col-lg-offset-10 {
    margin-left: 83.33333333%;
  }
  .col-lg-offset-9 {
    margin-left: 75%;
  }
  .col-lg-offset-8 {
    margin-left: 66.66666667%;
  }
  .col-lg-offset-7 {
    margin-left: 58.33333333%;
  }
  .col-lg-offset-6 {
    margin-left: 50%;
  }
  .col-lg-offset-5 {
    margin-left: 41.66666667%;
  }
  .col-lg-offset-4 {
    margin-left: 33.33333333%;
  }
  .col-lg-offset-3 {
    margin-left: 25%;
  }
  .col-lg-offset-2 {
    margin-left: 16.66666667%;
  }
  .col-lg-offset-1 {
    margin-left: 8.33333333%;
  }
  .col-lg-offset-0 {
    margin-left: 0%;
  }
}
table {
  background-color: transparent;
}
caption {
  padding-top: 8px;
  padding-bottom: 8px;
  color: #777777;
  text-align: left;
}
th {
  text-align: left;
}
.table {
  width: 100%;
  max-width: 100%;
  margin-bottom: 18px;
}
.table > thead > tr > th,
.table > tbody > tr > th,
.table > tfoot > tr > th,
.table > thead > tr > td,
.table > tbody > tr > td,
.table > tfoot > tr > td {
  padding: 8px;
  line-height: 1.42857143;
  vertical-align: top;
  border-top: 1px solid #ddd;
}
.table > thead > tr > th {
  vertical-align: bottom;
  border-bottom: 2px solid #ddd;
}
.table > caption + thead > tr:first-child > th,
.table > colgroup + thead > tr:first-child > th,
.table > thead:first-child > tr:first-child > th,
.table > caption + thead > tr:first-child > td,
.table > colgroup + thead > tr:first-child > td,
.table > thead:first-child > tr:first-child > td {
  border-top: 0;
}
.table > tbody + tbody {
  border-top: 2px solid #ddd;
}
.table .table {
  background-color: #fff;
}
.table-condensed > thead > tr > th,
.table-condensed > tbody > tr > th,
.table-condensed > tfoot > tr > th,
.table-condensed > thead > tr > td,
.table-condensed > tbody > tr > td,
.table-condensed > tfoot > tr > td {
  padding: 5px;
}
.table-bordered {
  border: 1px solid #ddd;
}
.table-bordered > thead > tr > th,
.table-bordered > tbody > tr > th,
.table-bordered > tfoot > tr > th,
.table-bordered > thead > tr > td,
.table-bordered > tbody > tr > td,
.table-bordered > tfoot > tr > td {
  border: 1px solid #ddd;
}
.table-bordered > thead > tr > th,
.table-bordered > thead > tr > td {
  border-bottom-width: 2px;
}
.table-striped > tbody > tr:nth-of-type(odd) {
  background-color: #f9f9f9;
}
.table-hover > tbody > tr:hover {
  background-color: #f5f5f5;
}
table col[class*="col-"] {
  position: static;
  float: none;
  display: table-column;
}
table td[class*="col-"],
table th[class*="col-"] {
  position: static;
  float: none;
  display: table-cell;
}
.table > thead > tr > td.active,
.table > tbody > tr > td.active,
.table > tfoot > tr > td.active,
.table > thead > tr > th.active,
.table > tbody > tr > th.active,
.table > tfoot > tr > th.active,
.table > thead > tr.active > td,
.table > tbody > tr.active > td,
.table > tfoot > tr.active > td,
.table > thead > tr.active > th,
.table > tbody > tr.active > th,
.table > tfoot > tr.active > th {
  background-color: #f5f5f5;
}
.table-hover > tbody > tr > td.active:hover,
.table-hover > tbody > tr > th.active:hover,
.table-hover > tbody > tr.active:hover > td,
.table-hover > tbody > tr:hover > .active,
.table-hover > tbody > tr.active:hover > th {
  background-color: #e8e8e8;
}
.table > thead > tr > td.success,
.table > tbody > tr > td.success,
.table > tfoot > tr > td.success,
.table > thead > tr > th.success,
.table > tbody > tr > th.success,
.table > tfoot > tr > th.success,
.table > thead > tr.success > td,
.table > tbody > tr.success > td,
.table > tfoot > tr.success > td,
.table > thead > tr.success > th,
.table > tbody > tr.success > th,
.table > tfoot > tr.success > th {
  background-color: #dff0d8;
}
.table-hover > tbody > tr > td.success:hover,
.table-hover > tbody > tr > th.success:hover,
.table-hover > tbody > tr.success:hover > td,
.table-hover > tbody > tr:hover > .success,
.table-hover > tbody > tr.success:hover > th {
  background-color: #d0e9c6;
}
.table > thead > tr > td.info,
.table > tbody > tr > td.info,
.table > tfoot > tr > td.info,
.table > thead > tr > th.info,
.table > tbody > tr > th.info,
.table > tfoot > tr > th.info,
.table > thead > tr.info > td,
.table > tbody > tr.info > td,
.table > tfoot > tr.info > td,
.table > thead > tr.info > th,
.table > tbody > tr.info > th,
.table > tfoot > tr.info > th {
  background-color: #d9edf7;
}
.table-hover > tbody > tr > td.info:hover,
.table-hover > tbody > tr > th.info:hover,
.table-hover > tbody > tr.info:hover > td,
.table-hover > tbody > tr:hover > .info,
.table-hover > tbody > tr.info:hover > th {
  background-color: #c4e3f3;
}
.table > thead > tr > td.warning,
.table > tbody > tr > td.warning,
.table > tfoot > tr > td.warning,
.table > thead > tr > th.warning,
.table > tbody > tr > th.warning,
.table > tfoot > tr > th.warning,
.table > thead > tr.warning > td,
.table > tbody > tr.warning > td,
.table > tfoot > tr.warning > td,
.table > thead > tr.warning > th,
.table > tbody > tr.warning > th,
.table > tfoot > tr.warning > th {
  background-color: #fcf8e3;
}
.table-hover > tbody > tr > td.warning:hover,
.table-hover > tbody > tr > th.warning:hover,
.table-hover > tbody > tr.warning:hover > td,
.table-hover > tbody > tr:hover > .warning,
.table-hover > tbody > tr.warning:hover > th {
  background-color: #faf2cc;
}
.table > thead > tr > td.danger,
.table > tbody > tr > td.danger,
.table > tfoot > tr > td.danger,
.table > thead > tr > th.danger,
.table > tbody > tr > th.danger,
.table > tfoot > tr > th.danger,
.table > thead > tr.danger > td,
.table > tbody > tr.danger > td,
.table > tfoot > tr.danger > td,
.table > thead > tr.danger > th,
.table > tbody > tr.danger > th,
.table > tfoot > tr.danger > th {
  background-color: #f2dede;
}
.table-hover > tbody > tr > td.danger:hover,
.table-hover > tbody > tr > th.danger:hover,
.table-hover > tbody > tr.danger:hover > td,
.table-hover > tbody > tr:hover > .danger,
.table-hover > tbody > tr.danger:hover > th {
  background-color: #ebcccc;
}
.table-responsive {
  overflow-x: auto;
  min-height: 0.01%;
}
@media screen and (max-width: 767px) {
  .table-responsive {
    width: 100%;
    margin-bottom: 13.5px;
    overflow-y: hidden;
    -ms-overflow-style: -ms-autohiding-scrollbar;
    border: 1px solid #ddd;
  }
  .table-responsive > .table {
    margin-bottom: 0;
  }
  .table-responsive > .table > thead > tr > th,
  .table-responsive > .table > tbody > tr > th,
  .table-responsive > .table > tfoot > tr > th,
  .table-responsive > .table > thead > tr > td,
  .table-responsive > .table > tbody > tr > td,
  .table-responsive > .table > tfoot > tr > td {
    white-space: nowrap;
  }
  .table-responsive > .table-bordered {
    border: 0;
  }
  .table-responsive > .table-bordered > thead > tr > th:first-child,
  .table-responsive > .table-bordered > tbody > tr > th:first-child,
  .table-responsive > .table-bordered > tfoot > tr > th:first-child,
  .table-responsive > .table-bordered > thead > tr > td:first-child,
  .table-responsive > .table-bordered > tbody > tr > td:first-child,
  .table-responsive > .table-bordered > tfoot > tr > td:first-child {
    border-left: 0;
  }
  .table-responsive > .table-bordered > thead > tr > th:last-child,
  .table-responsive > .table-bordered > tbody > tr > th:last-child,
  .table-responsive > .table-bordered > tfoot > tr > th:last-child,
  .table-responsive > .table-bordered > thead > tr > td:last-child,
  .table-responsive > .table-bordered > tbody > tr > td:last-child,
  .table-responsive > .table-bordered > tfoot > tr > td:last-child {
    border-right: 0;
  }
  .table-responsive > .table-bordered > tbody > tr:last-child > th,
  .table-responsive > .table-bordered > tfoot > tr:last-child > th,
  .table-responsive > .table-bordered > tbody > tr:last-child > td,
  .table-responsive > .table-bordered > tfoot > tr:last-child > td {
    border-bottom: 0;
  }
}
fieldset {
  padding: 0;
  margin: 0;
  border: 0;
  min-width: 0;
}
legend {
  display: block;
  width: 100%;
  padding: 0;
  margin-bottom: 18px;
  font-size: 19.5px;
  line-height: inherit;
  color: #333333;
  border: 0;
  border-bottom: 1px solid #e5e5e5;
}
label {
  display: inline-block;
  max-width: 100%;
  margin-bottom: 5px;
  font-weight: bold;
}
input[type="search"] {
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}
input[type="radio"],
input[type="checkbox"] {
  margin: 4px 0 0;
  margin-top: 1px \9;
  line-height: normal;
}
input[type="file"] {
  display: block;
}
input[type="range"] {
  display: block;
  width: 100%;
}
select[multiple],
select[size] {
  height: auto;
}
input[type="file"]:focus,
input[type="radio"]:focus,
input[type="checkbox"]:focus {
  outline: 5px auto -webkit-focus-ring-color;
  outline-offset: -2px;
}
output {
  display: block;
  padding-top: 7px;
  font-size: 13px;
  line-height: 1.42857143;
  color: #555555;
}
.form-control {
  display: block;
  width: 100%;
  height: 32px;
  padding: 6px 12px;
  font-size: 13px;
  line-height: 1.42857143;
  color: #555555;
  background-color: #fff;
  background-image: none;
  border: 1px solid #ccc;
  border-radius: 2px;
  -webkit-box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  -webkit-transition: border-color ease-in-out .15s, box-shadow ease-in-out .15s;
  -o-transition: border-color ease-in-out .15s, box-shadow ease-in-out .15s;
  transition: border-color ease-in-out .15s, box-shadow ease-in-out .15s;
}
.form-control:focus {
  border-color: #66afe9;
  outline: 0;
  -webkit-box-shadow: inset 0 1px 1px rgba(0,0,0,.075), 0 0 8px rgba(102, 175, 233, 0.6);
  box-shadow: inset 0 1px 1px rgba(0,0,0,.075), 0 0 8px rgba(102, 175, 233, 0.6);
}
.form-control::-moz-placeholder {
  color: #999;
  opacity: 1;
}
.form-control:-ms-input-placeholder {
  color: #999;
}
.form-control::-webkit-input-placeholder {
  color: #999;
}
.form-control::-ms-expand {
  border: 0;
  background-color: transparent;
}
.form-control[disabled],
.form-control[readonly],
fieldset[disabled] .form-control {
  background-color: #eeeeee;
  opacity: 1;
}
.form-control[disabled],
fieldset[disabled] .form-control {
  cursor: not-allowed;
}
textarea.form-control {
  height: auto;
}
input[type="search"] {
  -webkit-appearance: none;
}
@media screen and (-webkit-min-device-pixel-ratio: 0) {
  input[type="date"].form-control,
  input[type="time"].form-control,
  input[type="datetime-local"].form-control,
  input[type="month"].form-control {
    line-height: 32px;
  }
  input[type="date"].input-sm,
  input[type="time"].input-sm,
  input[type="datetime-local"].input-sm,
  input[type="month"].input-sm,
  .input-group-sm input[type="date"],
  .input-group-sm input[type="time"],
  .input-group-sm input[type="datetime-local"],
  .input-group-sm input[type="month"] {
    line-height: 30px;
  }
  input[type="date"].input-lg,
  input[type="time"].input-lg,
  input[type="datetime-local"].input-lg,
  input[type="month"].input-lg,
  .input-group-lg input[type="date"],
  .input-group-lg input[type="time"],
  .input-group-lg input[type="datetime-local"],
  .input-group-lg input[type="month"] {
    line-height: 45px;
  }
}
.form-group {
  margin-bottom: 15px;
}
.radio,
.checkbox {
  position: relative;
  display: block;
  margin-top: 10px;
  margin-bottom: 10px;
}
.radio label,
.checkbox label {
  min-height: 18px;
  padding-left: 20px;
  margin-bottom: 0;
  font-weight: normal;
  cursor: pointer;
}
.radio input[type="radio"],
.radio-inline input[type="radio"],
.checkbox input[type="checkbox"],
.checkbox-inline input[type="checkbox"] {
  position: absolute;
  margin-left: -20px;
  margin-top: 4px \9;
}
.radio + .radio,
.checkbox + .checkbox {
  margin-top: -5px;
}
.radio-inline,
.checkbox-inline {
  position: relative;
  display: inline-block;
  padding-left: 20px;
  margin-bottom: 0;
  vertical-align: middle;
  font-weight: normal;
  cursor: pointer;
}
.radio-inline + .radio-inline,
.checkbox-inline + .checkbox-inline {
  margin-top: 0;
  margin-left: 10px;
}
input[type="radio"][disabled],
input[type="checkbox"][disabled],
input[type="radio"].disabled,
input[type="checkbox"].disabled,
fieldset[disabled] input[type="radio"],
fieldset[disabled] input[type="checkbox"] {
  cursor: not-allowed;
}
.radio-inline.disabled,
.checkbox-inline.disabled,
fieldset[disabled] .radio-inline,
fieldset[disabled] .checkbox-inline {
  cursor: not-allowed;
}
.radio.disabled label,
.checkbox.disabled label,
fieldset[disabled] .radio label,
fieldset[disabled] .checkbox label {
  cursor: not-allowed;
}
.form-control-static {
  padding-top: 7px;
  padding-bottom: 7px;
  margin-bottom: 0;
  min-height: 31px;
}
.form-control-static.input-lg,
.form-control-static.input-sm {
  padding-left: 0;
  padding-right: 0;
}
.input-sm {
  height: 30px;
  padding: 5px 10px;
  font-size: 12px;
  line-height: 1.5;
  border-radius: 1px;
}
select.input-sm {
  height: 30px;
  line-height: 30px;
}
textarea.input-sm,
select[multiple].input-sm {
  height: auto;
}
.form-group-sm .form-control {
  height: 30px;
  padding: 5px 10px;
  font-size: 12px;
  line-height: 1.5;
  border-radius: 1px;
}
.form-group-sm select.form-control {
  height: 30px;
  line-height: 30px;
}
.form-group-sm textarea.form-control,
.form-group-sm select[multiple].form-control {
  height: auto;
}
.form-group-sm .form-control-static {
  height: 30px;
  min-height: 30px;
  padding: 6px 10px;
  font-size: 12px;
  line-height: 1.5;
}
.input-lg {
  height: 45px;
  padding: 10px 16px;
  font-size: 17px;
  line-height: 1.3333333;
  border-radius: 3px;
}
select.input-lg {
  height: 45px;
  line-height: 45px;
}
textarea.input-lg,
select[multiple].input-lg {
  height: auto;
}
.form-group-lg .form-control {
  height: 45px;
  padding: 10px 16px;
  font-size: 17px;
  line-height: 1.3333333;
  border-radius: 3px;
}
.form-group-lg select.form-control {
  height: 45px;
  line-height: 45px;
}
.form-group-lg textarea.form-control,
.form-group-lg select[multiple].form-control {
  height: auto;
}
.form-group-lg .form-control-static {
  height: 45px;
  min-height: 35px;
  padding: 11px 16px;
  font-size: 17px;
  line-height: 1.3333333;
}
.has-feedback {
  position: relative;
}
.has-feedback .form-control {
  padding-right: 40px;
}
.form-control-feedback {
  position: absolute;
  top: 0;
  right: 0;
  z-index: 2;
  display: block;
  width: 32px;
  height: 32px;
  line-height: 32px;
  text-align: center;
  pointer-events: none;
}
.input-lg + .form-control-feedback,
.input-group-lg + .form-control-feedback,
.form-group-lg .form-control + .form-control-feedback {
  width: 45px;
  height: 45px;
  line-height: 45px;
}
.input-sm + .form-control-feedback,
.input-group-sm + .form-control-feedback,
.form-group-sm .form-control + .form-control-feedback {
  width: 30px;
  height: 30px;
  line-height: 30px;
}
.has-success .help-block,
.has-success .control-label,
.has-success .radio,
.has-success .checkbox,
.has-success .radio-inline,
.has-success .checkbox-inline,
.has-success.radio label,
.has-success.checkbox label,
.has-success.radio-inline label,
.has-success.checkbox-inline label {
  color: #3c763d;
}
.has-success .form-control {
  border-color: #3c763d;
  -webkit-box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
}
.has-success .form-control:focus {
  border-color: #2b542c;
  -webkit-box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075), 0 0 6px #67b168;
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075), 0 0 6px #67b168;
}
.has-success .input-group-addon {
  color: #3c763d;
  border-color: #3c763d;
  background-color: #dff0d8;
}
.has-success .form-control-feedback {
  color: #3c763d;
}
.has-warning .help-block,
.has-warning .control-label,
.has-warning .radio,
.has-warning .checkbox,
.has-warning .radio-inline,
.has-warning .checkbox-inline,
.has-warning.radio label,
.has-warning.checkbox label,
.has-warning.radio-inline label,
.has-warning.checkbox-inline label {
  color: #8a6d3b;
}
.has-warning .form-control {
  border-color: #8a6d3b;
  -webkit-box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
}
.has-warning .form-control:focus {
  border-color: #66512c;
  -webkit-box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075), 0 0 6px #c0a16b;
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075), 0 0 6px #c0a16b;
}
.has-warning .input-group-addon {
  color: #8a6d3b;
  border-color: #8a6d3b;
  background-color: #fcf8e3;
}
.has-warning .form-control-feedback {
  color: #8a6d3b;
}
.has-error .help-block,
.has-error .control-label,
.has-error .radio,
.has-error .checkbox,
.has-error .radio-inline,
.has-error .checkbox-inline,
.has-error.radio label,
.has-error.checkbox label,
.has-error.radio-inline label,
.has-error.checkbox-inline label {
  color: #a94442;
}
.has-error .form-control {
  border-color: #a94442;
  -webkit-box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
}
.has-error .form-control:focus {
  border-color: #843534;
  -webkit-box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075), 0 0 6px #ce8483;
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075), 0 0 6px #ce8483;
}
.has-error .input-group-addon {
  color: #a94442;
  border-color: #a94442;
  background-color: #f2dede;
}
.has-error .form-control-feedback {
  color: #a94442;
}
.has-feedback label ~ .form-control-feedback {
  top: 23px;
}
.has-feedback label.sr-only ~ .form-control-feedback {
  top: 0;
}
.help-block {
  display: block;
  margin-top: 5px;
  margin-bottom: 10px;
  color: #404040;
}
@media (min-width: 768px) {
  .form-inline .form-group {
    display: inline-block;
    margin-bottom: 0;
    vertical-align: middle;
  }
  .form-inline .form-control {
    display: inline-block;
    width: auto;
    vertical-align: middle;
  }
  .form-inline .form-control-static {
    display: inline-block;
  }
  .form-inline .input-group {
    display: inline-table;
    vertical-align: middle;
  }
  .form-inline .input-group .input-group-addon,
  .form-inline .input-group .input-group-btn,
  .form-inline .input-group .form-control {
    width: auto;
  }
  .form-inline .input-group > .form-control {
    width: 100%;
  }
  .form-inline .control-label {
    margin-bottom: 0;
    vertical-align: middle;
  }
  .form-inline .radio,
  .form-inline .checkbox {
    display: inline-block;
    margin-top: 0;
    margin-bottom: 0;
    vertical-align: middle;
  }
  .form-inline .radio label,
  .form-inline .checkbox label {
    padding-left: 0;
  }
  .form-inline .radio input[type="radio"],
  .form-inline .checkbox input[type="checkbox"] {
    position: relative;
    margin-left: 0;
  }
  .form-inline .has-feedback .form-control-feedback {
    top: 0;
  }
}
.form-horizontal .radio,
.form-horizontal .checkbox,
.form-horizontal .radio-inline,
.form-horizontal .checkbox-inline {
  margin-top: 0;
  margin-bottom: 0;
  padding-top: 7px;
}
.form-horizontal .radio,
.form-horizontal .checkbox {
  min-height: 25px;
}
.form-horizontal .form-group {
  margin-left: 0px;
  margin-right: 0px;
}
@media (min-width: 768px) {
  .form-horizontal .control-label {
    text-align: right;
    margin-bottom: 0;
    padding-top: 7px;
  }
}
.form-horizontal .has-feedback .form-control-feedback {
  right: 0px;
}
@media (min-width: 768px) {
  .form-horizontal .form-group-lg .control-label {
    padding-top: 11px;
    font-size: 17px;
  }
}
@media (min-width: 768px) {
  .form-horizontal .form-group-sm .control-label {
    padding-top: 6px;
    font-size: 12px;
  }
}
.btn {
  display: inline-block;
  margin-bottom: 0;
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  touch-action: manipulation;
  cursor: pointer;
  background-image: none;
  border: 1px solid transparent;
  white-space: nowrap;
  padding: 6px 12px;
  font-size: 13px;
  line-height: 1.42857143;
  border-radius: 2px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}
.btn:focus,
.btn:active:focus,
.btn.active:focus,
.btn.focus,
.btn:active.focus,
.btn.active.focus {
  outline: 5px auto -webkit-focus-ring-color;
  outline-offset: -2px;
}
.btn:hover,
.btn:focus,
.btn.focus {
  color: #333;
  text-decoration: none;
}
.btn:active,
.btn.active {
  outline: 0;
  background-image: none;
  -webkit-box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
  box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
}
.btn.disabled,
.btn[disabled],
fieldset[disabled] .btn {
  cursor: not-allowed;
  opacity: 0.65;
  filter: alpha(opacity=65);
  -webkit-box-shadow: none;
  box-shadow: none;
}
a.btn.disabled,
fieldset[disabled] a.btn {
  pointer-events: none;
}
.btn-default {
  color: #333;
  background-color: #fff;
  border-color: #ccc;
}
.btn-default:focus,
.btn-default.focus {
  color: #333;
  background-color: #e6e6e6;
  border-color: #8c8c8c;
}
.btn-default:hover {
  color: #333;
  background-color: #e6e6e6;
  border-color: #adadad;
}
.btn-default:active,
.btn-default.active,
.open > .dropdown-toggle.btn-default {
  color: #333;
  background-color: #e6e6e6;
  border-color: #adadad;
}
.btn-default:active:hover,
.btn-default.active:hover,
.open > .dropdown-toggle.btn-default:hover,
.btn-default:active:focus,
.btn-default.active:focus,
.open > .dropdown-toggle.btn-default:focus,
.btn-default:active.focus,
.btn-default.active.focus,
.open > .dropdown-toggle.btn-default.focus {
  color: #333;
  background-color: #d4d4d4;
  border-color: #8c8c8c;
}
.btn-default:active,
.btn-default.active,
.open > .dropdown-toggle.btn-default {
  background-image: none;
}
.btn-default.disabled:hover,
.btn-default[disabled]:hover,
fieldset[disabled] .btn-default:hover,
.btn-default.disabled:focus,
.btn-default[disabled]:focus,
fieldset[disabled] .btn-default:focus,
.btn-default.disabled.focus,
.btn-default[disabled].focus,
fieldset[disabled] .btn-default.focus {
  background-color: #fff;
  border-color: #ccc;
}
.btn-default .badge {
  color: #fff;
  background-color: #333;
}
.btn-primary {
  color: #fff;
  background-color: #337ab7;
  border-color: #2e6da4;
}
.btn-primary:focus,
.btn-primary.focus {
  color: #fff;
  background-color: #286090;
  border-color: #122b40;
}
.btn-primary:hover {
  color: #fff;
  background-color: #286090;
  border-color: #204d74;
}
.btn-primary:active,
.btn-primary.active,
.open > .dropdown-toggle.btn-primary {
  color: #fff;
  background-color: #286090;
  border-color: #204d74;
}
.btn-primary:active:hover,
.btn-primary.active:hover,
.open > .dropdown-toggle.btn-primary:hover,
.btn-primary:active:focus,
.btn-primary.active:focus,
.open > .dropdown-toggle.btn-primary:focus,
.btn-primary:active.focus,
.btn-primary.active.focus,
.open > .dropdown-toggle.btn-primary.focus {
  color: #fff;
  background-color: #204d74;
  border-color: #122b40;
}
.btn-primary:active,
.btn-primary.active,
.open > .dropdown-toggle.btn-primary {
  background-image: none;
}
.btn-primary.disabled:hover,
.btn-primary[disabled]:hover,
fieldset[disabled] .btn-primary:hover,
.btn-primary.disabled:focus,
.btn-primary[disabled]:focus,
fieldset[disabled] .btn-primary:focus,
.btn-primary.disabled.focus,
.btn-primary[disabled].focus,
fieldset[disabled] .btn-primary.focus {
  background-color: #337ab7;
  border-color: #2e6da4;
}
.btn-primary .badge {
  color: #337ab7;
  background-color: #fff;
}
.btn-success {
  color: #fff;
  background-color: #5cb85c;
  border-color: #4cae4c;
}
.btn-success:focus,
.btn-success.focus {
  color: #fff;
  background-color: #449d44;
  border-color: #255625;
}
.btn-success:hover {
  color: #fff;
  background-color: #449d44;
  border-color: #398439;
}
.btn-success:active,
.btn-success.active,
.open > .dropdown-toggle.btn-success {
  color: #fff;
  background-color: #449d44;
  border-color: #398439;
}
.btn-success:active:hover,
.btn-success.active:hover,
.open > .dropdown-toggle.btn-success:hover,
.btn-success:active:focus,
.btn-success.active:focus,
.open > .dropdown-toggle.btn-success:focus,
.btn-success:active.focus,
.btn-success.active.focus,
.open > .dropdown-toggle.btn-success.focus {
  color: #fff;
  background-color: #398439;
  border-color: #255625;
}
.btn-success:active,
.btn-success.active,
.open > .dropdown-toggle.btn-success {
  background-image: none;
}
.btn-success.disabled:hover,
.btn-success[disabled]:hover,
fieldset[disabled] .btn-success:hover,
.btn-success.disabled:focus,
.btn-success[disabled]:focus,
fieldset[disabled] .btn-success:focus,
.btn-success.disabled.focus,
.btn-success[disabled].focus,
fieldset[disabled] .btn-success.focus {
  background-color: #5cb85c;
  border-color: #4cae4c;
}
.btn-success .badge {
  color: #5cb85c;
  background-color: #fff;
}
.btn-info {
  color: #fff;
  background-color: #5bc0de;
  border-color: #46b8da;
}
.btn-info:focus,
.btn-info.focus {
  color: #fff;
  background-color: #31b0d5;
  border-color: #1b6d85;
}
.btn-info:hover {
  color: #fff;
  background-color: #31b0d5;
  border-color: #269abc;
}
.btn-info:active,
.btn-info.active,
.open > .dropdown-toggle.btn-info {
  color: #fff;
  background-color: #31b0d5;
  border-color: #269abc;
}
.btn-info:active:hover,
.btn-info.active:hover,
.open > .dropdown-toggle.btn-info:hover,
.btn-info:active:focus,
.btn-info.active:focus,
.open > .dropdown-toggle.btn-info:focus,
.btn-info:active.focus,
.btn-info.active.focus,
.open > .dropdown-toggle.btn-info.focus {
  color: #fff;
  background-color: #269abc;
  border-color: #1b6d85;
}
.btn-info:active,
.btn-info.active,
.open > .dropdown-toggle.btn-info {
  background-image: none;
}
.btn-info.disabled:hover,
.btn-info[disabled]:hover,
fieldset[disabled] .btn-info:hover,
.btn-info.disabled:focus,
.btn-info[disabled]:focus,
fieldset[disabled] .btn-info:focus,
.btn-info.disabled.focus,
.btn-info[disabled].focus,
fieldset[disabled] .btn-info.focus {
  background-color: #5bc0de;
  border-color: #46b8da;
}
.btn-info .badge {
  color: #5bc0de;
  background-color: #fff;
}
.btn-warning {
  color: #fff;
  background-color: #f0ad4e;
  border-color: #eea236;
}
.btn-warning:focus,
.btn-warning.focus {
  color: #fff;
  background-color: #ec971f;
  border-color: #985f0d;
}
.btn-warning:hover {
  color: #fff;
  background-color: #ec971f;
  border-color: #d58512;
}
.btn-warning:active,
.btn-warning.active,
.open > .dropdown-toggle.btn-warning {
  color: #fff;
  background-color: #ec971f;
  border-color: #d58512;
}
.btn-warning:active:hover,
.btn-warning.active:hover,
.open > .dropdown-toggle.btn-warning:hover,
.btn-warning:active:focus,
.btn-warning.active:focus,
.open > .dropdown-toggle.btn-warning:focus,
.btn-warning:active.focus,
.btn-warning.active.focus,
.open > .dropdown-toggle.btn-warning.focus {
  color: #fff;
  background-color: #d58512;
  border-color: #985f0d;
}
.btn-warning:active,
.btn-warning.active,
.open > .dropdown-toggle.btn-warning {
  background-image: none;
}
.btn-warning.disabled:hover,
.btn-warning[disabled]:hover,
fieldset[disabled] .btn-warning:hover,
.btn-warning.disabled:focus,
.btn-warning[disabled]:focus,
fieldset[disabled] .btn-warning:focus,
.btn-warning.disabled.focus,
.btn-warning[disabled].focus,
fieldset[disabled] .btn-warning.focus {
  background-color: #f0ad4e;
  border-color: #eea236;
}
.btn-warning .badge {
  color: #f0ad4e;
  background-color: #fff;
}
.btn-danger {
  color: #fff;
  background-color: #d9534f;
  border-color: #d43f3a;
}
.btn-danger:focus,
.btn-danger.focus {
  color: #fff;
  background-color: #c9302c;
  border-color: #761c19;
}
.btn-danger:hover {
  color: #fff;
  background-color: #c9302c;
  border-color: #ac2925;
}
.btn-danger:active,
.btn-danger.active,
.open > .dropdown-toggle.btn-danger {
  color: #fff;
  background-color: #c9302c;
  border-color: #ac2925;
}
.btn-danger:active:hover,
.btn-danger.active:hover,
.open > .dropdown-toggle.btn-danger:hover,
.btn-danger:active:focus,
.btn-danger.active:focus,
.open > .dropdown-toggle.btn-danger:focus,
.btn-danger:active.focus,
.btn-danger.active.focus,
.open > .dropdown-toggle.btn-danger.focus {
  color: #fff;
  background-color: #ac2925;
  border-color: #761c19;
}
.btn-danger:active,
.btn-danger.active,
.open > .dropdown-toggle.btn-danger {
  background-image: none;
}
.btn-danger.disabled:hover,
.btn-danger[disabled]:hover,
fieldset[disabled] .btn-danger:hover,
.btn-danger.disabled:focus,
.btn-danger[disabled]:focus,
fieldset[disabled] .btn-danger:focus,
.btn-danger.disabled.focus,
.btn-danger[disabled].focus,
fieldset[disabled] .btn-danger.focus {
  background-color: #d9534f;
  border-color: #d43f3a;
}
.btn-danger .badge {
  color: #d9534f;
  background-color: #fff;
}
.btn-link {
  color: #337ab7;
  font-weight: normal;
  border-radius: 0;
}
.btn-link,
.btn-link:active,
.btn-link.active,
.btn-link[disabled],
fieldset[disabled] .btn-link {
  background-color: transparent;
  -webkit-box-shadow: none;
  box-shadow: none;
}
.btn-link,
.btn-link:hover,
.btn-link:focus,
.btn-link:active {
  border-color: transparent;
}
.btn-link:hover,
.btn-link:focus {
  color: #23527c;
  text-decoration: underline;
  background-color: transparent;
}
.btn-link[disabled]:hover,
fieldset[disabled] .btn-link:hover,
.btn-link[disabled]:focus,
fieldset[disabled] .btn-link:focus {
  color: #777777;
  text-decoration: none;
}
.btn-lg,
.btn-group-lg > .btn {
  padding: 10px 16px;
  font-size: 17px;
  line-height: 1.3333333;
  border-radius: 3px;
}
.btn-sm,
.btn-group-sm > .btn {
  padding: 5px 10px;
  font-size: 12px;
  line-height: 1.5;
  border-radius: 1px;
}
.btn-xs,
.btn-group-xs > .btn {
  padding: 1px 5px;
  font-size: 12px;
  line-height: 1.5;
  border-radius: 1px;
}
.btn-block {
  display: block;
  width: 100%;
}
.btn-block + .btn-block {
  margin-top: 5px;
}
input[type="submit"].btn-block,
input[type="reset"].btn-block,
input[type="button"].btn-block {
  width: 100%;
}
.fade {
  opacity: 0;
  -webkit-transition: opacity 0.15s linear;
  -o-transition: opacity 0.15s linear;
  transition: opacity 0.15s linear;
}
.fade.in {
  opacity: 1;
}
.collapse {
  display: none;
}
.collapse.in {
  display: block;
}
tr.collapse.in {
  display: table-row;
}
tbody.collapse.in {
  display: table-row-group;
}
.collapsing {
  position: relative;
  height: 0;
  overflow: hidden;
  -webkit-transition-property: height, visibility;
  transition-property: height, visibility;
  -webkit-transition-duration: 0.35s;
  transition-duration: 0.35s;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
}
.caret {
  display: inline-block;
  width: 0;
  height: 0;
  margin-left: 2px;
  vertical-align: middle;
  border-top: 4px dashed;
  border-top: 4px solid \9;
  border-right: 4px solid transparent;
  border-left: 4px solid transparent;
}
.dropup,
.dropdown {
  position: relative;
}
.dropdown-toggle:focus {
  outline: 0;
}
.dropdown-menu {
  position: absolute;
  top: 100%;
  left: 0;
  z-index: 1000;
  display: none;
  float: left;
  min-width: 160px;
  padding: 5px 0;
  margin: 2px 0 0;
  list-style: none;
  font-size: 13px;
  text-align: left;
  background-color: #fff;
  border: 1px solid #ccc;
  border: 1px solid rgba(0, 0, 0, 0.15);
  border-radius: 2px;
  -webkit-box-shadow: 0 6px 12px rgba(0, 0, 0, 0.175);
  box-shadow: 0 6px 12px rgba(0, 0, 0, 0.175);
  background-clip: padding-box;
}
.dropdown-menu.pull-right {
  right: 0;
  left: auto;
}
.dropdown-menu .divider {
  height: 1px;
  margin: 8px 0;
  overflow: hidden;
  background-color: #e5e5e5;
}
.dropdown-menu > li > a {
  display: block;
  padding: 3px 20px;
  clear: both;
  font-weight: normal;
  line-height: 1.42857143;
  color: #333333;
  white-space: nowrap;
}
.dropdown-menu > li > a:hover,
.dropdown-menu > li > a:focus {
  text-decoration: none;
  color: #262626;
  background-color: #f5f5f5;
}
.dropdown-menu > .active > a,
.dropdown-menu > .active > a:hover,
.dropdown-menu > .active > a:focus {
  color: #fff;
  text-decoration: none;
  outline: 0;
  background-color: #337ab7;
}
.dropdown-menu > .disabled > a,
.dropdown-menu > .disabled > a:hover,
.dropdown-menu > .disabled > a:focus {
  color: #777777;
}
.dropdown-menu > .disabled > a:hover,
.dropdown-menu > .disabled > a:focus {
  text-decoration: none;
  background-color: transparent;
  background-image: none;
  filter: progid:DXImageTransform.Microsoft.gradient(enabled = false);
  cursor: not-allowed;
}
.open > .dropdown-menu {
  display: block;
}
.open > a {
  outline: 0;
}
.dropdown-menu-right {
  left: auto;
  right: 0;
}
.dropdown-menu-left {
  left: 0;
  right: auto;
}
.dropdown-header {
  display: block;
  padding: 3px 20px;
  font-size: 12px;
  line-height: 1.42857143;
  color: #777777;
  white-space: nowrap;
}
.dropdown-backdrop {
  position: fixed;
  left: 0;
  right: 0;
  bottom: 0;
  top: 0;
  z-index: 990;
}
.pull-right > .dropdown-menu {
  right: 0;
  left: auto;
}
.dropup .caret,
.navbar-fixed-bottom .dropdown .caret {
  border-top: 0;
  border-bottom: 4px dashed;
  border-bottom: 4px solid \9;
  content: "";
}
.dropup .dropdown-menu,
.navbar-fixed-bottom .dropdown .dropdown-menu {
  top: auto;
  bottom: 100%;
  margin-bottom: 2px;
}
@media (min-width: 541px) {
  .navbar-right .dropdown-menu {
    left: auto;
    right: 0;
  }
  .navbar-right .dropdown-menu-left {
    left: 0;
    right: auto;
  }
}
.btn-group,
.btn-group-vertical {
  position: relative;
  display: inline-block;
  vertical-align: middle;
}
.btn-group > .btn,
.btn-group-vertical > .btn {
  position: relative;
  float: left;
}
.btn-group > .btn:hover,
.btn-group-vertical > .btn:hover,
.btn-group > .btn:focus,
.btn-group-vertical > .btn:focus,
.btn-group > .btn:active,
.btn-group-vertical > .btn:active,
.btn-group > .btn.active,
.btn-group-vertical > .btn.active {
  z-index: 2;
}
.btn-group .btn + .btn,
.btn-group .btn + .btn-group,
.btn-group .btn-group + .btn,
.btn-group .btn-group + .btn-group {
  margin-left: -1px;
}
.btn-toolbar {
  margin-left: -5px;
}
.btn-toolbar .btn,
.btn-toolbar .btn-group,
.btn-toolbar .input-group {
  float: left;
}
.btn-toolbar > .btn,
.btn-toolbar > .btn-group,
.btn-toolbar > .input-group {
  margin-left: 5px;
}
.btn-group > .btn:not(:first-child):not(:last-child):not(.dropdown-toggle) {
  border-radius: 0;
}
.btn-group > .btn:first-child {
  margin-left: 0;
}
.btn-group > .btn:first-child:not(:last-child):not(.dropdown-toggle) {
  border-bottom-right-radius: 0;
  border-top-right-radius: 0;
}
.btn-group > .btn:last-child:not(:first-child),
.btn-group > .dropdown-toggle:not(:first-child) {
  border-bottom-left-radius: 0;
  border-top-left-radius: 0;
}
.btn-group > .btn-group {
  float: left;
}
.btn-group > .btn-group:not(:first-child):not(:last-child) > .btn {
  border-radius: 0;
}
.btn-group > .btn-group:first-child:not(:last-child) > .btn:last-child,
.btn-group > .btn-group:first-child:not(:last-child) > .dropdown-toggle {
  border-bottom-right-radius: 0;
  border-top-right-radius: 0;
}
.btn-group > .btn-group:last-child:not(:first-child) > .btn:first-child {
  border-bottom-left-radius: 0;
  border-top-left-radius: 0;
}
.btn-group .dropdown-toggle:active,
.btn-group.open .dropdown-toggle {
  outline: 0;
}
.btn-group > .btn + .dropdown-toggle {
  padding-left: 8px;
  padding-right: 8px;
}
.btn-group > .btn-lg + .dropdown-toggle {
  padding-left: 12px;
  padding-right: 12px;
}
.btn-group.open .dropdown-toggle {
  -webkit-box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
  box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
}
.btn-group.open .dropdown-toggle.btn-link {
  -webkit-box-shadow: none;
  box-shadow: none;
}
.btn .caret {
  margin-left: 0;
}
.btn-lg .caret {
  border-width: 5px 5px 0;
  border-bottom-width: 0;
}
.dropup .btn-lg .caret {
  border-width: 0 5px 5px;
}
.btn-group-vertical > .btn,
.btn-group-vertical > .btn-group,
.btn-group-vertical > .btn-group > .btn {
  display: block;
  float: none;
  width: 100%;
  max-width: 100%;
}
.btn-group-vertical > .btn-group > .btn {
  float: none;
}
.btn-group-vertical > .btn + .btn,
.btn-group-vertical > .btn + .btn-group,
.btn-group-vertical > .btn-group + .btn,
.btn-group-vertical > .btn-group + .btn-group {
  margin-top: -1px;
  margin-left: 0;
}
.btn-group-vertical > .btn:not(:first-child):not(:last-child) {
  border-radius: 0;
}
.btn-group-vertical > .btn:first-child:not(:last-child) {
  border-top-right-radius: 2px;
  border-top-left-radius: 2px;
  border-bottom-right-radius: 0;
  border-bottom-left-radius: 0;
}
.btn-group-vertical > .btn:last-child:not(:first-child) {
  border-top-right-radius: 0;
  border-top-left-radius: 0;
  border-bottom-right-radius: 2px;
  border-bottom-left-radius: 2px;
}
.btn-group-vertical > .btn-group:not(:first-child):not(:last-child) > .btn {
  border-radius: 0;
}
.btn-group-vertical > .btn-group:first-child:not(:last-child) > .btn:last-child,
.btn-group-vertical > .btn-group:first-child:not(:last-child) > .dropdown-toggle {
  border-bottom-right-radius: 0;
  border-bottom-left-radius: 0;
}
.btn-group-vertical > .btn-group:last-child:not(:first-child) > .btn:first-child {
  border-top-right-radius: 0;
  border-top-left-radius: 0;
}
.btn-group-justified {
  display: table;
  width: 100%;
  table-layout: fixed;
  border-collapse: separate;
}
.btn-group-justified > .btn,
.btn-group-justified > .btn-group {
  float: none;
  display: table-cell;
  width: 1%;
}
.btn-group-justified > .btn-group .btn {
  width: 100%;
}
.btn-group-justified > .btn-group .dropdown-menu {
  left: auto;
}
[data-toggle="buttons"] > .btn input[type="radio"],
[data-toggle="buttons"] > .btn-group > .btn input[type="radio"],
[data-toggle="buttons"] > .btn input[type="checkbox"],
[data-toggle="buttons"] > .btn-group > .btn input[type="checkbox"] {
  position: absolute;
  clip: rect(0, 0, 0, 0);
  pointer-events: none;
}
.input-group {
  position: relative;
  display: table;
  border-collapse: separate;
}
.input-group[class*="col-"] {
  float: none;
  padding-left: 0;
  padding-right: 0;
}
.input-group .form-control {
  position: relative;
  z-index: 2;
  float: left;
  width: 100%;
  margin-bottom: 0;
}
.input-group .form-control:focus {
  z-index: 3;
}
.input-group-lg > .form-control,
.input-group-lg > .input-group-addon,
.input-group-lg > .input-group-btn > .btn {
  height: 45px;
  padding: 10px 16px;
  font-size: 17px;
  line-height: 1.3333333;
  border-radius: 3px;
}
select.input-group-lg > .form-control,
select.input-group-lg > .input-group-addon,
select.input-group-lg > .input-group-btn > .btn {
  height: 45px;
  line-height: 45px;
}
textarea.input-group-lg > .form-control,
textarea.input-group-lg > .input-group-addon,
textarea.input-group-lg > .input-group-btn > .btn,
select[multiple].input-group-lg > .form-control,
select[multiple].input-group-lg > .input-group-addon,
select[multiple].input-group-lg > .input-group-btn > .btn {
  height: auto;
}
.input-group-sm > .form-control,
.input-group-sm > .input-group-addon,
.input-group-sm > .input-group-btn > .btn {
  height: 30px;
  padding: 5px 10px;
  font-size: 12px;
  line-height: 1.5;
  border-radius: 1px;
}
select.input-group-sm > .form-control,
select.input-group-sm > .input-group-addon,
select.input-group-sm > .input-group-btn > .btn {
  height: 30px;
  line-height: 30px;
}
textarea.input-group-sm > .form-control,
textarea.input-group-sm > .input-group-addon,
textarea.input-group-sm > .input-group-btn > .btn,
select[multiple].input-group-sm > .form-control,
select[multiple].input-group-sm > .input-group-addon,
select[multiple].input-group-sm > .input-group-btn > .btn {
  height: auto;
}
.input-group-addon,
.input-group-btn,
.input-group .form-control {
  display: table-cell;
}
.input-group-addon:not(:first-child):not(:last-child),
.input-group-btn:not(:first-child):not(:last-child),
.input-group .form-control:not(:first-child):not(:last-child) {
  border-radius: 0;
}
.input-group-addon,
.input-group-btn {
  width: 1%;
  white-space: nowrap;
  vertical-align: middle;
}
.input-group-addon {
  padding: 6px 12px;
  font-size: 13px;
  font-weight: normal;
  line-height: 1;
  color: #555555;
  text-align: center;
  background-color: #eeeeee;
  border: 1px solid #ccc;
  border-radius: 2px;
}
.input-group-addon.input-sm {
  padding: 5px 10px;
  font-size: 12px;
  border-radius: 1px;
}
.input-group-addon.input-lg {
  padding: 10px 16px;
  font-size: 17px;
  border-radius: 3px;
}
.input-group-addon input[type="radio"],
.input-group-addon input[type="checkbox"] {
  margin-top: 0;
}
.input-group .form-control:first-child,
.input-group-addon:first-child,
.input-group-btn:first-child > .btn,
.input-group-btn:first-child > .btn-group > .btn,
.input-group-btn:first-child > .dropdown-toggle,
.input-group-btn:last-child > .btn:not(:last-child):not(.dropdown-toggle),
.input-group-btn:last-child > .btn-group:not(:last-child) > .btn {
  border-bottom-right-radius: 0;
  border-top-right-radius: 0;
}
.input-group-addon:first-child {
  border-right: 0;
}
.input-group .form-control:last-child,
.input-group-addon:last-child,
.input-group-btn:last-child > .btn,
.input-group-btn:last-child > .btn-group > .btn,
.input-group-btn:last-child > .dropdown-toggle,
.input-group-btn:first-child > .btn:not(:first-child),
.input-group-btn:first-child > .btn-group:not(:first-child) > .btn {
  border-bottom-left-radius: 0;
  border-top-left-radius: 0;
}
.input-group-addon:last-child {
  border-left: 0;
}
.input-group-btn {
  position: relative;
  font-size: 0;
  white-space: nowrap;
}
.input-group-btn > .btn {
  position: relative;
}
.input-group-btn > .btn + .btn {
  margin-left: -1px;
}
.input-group-btn > .btn:hover,
.input-group-btn > .btn:focus,
.input-group-btn > .btn:active {
  z-index: 2;
}
.input-group-btn:first-child > .btn,
.input-group-btn:first-child > .btn-group {
  margin-right: -1px;
}
.input-group-btn:last-child > .btn,
.input-group-btn:last-child > .btn-group {
  z-index: 2;
  margin-left: -1px;
}
.nav {
  margin-bottom: 0;
  padding-left: 0;
  list-style: none;
}
.nav > li {
  position: relative;
  display: block;
}
.nav > li > a {
  position: relative;
  display: block;
  padding: 10px 15px;
}
.nav > li > a:hover,
.nav > li > a:focus {
  text-decoration: none;
  background-color: #eeeeee;
}
.nav > li.disabled > a {
  color: #777777;
}
.nav > li.disabled > a:hover,
.nav > li.disabled > a:focus {
  color: #777777;
  text-decoration: none;
  background-color: transparent;
  cursor: not-allowed;
}
.nav .open > a,
.nav .open > a:hover,
.nav .open > a:focus {
  background-color: #eeeeee;
  border-color: #337ab7;
}
.nav .nav-divider {
  height: 1px;
  margin: 8px 0;
  overflow: hidden;
  background-color: #e5e5e5;
}
.nav > li > a > img {
  max-width: none;
}
.nav-tabs {
  border-bottom: 1px solid #ddd;
}
.nav-tabs > li {
  float: left;
  margin-bottom: -1px;
}
.nav-tabs > li > a {
  margin-right: 2px;
  line-height: 1.42857143;
  border: 1px solid transparent;
  border-radius: 2px 2px 0 0;
}
.nav-tabs > li > a:hover {
  border-color: #eeeeee #eeeeee #ddd;
}
.nav-tabs > li.active > a,
.nav-tabs > li.active > a:hover,
.nav-tabs > li.active > a:focus {
  color: #555555;
  background-color: #fff;
  border: 1px solid #ddd;
  border-bottom-color: transparent;
  cursor: default;
}
.nav-tabs.nav-justified {
  width: 100%;
  border-bottom: 0;
}
.nav-tabs.nav-justified > li {
  float: none;
}
.nav-tabs.nav-justified > li > a {
  text-align: center;
  margin-bottom: 5px;
}
.nav-tabs.nav-justified > .dropdown .dropdown-menu {
  top: auto;
  left: auto;
}
@media (min-width: 768px) {
  .nav-tabs.nav-justified > li {
    display: table-cell;
    width: 1%;
  }
  .nav-tabs.nav-justified > li > a {
    margin-bottom: 0;
  }
}
.nav-tabs.nav-justified > li > a {
  margin-right: 0;
  border-radius: 2px;
}
.nav-tabs.nav-justified > .active > a,
.nav-tabs.nav-justified > .active > a:hover,
.nav-tabs.nav-justified > .active > a:focus {
  border: 1px solid #ddd;
}
@media (min-width: 768px) {
  .nav-tabs.nav-justified > li > a {
    border-bottom: 1px solid #ddd;
    border-radius: 2px 2px 0 0;
  }
  .nav-tabs.nav-justified > .active > a,
  .nav-tabs.nav-justified > .active > a:hover,
  .nav-tabs.nav-justified > .active > a:focus {
    border-bottom-color: #fff;
  }
}
.nav-pills > li {
  float: left;
}
.nav-pills > li > a {
  border-radius: 2px;
}
.nav-pills > li + li {
  margin-left: 2px;
}
.nav-pills > li.active > a,
.nav-pills > li.active > a:hover,
.nav-pills > li.active > a:focus {
  color: #fff;
  background-color: #337ab7;
}
.nav-stacked > li {
  float: none;
}
.nav-stacked > li + li {
  margin-top: 2px;
  margin-left: 0;
}
.nav-justified {
  width: 100%;
}
.nav-justified > li {
  float: none;
}
.nav-justified > li > a {
  text-align: center;
  margin-bottom: 5px;
}
.nav-justified > .dropdown .dropdown-menu {
  top: auto;
  left: auto;
}
@media (min-width: 768px) {
  .nav-justified > li {
    display: table-cell;
    width: 1%;
  }
  .nav-justified > li > a {
    margin-bottom: 0;
  }
}
.nav-tabs-justified {
  border-bottom: 0;
}
.nav-tabs-justified > li > a {
  margin-right: 0;
  border-radius: 2px;
}
.nav-tabs-justified > .active > a,
.nav-tabs-justified > .active > a:hover,
.nav-tabs-justified > .active > a:focus {
  border: 1px solid #ddd;
}
@media (min-width: 768px) {
  .nav-tabs-justified > li > a {
    border-bottom: 1px solid #ddd;
    border-radius: 2px 2px 0 0;
  }
  .nav-tabs-justified > .active > a,
  .nav-tabs-justified > .active > a:hover,
  .nav-tabs-justified > .active > a:focus {
    border-bottom-color: #fff;
  }
}
.tab-content > .tab-pane {
  display: none;
}
.tab-content > .active {
  display: block;
}
.nav-tabs .dropdown-menu {
  margin-top: -1px;
  border-top-right-radius: 0;
  border-top-left-radius: 0;
}
.navbar {
  position: relative;
  min-height: 30px;
  margin-bottom: 18px;
  border: 1px solid transparent;
}
@media (min-width: 541px) {
  .navbar {
    border-radius: 2px;
  }
}
@media (min-width: 541px) {
  .navbar-header {
    float: left;
  }
}
.navbar-collapse {
  overflow-x: visible;
  padding-right: 0px;
  padding-left: 0px;
  border-top: 1px solid transparent;
  box-shadow: inset 0 1px 0 rgba(255, 255, 255, 0.1);
  -webkit-overflow-scrolling: touch;
}
.navbar-collapse.in {
  overflow-y: auto;
}
@media (min-width: 541px) {
  .navbar-collapse {
    width: auto;
    border-top: 0;
    box-shadow: none;
  }
  .navbar-collapse.collapse {
    display: block !important;
    height: auto !important;
    padding-bottom: 0;
    overflow: visible !important;
  }
  .navbar-collapse.in {
    overflow-y: visible;
  }
  .navbar-fixed-top .navbar-collapse,
  .navbar-static-top .navbar-collapse,
  .navbar-fixed-bottom .navbar-collapse {
    padding-left: 0;
    padding-right: 0;
  }
}
.navbar-fixed-top .navbar-collapse,
.navbar-fixed-bottom .navbar-collapse {
  max-height: 340px;
}
@media (max-device-width: 540px) and (orientation: landscape) {
  .navbar-fixed-top .navbar-collapse,
  .navbar-fixed-bottom .navbar-collapse {
    max-height: 200px;
  }
}
.container > .navbar-header,
.container-fluid > .navbar-header,
.container > .navbar-collapse,
.container-fluid > .navbar-collapse {
  margin-right: 0px;
  margin-left: 0px;
}
@media (min-width: 541px) {
  .container > .navbar-header,
  .container-fluid > .navbar-header,
  .container > .navbar-collapse,
  .container-fluid > .navbar-collapse {
    margin-right: 0;
    margin-left: 0;
  }
}
.navbar-static-top {
  z-index: 1000;
  border-width: 0 0 1px;
}
@media (min-width: 541px) {
  .navbar-static-top {
    border-radius: 0;
  }
}
.navbar-fixed-top,
.navbar-fixed-bottom {
  position: fixed;
  right: 0;
  left: 0;
  z-index: 1030;
}
@media (min-width: 541px) {
  .navbar-fixed-top,
  .navbar-fixed-bottom {
    border-radius: 0;
  }
}
.navbar-fixed-top {
  top: 0;
  border-width: 0 0 1px;
}
.navbar-fixed-bottom {
  bottom: 0;
  margin-bottom: 0;
  border-width: 1px 0 0;
}
.navbar-brand {
  float: left;
  padding: 6px 0px;
  font-size: 17px;
  line-height: 18px;
  height: 30px;
}
.navbar-brand:hover,
.navbar-brand:focus {
  text-decoration: none;
}
.navbar-brand > img {
  display: block;
}
@media (min-width: 541px) {
  .navbar > .container .navbar-brand,
  .navbar > .container-fluid .navbar-brand {
    margin-left: 0px;
  }
}
.navbar-toggle {
  position: relative;
  float: right;
  margin-right: 0px;
  padding: 9px 10px;
  margin-top: -2px;
  margin-bottom: -2px;
  background-color: transparent;
  background-image: none;
  border: 1px solid transparent;
  border-radius: 2px;
}
.navbar-toggle:focus {
  outline: 0;
}
.navbar-toggle .icon-bar {
  display: block;
  width: 22px;
  height: 2px;
  border-radius: 1px;
}
.navbar-toggle .icon-bar + .icon-bar {
  margin-top: 4px;
}
@media (min-width: 541px) {
  .navbar-toggle {
    display: none;
  }
}
.navbar-nav {
  margin: 3px 0px;
}
.navbar-nav > li > a {
  padding-top: 10px;
  padding-bottom: 10px;
  line-height: 18px;
}
@media (max-width: 540px) {
  .navbar-nav .open .dropdown-menu {
    position: static;
    float: none;
    width: auto;
    margin-top: 0;
    background-color: transparent;
    border: 0;
    box-shadow: none;
  }
  .navbar-nav .open .dropdown-menu > li > a,
  .navbar-nav .open .dropdown-menu .dropdown-header {
    padding: 5px 15px 5px 25px;
  }
  .navbar-nav .open .dropdown-menu > li > a {
    line-height: 18px;
  }
  .navbar-nav .open .dropdown-menu > li > a:hover,
  .navbar-nav .open .dropdown-menu > li > a:focus {
    background-image: none;
  }
}
@media (min-width: 541px) {
  .navbar-nav {
    float: left;
    margin: 0;
  }
  .navbar-nav > li {
    float: left;
  }
  .navbar-nav > li > a {
    padding-top: 6px;
    padding-bottom: 6px;
  }
}
.navbar-form {
  margin-left: 0px;
  margin-right: 0px;
  padding: 10px 0px;
  border-top: 1px solid transparent;
  border-bottom: 1px solid transparent;
  -webkit-box-shadow: inset 0 1px 0 rgba(255, 255, 255, 0.1), 0 1px 0 rgba(255, 255, 255, 0.1);
  box-shadow: inset 0 1px 0 rgba(255, 255, 255, 0.1), 0 1px 0 rgba(255, 255, 255, 0.1);
  margin-top: -1px;
  margin-bottom: -1px;
}
@media (min-width: 768px) {
  .navbar-form .form-group {
    display: inline-block;
    margin-bottom: 0;
    vertical-align: middle;
  }
  .navbar-form .form-control {
    display: inline-block;
    width: auto;
    vertical-align: middle;
  }
  .navbar-form .form-control-static {
    display: inline-block;
  }
  .navbar-form .input-group {
    display: inline-table;
    vertical-align: middle;
  }
  .navbar-form .input-group .input-group-addon,
  .navbar-form .input-group .input-group-btn,
  .navbar-form .input-group .form-control {
    width: auto;
  }
  .navbar-form .input-group > .form-control {
    width: 100%;
  }
  .navbar-form .control-label {
    margin-bottom: 0;
    vertical-align: middle;
  }
  .navbar-form .radio,
  .navbar-form .checkbox {
    display: inline-block;
    margin-top: 0;
    margin-bottom: 0;
    vertical-align: middle;
  }
  .navbar-form .radio label,
  .navbar-form .checkbox label {
    padding-left: 0;
  }
  .navbar-form .radio input[type="radio"],
  .navbar-form .checkbox input[type="checkbox"] {
    position: relative;
    margin-left: 0;
  }
  .navbar-form .has-feedback .form-control-feedback {
    top: 0;
  }
}
@media (max-width: 540px) {
  .navbar-form .form-group {
    margin-bottom: 5px;
  }
  .navbar-form .form-group:last-child {
    margin-bottom: 0;
  }
}
@media (min-width: 541px) {
  .navbar-form {
    width: auto;
    border: 0;
    margin-left: 0;
    margin-right: 0;
    padding-top: 0;
    padding-bottom: 0;
    -webkit-box-shadow: none;
    box-shadow: none;
  }
}
.navbar-nav > li > .dropdown-menu {
  margin-top: 0;
  border-top-right-radius: 0;
  border-top-left-radius: 0;
}
.navbar-fixed-bottom .navbar-nav > li > .dropdown-menu {
  margin-bottom: 0;
  border-top-right-radius: 2px;
  border-top-left-radius: 2px;
  border-bottom-right-radius: 0;
  border-bottom-left-radius: 0;
}
.navbar-btn {
  margin-top: -1px;
  margin-bottom: -1px;
}
.navbar-btn.btn-sm {
  margin-top: 0px;
  margin-bottom: 0px;
}
.navbar-btn.btn-xs {
  margin-top: 4px;
  margin-bottom: 4px;
}
.navbar-text {
  margin-top: 6px;
  margin-bottom: 6px;
}
@media (min-width: 541px) {
  .navbar-text {
    float: left;
    margin-left: 0px;
    margin-right: 0px;
  }
}
@media (min-width: 541px) {
  .navbar-left {
    float: left !important;
    float: left;
  }
  .navbar-right {
    float: right !important;
    float: right;
    margin-right: 0px;
  }
  .navbar-right ~ .navbar-right {
    margin-right: 0;
  }
}
.navbar-default {
  background-color: #f8f8f8;
  border-color: #e7e7e7;
}
.navbar-default .navbar-brand {
  color: #777;
}
.navbar-default .navbar-brand:hover,
.navbar-default .navbar-brand:focus {
  color: #5e5e5e;
  background-color: transparent;
}
.navbar-default .navbar-text {
  color: #777;
}
.navbar-default .navbar-nav > li > a {
  color: #777;
}
.navbar-default .navbar-nav > li > a:hover,
.navbar-default .navbar-nav > li > a:focus {
  color: #333;
  background-color: transparent;
}
.navbar-default .navbar-nav > .active > a,
.navbar-default .navbar-nav > .active > a:hover,
.navbar-default .navbar-nav > .active > a:focus {
  color: #555;
  background-color: #e7e7e7;
}
.navbar-default .navbar-nav > .disabled > a,
.navbar-default .navbar-nav > .disabled > a:hover,
.navbar-default .navbar-nav > .disabled > a:focus {
  color: #ccc;
  background-color: transparent;
}
.navbar-default .navbar-toggle {
  border-color: #ddd;
}
.navbar-default .navbar-toggle:hover,
.navbar-default .navbar-toggle:focus {
  background-color: #ddd;
}
.navbar-default .navbar-toggle .icon-bar {
  background-color: #888;
}
.navbar-default .navbar-collapse,
.navbar-default .navbar-form {
  border-color: #e7e7e7;
}
.navbar-default .navbar-nav > .open > a,
.navbar-default .navbar-nav > .open > a:hover,
.navbar-default .navbar-nav > .open > a:focus {
  background-color: #e7e7e7;
  color: #555;
}
@media (max-width: 540px) {
  .navbar-default .navbar-nav .open .dropdown-menu > li > a {
    color: #777;
  }
  .navbar-default .navbar-nav .open .dropdown-menu > li > a:hover,
  .navbar-default .navbar-nav .open .dropdown-menu > li > a:focus {
    color: #333;
    background-color: transparent;
  }
  .navbar-default .navbar-nav .open .dropdown-menu > .active > a,
  .navbar-default .navbar-nav .open .dropdown-menu > .active > a:hover,
  .navbar-default .navbar-nav .open .dropdown-menu > .active > a:focus {
    color: #555;
    background-color: #e7e7e7;
  }
  .navbar-default .navbar-nav .open .dropdown-menu > .disabled > a,
  .navbar-default .navbar-nav .open .dropdown-menu > .disabled > a:hover,
  .navbar-default .navbar-nav .open .dropdown-menu > .disabled > a:focus {
    color: #ccc;
    background-color: transparent;
  }
}
.navbar-default .navbar-link {
  color: #777;
}
.navbar-default .navbar-link:hover {
  color: #333;
}
.navbar-default .btn-link {
  color: #777;
}
.navbar-default .btn-link:hover,
.navbar-default .btn-link:focus {
  color: #333;
}
.navbar-default .btn-link[disabled]:hover,
fieldset[disabled] .navbar-default .btn-link:hover,
.navbar-default .btn-link[disabled]:focus,
fieldset[disabled] .navbar-default .btn-link:focus {
  color: #ccc;
}
.navbar-inverse {
  background-color: #222;
  border-color: #080808;
}
.navbar-inverse .navbar-brand {
  color: #9d9d9d;
}
.navbar-inverse .navbar-brand:hover,
.navbar-inverse .navbar-brand:focus {
  color: #fff;
  background-color: transparent;
}
.navbar-inverse .navbar-text {
  color: #9d9d9d;
}
.navbar-inverse .navbar-nav > li > a {
  color: #9d9d9d;
}
.navbar-inverse .navbar-nav > li > a:hover,
.navbar-inverse .navbar-nav > li > a:focus {
  color: #fff;
  background-color: transparent;
}
.navbar-inverse .navbar-nav > .active > a,
.navbar-inverse .navbar-nav > .active > a:hover,
.navbar-inverse .navbar-nav > .active > a:focus {
  color: #fff;
  background-color: #080808;
}
.navbar-inverse .navbar-nav > .disabled > a,
.navbar-inverse .navbar-nav > .disabled > a:hover,
.navbar-inverse .navbar-nav > .disabled > a:focus {
  color: #444;
  background-color: transparent;
}
.navbar-inverse .navbar-toggle {
  border-color: #333;
}
.navbar-inverse .navbar-toggle:hover,
.navbar-inverse .navbar-toggle:focus {
  background-color: #333;
}
.navbar-inverse .navbar-toggle .icon-bar {
  background-color: #fff;
}
.navbar-inverse .navbar-collapse,
.navbar-inverse .navbar-form {
  border-color: #101010;
}
.navbar-inverse .navbar-nav > .open > a,
.navbar-inverse .navbar-nav > .open > a:hover,
.navbar-inverse .navbar-nav > .open > a:focus {
  background-color: #080808;
  color: #fff;
}
@media (max-width: 540px) {
  .navbar-inverse .navbar-nav .open .dropdown-menu > .dropdown-header {
    border-color: #080808;
  }
  .navbar-inverse .navbar-nav .open .dropdown-menu .divider {
    background-color: #080808;
  }
  .navbar-inverse .navbar-nav .open .dropdown-menu > li > a {
    color: #9d9d9d;
  }
  .navbar-inverse .navbar-nav .open .dropdown-menu > li > a:hover,
  .navbar-inverse .navbar-nav .open .dropdown-menu > li > a:focus {
    color: #fff;
    background-color: transparent;
  }
  .navbar-inverse .navbar-nav .open .dropdown-menu > .active > a,
  .navbar-inverse .navbar-nav .open .dropdown-menu > .active > a:hover,
  .navbar-inverse .navbar-nav .open .dropdown-menu > .active > a:focus {
    color: #fff;
    background-color: #080808;
  }
  .navbar-inverse .navbar-nav .open .dropdown-menu > .disabled > a,
  .navbar-inverse .navbar-nav .open .dropdown-menu > .disabled > a:hover,
  .navbar-inverse .navbar-nav .open .dropdown-menu > .disabled > a:focus {
    color: #444;
    background-color: transparent;
  }
}
.navbar-inverse .navbar-link {
  color: #9d9d9d;
}
.navbar-inverse .navbar-link:hover {
  color: #fff;
}
.navbar-inverse .btn-link {
  color: #9d9d9d;
}
.navbar-inverse .btn-link:hover,
.navbar-inverse .btn-link:focus {
  color: #fff;
}
.navbar-inverse .btn-link[disabled]:hover,
fieldset[disabled] .navbar-inverse .btn-link:hover,
.navbar-inverse .btn-link[disabled]:focus,
fieldset[disabled] .navbar-inverse .btn-link:focus {
  color: #444;
}
.breadcrumb {
  padding: 8px 15px;
  margin-bottom: 18px;
  list-style: none;
  background-color: #f5f5f5;
  border-radius: 2px;
}
.breadcrumb > li {
  display: inline-block;
}
.breadcrumb > li + li:before {
  content: "/\00a0";
  padding: 0 5px;
  color: #5e5e5e;
}
.breadcrumb > .active {
  color: #777777;
}
.pagination {
  display: inline-block;
  padding-left: 0;
  margin: 18px 0;
  border-radius: 2px;
}
.pagination > li {
  display: inline;
}
.pagination > li > a,
.pagination > li > span {
  position: relative;
  float: left;
  padding: 6px 12px;
  line-height: 1.42857143;
  text-decoration: none;
  color: #337ab7;
  background-color: #fff;
  border: 1px solid #ddd;
  margin-left: -1px;
}
.pagination > li:first-child > a,
.pagination > li:first-child > span {
  margin-left: 0;
  border-bottom-left-radius: 2px;
  border-top-left-radius: 2px;
}
.pagination > li:last-child > a,
.pagination > li:last-child > span {
  border-bottom-right-radius: 2px;
  border-top-right-radius: 2px;
}
.pagination > li > a:hover,
.pagination > li > span:hover,
.pagination > li > a:focus,
.pagination > li > span:focus {
  z-index: 2;
  color: #23527c;
  background-color: #eeeeee;
  border-color: #ddd;
}
.pagination > .active > a,
.pagination > .active > span,
.pagination > .active > a:hover,
.pagination > .active > span:hover,
.pagination > .active > a:focus,
.pagination > .active > span:focus {
  z-index: 3;
  color: #fff;
  background-color: #337ab7;
  border-color: #337ab7;
  cursor: default;
}
.pagination > .disabled > span,
.pagination > .disabled > span:hover,
.pagination > .disabled > span:focus,
.pagination > .disabled > a,
.pagination > .disabled > a:hover,
.pagination > .disabled > a:focus {
  color: #777777;
  background-color: #fff;
  border-color: #ddd;
  cursor: not-allowed;
}
.pagination-lg > li > a,
.pagination-lg > li > span {
  padding: 10px 16px;
  font-size: 17px;
  line-height: 1.3333333;
}
.pagination-lg > li:first-child > a,
.pagination-lg > li:first-child > span {
  border-bottom-left-radius: 3px;
  border-top-left-radius: 3px;
}
.pagination-lg > li:last-child > a,
.pagination-lg > li:last-child > span {
  border-bottom-right-radius: 3px;
  border-top-right-radius: 3px;
}
.pagination-sm > li > a,
.pagination-sm > li > span {
  padding: 5px 10px;
  font-size: 12px;
  line-height: 1.5;
}
.pagination-sm > li:first-child > a,
.pagination-sm > li:first-child > span {
  border-bottom-left-radius: 1px;
  border-top-left-radius: 1px;
}
.pagination-sm > li:last-child > a,
.pagination-sm > li:last-child > span {
  border-bottom-right-radius: 1px;
  border-top-right-radius: 1px;
}
.pager {
  padding-left: 0;
  margin: 18px 0;
  list-style: none;
  text-align: center;
}
.pager li {
  display: inline;
}
.pager li > a,
.pager li > span {
  display: inline-block;
  padding: 5px 14px;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 15px;
}
.pager li > a:hover,
.pager li > a:focus {
  text-decoration: none;
  background-color: #eeeeee;
}
.pager .next > a,
.pager .next > span {
  float: right;
}
.pager .previous > a,
.pager .previous > span {
  float: left;
}
.pager .disabled > a,
.pager .disabled > a:hover,
.pager .disabled > a:focus,
.pager .disabled > span {
  color: #777777;
  background-color: #fff;
  cursor: not-allowed;
}
.label {
  display: inline;
  padding: .2em .6em .3em;
  font-size: 75%;
  font-weight: bold;
  line-height: 1;
  color: #fff;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: .25em;
}
a.label:hover,
a.label:focus {
  color: #fff;
  text-decoration: none;
  cursor: pointer;
}
.label:empty {
  display: none;
}
.btn .label {
  position: relative;
  top: -1px;
}
.label-default {
  background-color: #777777;
}
.label-default[href]:hover,
.label-default[href]:focus {
  background-color: #5e5e5e;
}
.label-primary {
  background-color: #337ab7;
}
.label-primary[href]:hover,
.label-primary[href]:focus {
  background-color: #286090;
}
.label-success {
  background-color: #5cb85c;
}
.label-success[href]:hover,
.label-success[href]:focus {
  background-color: #449d44;
}
.label-info {
  background-color: #5bc0de;
}
.label-info[href]:hover,
.label-info[href]:focus {
  background-color: #31b0d5;
}
.label-warning {
  background-color: #f0ad4e;
}
.label-warning[href]:hover,
.label-warning[href]:focus {
  background-color: #ec971f;
}
.label-danger {
  background-color: #d9534f;
}
.label-danger[href]:hover,
.label-danger[href]:focus {
  background-color: #c9302c;
}
.badge {
  display: inline-block;
  min-width: 10px;
  padding: 3px 7px;
  font-size: 12px;
  font-weight: bold;
  color: #fff;
  line-height: 1;
  vertical-align: middle;
  white-space: nowrap;
  text-align: center;
  background-color: #777777;
  border-radius: 10px;
}
.badge:empty {
  display: none;
}
.btn .badge {
  position: relative;
  top: -1px;
}
.btn-xs .badge,
.btn-group-xs > .btn .badge {
  top: 0;
  padding: 1px 5px;
}
a.badge:hover,
a.badge:focus {
  color: #fff;
  text-decoration: none;
  cursor: pointer;
}
.list-group-item.active > .badge,
.nav-pills > .active > a > .badge {
  color: #337ab7;
  background-color: #fff;
}
.list-group-item > .badge {
  float: right;
}
.list-group-item > .badge + .badge {
  margin-right: 5px;
}
.nav-pills > li > a > .badge {
  margin-left: 3px;
}
.jumbotron {
  padding-top: 30px;
  padding-bottom: 30px;
  margin-bottom: 30px;
  color: inherit;
  background-color: #eeeeee;
}
.jumbotron h1,
.jumbotron .h1 {
  color: inherit;
}
.jumbotron p {
  margin-bottom: 15px;
  font-size: 20px;
  font-weight: 200;
}
.jumbotron > hr {
  border-top-color: #d5d5d5;
}
.container .jumbotron,
.container-fluid .jumbotron {
  border-radius: 3px;
  padding-left: 0px;
  padding-right: 0px;
}
.jumbotron .container {
  max-width: 100%;
}
@media screen and (min-width: 768px) {
  .jumbotron {
    padding-top: 48px;
    padding-bottom: 48px;
  }
  .container .jumbotron,
  .container-fluid .jumbotron {
    padding-left: 60px;
    padding-right: 60px;
  }
  .jumbotron h1,
  .jumbotron .h1 {
    font-size: 59px;
  }
}
.thumbnail {
  display: block;
  padding: 4px;
  margin-bottom: 18px;
  line-height: 1.42857143;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 2px;
  -webkit-transition: border 0.2s ease-in-out;
  -o-transition: border 0.2s ease-in-out;
  transition: border 0.2s ease-in-out;
}
.thumbnail > img,
.thumbnail a > img {
  margin-left: auto;
  margin-right: auto;
}
a.thumbnail:hover,
a.thumbnail:focus,
a.thumbnail.active {
  border-color: #337ab7;
}
.thumbnail .caption {
  padding: 9px;
  color: #000;
}
.alert {
  padding: 15px;
  margin-bottom: 18px;
  border: 1px solid transparent;
  border-radius: 2px;
}
.alert h4 {
  margin-top: 0;
  color: inherit;
}
.alert .alert-link {
  font-weight: bold;
}
.alert > p,
.alert > ul {
  margin-bottom: 0;
}
.alert > p + p {
  margin-top: 5px;
}
.alert-dismissable,
.alert-dismissible {
  padding-right: 35px;
}
.alert-dismissable .close,
.alert-dismissible .close {
  position: relative;
  top: -2px;
  right: -21px;
  color: inherit;
}
.alert-success {
  background-color: #dff0d8;
  border-color: #d6e9c6;
  color: #3c763d;
}
.alert-success hr {
  border-top-color: #c9e2b3;
}
.alert-success .alert-link {
  color: #2b542c;
}
.alert-info {
  background-color: #d9edf7;
  border-color: #bce8f1;
  color: #31708f;
}
.alert-info hr {
  border-top-color: #a6e1ec;
}
.alert-info .alert-link {
  color: #245269;
}
.alert-warning {
  background-color: #fcf8e3;
  border-color: #faebcc;
  color: #8a6d3b;
}
.alert-warning hr {
  border-top-color: #f7e1b5;
}
.alert-warning .alert-link {
  color: #66512c;
}
.alert-danger {
  background-color: #f2dede;
  border-color: #ebccd1;
  color: #a94442;
}
.alert-danger hr {
  border-top-color: #e4b9c0;
}
.alert-danger .alert-link {
  color: #843534;
}
@-webkit-keyframes progress-bar-stripes {
  from {
    background-position: 40px 0;
  }
  to {
    background-position: 0 0;
  }
}
@keyframes progress-bar-stripes {
  from {
    background-position: 40px 0;
  }
  to {
    background-position: 0 0;
  }
}
.progress {
  overflow: hidden;
  height: 18px;
  margin-bottom: 18px;
  background-color: #f5f5f5;
  border-radius: 2px;
  -webkit-box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.1);
  box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.1);
}
.progress-bar {
  float: left;
  width: 0%;
  height: 100%;
  font-size: 12px;
  line-height: 18px;
  color: #fff;
  text-align: center;
  background-color: #337ab7;
  -webkit-box-shadow: inset 0 -1px 0 rgba(0, 0, 0, 0.15);
  box-shadow: inset 0 -1px 0 rgba(0, 0, 0, 0.15);
  -webkit-transition: width 0.6s ease;
  -o-transition: width 0.6s ease;
  transition: width 0.6s ease;
}
.progress-striped .progress-bar,
.progress-bar-striped {
  background-image: -webkit-linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
  background-image: -o-linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
  background-image: linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
  background-size: 40px 40px;
}
.progress.active .progress-bar,
.progress-bar.active {
  -webkit-animation: progress-bar-stripes 2s linear infinite;
  -o-animation: progress-bar-stripes 2s linear infinite;
  animation: progress-bar-stripes 2s linear infinite;
}
.progress-bar-success {
  background-color: #5cb85c;
}
.progress-striped .progress-bar-success {
  background-image: -webkit-linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
  background-image: -o-linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
  background-image: linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
}
.progress-bar-info {
  background-color: #5bc0de;
}
.progress-striped .progress-bar-info {
  background-image: -webkit-linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
  background-image: -o-linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
  background-image: linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
}
.progress-bar-warning {
  background-color: #f0ad4e;
}
.progress-striped .progress-bar-warning {
  background-image: -webkit-linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
  background-image: -o-linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
  background-image: linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
}
.progress-bar-danger {
  background-color: #d9534f;
}
.progress-striped .progress-bar-danger {
  background-image: -webkit-linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
  background-image: -o-linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
  background-image: linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
}
.media {
  margin-top: 15px;
}
.media:first-child {
  margin-top: 0;
}
.media,
.media-body {
  zoom: 1;
  overflow: hidden;
}
.media-body {
  width: 10000px;
}
.media-object {
  display: block;
}
.media-object.img-thumbnail {
  max-width: none;
}
.media-right,
.media > .pull-right {
  padding-left: 10px;
}
.media-left,
.media > .pull-left {
  padding-right: 10px;
}
.media-left,
.media-right,
.media-body {
  display: table-cell;
  vertical-align: top;
}
.media-middle {
  vertical-align: middle;
}
.media-bottom {
  vertical-align: bottom;
}
.media-heading {
  margin-top: 0;
  margin-bottom: 5px;
}
.media-list {
  padding-left: 0;
  list-style: none;
}
.list-group {
  margin-bottom: 20px;
  padding-left: 0;
}
.list-group-item {
  position: relative;
  display: block;
  padding: 10px 15px;
  margin-bottom: -1px;
  background-color: #fff;
  border: 1px solid #ddd;
}
.list-group-item:first-child {
  border-top-right-radius: 2px;
  border-top-left-radius: 2px;
}
.list-group-item:last-child {
  margin-bottom: 0;
  border-bottom-right-radius: 2px;
  border-bottom-left-radius: 2px;
}
a.list-group-item,
button.list-group-item {
  color: #555;
}
a.list-group-item .list-group-item-heading,
button.list-group-item .list-group-item-heading {
  color: #333;
}
a.list-group-item:hover,
button.list-group-item:hover,
a.list-group-item:focus,
button.list-group-item:focus {
  text-decoration: none;
  color: #555;
  background-color: #f5f5f5;
}
button.list-group-item {
  width: 100%;
  text-align: left;
}
.list-group-item.disabled,
.list-group-item.disabled:hover,
.list-group-item.disabled:focus {
  background-color: #eeeeee;
  color: #777777;
  cursor: not-allowed;
}
.list-group-item.disabled .list-group-item-heading,
.list-group-item.disabled:hover .list-group-item-heading,
.list-group-item.disabled:focus .list-group-item-heading {
  color: inherit;
}
.list-group-item.disabled .list-group-item-text,
.list-group-item.disabled:hover .list-group-item-text,
.list-group-item.disabled:focus .list-group-item-text {
  color: #777777;
}
.list-group-item.active,
.list-group-item.active:hover,
.list-group-item.active:focus {
  z-index: 2;
  color: #fff;
  background-color: #337ab7;
  border-color: #337ab7;
}
.list-group-item.active .list-group-item-heading,
.list-group-item.active:hover .list-group-item-heading,
.list-group-item.active:focus .list-group-item-heading,
.list-group-item.active .list-group-item-heading > small,
.list-group-item.active:hover .list-group-item-heading > small,
.list-group-item.active:focus .list-group-item-heading > small,
.list-group-item.active .list-group-item-heading > .small,
.list-group-item.active:hover .list-group-item-heading > .small,
.list-group-item.active:focus .list-group-item-heading > .small {
  color: inherit;
}
.list-group-item.active .list-group-item-text,
.list-group-item.active:hover .list-group-item-text,
.list-group-item.active:focus .list-group-item-text {
  color: #c7ddef;
}
.list-group-item-success {
  color: #3c763d;
  background-color: #dff0d8;
}
a.list-group-item-success,
button.list-group-item-success {
  color: #3c763d;
}
a.list-group-item-success .list-group-item-heading,
button.list-group-item-success .list-group-item-heading {
  color: inherit;
}
a.list-group-item-success:hover,
button.list-group-item-success:hover,
a.list-group-item-success:focus,
button.list-group-item-success:focus {
  color: #3c763d;
  background-color: #d0e9c6;
}
a.list-group-item-success.active,
button.list-group-item-success.active,
a.list-group-item-success.active:hover,
button.list-group-item-success.active:hover,
a.list-group-item-success.active:focus,
button.list-group-item-success.active:focus {
  color: #fff;
  background-color: #3c763d;
  border-color: #3c763d;
}
.list-group-item-info {
  color: #31708f;
  background-color: #d9edf7;
}
a.list-group-item-info,
button.list-group-item-info {
  color: #31708f;
}
a.list-group-item-info .list-group-item-heading,
button.list-group-item-info .list-group-item-heading {
  color: inherit;
}
a.list-group-item-info:hover,
button.list-group-item-info:hover,
a.list-group-item-info:focus,
button.list-group-item-info:focus {
  color: #31708f;
  background-color: #c4e3f3;
}
a.list-group-item-info.active,
button.list-group-item-info.active,
a.list-group-item-info.active:hover,
button.list-group-item-info.active:hover,
a.list-group-item-info.active:focus,
button.list-group-item-info.active:focus {
  color: #fff;
  background-color: #31708f;
  border-color: #31708f;
}
.list-group-item-warning {
  color: #8a6d3b;
  background-color: #fcf8e3;
}
a.list-group-item-warning,
button.list-group-item-warning {
  color: #8a6d3b;
}
a.list-group-item-warning .list-group-item-heading,
button.list-group-item-warning .list-group-item-heading {
  color: inherit;
}
a.list-group-item-warning:hover,
button.list-group-item-warning:hover,
a.list-group-item-warning:focus,
button.list-group-item-warning:focus {
  color: #8a6d3b;
  background-color: #faf2cc;
}
a.list-group-item-warning.active,
button.list-group-item-warning.active,
a.list-group-item-warning.active:hover,
button.list-group-item-warning.active:hover,
a.list-group-item-warning.active:focus,
button.list-group-item-warning.active:focus {
  color: #fff;
  background-color: #8a6d3b;
  border-color: #8a6d3b;
}
.list-group-item-danger {
  color: #a94442;
  background-color: #f2dede;
}
a.list-group-item-danger,
button.list-group-item-danger {
  color: #a94442;
}
a.list-group-item-danger .list-group-item-heading,
button.list-group-item-danger .list-group-item-heading {
  color: inherit;
}
a.list-group-item-danger:hover,
button.list-group-item-danger:hover,
a.list-group-item-danger:focus,
button.list-group-item-danger:focus {
  color: #a94442;
  background-color: #ebcccc;
}
a.list-group-item-danger.active,
button.list-group-item-danger.active,
a.list-group-item-danger.active:hover,
button.list-group-item-danger.active:hover,
a.list-group-item-danger.active:focus,
button.list-group-item-danger.active:focus {
  color: #fff;
  background-color: #a94442;
  border-color: #a94442;
}
.list-group-item-heading {
  margin-top: 0;
  margin-bottom: 5px;
}
.list-group-item-text {
  margin-bottom: 0;
  line-height: 1.3;
}
.panel {
  margin-bottom: 18px;
  background-color: #fff;
  border: 1px solid transparent;
  border-radius: 2px;
  -webkit-box-shadow: 0 1px 1px rgba(0, 0, 0, 0.05);
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.05);
}
.panel-body {
  padding: 15px;
}
.panel-heading {
  padding: 10px 15px;
  border-bottom: 1px solid transparent;
  border-top-right-radius: 1px;
  border-top-left-radius: 1px;
}
.panel-heading > .dropdown .dropdown-toggle {
  color: inherit;
}
.panel-title {
  margin-top: 0;
  margin-bottom: 0;
  font-size: 15px;
  color: inherit;
}
.panel-title > a,
.panel-title > small,
.panel-title > .small,
.panel-title > small > a,
.panel-title > .small > a {
  color: inherit;
}
.panel-footer {
  padding: 10px 15px;
  background-color: #f5f5f5;
  border-top: 1px solid #ddd;
  border-bottom-right-radius: 1px;
  border-bottom-left-radius: 1px;
}
.panel > .list-group,
.panel > .panel-collapse > .list-group {
  margin-bottom: 0;
}
.panel > .list-group .list-group-item,
.panel > .panel-collapse > .list-group .list-group-item {
  border-width: 1px 0;
  border-radius: 0;
}
.panel > .list-group:first-child .list-group-item:first-child,
.panel > .panel-collapse > .list-group:first-child .list-group-item:first-child {
  border-top: 0;
  border-top-right-radius: 1px;
  border-top-left-radius: 1px;
}
.panel > .list-group:last-child .list-group-item:last-child,
.panel > .panel-collapse > .list-group:last-child .list-group-item:last-child {
  border-bottom: 0;
  border-bottom-right-radius: 1px;
  border-bottom-left-radius: 1px;
}
.panel > .panel-heading + .panel-collapse > .list-group .list-group-item:first-child {
  border-top-right-radius: 0;
  border-top-left-radius: 0;
}
.panel-heading + .list-group .list-group-item:first-child {
  border-top-width: 0;
}
.list-group + .panel-footer {
  border-top-width: 0;
}
.panel > .table,
.panel > .table-responsive > .table,
.panel > .panel-collapse > .table {
  margin-bottom: 0;
}
.panel > .table caption,
.panel > .table-responsive > .table caption,
.panel > .panel-collapse > .table caption {
  padding-left: 15px;
  padding-right: 15px;
}
.panel > .table:first-child,
.panel > .table-responsive:first-child > .table:first-child {
  border-top-right-radius: 1px;
  border-top-left-radius: 1px;
}
.panel > .table:first-child > thead:first-child > tr:first-child,
.panel > .table-responsive:first-child > .table:first-child > thead:first-child > tr:first-child,
.panel > .table:first-child > tbody:first-child > tr:first-child,
.panel > .table-responsive:first-child > .table:first-child > tbody:first-child > tr:first-child {
  border-top-left-radius: 1px;
  border-top-right-radius: 1px;
}
.panel > .table:first-child > thead:first-child > tr:first-child td:first-child,
.panel > .table-responsive:first-child > .table:first-child > thead:first-child > tr:first-child td:first-child,
.panel > .table:first-child > tbody:first-child > tr:first-child td:first-child,
.panel > .table-responsive:first-child > .table:first-child > tbody:first-child > tr:first-child td:first-child,
.panel > .table:first-child > thead:first-child > tr:first-child th:first-child,
.panel > .table-responsive:first-child > .table:first-child > thead:first-child > tr:first-child th:first-child,
.panel > .table:first-child > tbody:first-child > tr:first-child th:first-child,
.panel > .table-responsive:first-child > .table:first-child > tbody:first-child > tr:first-child th:first-child {
  border-top-left-radius: 1px;
}
.panel > .table:first-child > thead:first-child > tr:first-child td:last-child,
.panel > .table-responsive:first-child > .table:first-child > thead:first-child > tr:first-child td:last-child,
.panel > .table:first-child > tbody:first-child > tr:first-child td:last-child,
.panel > .table-responsive:first-child > .table:first-child > tbody:first-child > tr:first-child td:last-child,
.panel > .table:first-child > thead:first-child > tr:first-child th:last-child,
.panel > .table-responsive:first-child > .table:first-child > thead:first-child > tr:first-child th:last-child,
.panel > .table:first-child > tbody:first-child > tr:first-child th:last-child,
.panel > .table-responsive:first-child > .table:first-child > tbody:first-child > tr:first-child th:last-child {
  border-top-right-radius: 1px;
}
.panel > .table:last-child,
.panel > .table-responsive:last-child > .table:last-child {
  border-bottom-right-radius: 1px;
  border-bottom-left-radius: 1px;
}
.panel > .table:last-child > tbody:last-child > tr:last-child,
.panel > .table-responsive:last-child > .table:last-child > tbody:last-child > tr:last-child,
.panel > .table:last-child > tfoot:last-child > tr:last-child,
.panel > .table-responsive:last-child > .table:last-child > tfoot:last-child > tr:last-child {
  border-bottom-left-radius: 1px;
  border-bottom-right-radius: 1px;
}
.panel > .table:last-child > tbody:last-child > tr:last-child td:first-child,
.panel > .table-responsive:last-child > .table:last-child > tbody:last-child > tr:last-child td:first-child,
.panel > .table:last-child > tfoot:last-child > tr:last-child td:first-child,
.panel > .table-responsive:last-child > .table:last-child > tfoot:last-child > tr:last-child td:first-child,
.panel > .table:last-child > tbody:last-child > tr:last-child th:first-child,
.panel > .table-responsive:last-child > .table:last-child > tbody:last-child > tr:last-child th:first-child,
.panel > .table:last-child > tfoot:last-child > tr:last-child th:first-child,
.panel > .table-responsive:last-child > .table:last-child > tfoot:last-child > tr:last-child th:first-child {
  border-bottom-left-radius: 1px;
}
.panel > .table:last-child > tbody:last-child > tr:last-child td:last-child,
.panel > .table-responsive:last-child > .table:last-child > tbody:last-child > tr:last-child td:last-child,
.panel > .table:last-child > tfoot:last-child > tr:last-child td:last-child,
.panel > .table-responsive:last-child > .table:last-child > tfoot:last-child > tr:last-child td:last-child,
.panel > .table:last-child > tbody:last-child > tr:last-child th:last-child,
.panel > .table-responsive:last-child > .table:last-child > tbody:last-child > tr:last-child th:last-child,
.panel > .table:last-child > tfoot:last-child > tr:last-child th:last-child,
.panel > .table-responsive:last-child > .table:last-child > tfoot:last-child > tr:last-child th:last-child {
  border-bottom-right-radius: 1px;
}
.panel > .panel-body + .table,
.panel > .panel-body + .table-responsive,
.panel > .table + .panel-body,
.panel > .table-responsive + .panel-body {
  border-top: 1px solid #ddd;
}
.panel > .table > tbody:first-child > tr:first-child th,
.panel > .table > tbody:first-child > tr:first-child td {
  border-top: 0;
}
.panel > .table-bordered,
.panel > .table-responsive > .table-bordered {
  border: 0;
}
.panel > .table-bordered > thead > tr > th:first-child,
.panel > .table-responsive > .table-bordered > thead > tr > th:first-child,
.panel > .table-bordered > tbody > tr > th:first-child,
.panel > .table-responsive > .table-bordered > tbody > tr > th:first-child,
.panel > .table-bordered > tfoot > tr > th:first-child,
.panel > .table-responsive > .table-bordered > tfoot > tr > th:first-child,
.panel > .table-bordered > thead > tr > td:first-child,
.panel > .table-responsive > .table-bordered > thead > tr > td:first-child,
.panel > .table-bordered > tbody > tr > td:first-child,
.panel > .table-responsive > .table-bordered > tbody > tr > td:first-child,
.panel > .table-bordered > tfoot > tr > td:first-child,
.panel > .table-responsive > .table-bordered > tfoot > tr > td:first-child {
  border-left: 0;
}
.panel > .table-bordered > thead > tr > th:last-child,
.panel > .table-responsive > .table-bordered > thead > tr > th:last-child,
.panel > .table-bordered > tbody > tr > th:last-child,
.panel > .table-responsive > .table-bordered > tbody > tr > th:last-child,
.panel > .table-bordered > tfoot > tr > th:last-child,
.panel > .table-responsive > .table-bordered > tfoot > tr > th:last-child,
.panel > .table-bordered > thead > tr > td:last-child,
.panel > .table-responsive > .table-bordered > thead > tr > td:last-child,
.panel > .table-bordered > tbody > tr > td:last-child,
.panel > .table-responsive > .table-bordered > tbody > tr > td:last-child,
.panel > .table-bordered > tfoot > tr > td:last-child,
.panel > .table-responsive > .table-bordered > tfoot > tr > td:last-child {
  border-right: 0;
}
.panel > .table-bordered > thead > tr:first-child > td,
.panel > .table-responsive > .table-bordered > thead > tr:first-child > td,
.panel > .table-bordered > tbody > tr:first-child > td,
.panel > .table-responsive > .table-bordered > tbody > tr:first-child > td,
.panel > .table-bordered > thead > tr:first-child > th,
.panel > .table-responsive > .table-bordered > thead > tr:first-child > th,
.panel > .table-bordered > tbody > tr:first-child > th,
.panel > .table-responsive > .table-bordered > tbody > tr:first-child > th {
  border-bottom: 0;
}
.panel > .table-bordered > tbody > tr:last-child > td,
.panel > .table-responsive > .table-bordered > tbody > tr:last-child > td,
.panel > .table-bordered > tfoot > tr:last-child > td,
.panel > .table-responsive > .table-bordered > tfoot > tr:last-child > td,
.panel > .table-bordered > tbody > tr:last-child > th,
.panel > .table-responsive > .table-bordered > tbody > tr:last-child > th,
.panel > .table-bordered > tfoot > tr:last-child > th,
.panel > .table-responsive > .table-bordered > tfoot > tr:last-child > th {
  border-bottom: 0;
}
.panel > .table-responsive {
  border: 0;
  margin-bottom: 0;
}
.panel-group {
  margin-bottom: 18px;
}
.panel-group .panel {
  margin-bottom: 0;
  border-radius: 2px;
}
.panel-group .panel + .panel {
  margin-top: 5px;
}
.panel-group .panel-heading {
  border-bottom: 0;
}
.panel-group .panel-heading + .panel-collapse > .panel-body,
.panel-group .panel-heading + .panel-collapse > .list-group {
  border-top: 1px solid #ddd;
}
.panel-group .panel-footer {
  border-top: 0;
}
.panel-group .panel-footer + .panel-collapse .panel-body {
  border-bottom: 1px solid #ddd;
}
.panel-default {
  border-color: #ddd;
}
.panel-default > .panel-heading {
  color: #333333;
  background-color: #f5f5f5;
  border-color: #ddd;
}
.panel-default > .panel-heading + .panel-collapse > .panel-body {
  border-top-color: #ddd;
}
.panel-default > .panel-heading .badge {
  color: #f5f5f5;
  background-color: #333333;
}
.panel-default > .panel-footer + .panel-collapse > .panel-body {
  border-bottom-color: #ddd;
}
.panel-primary {
  border-color: #337ab7;
}
.panel-primary > .panel-heading {
  color: #fff;
  background-color: #337ab7;
  border-color: #337ab7;
}
.panel-primary > .panel-heading + .panel-collapse > .panel-body {
  border-top-color: #337ab7;
}
.panel-primary > .panel-heading .badge {
  color: #337ab7;
  background-color: #fff;
}
.panel-primary > .panel-footer + .panel-collapse > .panel-body {
  border-bottom-color: #337ab7;
}
.panel-success {
  border-color: #d6e9c6;
}
.panel-success > .panel-heading {
  color: #3c763d;
  background-color: #dff0d8;
  border-color: #d6e9c6;
}
.panel-success > .panel-heading + .panel-collapse > .panel-body {
  border-top-color: #d6e9c6;
}
.panel-success > .panel-heading .badge {
  color: #dff0d8;
  background-color: #3c763d;
}
.panel-success > .panel-footer + .panel-collapse > .panel-body {
  border-bottom-color: #d6e9c6;
}
.panel-info {
  border-color: #bce8f1;
}
.panel-info > .panel-heading {
  color: #31708f;
  background-color: #d9edf7;
  border-color: #bce8f1;
}
.panel-info > .panel-heading + .panel-collapse > .panel-body {
  border-top-color: #bce8f1;
}
.panel-info > .panel-heading .badge {
  color: #d9edf7;
  background-color: #31708f;
}
.panel-info > .panel-footer + .panel-collapse > .panel-body {
  border-bottom-color: #bce8f1;
}
.panel-warning {
  border-color: #faebcc;
}
.panel-warning > .panel-heading {
  color: #8a6d3b;
  background-color: #fcf8e3;
  border-color: #faebcc;
}
.panel-warning > .panel-heading + .panel-collapse > .panel-body {
  border-top-color: #faebcc;
}
.panel-warning > .panel-heading .badge {
  color: #fcf8e3;
  background-color: #8a6d3b;
}
.panel-warning > .panel-footer + .panel-collapse > .panel-body {
  border-bottom-color: #faebcc;
}
.panel-danger {
  border-color: #ebccd1;
}
.panel-danger > .panel-heading {
  color: #a94442;
  background-color: #f2dede;
  border-color: #ebccd1;
}
.panel-danger > .panel-heading + .panel-collapse > .panel-body {
  border-top-color: #ebccd1;
}
.panel-danger > .panel-heading .badge {
  color: #f2dede;
  background-color: #a94442;
}
.panel-danger > .panel-footer + .panel-collapse > .panel-body {
  border-bottom-color: #ebccd1;
}
.embed-responsive {
  position: relative;
  display: block;
  height: 0;
  padding: 0;
  overflow: hidden;
}
.embed-responsive .embed-responsive-item,
.embed-responsive iframe,
.embed-responsive embed,
.embed-responsive object,
.embed-responsive video {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  height: 100%;
  width: 100%;
  border: 0;
}
.embed-responsive-16by9 {
  padding-bottom: 56.25%;
}
.embed-responsive-4by3 {
  padding-bottom: 75%;
}
.well {
  min-height: 20px;
  padding: 19px;
  margin-bottom: 20px;
  background-color: #f5f5f5;
  border: 1px solid #e3e3e3;
  border-radius: 2px;
  -webkit-box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.05);
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.05);
}
.well blockquote {
  border-color: #ddd;
  border-color: rgba(0, 0, 0, 0.15);
}
.well-lg {
  padding: 24px;
  border-radius: 3px;
}
.well-sm {
  padding: 9px;
  border-radius: 1px;
}
.close {
  float: right;
  font-size: 19.5px;
  font-weight: bold;
  line-height: 1;
  color: #000;
  text-shadow: 0 1px 0 #fff;
  opacity: 0.2;
  filter: alpha(opacity=20);
}
.close:hover,
.close:focus {
  color: #000;
  text-decoration: none;
  cursor: pointer;
  opacity: 0.5;
  filter: alpha(opacity=50);
}
button.close {
  padding: 0;
  cursor: pointer;
  background: transparent;
  border: 0;
  -webkit-appearance: none;
}
.modal-open {
  overflow: hidden;
}
.modal {
  display: none;
  overflow: hidden;
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1050;
  -webkit-overflow-scrolling: touch;
  outline: 0;
}
.modal.fade .modal-dialog {
  -webkit-transform: translate(0, -25%);
  -ms-transform: translate(0, -25%);
  -o-transform: translate(0, -25%);
  transform: translate(0, -25%);
  -webkit-transition: -webkit-transform 0.3s ease-out;
  -moz-transition: -moz-transform 0.3s ease-out;
  -o-transition: -o-transform 0.3s ease-out;
  transition: transform 0.3s ease-out;
}
.modal.in .modal-dialog {
  -webkit-transform: translate(0, 0);
  -ms-transform: translate(0, 0);
  -o-transform: translate(0, 0);
  transform: translate(0, 0);
}
.modal-open .modal {
  overflow-x: hidden;
  overflow-y: auto;
}
.modal-dialog {
  position: relative;
  width: auto;
  margin: 10px;
}
.modal-content {
  position: relative;
  background-color: #fff;
  border: 1px solid #999;
  border: 1px solid rgba(0, 0, 0, 0.2);
  border-radius: 3px;
  -webkit-box-shadow: 0 3px 9px rgba(0, 0, 0, 0.5);
  box-shadow: 0 3px 9px rgba(0, 0, 0, 0.5);
  background-clip: padding-box;
  outline: 0;
}
.modal-backdrop {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1040;
  background-color: #000;
}
.modal-backdrop.fade {
  opacity: 0;
  filter: alpha(opacity=0);
}
.modal-backdrop.in {
  opacity: 0.5;
  filter: alpha(opacity=50);
}
.modal-header {
  padding: 15px;
  border-bottom: 1px solid #e5e5e5;
}
.modal-header .close {
  margin-top: -2px;
}
.modal-title {
  margin: 0;
  line-height: 1.42857143;
}
.modal-body {
  position: relative;
  padding: 15px;
}
.modal-footer {
  padding: 15px;
  text-align: right;
  border-top: 1px solid #e5e5e5;
}
.modal-footer .btn + .btn {
  margin-left: 5px;
  margin-bottom: 0;
}
.modal-footer .btn-group .btn + .btn {
  margin-left: -1px;
}
.modal-footer .btn-block + .btn-block {
  margin-left: 0;
}
.modal-scrollbar-measure {
  position: absolute;
  top: -9999px;
  width: 50px;
  height: 50px;
  overflow: scroll;
}
@media (min-width: 768px) {
  .modal-dialog {
    width: 600px;
    margin: 30px auto;
  }
  .modal-content {
    -webkit-box-shadow: 0 5px 15px rgba(0, 0, 0, 0.5);
    box-shadow: 0 5px 15px rgba(0, 0, 0, 0.5);
  }
  .modal-sm {
    width: 300px;
  }
}
@media (min-width: 992px) {
  .modal-lg {
    width: 900px;
  }
}
.tooltip {
  position: absolute;
  z-index: 1070;
  display: block;
  font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
  font-style: normal;
  font-weight: normal;
  letter-spacing: normal;
  line-break: auto;
  line-height: 1.42857143;
  text-align: left;
  text-align: start;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  white-space: normal;
  word-break: normal;
  word-spacing: normal;
  word-wrap: normal;
  font-size: 12px;
  opacity: 0;
  filter: alpha(opacity=0);
}
.tooltip.in {
  opacity: 0.9;
  filter: alpha(opacity=90);
}
.tooltip.top {
  margin-top: -3px;
  padding: 5px 0;
}
.tooltip.right {
  margin-left: 3px;
  padding: 0 5px;
}
.tooltip.bottom {
  margin-top: 3px;
  padding: 5px 0;
}
.tooltip.left {
  margin-left: -3px;
  padding: 0 5px;
}
.tooltip-inner {
  max-width: 200px;
  padding: 3px 8px;
  color: #fff;
  text-align: center;
  background-color: #000;
  border-radius: 2px;
}
.tooltip-arrow {
  position: absolute;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
}
.tooltip.top .tooltip-arrow {
  bottom: 0;
  left: 50%;
  margin-left: -5px;
  border-width: 5px 5px 0;
  border-top-color: #000;
}
.tooltip.top-left .tooltip-arrow {
  bottom: 0;
  right: 5px;
  margin-bottom: -5px;
  border-width: 5px 5px 0;
  border-top-color: #000;
}
.tooltip.top-right .tooltip-arrow {
  bottom: 0;
  left: 5px;
  margin-bottom: -5px;
  border-width: 5px 5px 0;
  border-top-color: #000;
}
.tooltip.right .tooltip-arrow {
  top: 50%;
  left: 0;
  margin-top: -5px;
  border-width: 5px 5px 5px 0;
  border-right-color: #000;
}
.tooltip.left .tooltip-arrow {
  top: 50%;
  right: 0;
  margin-top: -5px;
  border-width: 5px 0 5px 5px;
  border-left-color: #000;
}
.tooltip.bottom .tooltip-arrow {
  top: 0;
  left: 50%;
  margin-left: -5px;
  border-width: 0 5px 5px;
  border-bottom-color: #000;
}
.tooltip.bottom-left .tooltip-arrow {
  top: 0;
  right: 5px;
  margin-top: -5px;
  border-width: 0 5px 5px;
  border-bottom-color: #000;
}
.tooltip.bottom-right .tooltip-arrow {
  top: 0;
  left: 5px;
  margin-top: -5px;
  border-width: 0 5px 5px;
  border-bottom-color: #000;
}
.popover {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1060;
  display: none;
  max-width: 276px;
  padding: 1px;
  font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
  font-style: normal;
  font-weight: normal;
  letter-spacing: normal;
  line-break: auto;
  line-height: 1.42857143;
  text-align: left;
  text-align: start;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  white-space: normal;
  word-break: normal;
  word-spacing: normal;
  word-wrap: normal;
  font-size: 13px;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ccc;
  border: 1px solid rgba(0, 0, 0, 0.2);
  border-radius: 3px;
  -webkit-box-shadow: 0 5px 10px rgba(0, 0, 0, 0.2);
  box-shadow: 0 5px 10px rgba(0, 0, 0, 0.2);
}
.popover.top {
  margin-top: -10px;
}
.popover.right {
  margin-left: 10px;
}
.popover.bottom {
  margin-top: 10px;
}
.popover.left {
  margin-left: -10px;
}
.popover-title {
  margin: 0;
  padding: 8px 14px;
  font-size: 13px;
  background-color: #f7f7f7;
  border-bottom: 1px solid #ebebeb;
  border-radius: 2px 2px 0 0;
}
.popover-content {
  padding: 9px 14px;
}
.popover > .arrow,
.popover > .arrow:after {
  position: absolute;
  display: block;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
}
.popover > .arrow {
  border-width: 11px;
}
.popover > .arrow:after {
  border-width: 10px;
  content: "";
}
.popover.top > .arrow {
  left: 50%;
  margin-left: -11px;
  border-bottom-width: 0;
  border-top-color: #999999;
  border-top-color: rgba(0, 0, 0, 0.25);
  bottom: -11px;
}
.popover.top > .arrow:after {
  content: " ";
  bottom: 1px;
  margin-left: -10px;
  border-bottom-width: 0;
  border-top-color: #fff;
}
.popover.right > .arrow {
  top: 50%;
  left: -11px;
  margin-top: -11px;
  border-left-width: 0;
  border-right-color: #999999;
  border-right-color: rgba(0, 0, 0, 0.25);
}
.popover.right > .arrow:after {
  content: " ";
  left: 1px;
  bottom: -10px;
  border-left-width: 0;
  border-right-color: #fff;
}
.popover.bottom > .arrow {
  left: 50%;
  margin-left: -11px;
  border-top-width: 0;
  border-bottom-color: #999999;
  border-bottom-color: rgba(0, 0, 0, 0.25);
  top: -11px;
}
.popover.bottom > .arrow:after {
  content: " ";
  top: 1px;
  margin-left: -10px;
  border-top-width: 0;
  border-bottom-color: #fff;
}
.popover.left > .arrow {
  top: 50%;
  right: -11px;
  margin-top: -11px;
  border-right-width: 0;
  border-left-color: #999999;
  border-left-color: rgba(0, 0, 0, 0.25);
}
.popover.left > .arrow:after {
  content: " ";
  right: 1px;
  border-right-width: 0;
  border-left-color: #fff;
  bottom: -10px;
}
.carousel {
  position: relative;
}
.carousel-inner {
  position: relative;
  overflow: hidden;
  width: 100%;
}
.carousel-inner > .item {
  display: none;
  position: relative;
  -webkit-transition: 0.6s ease-in-out left;
  -o-transition: 0.6s ease-in-out left;
  transition: 0.6s ease-in-out left;
}
.carousel-inner > .item > img,
.carousel-inner > .item > a > img {
  line-height: 1;
}
@media all and (transform-3d), (-webkit-transform-3d) {
  .carousel-inner > .item {
    -webkit-transition: -webkit-transform 0.6s ease-in-out;
    -moz-transition: -moz-transform 0.6s ease-in-out;
    -o-transition: -o-transform 0.6s ease-in-out;
    transition: transform 0.6s ease-in-out;
    -webkit-backface-visibility: hidden;
    -moz-backface-visibility: hidden;
    backface-visibility: hidden;
    -webkit-perspective: 1000px;
    -moz-perspective: 1000px;
    perspective: 1000px;
  }
  .carousel-inner > .item.next,
  .carousel-inner > .item.active.right {
    -webkit-transform: translate3d(100%, 0, 0);
    transform: translate3d(100%, 0, 0);
    left: 0;
  }
  .carousel-inner > .item.prev,
  .carousel-inner > .item.active.left {
    -webkit-transform: translate3d(-100%, 0, 0);
    transform: translate3d(-100%, 0, 0);
    left: 0;
  }
  .carousel-inner > .item.next.left,
  .carousel-inner > .item.prev.right,
  .carousel-inner > .item.active {
    -webkit-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
    left: 0;
  }
}
.carousel-inner > .active,
.carousel-inner > .next,
.carousel-inner > .prev {
  display: block;
}
.carousel-inner > .active {
  left: 0;
}
.carousel-inner > .next,
.carousel-inner > .prev {
  position: absolute;
  top: 0;
  width: 100%;
}
.carousel-inner > .next {
  left: 100%;
}
.carousel-inner > .prev {
  left: -100%;
}
.carousel-inner > .next.left,
.carousel-inner > .prev.right {
  left: 0;
}
.carousel-inner > .active.left {
  left: -100%;
}
.carousel-inner > .active.right {
  left: 100%;
}
.carousel-control {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  width: 15%;
  opacity: 0.5;
  filter: alpha(opacity=50);
  font-size: 20px;
  color: #fff;
  text-align: center;
  text-shadow: 0 1px 2px rgba(0, 0, 0, 0.6);
  background-color: rgba(0, 0, 0, 0);
}
.carousel-control.left {
  background-image: -webkit-linear-gradient(left, rgba(0, 0, 0, 0.5) 0%, rgba(0, 0, 0, 0.0001) 100%);
  background-image: -o-linear-gradient(left, rgba(0, 0, 0, 0.5) 0%, rgba(0, 0, 0, 0.0001) 100%);
  background-image: linear-gradient(to right, rgba(0, 0, 0, 0.5) 0%, rgba(0, 0, 0, 0.0001) 100%);
  background-repeat: repeat-x;
  filter: progid:DXImageTransform.Microsoft.gradient(startColorstr='#80000000', endColorstr='#00000000', GradientType=1);
}
.carousel-control.right {
  left: auto;
  right: 0;
  background-image: -webkit-linear-gradient(left, rgba(0, 0, 0, 0.0001) 0%, rgba(0, 0, 0, 0.5) 100%);
  background-image: -o-linear-gradient(left, rgba(0, 0, 0, 0.0001) 0%, rgba(0, 0, 0, 0.5) 100%);
  background-image: linear-gradient(to right, rgba(0, 0, 0, 0.0001) 0%, rgba(0, 0, 0, 0.5) 100%);
  background-repeat: repeat-x;
  filter: progid:DXImageTransform.Microsoft.gradient(startColorstr='#00000000', endColorstr='#80000000', GradientType=1);
}
.carousel-control:hover,
.carousel-control:focus {
  outline: 0;
  color: #fff;
  text-decoration: none;
  opacity: 0.9;
  filter: alpha(opacity=90);
}
.carousel-control .icon-prev,
.carousel-control .icon-next,
.carousel-control .glyphicon-chevron-left,
.carousel-control .glyphicon-chevron-right {
  position: absolute;
  top: 50%;
  margin-top: -10px;
  z-index: 5;
  display: inline-block;
}
.carousel-control .icon-prev,
.carousel-control .glyphicon-chevron-left {
  left: 50%;
  margin-left: -10px;
}
.carousel-control .icon-next,
.carousel-control .glyphicon-chevron-right {
  right: 50%;
  margin-right: -10px;
}
.carousel-control .icon-prev,
.carousel-control .icon-next {
  width: 20px;
  height: 20px;
  line-height: 1;
  font-family: serif;
}
.carousel-control .icon-prev:before {
  content: '\2039';
}
.carousel-control .icon-next:before {
  content: '\203a';
}
.carousel-indicators {
  position: absolute;
  bottom: 10px;
  left: 50%;
  z-index: 15;
  width: 60%;
  margin-left: -30%;
  padding-left: 0;
  list-style: none;
  text-align: center;
}
.carousel-indicators li {
  display: inline-block;
  width: 10px;
  height: 10px;
  margin: 1px;
  text-indent: -999px;
  border: 1px solid #fff;
  border-radius: 10px;
  cursor: pointer;
  background-color: #000 \9;
  background-color: rgba(0, 0, 0, 0);
}
.carousel-indicators .active {
  margin: 0;
  width: 12px;
  height: 12px;
  background-color: #fff;
}
.carousel-caption {
  position: absolute;
  left: 15%;
  right: 15%;
  bottom: 20px;
  z-index: 10;
  padding-top: 20px;
  padding-bottom: 20px;
  color: #fff;
  text-align: center;
  text-shadow: 0 1px 2px rgba(0, 0, 0, 0.6);
}
.carousel-caption .btn {
  text-shadow: none;
}
@media screen and (min-width: 768px) {
  .carousel-control .glyphicon-chevron-left,
  .carousel-control .glyphicon-chevron-right,
  .carousel-control .icon-prev,
  .carousel-control .icon-next {
    width: 30px;
    height: 30px;
    margin-top: -10px;
    font-size: 30px;
  }
  .carousel-control .glyphicon-chevron-left,
  .carousel-control .icon-prev {
    margin-left: -10px;
  }
  .carousel-control .glyphicon-chevron-right,
  .carousel-control .icon-next {
    margin-right: -10px;
  }
  .carousel-caption {
    left: 20%;
    right: 20%;
    padding-bottom: 30px;
  }
  .carousel-indicators {
    bottom: 20px;
  }
}
.clearfix:before,
.clearfix:after,
.dl-horizontal dd:before,
.dl-horizontal dd:after,
.container:before,
.container:after,
.container-fluid:before,
.container-fluid:after,
.row:before,
.row:after,
.form-horizontal .form-group:before,
.form-horizontal .form-group:after,
.btn-toolbar:before,
.btn-toolbar:after,
.btn-group-vertical > .btn-group:before,
.btn-group-vertical > .btn-group:after,
.nav:before,
.nav:after,
.navbar:before,
.navbar:after,
.navbar-header:before,
.navbar-header:after,
.navbar-collapse:before,
.navbar-collapse:after,
.pager:before,
.pager:after,
.panel-body:before,
.panel-body:after,
.modal-header:before,
.modal-header:after,
.modal-footer:before,
.modal-footer:after,
.item_buttons:before,
.item_buttons:after {
  content: " ";
  display: table;
}
.clearfix:after,
.dl-horizontal dd:after,
.container:after,
.container-fluid:after,
.row:after,
.form-horizontal .form-group:after,
.btn-toolbar:after,
.btn-group-vertical > .btn-group:after,
.nav:after,
.navbar:after,
.navbar-header:after,
.navbar-collapse:after,
.pager:after,
.panel-body:after,
.modal-header:after,
.modal-footer:after,
.item_buttons:after {
  clear: both;
}
.center-block {
  display: block;
  margin-left: auto;
  margin-right: auto;
}
.pull-right {
  float: right !important;
}
.pull-left {
  float: left !important;
}
.hide {
  display: none !important;
}
.show {
  display: block !important;
}
.invisible {
  visibility: hidden;
}
.text-hide {
  font: 0/0 a;
  color: transparent;
  text-shadow: none;
  background-color: transparent;
  border: 0;
}
.hidden {
  display: none !important;
}
.affix {
  position: fixed;
}
@-ms-viewport {
  width: device-width;
}
.visible-xs,
.visible-sm,
.visible-md,
.visible-lg {
  display: none !important;
}
.visible-xs-block,
.visible-xs-inline,
.visible-xs-inline-block,
.visible-sm-block,
.visible-sm-inline,
.visible-sm-inline-block,
.visible-md-block,
.visible-md-inline,
.visible-md-inline-block,
.visible-lg-block,
.visible-lg-inline,
.visible-lg-inline-block {
  display: none !important;
}
@media (max-width: 767px) {
  .visible-xs {
    display: block !important;
  }
  table.visible-xs {
    display: table !important;
  }
  tr.visible-xs {
    display: table-row !important;
  }
  th.visible-xs,
  td.visible-xs {
    display: table-cell !important;
  }
}
@media (max-width: 767px) {
  .visible-xs-block {
    display: block !important;
  }
}
@media (max-width: 767px) {
  .visible-xs-inline {
    display: inline !important;
  }
}
@media (max-width: 767px) {
  .visible-xs-inline-block {
    display: inline-block !important;
  }
}
@media (min-width: 768px) and (max-width: 991px) {
  .visible-sm {
    display: block !important;
  }
  table.visible-sm {
    display: table !important;
  }
  tr.visible-sm {
    display: table-row !important;
  }
  th.visible-sm,
  td.visible-sm {
    display: table-cell !important;
  }
}
@media (min-width: 768px) and (max-width: 991px) {
  .visible-sm-block {
    display: block !important;
  }
}
@media (min-width: 768px) and (max-width: 991px) {
  .visible-sm-inline {
    display: inline !important;
  }
}
@media (min-width: 768px) and (max-width: 991px) {
  .visible-sm-inline-block {
    display: inline-block !important;
  }
}
@media (min-width: 992px) and (max-width: 1199px) {
  .visible-md {
    display: block !important;
  }
  table.visible-md {
    display: table !important;
  }
  tr.visible-md {
    display: table-row !important;
  }
  th.visible-md,
  td.visible-md {
    display: table-cell !important;
  }
}
@media (min-width: 992px) and (max-width: 1199px) {
  .visible-md-block {
    display: block !important;
  }
}
@media (min-width: 992px) and (max-width: 1199px) {
  .visible-md-inline {
    display: inline !important;
  }
}
@media (min-width: 992px) and (max-width: 1199px) {
  .visible-md-inline-block {
    display: inline-block !important;
  }
}
@media (min-width: 1200px) {
  .visible-lg {
    display: block !important;
  }
  table.visible-lg {
    display: table !important;
  }
  tr.visible-lg {
    display: table-row !important;
  }
  th.visible-lg,
  td.visible-lg {
    display: table-cell !important;
  }
}
@media (min-width: 1200px) {
  .visible-lg-block {
    display: block !important;
  }
}
@media (min-width: 1200px) {
  .visible-lg-inline {
    display: inline !important;
  }
}
@media (min-width: 1200px) {
  .visible-lg-inline-block {
    display: inline-block !important;
  }
}
@media (max-width: 767px) {
  .hidden-xs {
    display: none !important;
  }
}
@media (min-width: 768px) and (max-width: 991px) {
  .hidden-sm {
    display: none !important;
  }
}
@media (min-width: 992px) and (max-width: 1199px) {
  .hidden-md {
    display: none !important;
  }
}
@media (min-width: 1200px) {
  .hidden-lg {
    display: none !important;
  }
}
.visible-print {
  display: none !important;
}
@media print {
  .visible-print {
    display: block !important;
  }
  table.visible-print {
    display: table !important;
  }
  tr.visible-print {
    display: table-row !important;
  }
  th.visible-print,
  td.visible-print {
    display: table-cell !important;
  }
}
.visible-print-block {
  display: none !important;
}
@media print {
  .visible-print-block {
    display: block !important;
  }
}
.visible-print-inline {
  display: none !important;
}
@media print {
  .visible-print-inline {
    display: inline !important;
  }
}
.visible-print-inline-block {
  display: none !important;
}
@media print {
  .visible-print-inline-block {
    display: inline-block !important;
  }
}
@media print {
  .hidden-print {
    display: none !important;
  }
}
/*!
*
* Font Awesome
*
*/
/*!
 *  Font Awesome 4.7.0 by @davegandy - http://fontawesome.io - @fontawesome
 *  License - http://fontawesome.io/license (Font: SIL OFL 1.1, CSS: MIT License)
 */
/* FONT PATH
 * -------------------------- */
@font-face {
  font-family: 'FontAwesome';
  src: url('../components/font-awesome/fonts/fontawesome-webfont.eot?v=4.7.0');
  src: url('../components/font-awesome/fonts/fontawesome-webfont.eot?#iefix&v=4.7.0') format('embedded-opentype'), url('../components/font-awesome/fonts/fontawesome-webfont.woff2?v=4.7.0') format('woff2'), url('../components/font-awesome/fonts/fontawesome-webfont.woff?v=4.7.0') format('woff'), url('../components/font-awesome/fonts/fontawesome-webfont.ttf?v=4.7.0') format('truetype'), url('../components/font-awesome/fonts/fontawesome-webfont.svg?v=4.7.0#fontawesomeregular') format('svg');
  font-weight: normal;
  font-style: normal;
}
.fa {
  display: inline-block;
  font: normal normal normal 14px/1 FontAwesome;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
/* makes the font 33% larger relative to the icon container */
.fa-lg {
  font-size: 1.33333333em;
  line-height: 0.75em;
  vertical-align: -15%;
}
.fa-2x {
  font-size: 2em;
}
.fa-3x {
  font-size: 3em;
}
.fa-4x {
  font-size: 4em;
}
.fa-5x {
  font-size: 5em;
}
.fa-fw {
  width: 1.28571429em;
  text-align: center;
}
.fa-ul {
  padding-left: 0;
  margin-left: 2.14285714em;
  list-style-type: none;
}
.fa-ul > li {
  position: relative;
}
.fa-li {
  position: absolute;
  left: -2.14285714em;
  width: 2.14285714em;
  top: 0.14285714em;
  text-align: center;
}
.fa-li.fa-lg {
  left: -1.85714286em;
}
.fa-border {
  padding: .2em .25em .15em;
  border: solid 0.08em #eee;
  border-radius: .1em;
}
.fa-pull-left {
  float: left;
}
.fa-pull-right {
  float: right;
}
.fa.fa-pull-left {
  margin-right: .3em;
}
.fa.fa-pull-right {
  margin-left: .3em;
}
/* Deprecated as of 4.4.0 */
.pull-right {
  float: right;
}
.pull-left {
  float: left;
}
.fa.pull-left {
  margin-right: .3em;
}
.fa.pull-right {
  margin-left: .3em;
}
.fa-spin {
  -webkit-animation: fa-spin 2s infinite linear;
  animation: fa-spin 2s infinite linear;
}
.fa-pulse {
  -webkit-animation: fa-spin 1s infinite steps(8);
  animation: fa-spin 1s infinite steps(8);
}
@-webkit-keyframes fa-spin {
  0% {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(359deg);
    transform: rotate(359deg);
  }
}
@keyframes fa-spin {
  0% {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(359deg);
    transform: rotate(359deg);
  }
}
.fa-rotate-90 {
  -ms-filter: "progid:DXImageTransform.Microsoft.BasicImage(rotation=1)";
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}
.fa-rotate-180 {
  -ms-filter: "progid:DXImageTransform.Microsoft.BasicImage(rotation=2)";
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}
.fa-rotate-270 {
  -ms-filter: "progid:DXImageTransform.Microsoft.BasicImage(rotation=3)";
  -webkit-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  transform: rotate(270deg);
}
.fa-flip-horizontal {
  -ms-filter: "progid:DXImageTransform.Microsoft.BasicImage(rotation=0, mirror=1)";
  -webkit-transform: scale(-1, 1);
  -ms-transform: scale(-1, 1);
  transform: scale(-1, 1);
}
.fa-flip-vertical {
  -ms-filter: "progid:DXImageTransform.Microsoft.BasicImage(rotation=2, mirror=1)";
  -webkit-transform: scale(1, -1);
  -ms-transform: scale(1, -1);
  transform: scale(1, -1);
}
:root .fa-rotate-90,
:root .fa-rotate-180,
:root .fa-rotate-270,
:root .fa-flip-horizontal,
:root .fa-flip-vertical {
  filter: none;
}
.fa-stack {
  position: relative;
  display: inline-block;
  width: 2em;
  height: 2em;
  line-height: 2em;
  vertical-align: middle;
}
.fa-stack-1x,
.fa-stack-2x {
  position: absolute;
  left: 0;
  width: 100%;
  text-align: center;
}
.fa-stack-1x {
  line-height: inherit;
}
.fa-stack-2x {
  font-size: 2em;
}
.fa-inverse {
  color: #fff;
}
/* Font Awesome uses the Unicode Private Use Area (PUA) to ensure screen
   readers do not read off random characters that represent icons */
.fa-glass:before {
  content: "\f000";
}
.fa-music:before {
  content: "\f001";
}
.fa-search:before {
  content: "\f002";
}
.fa-envelope-o:before {
  content: "\f003";
}
.fa-heart:before {
  content: "\f004";
}
.fa-star:before {
  content: "\f005";
}
.fa-star-o:before {
  content: "\f006";
}
.fa-user:before {
  content: "\f007";
}
.fa-film:before {
  content: "\f008";
}
.fa-th-large:before {
  content: "\f009";
}
.fa-th:before {
  content: "\f00a";
}
.fa-th-list:before {
  content: "\f00b";
}
.fa-check:before {
  content: "\f00c";
}
.fa-remove:before,
.fa-close:before,
.fa-times:before {
  content: "\f00d";
}
.fa-search-plus:before {
  content: "\f00e";
}
.fa-search-minus:before {
  content: "\f010";
}
.fa-power-off:before {
  content: "\f011";
}
.fa-signal:before {
  content: "\f012";
}
.fa-gear:before,
.fa-cog:before {
  content: "\f013";
}
.fa-trash-o:before {
  content: "\f014";
}
.fa-home:before {
  content: "\f015";
}
.fa-file-o:before {
  content: "\f016";
}
.fa-clock-o:before {
  content: "\f017";
}
.fa-road:before {
  content: "\f018";
}
.fa-download:before {
  content: "\f019";
}
.fa-arrow-circle-o-down:before {
  content: "\f01a";
}
.fa-arrow-circle-o-up:before {
  content: "\f01b";
}
.fa-inbox:before {
  content: "\f01c";
}
.fa-play-circle-o:before {
  content: "\f01d";
}
.fa-rotate-right:before,
.fa-repeat:before {
  content: "\f01e";
}
.fa-refresh:before {
  content: "\f021";
}
.fa-list-alt:before {
  content: "\f022";
}
.fa-lock:before {
  content: "\f023";
}
.fa-flag:before {
  content: "\f024";
}
.fa-headphones:before {
  content: "\f025";
}
.fa-volume-off:before {
  content: "\f026";
}
.fa-volume-down:before {
  content: "\f027";
}
.fa-volume-up:before {
  content: "\f028";
}
.fa-qrcode:before {
  content: "\f029";
}
.fa-barcode:before {
  content: "\f02a";
}
.fa-tag:before {
  content: "\f02b";
}
.fa-tags:before {
  content: "\f02c";
}
.fa-book:before {
  content: "\f02d";
}
.fa-bookmark:before {
  content: "\f02e";
}
.fa-print:before {
  content: "\f02f";
}
.fa-camera:before {
  content: "\f030";
}
.fa-font:before {
  content: "\f031";
}
.fa-bold:before {
  content: "\f032";
}
.fa-italic:before {
  content: "\f033";
}
.fa-text-height:before {
  content: "\f034";
}
.fa-text-width:before {
  content: "\f035";
}
.fa-align-left:before {
  content: "\f036";
}
.fa-align-center:before {
  content: "\f037";
}
.fa-align-right:before {
  content: "\f038";
}
.fa-align-justify:before {
  content: "\f039";
}
.fa-list:before {
  content: "\f03a";
}
.fa-dedent:before,
.fa-outdent:before {
  content: "\f03b";
}
.fa-indent:before {
  content: "\f03c";
}
.fa-video-camera:before {
  content: "\f03d";
}
.fa-photo:before,
.fa-image:before,
.fa-picture-o:before {
  content: "\f03e";
}
.fa-pencil:before {
  content: "\f040";
}
.fa-map-marker:before {
  content: "\f041";
}
.fa-adjust:before {
  content: "\f042";
}
.fa-tint:before {
  content: "\f043";
}
.fa-edit:before,
.fa-pencil-square-o:before {
  content: "\f044";
}
.fa-share-square-o:before {
  content: "\f045";
}
.fa-check-square-o:before {
  content: "\f046";
}
.fa-arrows:before {
  content: "\f047";
}
.fa-step-backward:before {
  content: "\f048";
}
.fa-fast-backward:before {
  content: "\f049";
}
.fa-backward:before {
  content: "\f04a";
}
.fa-play:before {
  content: "\f04b";
}
.fa-pause:before {
  content: "\f04c";
}
.fa-stop:before {
  content: "\f04d";
}
.fa-forward:before {
  content: "\f04e";
}
.fa-fast-forward:before {
  content: "\f050";
}
.fa-step-forward:before {
  content: "\f051";
}
.fa-eject:before {
  content: "\f052";
}
.fa-chevron-left:before {
  content: "\f053";
}
.fa-chevron-right:before {
  content: "\f054";
}
.fa-plus-circle:before {
  content: "\f055";
}
.fa-minus-circle:before {
  content: "\f056";
}
.fa-times-circle:before {
  content: "\f057";
}
.fa-check-circle:before {
  content: "\f058";
}
.fa-question-circle:before {
  content: "\f059";
}
.fa-info-circle:before {
  content: "\f05a";
}
.fa-crosshairs:before {
  content: "\f05b";
}
.fa-times-circle-o:before {
  content: "\f05c";
}
.fa-check-circle-o:before {
  content: "\f05d";
}
.fa-ban:before {
  content: "\f05e";
}
.fa-arrow-left:before {
  content: "\f060";
}
.fa-arrow-right:before {
  content: "\f061";
}
.fa-arrow-up:before {
  content: "\f062";
}
.fa-arrow-down:before {
  content: "\f063";
}
.fa-mail-forward:before,
.fa-share:before {
  content: "\f064";
}
.fa-expand:before {
  content: "\f065";
}
.fa-compress:before {
  content: "\f066";
}
.fa-plus:before {
  content: "\f067";
}
.fa-minus:before {
  content: "\f068";
}
.fa-asterisk:before {
  content: "\f069";
}
.fa-exclamation-circle:before {
  content: "\f06a";
}
.fa-gift:before {
  content: "\f06b";
}
.fa-leaf:before {
  content: "\f06c";
}
.fa-fire:before {
  content: "\f06d";
}
.fa-eye:before {
  content: "\f06e";
}
.fa-eye-slash:before {
  content: "\f070";
}
.fa-warning:before,
.fa-exclamation-triangle:before {
  content: "\f071";
}
.fa-plane:before {
  content: "\f072";
}
.fa-calendar:before {
  content: "\f073";
}
.fa-random:before {
  content: "\f074";
}
.fa-comment:before {
  content: "\f075";
}
.fa-magnet:before {
  content: "\f076";
}
.fa-chevron-up:before {
  content: "\f077";
}
.fa-chevron-down:before {
  content: "\f078";
}
.fa-retweet:before {
  content: "\f079";
}
.fa-shopping-cart:before {
  content: "\f07a";
}
.fa-folder:before {
  content: "\f07b";
}
.fa-folder-open:before {
  content: "\f07c";
}
.fa-arrows-v:before {
  content: "\f07d";
}
.fa-arrows-h:before {
  content: "\f07e";
}
.fa-bar-chart-o:before,
.fa-bar-chart:before {
  content: "\f080";
}
.fa-twitter-square:before {
  content: "\f081";
}
.fa-facebook-square:before {
  content: "\f082";
}
.fa-camera-retro:before {
  content: "\f083";
}
.fa-key:before {
  content: "\f084";
}
.fa-gears:before,
.fa-cogs:before {
  content: "\f085";
}
.fa-comments:before {
  content: "\f086";
}
.fa-thumbs-o-up:before {
  content: "\f087";
}
.fa-thumbs-o-down:before {
  content: "\f088";
}
.fa-star-half:before {
  content: "\f089";
}
.fa-heart-o:before {
  content: "\f08a";
}
.fa-sign-out:before {
  content: "\f08b";
}
.fa-linkedin-square:before {
  content: "\f08c";
}
.fa-thumb-tack:before {
  content: "\f08d";
}
.fa-external-link:before {
  content: "\f08e";
}
.fa-sign-in:before {
  content: "\f090";
}
.fa-trophy:before {
  content: "\f091";
}
.fa-github-square:before {
  content: "\f092";
}
.fa-upload:before {
  content: "\f093";
}
.fa-lemon-o:before {
  content: "\f094";
}
.fa-phone:before {
  content: "\f095";
}
.fa-square-o:before {
  content: "\f096";
}
.fa-bookmark-o:before {
  content: "\f097";
}
.fa-phone-square:before {
  content: "\f098";
}
.fa-twitter:before {
  content: "\f099";
}
.fa-facebook-f:before,
.fa-facebook:before {
  content: "\f09a";
}
.fa-github:before {
  content: "\f09b";
}
.fa-unlock:before {
  content: "\f09c";
}
.fa-credit-card:before {
  content: "\f09d";
}
.fa-feed:before,
.fa-rss:before {
  content: "\f09e";
}
.fa-hdd-o:before {
  content: "\f0a0";
}
.fa-bullhorn:before {
  content: "\f0a1";
}
.fa-bell:before {
  content: "\f0f3";
}
.fa-certificate:before {
  content: "\f0a3";
}
.fa-hand-o-right:before {
  content: "\f0a4";
}
.fa-hand-o-left:before {
  content: "\f0a5";
}
.fa-hand-o-up:before {
  content: "\f0a6";
}
.fa-hand-o-down:before {
  content: "\f0a7";
}
.fa-arrow-circle-left:before {
  content: "\f0a8";
}
.fa-arrow-circle-right:before {
  content: "\f0a9";
}
.fa-arrow-circle-up:before {
  content: "\f0aa";
}
.fa-arrow-circle-down:before {
  content: "\f0ab";
}
.fa-globe:before {
  content: "\f0ac";
}
.fa-wrench:before {
  content: "\f0ad";
}
.fa-tasks:before {
  content: "\f0ae";
}
.fa-filter:before {
  content: "\f0b0";
}
.fa-briefcase:before {
  content: "\f0b1";
}
.fa-arrows-alt:before {
  content: "\f0b2";
}
.fa-group:before,
.fa-users:before {
  content: "\f0c0";
}
.fa-chain:before,
.fa-link:before {
  content: "\f0c1";
}
.fa-cloud:before {
  content: "\f0c2";
}
.fa-flask:before {
  content: "\f0c3";
}
.fa-cut:before,
.fa-scissors:before {
  content: "\f0c4";
}
.fa-copy:before,
.fa-files-o:before {
  content: "\f0c5";
}
.fa-paperclip:before {
  content: "\f0c6";
}
.fa-save:before,
.fa-floppy-o:before {
  content: "\f0c7";
}
.fa-square:before {
  content: "\f0c8";
}
.fa-navicon:before,
.fa-reorder:before,
.fa-bars:before {
  content: "\f0c9";
}
.fa-list-ul:before {
  content: "\f0ca";
}
.fa-list-ol:before {
  content: "\f0cb";
}
.fa-strikethrough:before {
  content: "\f0cc";
}
.fa-underline:before {
  content: "\f0cd";
}
.fa-table:before {
  content: "\f0ce";
}
.fa-magic:before {
  content: "\f0d0";
}
.fa-truck:before {
  content: "\f0d1";
}
.fa-pinterest:before {
  content: "\f0d2";
}
.fa-pinterest-square:before {
  content: "\f0d3";
}
.fa-google-plus-square:before {
  content: "\f0d4";
}
.fa-google-plus:before {
  content: "\f0d5";
}
.fa-money:before {
  content: "\f0d6";
}
.fa-caret-down:before {
  content: "\f0d7";
}
.fa-caret-up:before {
  content: "\f0d8";
}
.fa-caret-left:before {
  content: "\f0d9";
}
.fa-caret-right:before {
  content: "\f0da";
}
.fa-columns:before {
  content: "\f0db";
}
.fa-unsorted:before,
.fa-sort:before {
  content: "\f0dc";
}
.fa-sort-down:before,
.fa-sort-desc:before {
  content: "\f0dd";
}
.fa-sort-up:before,
.fa-sort-asc:before {
  content: "\f0de";
}
.fa-envelope:before {
  content: "\f0e0";
}
.fa-linkedin:before {
  content: "\f0e1";
}
.fa-rotate-left:before,
.fa-undo:before {
  content: "\f0e2";
}
.fa-legal:before,
.fa-gavel:before {
  content: "\f0e3";
}
.fa-dashboard:before,
.fa-tachometer:before {
  content: "\f0e4";
}
.fa-comment-o:before {
  content: "\f0e5";
}
.fa-comments-o:before {
  content: "\f0e6";
}
.fa-flash:before,
.fa-bolt:before {
  content: "\f0e7";
}
.fa-sitemap:before {
  content: "\f0e8";
}
.fa-umbrella:before {
  content: "\f0e9";
}
.fa-paste:before,
.fa-clipboard:before {
  content: "\f0ea";
}
.fa-lightbulb-o:before {
  content: "\f0eb";
}
.fa-exchange:before {
  content: "\f0ec";
}
.fa-cloud-download:before {
  content: "\f0ed";
}
.fa-cloud-upload:before {
  content: "\f0ee";
}
.fa-user-md:before {
  content: "\f0f0";
}
.fa-stethoscope:before {
  content: "\f0f1";
}
.fa-suitcase:before {
  content: "\f0f2";
}
.fa-bell-o:before {
  content: "\f0a2";
}
.fa-coffee:before {
  content: "\f0f4";
}
.fa-cutlery:before {
  content: "\f0f5";
}
.fa-file-text-o:before {
  content: "\f0f6";
}
.fa-building-o:before {
  content: "\f0f7";
}
.fa-hospital-o:before {
  content: "\f0f8";
}
.fa-ambulance:before {
  content: "\f0f9";
}
.fa-medkit:before {
  content: "\f0fa";
}
.fa-fighter-jet:before {
  content: "\f0fb";
}
.fa-beer:before {
  content: "\f0fc";
}
.fa-h-square:before {
  content: "\f0fd";
}
.fa-plus-square:before {
  content: "\f0fe";
}
.fa-angle-double-left:before {
  content: "\f100";
}
.fa-angle-double-right:before {
  content: "\f101";
}
.fa-angle-double-up:before {
  content: "\f102";
}
.fa-angle-double-down:before {
  content: "\f103";
}
.fa-angle-left:before {
  content: "\f104";
}
.fa-angle-right:before {
  content: "\f105";
}
.fa-angle-up:before {
  content: "\f106";
}
.fa-angle-down:before {
  content: "\f107";
}
.fa-desktop:before {
  content: "\f108";
}
.fa-laptop:before {
  content: "\f109";
}
.fa-tablet:before {
  content: "\f10a";
}
.fa-mobile-phone:before,
.fa-mobile:before {
  content: "\f10b";
}
.fa-circle-o:before {
  content: "\f10c";
}
.fa-quote-left:before {
  content: "\f10d";
}
.fa-quote-right:before {
  content: "\f10e";
}
.fa-spinner:before {
  content: "\f110";
}
.fa-circle:before {
  content: "\f111";
}
.fa-mail-reply:before,
.fa-reply:before {
  content: "\f112";
}
.fa-github-alt:before {
  content: "\f113";
}
.fa-folder-o:before {
  content: "\f114";
}
.fa-folder-open-o:before {
  content: "\f115";
}
.fa-smile-o:before {
  content: "\f118";
}
.fa-frown-o:before {
  content: "\f119";
}
.fa-meh-o:before {
  content: "\f11a";
}
.fa-gamepad:before {
  content: "\f11b";
}
.fa-keyboard-o:before {
  content: "\f11c";
}
.fa-flag-o:before {
  content: "\f11d";
}
.fa-flag-checkered:before {
  content: "\f11e";
}
.fa-terminal:before {
  content: "\f120";
}
.fa-code:before {
  content: "\f121";
}
.fa-mail-reply-all:before,
.fa-reply-all:before {
  content: "\f122";
}
.fa-star-half-empty:before,
.fa-star-half-full:before,
.fa-star-half-o:before {
  content: "\f123";
}
.fa-location-arrow:before {
  content: "\f124";
}
.fa-crop:before {
  content: "\f125";
}
.fa-code-fork:before {
  content: "\f126";
}
.fa-unlink:before,
.fa-chain-broken:before {
  content: "\f127";
}
.fa-question:before {
  content: "\f128";
}
.fa-info:before {
  content: "\f129";
}
.fa-exclamation:before {
  content: "\f12a";
}
.fa-superscript:before {
  content: "\f12b";
}
.fa-subscript:before {
  content: "\f12c";
}
.fa-eraser:before {
  content: "\f12d";
}
.fa-puzzle-piece:before {
  content: "\f12e";
}
.fa-microphone:before {
  content: "\f130";
}
.fa-microphone-slash:before {
  content: "\f131";
}
.fa-shield:before {
  content: "\f132";
}
.fa-calendar-o:before {
  content: "\f133";
}
.fa-fire-extinguisher:before {
  content: "\f134";
}
.fa-rocket:before {
  content: "\f135";
}
.fa-maxcdn:before {
  content: "\f136";
}
.fa-chevron-circle-left:before {
  content: "\f137";
}
.fa-chevron-circle-right:before {
  content: "\f138";
}
.fa-chevron-circle-up:before {
  content: "\f139";
}
.fa-chevron-circle-down:before {
  content: "\f13a";
}
.fa-html5:before {
  content: "\f13b";
}
.fa-css3:before {
  content: "\f13c";
}
.fa-anchor:before {
  content: "\f13d";
}
.fa-unlock-alt:before {
  content: "\f13e";
}
.fa-bullseye:before {
  content: "\f140";
}
.fa-ellipsis-h:before {
  content: "\f141";
}
.fa-ellipsis-v:before {
  content: "\f142";
}
.fa-rss-square:before {
  content: "\f143";
}
.fa-play-circle:before {
  content: "\f144";
}
.fa-ticket:before {
  content: "\f145";
}
.fa-minus-square:before {
  content: "\f146";
}
.fa-minus-square-o:before {
  content: "\f147";
}
.fa-level-up:before {
  content: "\f148";
}
.fa-level-down:before {
  content: "\f149";
}
.fa-check-square:before {
  content: "\f14a";
}
.fa-pencil-square:before {
  content: "\f14b";
}
.fa-external-link-square:before {
  content: "\f14c";
}
.fa-share-square:before {
  content: "\f14d";
}
.fa-compass:before {
  content: "\f14e";
}
.fa-toggle-down:before,
.fa-caret-square-o-down:before {
  content: "\f150";
}
.fa-toggle-up:before,
.fa-caret-square-o-up:before {
  content: "\f151";
}
.fa-toggle-right:before,
.fa-caret-square-o-right:before {
  content: "\f152";
}
.fa-euro:before,
.fa-eur:before {
  content: "\f153";
}
.fa-gbp:before {
  content: "\f154";
}
.fa-dollar:before,
.fa-usd:before {
  content: "\f155";
}
.fa-rupee:before,
.fa-inr:before {
  content: "\f156";
}
.fa-cny:before,
.fa-rmb:before,
.fa-yen:before,
.fa-jpy:before {
  content: "\f157";
}
.fa-ruble:before,
.fa-rouble:before,
.fa-rub:before {
  content: "\f158";
}
.fa-won:before,
.fa-krw:before {
  content: "\f159";
}
.fa-bitcoin:before,
.fa-btc:before {
  content: "\f15a";
}
.fa-file:before {
  content: "\f15b";
}
.fa-file-text:before {
  content: "\f15c";
}
.fa-sort-alpha-asc:before {
  content: "\f15d";
}
.fa-sort-alpha-desc:before {
  content: "\f15e";
}
.fa-sort-amount-asc:before {
  content: "\f160";
}
.fa-sort-amount-desc:before {
  content: "\f161";
}
.fa-sort-numeric-asc:before {
  content: "\f162";
}
.fa-sort-numeric-desc:before {
  content: "\f163";
}
.fa-thumbs-up:before {
  content: "\f164";
}
.fa-thumbs-down:before {
  content: "\f165";
}
.fa-youtube-square:before {
  content: "\f166";
}
.fa-youtube:before {
  content: "\f167";
}
.fa-xing:before {
  content: "\f168";
}
.fa-xing-square:before {
  content: "\f169";
}
.fa-youtube-play:before {
  content: "\f16a";
}
.fa-dropbox:before {
  content: "\f16b";
}
.fa-stack-overflow:before {
  content: "\f16c";
}
.fa-instagram:before {
  content: "\f16d";
}
.fa-flickr:before {
  content: "\f16e";
}
.fa-adn:before {
  content: "\f170";
}
.fa-bitbucket:before {
  content: "\f171";
}
.fa-bitbucket-square:before {
  content: "\f172";
}
.fa-tumblr:before {
  content: "\f173";
}
.fa-tumblr-square:before {
  content: "\f174";
}
.fa-long-arrow-down:before {
  content: "\f175";
}
.fa-long-arrow-up:before {
  content: "\f176";
}
.fa-long-arrow-left:before {
  content: "\f177";
}
.fa-long-arrow-right:before {
  content: "\f178";
}
.fa-apple:before {
  content: "\f179";
}
.fa-windows:before {
  content: "\f17a";
}
.fa-android:before {
  content: "\f17b";
}
.fa-linux:before {
  content: "\f17c";
}
.fa-dribbble:before {
  content: "\f17d";
}
.fa-skype:before {
  content: "\f17e";
}
.fa-foursquare:before {
  content: "\f180";
}
.fa-trello:before {
  content: "\f181";
}
.fa-female:before {
  content: "\f182";
}
.fa-male:before {
  content: "\f183";
}
.fa-gittip:before,
.fa-gratipay:before {
  content: "\f184";
}
.fa-sun-o:before {
  content: "\f185";
}
.fa-moon-o:before {
  content: "\f186";
}
.fa-archive:before {
  content: "\f187";
}
.fa-bug:before {
  content: "\f188";
}
.fa-vk:before {
  content: "\f189";
}
.fa-weibo:before {
  content: "\f18a";
}
.fa-renren:before {
  content: "\f18b";
}
.fa-pagelines:before {
  content: "\f18c";
}
.fa-stack-exchange:before {
  content: "\f18d";
}
.fa-arrow-circle-o-right:before {
  content: "\f18e";
}
.fa-arrow-circle-o-left:before {
  content: "\f190";
}
.fa-toggle-left:before,
.fa-caret-square-o-left:before {
  content: "\f191";
}
.fa-dot-circle-o:before {
  content: "\f192";
}
.fa-wheelchair:before {
  content: "\f193";
}
.fa-vimeo-square:before {
  content: "\f194";
}
.fa-turkish-lira:before,
.fa-try:before {
  content: "\f195";
}
.fa-plus-square-o:before {
  content: "\f196";
}
.fa-space-shuttle:before {
  content: "\f197";
}
.fa-slack:before {
  content: "\f198";
}
.fa-envelope-square:before {
  content: "\f199";
}
.fa-wordpress:before {
  content: "\f19a";
}
.fa-openid:before {
  content: "\f19b";
}
.fa-institution:before,
.fa-bank:before,
.fa-university:before {
  content: "\f19c";
}
.fa-mortar-board:before,
.fa-graduation-cap:before {
  content: "\f19d";
}
.fa-yahoo:before {
  content: "\f19e";
}
.fa-google:before {
  content: "\f1a0";
}
.fa-reddit:before {
  content: "\f1a1";
}
.fa-reddit-square:before {
  content: "\f1a2";
}
.fa-stumbleupon-circle:before {
  content: "\f1a3";
}
.fa-stumbleupon:before {
  content: "\f1a4";
}
.fa-delicious:before {
  content: "\f1a5";
}
.fa-digg:before {
  content: "\f1a6";
}
.fa-pied-piper-pp:before {
  content: "\f1a7";
}
.fa-pied-piper-alt:before {
  content: "\f1a8";
}
.fa-drupal:before {
  content: "\f1a9";
}
.fa-joomla:before {
  content: "\f1aa";
}
.fa-language:before {
  content: "\f1ab";
}
.fa-fax:before {
  content: "\f1ac";
}
.fa-building:before {
  content: "\f1ad";
}
.fa-child:before {
  content: "\f1ae";
}
.fa-paw:before {
  content: "\f1b0";
}
.fa-spoon:before {
  content: "\f1b1";
}
.fa-cube:before {
  content: "\f1b2";
}
.fa-cubes:before {
  content: "\f1b3";
}
.fa-behance:before {
  content: "\f1b4";
}
.fa-behance-square:before {
  content: "\f1b5";
}
.fa-steam:before {
  content: "\f1b6";
}
.fa-steam-square:before {
  content: "\f1b7";
}
.fa-recycle:before {
  content: "\f1b8";
}
.fa-automobile:before,
.fa-car:before {
  content: "\f1b9";
}
.fa-cab:before,
.fa-taxi:before {
  content: "\f1ba";
}
.fa-tree:before {
  content: "\f1bb";
}
.fa-spotify:before {
  content: "\f1bc";
}
.fa-deviantart:before {
  content: "\f1bd";
}
.fa-soundcloud:before {
  content: "\f1be";
}
.fa-database:before {
  content: "\f1c0";
}
.fa-file-pdf-o:before {
  content: "\f1c1";
}
.fa-file-word-o:before {
  content: "\f1c2";
}
.fa-file-excel-o:before {
  content: "\f1c3";
}
.fa-file-powerpoint-o:before {
  content: "\f1c4";
}
.fa-file-photo-o:before,
.fa-file-picture-o:before,
.fa-file-image-o:before {
  content: "\f1c5";
}
.fa-file-zip-o:before,
.fa-file-archive-o:before {
  content: "\f1c6";
}
.fa-file-sound-o:before,
.fa-file-audio-o:before {
  content: "\f1c7";
}
.fa-file-movie-o:before,
.fa-file-video-o:before {
  content: "\f1c8";
}
.fa-file-code-o:before {
  content: "\f1c9";
}
.fa-vine:before {
  content: "\f1ca";
}
.fa-codepen:before {
  content: "\f1cb";
}
.fa-jsfiddle:before {
  content: "\f1cc";
}
.fa-life-bouy:before,
.fa-life-buoy:before,
.fa-life-saver:before,
.fa-support:before,
.fa-life-ring:before {
  content: "\f1cd";
}
.fa-circle-o-notch:before {
  content: "\f1ce";
}
.fa-ra:before,
.fa-resistance:before,
.fa-rebel:before {
  content: "\f1d0";
}
.fa-ge:before,
.fa-empire:before {
  content: "\f1d1";
}
.fa-git-square:before {
  content: "\f1d2";
}
.fa-git:before {
  content: "\f1d3";
}
.fa-y-combinator-square:before,
.fa-yc-square:before,
.fa-hacker-news:before {
  content: "\f1d4";
}
.fa-tencent-weibo:before {
  content: "\f1d5";
}
.fa-qq:before {
  content: "\f1d6";
}
.fa-wechat:before,
.fa-weixin:before {
  content: "\f1d7";
}
.fa-send:before,
.fa-paper-plane:before {
  content: "\f1d8";
}
.fa-send-o:before,
.fa-paper-plane-o:before {
  content: "\f1d9";
}
.fa-history:before {
  content: "\f1da";
}
.fa-circle-thin:before {
  content: "\f1db";
}
.fa-header:before {
  content: "\f1dc";
}
.fa-paragraph:before {
  content: "\f1dd";
}
.fa-sliders:before {
  content: "\f1de";
}
.fa-share-alt:before {
  content: "\f1e0";
}
.fa-share-alt-square:before {
  content: "\f1e1";
}
.fa-bomb:before {
  content: "\f1e2";
}
.fa-soccer-ball-o:before,
.fa-futbol-o:before {
  content: "\f1e3";
}
.fa-tty:before {
  content: "\f1e4";
}
.fa-binoculars:before {
  content: "\f1e5";
}
.fa-plug:before {
  content: "\f1e6";
}
.fa-slideshare:before {
  content: "\f1e7";
}
.fa-twitch:before {
  content: "\f1e8";
}
.fa-yelp:before {
  content: "\f1e9";
}
.fa-newspaper-o:before {
  content: "\f1ea";
}
.fa-wifi:before {
  content: "\f1eb";
}
.fa-calculator:before {
  content: "\f1ec";
}
.fa-paypal:before {
  content: "\f1ed";
}
.fa-google-wallet:before {
  content: "\f1ee";
}
.fa-cc-visa:before {
  content: "\f1f0";
}
.fa-cc-mastercard:before {
  content: "\f1f1";
}
.fa-cc-discover:before {
  content: "\f1f2";
}
.fa-cc-amex:before {
  content: "\f1f3";
}
.fa-cc-paypal:before {
  content: "\f1f4";
}
.fa-cc-stripe:before {
  content: "\f1f5";
}
.fa-bell-slash:before {
  content: "\f1f6";
}
.fa-bell-slash-o:before {
  content: "\f1f7";
}
.fa-trash:before {
  content: "\f1f8";
}
.fa-copyright:before {
  content: "\f1f9";
}
.fa-at:before {
  content: "\f1fa";
}
.fa-eyedropper:before {
  content: "\f1fb";
}
.fa-paint-brush:before {
  content: "\f1fc";
}
.fa-birthday-cake:before {
  content: "\f1fd";
}
.fa-area-chart:before {
  content: "\f1fe";
}
.fa-pie-chart:before {
  content: "\f200";
}
.fa-line-chart:before {
  content: "\f201";
}
.fa-lastfm:before {
  content: "\f202";
}
.fa-lastfm-square:before {
  content: "\f203";
}
.fa-toggle-off:before {
  content: "\f204";
}
.fa-toggle-on:before {
  content: "\f205";
}
.fa-bicycle:before {
  content: "\f206";
}
.fa-bus:before {
  content: "\f207";
}
.fa-ioxhost:before {
  content: "\f208";
}
.fa-angellist:before {
  content: "\f209";
}
.fa-cc:before {
  content: "\f20a";
}
.fa-shekel:before,
.fa-sheqel:before,
.fa-ils:before {
  content: "\f20b";
}
.fa-meanpath:before {
  content: "\f20c";
}
.fa-buysellads:before {
  content: "\f20d";
}
.fa-connectdevelop:before {
  content: "\f20e";
}
.fa-dashcube:before {
  content: "\f210";
}
.fa-forumbee:before {
  content: "\f211";
}
.fa-leanpub:before {
  content: "\f212";
}
.fa-sellsy:before {
  content: "\f213";
}
.fa-shirtsinbulk:before {
  content: "\f214";
}
.fa-simplybuilt:before {
  content: "\f215";
}
.fa-skyatlas:before {
  content: "\f216";
}
.fa-cart-plus:before {
  content: "\f217";
}
.fa-cart-arrow-down:before {
  content: "\f218";
}
.fa-diamond:before {
  content: "\f219";
}
.fa-ship:before {
  content: "\f21a";
}
.fa-user-secret:before {
  content: "\f21b";
}
.fa-motorcycle:before {
  content: "\f21c";
}
.fa-street-view:before {
  content: "\f21d";
}
.fa-heartbeat:before {
  content: "\f21e";
}
.fa-venus:before {
  content: "\f221";
}
.fa-mars:before {
  content: "\f222";
}
.fa-mercury:before {
  content: "\f223";
}
.fa-intersex:before,
.fa-transgender:before {
  content: "\f224";
}
.fa-transgender-alt:before {
  content: "\f225";
}
.fa-venus-double:before {
  content: "\f226";
}
.fa-mars-double:before {
  content: "\f227";
}
.fa-venus-mars:before {
  content: "\f228";
}
.fa-mars-stroke:before {
  content: "\f229";
}
.fa-mars-stroke-v:before {
  content: "\f22a";
}
.fa-mars-stroke-h:before {
  content: "\f22b";
}
.fa-neuter:before {
  content: "\f22c";
}
.fa-genderless:before {
  content: "\f22d";
}
.fa-facebook-official:before {
  content: "\f230";
}
.fa-pinterest-p:before {
  content: "\f231";
}
.fa-whatsapp:before {
  content: "\f232";
}
.fa-server:before {
  content: "\f233";
}
.fa-user-plus:before {
  content: "\f234";
}
.fa-user-times:before {
  content: "\f235";
}
.fa-hotel:before,
.fa-bed:before {
  content: "\f236";
}
.fa-viacoin:before {
  content: "\f237";
}
.fa-train:before {
  content: "\f238";
}
.fa-subway:before {
  content: "\f239";
}
.fa-medium:before {
  content: "\f23a";
}
.fa-yc:before,
.fa-y-combinator:before {
  content: "\f23b";
}
.fa-optin-monster:before {
  content: "\f23c";
}
.fa-opencart:before {
  content: "\f23d";
}
.fa-expeditedssl:before {
  content: "\f23e";
}
.fa-battery-4:before,
.fa-battery:before,
.fa-battery-full:before {
  content: "\f240";
}
.fa-battery-3:before,
.fa-battery-three-quarters:before {
  content: "\f241";
}
.fa-battery-2:before,
.fa-battery-half:before {
  content: "\f242";
}
.fa-battery-1:before,
.fa-battery-quarter:before {
  content: "\f243";
}
.fa-battery-0:before,
.fa-battery-empty:before {
  content: "\f244";
}
.fa-mouse-pointer:before {
  content: "\f245";
}
.fa-i-cursor:before {
  content: "\f246";
}
.fa-object-group:before {
  content: "\f247";
}
.fa-object-ungroup:before {
  content: "\f248";
}
.fa-sticky-note:before {
  content: "\f249";
}
.fa-sticky-note-o:before {
  content: "\f24a";
}
.fa-cc-jcb:before {
  content: "\f24b";
}
.fa-cc-diners-club:before {
  content: "\f24c";
}
.fa-clone:before {
  content: "\f24d";
}
.fa-balance-scale:before {
  content: "\f24e";
}
.fa-hourglass-o:before {
  content: "\f250";
}
.fa-hourglass-1:before,
.fa-hourglass-start:before {
  content: "\f251";
}
.fa-hourglass-2:before,
.fa-hourglass-half:before {
  content: "\f252";
}
.fa-hourglass-3:before,
.fa-hourglass-end:before {
  content: "\f253";
}
.fa-hourglass:before {
  content: "\f254";
}
.fa-hand-grab-o:before,
.fa-hand-rock-o:before {
  content: "\f255";
}
.fa-hand-stop-o:before,
.fa-hand-paper-o:before {
  content: "\f256";
}
.fa-hand-scissors-o:before {
  content: "\f257";
}
.fa-hand-lizard-o:before {
  content: "\f258";
}
.fa-hand-spock-o:before {
  content: "\f259";
}
.fa-hand-pointer-o:before {
  content: "\f25a";
}
.fa-hand-peace-o:before {
  content: "\f25b";
}
.fa-trademark:before {
  content: "\f25c";
}
.fa-registered:before {
  content: "\f25d";
}
.fa-creative-commons:before {
  content: "\f25e";
}
.fa-gg:before {
  content: "\f260";
}
.fa-gg-circle:before {
  content: "\f261";
}
.fa-tripadvisor:before {
  content: "\f262";
}
.fa-odnoklassniki:before {
  content: "\f263";
}
.fa-odnoklassniki-square:before {
  content: "\f264";
}
.fa-get-pocket:before {
  content: "\f265";
}
.fa-wikipedia-w:before {
  content: "\f266";
}
.fa-safari:before {
  content: "\f267";
}
.fa-chrome:before {
  content: "\f268";
}
.fa-firefox:before {
  content: "\f269";
}
.fa-opera:before {
  content: "\f26a";
}
.fa-internet-explorer:before {
  content: "\f26b";
}
.fa-tv:before,
.fa-television:before {
  content: "\f26c";
}
.fa-contao:before {
  content: "\f26d";
}
.fa-500px:before {
  content: "\f26e";
}
.fa-amazon:before {
  content: "\f270";
}
.fa-calendar-plus-o:before {
  content: "\f271";
}
.fa-calendar-minus-o:before {
  content: "\f272";
}
.fa-calendar-times-o:before {
  content: "\f273";
}
.fa-calendar-check-o:before {
  content: "\f274";
}
.fa-industry:before {
  content: "\f275";
}
.fa-map-pin:before {
  content: "\f276";
}
.fa-map-signs:before {
  content: "\f277";
}
.fa-map-o:before {
  content: "\f278";
}
.fa-map:before {
  content: "\f279";
}
.fa-commenting:before {
  content: "\f27a";
}
.fa-commenting-o:before {
  content: "\f27b";
}
.fa-houzz:before {
  content: "\f27c";
}
.fa-vimeo:before {
  content: "\f27d";
}
.fa-black-tie:before {
  content: "\f27e";
}
.fa-fonticons:before {
  content: "\f280";
}
.fa-reddit-alien:before {
  content: "\f281";
}
.fa-edge:before {
  content: "\f282";
}
.fa-credit-card-alt:before {
  content: "\f283";
}
.fa-codiepie:before {
  content: "\f284";
}
.fa-modx:before {
  content: "\f285";
}
.fa-fort-awesome:before {
  content: "\f286";
}
.fa-usb:before {
  content: "\f287";
}
.fa-product-hunt:before {
  content: "\f288";
}
.fa-mixcloud:before {
  content: "\f289";
}
.fa-scribd:before {
  content: "\f28a";
}
.fa-pause-circle:before {
  content: "\f28b";
}
.fa-pause-circle-o:before {
  content: "\f28c";
}
.fa-stop-circle:before {
  content: "\f28d";
}
.fa-stop-circle-o:before {
  content: "\f28e";
}
.fa-shopping-bag:before {
  content: "\f290";
}
.fa-shopping-basket:before {
  content: "\f291";
}
.fa-hashtag:before {
  content: "\f292";
}
.fa-bluetooth:before {
  content: "\f293";
}
.fa-bluetooth-b:before {
  content: "\f294";
}
.fa-percent:before {
  content: "\f295";
}
.fa-gitlab:before {
  content: "\f296";
}
.fa-wpbeginner:before {
  content: "\f297";
}
.fa-wpforms:before {
  content: "\f298";
}
.fa-envira:before {
  content: "\f299";
}
.fa-universal-access:before {
  content: "\f29a";
}
.fa-wheelchair-alt:before {
  content: "\f29b";
}
.fa-question-circle-o:before {
  content: "\f29c";
}
.fa-blind:before {
  content: "\f29d";
}
.fa-audio-description:before {
  content: "\f29e";
}
.fa-volume-control-phone:before {
  content: "\f2a0";
}
.fa-braille:before {
  content: "\f2a1";
}
.fa-assistive-listening-systems:before {
  content: "\f2a2";
}
.fa-asl-interpreting:before,
.fa-american-sign-language-interpreting:before {
  content: "\f2a3";
}
.fa-deafness:before,
.fa-hard-of-hearing:before,
.fa-deaf:before {
  content: "\f2a4";
}
.fa-glide:before {
  content: "\f2a5";
}
.fa-glide-g:before {
  content: "\f2a6";
}
.fa-signing:before,
.fa-sign-language:before {
  content: "\f2a7";
}
.fa-low-vision:before {
  content: "\f2a8";
}
.fa-viadeo:before {
  content: "\f2a9";
}
.fa-viadeo-square:before {
  content: "\f2aa";
}
.fa-snapchat:before {
  content: "\f2ab";
}
.fa-snapchat-ghost:before {
  content: "\f2ac";
}
.fa-snapchat-square:before {
  content: "\f2ad";
}
.fa-pied-piper:before {
  content: "\f2ae";
}
.fa-first-order:before {
  content: "\f2b0";
}
.fa-yoast:before {
  content: "\f2b1";
}
.fa-themeisle:before {
  content: "\f2b2";
}
.fa-google-plus-circle:before,
.fa-google-plus-official:before {
  content: "\f2b3";
}
.fa-fa:before,
.fa-font-awesome:before {
  content: "\f2b4";
}
.fa-handshake-o:before {
  content: "\f2b5";
}
.fa-envelope-open:before {
  content: "\f2b6";
}
.fa-envelope-open-o:before {
  content: "\f2b7";
}
.fa-linode:before {
  content: "\f2b8";
}
.fa-address-book:before {
  content: "\f2b9";
}
.fa-address-book-o:before {
  content: "\f2ba";
}
.fa-vcard:before,
.fa-address-card:before {
  content: "\f2bb";
}
.fa-vcard-o:before,
.fa-address-card-o:before {
  content: "\f2bc";
}
.fa-user-circle:before {
  content: "\f2bd";
}
.fa-user-circle-o:before {
  content: "\f2be";
}
.fa-user-o:before {
  content: "\f2c0";
}
.fa-id-badge:before {
  content: "\f2c1";
}
.fa-drivers-license:before,
.fa-id-card:before {
  content: "\f2c2";
}
.fa-drivers-license-o:before,
.fa-id-card-o:before {
  content: "\f2c3";
}
.fa-quora:before {
  content: "\f2c4";
}
.fa-free-code-camp:before {
  content: "\f2c5";
}
.fa-telegram:before {
  content: "\f2c6";
}
.fa-thermometer-4:before,
.fa-thermometer:before,
.fa-thermometer-full:before {
  content: "\f2c7";
}
.fa-thermometer-3:before,
.fa-thermometer-three-quarters:before {
  content: "\f2c8";
}
.fa-thermometer-2:before,
.fa-thermometer-half:before {
  content: "\f2c9";
}
.fa-thermometer-1:before,
.fa-thermometer-quarter:before {
  content: "\f2ca";
}
.fa-thermometer-0:before,
.fa-thermometer-empty:before {
  content: "\f2cb";
}
.fa-shower:before {
  content: "\f2cc";
}
.fa-bathtub:before,
.fa-s15:before,
.fa-bath:before {
  content: "\f2cd";
}
.fa-podcast:before {
  content: "\f2ce";
}
.fa-window-maximize:before {
  content: "\f2d0";
}
.fa-window-minimize:before {
  content: "\f2d1";
}
.fa-window-restore:before {
  content: "\f2d2";
}
.fa-times-rectangle:before,
.fa-window-close:before {
  content: "\f2d3";
}
.fa-times-rectangle-o:before,
.fa-window-close-o:before {
  content: "\f2d4";
}
.fa-bandcamp:before {
  content: "\f2d5";
}
.fa-grav:before {
  content: "\f2d6";
}
.fa-etsy:before {
  content: "\f2d7";
}
.fa-imdb:before {
  content: "\f2d8";
}
.fa-ravelry:before {
  content: "\f2d9";
}
.fa-eercast:before {
  content: "\f2da";
}
.fa-microchip:before {
  content: "\f2db";
}
.fa-snowflake-o:before {
  content: "\f2dc";
}
.fa-superpowers:before {
  content: "\f2dd";
}
.fa-wpexplorer:before {
  content: "\f2de";
}
.fa-meetup:before {
  content: "\f2e0";
}
.sr-only {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  border: 0;
}
.sr-only-focusable:active,
.sr-only-focusable:focus {
  position: static;
  width: auto;
  height: auto;
  margin: 0;
  overflow: visible;
  clip: auto;
}
.sr-only-focusable:active,
.sr-only-focusable:focus {
  position: static;
  width: auto;
  height: auto;
  margin: 0;
  overflow: visible;
  clip: auto;
}
/*!
*
* IPython base
*
*/
.modal.fade .modal-dialog {
  -webkit-transform: translate(0, 0);
  -ms-transform: translate(0, 0);
  -o-transform: translate(0, 0);
  transform: translate(0, 0);
}
code {
  color: #000;
}
pre {
  font-size: inherit;
  line-height: inherit;
}
label {
  font-weight: normal;
}
/* Make the page background atleast 100% the height of the view port */
/* Make the page itself atleast 70% the height of the view port */
.border-box-sizing {
  box-sizing: border-box;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
}
.corner-all {
  border-radius: 2px;
}
.no-padding {
  padding: 0px;
}
/* Flexible box model classes */
/* Taken from Alex Russell http://infrequently.org/2009/08/css-3-progress/ */
/* This file is a compatability layer.  It allows the usage of flexible box 
model layouts accross multiple browsers, including older browsers.  The newest,
universal implementation of the flexible box model is used when available (see
`Modern browsers` comments below).  Browsers that are known to implement this 
new spec completely include:

    Firefox 28.0+
    Chrome 29.0+
    Internet Explorer 11+ 
    Opera 17.0+

Browsers not listed, including Safari, are supported via the styling under the
`Old browsers` comments below.
*/
.hbox {
  /* Old browsers */
  display: -webkit-box;
  -webkit-box-orient: horizontal;
  -webkit-box-align: stretch;
  display: -moz-box;
  -moz-box-orient: horizontal;
  -moz-box-align: stretch;
  display: box;
  box-orient: horizontal;
  box-align: stretch;
  /* Modern browsers */
  display: flex;
  flex-direction: row;
  align-items: stretch;
}
.hbox > * {
  /* Old browsers */
  -webkit-box-flex: 0;
  -moz-box-flex: 0;
  box-flex: 0;
  /* Modern browsers */
  flex: none;
}
.vbox {
  /* Old browsers */
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-box-align: stretch;
  display: -moz-box;
  -moz-box-orient: vertical;
  -moz-box-align: stretch;
  display: box;
  box-orient: vertical;
  box-align: stretch;
  /* Modern browsers */
  display: flex;
  flex-direction: column;
  align-items: stretch;
}
.vbox > * {
  /* Old browsers */
  -webkit-box-flex: 0;
  -moz-box-flex: 0;
  box-flex: 0;
  /* Modern browsers */
  flex: none;
}
.hbox.reverse,
.vbox.reverse,
.reverse {
  /* Old browsers */
  -webkit-box-direction: reverse;
  -moz-box-direction: reverse;
  box-direction: reverse;
  /* Modern browsers */
  flex-direction: row-reverse;
}
.hbox.box-flex0,
.vbox.box-flex0,
.box-flex0 {
  /* Old browsers */
  -webkit-box-flex: 0;
  -moz-box-flex: 0;
  box-flex: 0;
  /* Modern browsers */
  flex: none;
  width: auto;
}
.hbox.box-flex1,
.vbox.box-flex1,
.box-flex1 {
  /* Old browsers */
  -webkit-box-flex: 1;
  -moz-box-flex: 1;
  box-flex: 1;
  /* Modern browsers */
  flex: 1;
}
.hbox.box-flex,
.vbox.box-flex,
.box-flex {
  /* Old browsers */
  /* Old browsers */
  -webkit-box-flex: 1;
  -moz-box-flex: 1;
  box-flex: 1;
  /* Modern browsers */
  flex: 1;
}
.hbox.box-flex2,
.vbox.box-flex2,
.box-flex2 {
  /* Old browsers */
  -webkit-box-flex: 2;
  -moz-box-flex: 2;
  box-flex: 2;
  /* Modern browsers */
  flex: 2;
}
.box-group1 {
  /*  Deprecated */
  -webkit-box-flex-group: 1;
  -moz-box-flex-group: 1;
  box-flex-group: 1;
}
.box-group2 {
  /* Deprecated */
  -webkit-box-flex-group: 2;
  -moz-box-flex-group: 2;
  box-flex-group: 2;
}
.hbox.start,
.vbox.start,
.start {
  /* Old browsers */
  -webkit-box-pack: start;
  -moz-box-pack: start;
  box-pack: start;
  /* Modern browsers */
  justify-content: flex-start;
}
.hbox.end,
.vbox.end,
.end {
  /* Old browsers */
  -webkit-box-pack: end;
  -moz-box-pack: end;
  box-pack: end;
  /* Modern browsers */
  justify-content: flex-end;
}
.hbox.center,
.vbox.center,
.center {
  /* Old browsers */
  -webkit-box-pack: center;
  -moz-box-pack: center;
  box-pack: center;
  /* Modern browsers */
  justify-content: center;
}
.hbox.baseline,
.vbox.baseline,
.baseline {
  /* Old browsers */
  -webkit-box-pack: baseline;
  -moz-box-pack: baseline;
  box-pack: baseline;
  /* Modern browsers */
  justify-content: baseline;
}
.hbox.stretch,
.vbox.stretch,
.stretch {
  /* Old browsers */
  -webkit-box-pack: stretch;
  -moz-box-pack: stretch;
  box-pack: stretch;
  /* Modern browsers */
  justify-content: stretch;
}
.hbox.align-start,
.vbox.align-start,
.align-start {
  /* Old browsers */
  -webkit-box-align: start;
  -moz-box-align: start;
  box-align: start;
  /* Modern browsers */
  align-items: flex-start;
}
.hbox.align-end,
.vbox.align-end,
.align-end {
  /* Old browsers */
  -webkit-box-align: end;
  -moz-box-align: end;
  box-align: end;
  /* Modern browsers */
  align-items: flex-end;
}
.hbox.align-center,
.vbox.align-center,
.align-center {
  /* Old browsers */
  -webkit-box-align: center;
  -moz-box-align: center;
  box-align: center;
  /* Modern browsers */
  align-items: center;
}
.hbox.align-baseline,
.vbox.align-baseline,
.align-baseline {
  /* Old browsers */
  -webkit-box-align: baseline;
  -moz-box-align: baseline;
  box-align: baseline;
  /* Modern browsers */
  align-items: baseline;
}
.hbox.align-stretch,
.vbox.align-stretch,
.align-stretch {
  /* Old browsers */
  -webkit-box-align: stretch;
  -moz-box-align: stretch;
  box-align: stretch;
  /* Modern browsers */
  align-items: stretch;
}
div.error {
  margin: 2em;
  text-align: center;
}
div.error > h1 {
  font-size: 500%;
  line-height: normal;
}
div.error > p {
  font-size: 200%;
  line-height: normal;
}
div.traceback-wrapper {
  text-align: left;
  max-width: 800px;
  margin: auto;
}
div.traceback-wrapper pre.traceback {
  max-height: 600px;
  overflow: auto;
}
/**
 * Primary styles
 *
 * Author: Jupyter Development Team
 */
body {
  background-color: #fff;
  /* This makes sure that the body covers the entire window and needs to
       be in a different element than the display: box in wrapper below */
  position: absolute;
  left: 0px;
  right: 0px;
  top: 0px;
  bottom: 0px;
  overflow: visible;
}
body > #header {
  /* Initially hidden to prevent FLOUC */
  display: none;
  background-color: #fff;
  /* Display over codemirror */
  position: relative;
  z-index: 100;
}
body > #header #header-container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 5px;
  padding-bottom: 5px;
  padding-top: 5px;
  box-sizing: border-box;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
}
body > #header .header-bar {
  width: 100%;
  height: 1px;
  background: #e7e7e7;
  margin-bottom: -1px;
}
@media print {
  body > #header {
    display: none !important;
  }
}
#header-spacer {
  width: 100%;
  visibility: hidden;
}
@media print {
  #header-spacer {
    display: none;
  }
}
#ipython_notebook {
  padding-left: 0px;
  padding-top: 1px;
  padding-bottom: 1px;
}
[dir="rtl"] #ipython_notebook {
  margin-right: 10px;
  margin-left: 0;
}
[dir="rtl"] #ipython_notebook.pull-left {
  float: right !important;
  float: right;
}
.flex-spacer {
  flex: 1;
}
#noscript {
  width: auto;
  padding-top: 16px;
  padding-bottom: 16px;
  text-align: center;
  font-size: 22px;
  color: red;
  font-weight: bold;
}
#ipython_notebook img {
  height: 28px;
}
#site {
  width: 100%;
  display: none;
  box-sizing: border-box;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  overflow: auto;
}
@media print {
  #site {
    height: auto !important;
  }
}
/* Smaller buttons */
.ui-button .ui-button-text {
  padding: 0.2em 0.8em;
  font-size: 77%;
}
input.ui-button {
  padding: 0.3em 0.9em;
}
span#kernel_logo_widget {
  margin: 0 10px;
}
span#login_widget {
  float: right;
}
[dir="rtl"] span#login_widget {
  float: left;
}
span#login_widget > .button,
#logout {
  color: #333;
  background-color: #fff;
  border-color: #ccc;
}
span#login_widget > .button:focus,
#logout:focus,
span#login_widget > .button.focus,
#logout.focus {
  color: #333;
  background-color: #e6e6e6;
  border-color: #8c8c8c;
}
span#login_widget > .button:hover,
#logout:hover {
  color: #333;
  background-color: #e6e6e6;
  border-color: #adadad;
}
span#login_widget > .button:active,
#logout:active,
span#login_widget > .button.active,
#logout.active,
.open > .dropdown-togglespan#login_widget > .button,
.open > .dropdown-toggle#logout {
  color: #333;
  background-color: #e6e6e6;
  border-color: #adadad;
}
span#login_widget > .button:active:hover,
#logout:active:hover,
span#login_widget > .button.active:hover,
#logout.active:hover,
.open > .dropdown-togglespan#login_widget > .button:hover,
.open > .dropdown-toggle#logout:hover,
span#login_widget > .button:active:focus,
#logout:active:focus,
span#login_widget > .button.active:focus,
#logout.active:focus,
.open > .dropdown-togglespan#login_widget > .button:focus,
.open > .dropdown-toggle#logout:focus,
span#login_widget > .button:active.focus,
#logout:active.focus,
span#login_widget > .button.active.focus,
#logout.active.focus,
.open > .dropdown-togglespan#login_widget > .button.focus,
.open > .dropdown-toggle#logout.focus {
  color: #333;
  background-color: #d4d4d4;
  border-color: #8c8c8c;
}
span#login_widget > .button:active,
#logout:active,
span#login_widget > .button.active,
#logout.active,
.open > .dropdown-togglespan#login_widget > .button,
.open > .dropdown-toggle#logout {
  background-image: none;
}
span#login_widget > .button.disabled:hover,
#logout.disabled:hover,
span#login_widget > .button[disabled]:hover,
#logout[disabled]:hover,
fieldset[disabled] span#login_widget > .button:hover,
fieldset[disabled] #logout:hover,
span#login_widget > .button.disabled:focus,
#logout.disabled:focus,
span#login_widget > .button[disabled]:focus,
#logout[disabled]:focus,
fieldset[disabled] span#login_widget > .button:focus,
fieldset[disabled] #logout:focus,
span#login_widget > .button.disabled.focus,
#logout.disabled.focus,
span#login_widget > .button[disabled].focus,
#logout[disabled].focus,
fieldset[disabled] span#login_widget > .button.focus,
fieldset[disabled] #logout.focus {
  background-color: #fff;
  border-color: #ccc;
}
span#login_widget > .button .badge,
#logout .badge {
  color: #fff;
  background-color: #333;
}
.nav-header {
  text-transform: none;
}
#header > span {
  margin-top: 10px;
}
.modal_stretch .modal-dialog {
  /* Old browsers */
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-box-align: stretch;
  display: -moz-box;
  -moz-box-orient: vertical;
  -moz-box-align: stretch;
  display: box;
  box-orient: vertical;
  box-align: stretch;
  /* Modern browsers */
  display: flex;
  flex-direction: column;
  align-items: stretch;
  min-height: 80vh;
}
.modal_stretch .modal-dialog .modal-body {
  max-height: calc(100vh - 200px);
  overflow: auto;
  flex: 1;
}
.modal-header {
  cursor: move;
}
@media (min-width: 768px) {
  .modal .modal-dialog {
    width: 700px;
  }
}
@media (min-width: 768px) {
  select.form-control {
    margin-left: 12px;
    margin-right: 12px;
  }
}
/*!
*
* IPython auth
*
*/
.center-nav {
  display: inline-block;
  margin-bottom: -4px;
}
[dir="rtl"] .center-nav form.pull-left {
  float: right !important;
  float: right;
}
[dir="rtl"] .center-nav .navbar-text {
  float: right;
}
[dir="rtl"] .navbar-inner {
  text-align: right;
}
[dir="rtl"] div.text-left {
  text-align: right;
}
/*!
*
* IPython tree view
*
*/
/* We need an invisible input field on top of the sentense*/
/* "Drag file onto the list ..." */
.alternate_upload {
  background-color: none;
  display: inline;
}
.alternate_upload.form {
  padding: 0;
  margin: 0;
}
.alternate_upload input.fileinput {
  position: absolute;
  display: block;
  width: 100%;
  height: 100%;
  overflow: hidden;
  cursor: pointer;
  opacity: 0;
  z-index: 2;
}
.alternate_upload .btn-xs > input.fileinput {
  margin: -1px -5px;
}
.alternate_upload .btn-upload {
  position: relative;
  height: 22px;
}
::-webkit-file-upload-button {
  cursor: pointer;
}
/**
 * Primary styles
 *
 * Author: Jupyter Development Team
 */
ul#tabs {
  margin-bottom: 4px;
}
ul#tabs a {
  padding-top: 6px;
  padding-bottom: 4px;
}
[dir="rtl"] ul#tabs.nav-tabs > li {
  float: right;
}
[dir="rtl"] ul#tabs.nav.nav-tabs {
  padding-right: 0;
}
ul.breadcrumb a:focus,
ul.breadcrumb a:hover {
  text-decoration: none;
}
ul.breadcrumb i.icon-home {
  font-size: 16px;
  margin-right: 4px;
}
ul.breadcrumb span {
  color: #5e5e5e;
}
.list_toolbar {
  padding: 4px 0 4px 0;
  vertical-align: middle;
}
.list_toolbar .tree-buttons {
  padding-top: 1px;
}
[dir="rtl"] .list_toolbar .tree-buttons .pull-right {
  float: left !important;
  float: left;
}
[dir="rtl"] .list_toolbar .col-sm-4,
[dir="rtl"] .list_toolbar .col-sm-8 {
  float: right;
}
.dynamic-buttons {
  padding-top: 3px;
  display: inline-block;
}
.list_toolbar [class*="span"] {
  min-height: 24px;
}
.list_header {
  font-weight: bold;
  background-color: #EEE;
}
.list_placeholder {
  font-weight: bold;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 7px;
  padding-right: 7px;
}
.list_container {
  margin-top: 4px;
  margin-bottom: 20px;
  border: 1px solid #ddd;
  border-radius: 2px;
}
.list_container > div {
  border-bottom: 1px solid #ddd;
}
.list_container > div:hover .list-item {
  background-color: red;
}
.list_container > div:last-child {
  border: none;
}
.list_item:hover .list_item {
  background-color: #ddd;
}
.list_item a {
  text-decoration: none;
}
.list_item:hover {
  background-color: #fafafa;
}
.list_header > div,
.list_item > div {
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 7px;
  padding-right: 7px;
  line-height: 22px;
}
.list_header > div input,
.list_item > div input {
  margin-right: 7px;
  margin-left: 14px;
  vertical-align: text-bottom;
  line-height: 22px;
  position: relative;
  top: -1px;
}
.list_header > div .item_link,
.list_item > div .item_link {
  margin-left: -1px;
  vertical-align: baseline;
  line-height: 22px;
}
[dir="rtl"] .list_item > div input {
  margin-right: 0;
}
.new-file input[type=checkbox] {
  visibility: hidden;
}
.item_name {
  line-height: 22px;
  height: 24px;
}
.item_icon {
  font-size: 14px;
  color: #5e5e5e;
  margin-right: 7px;
  margin-left: 7px;
  line-height: 22px;
  vertical-align: baseline;
}
.item_modified {
  margin-right: 7px;
  margin-left: 7px;
}
[dir="rtl"] .item_modified.pull-right {
  float: left !important;
  float: left;
}
.item_buttons {
  line-height: 1em;
  margin-left: -5px;
}
.item_buttons .btn,
.item_buttons .btn-group,
.item_buttons .input-group {
  float: left;
}
.item_buttons > .btn,
.item_buttons > .btn-group,
.item_buttons > .input-group {
  margin-left: 5px;
}
.item_buttons .btn {
  min-width: 13ex;
}
.item_buttons .running-indicator {
  padding-top: 4px;
  color: #5cb85c;
}
.item_buttons .kernel-name {
  padding-top: 4px;
  color: #5bc0de;
  margin-right: 7px;
  float: left;
}
[dir="rtl"] .item_buttons.pull-right {
  float: left !important;
  float: left;
}
[dir="rtl"] .item_buttons .kernel-name {
  margin-left: 7px;
  float: right;
}
.toolbar_info {
  height: 24px;
  line-height: 24px;
}
.list_item input:not([type=checkbox]) {
  padding-top: 3px;
  padding-bottom: 3px;
  height: 22px;
  line-height: 14px;
  margin: 0px;
}
.highlight_text {
  color: blue;
}
#project_name {
  display: inline-block;
  padding-left: 7px;
  margin-left: -2px;
}
#project_name > .breadcrumb {
  padding: 0px;
  margin-bottom: 0px;
  background-color: transparent;
  font-weight: bold;
}
.sort_button {
  display: inline-block;
  padding-left: 7px;
}
[dir="rtl"] .sort_button.pull-right {
  float: left !important;
  float: left;
}
#tree-selector {
  padding-right: 0px;
}
#button-select-all {
  min-width: 50px;
}
[dir="rtl"] #button-select-all.btn {
  float: right ;
}
#select-all {
  margin-left: 7px;
  margin-right: 2px;
  margin-top: 2px;
  height: 16px;
}
[dir="rtl"] #select-all.pull-left {
  float: right !important;
  float: right;
}
.menu_icon {
  margin-right: 2px;
}
.tab-content .row {
  margin-left: 0px;
  margin-right: 0px;
}
.folder_icon:before {
  display: inline-block;
  font: normal normal normal 14px/1 FontAwesome;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  content: "\f114";
}
.folder_icon:before.fa-pull-left {
  margin-right: .3em;
}
.folder_icon:before.fa-pull-right {
  margin-left: .3em;
}
.folder_icon:before.pull-left {
  margin-right: .3em;
}
.folder_icon:before.pull-right {
  margin-left: .3em;
}
.notebook_icon:before {
  display: inline-block;
  font: normal normal normal 14px/1 FontAwesome;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  content: "\f02d";
  position: relative;
  top: -1px;
}
.notebook_icon:before.fa-pull-left {
  margin-right: .3em;
}
.notebook_icon:before.fa-pull-right {
  margin-left: .3em;
}
.notebook_icon:before.pull-left {
  margin-right: .3em;
}
.notebook_icon:before.pull-right {
  margin-left: .3em;
}
.running_notebook_icon:before {
  display: inline-block;
  font: normal normal normal 14px/1 FontAwesome;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  content: "\f02d";
  position: relative;
  top: -1px;
  color: #5cb85c;
}
.running_notebook_icon:before.fa-pull-left {
  margin-right: .3em;
}
.running_notebook_icon:before.fa-pull-right {
  margin-left: .3em;
}
.running_notebook_icon:before.pull-left {
  margin-right: .3em;
}
.running_notebook_icon:before.pull-right {
  margin-left: .3em;
}
.file_icon:before {
  display: inline-block;
  font: normal normal normal 14px/1 FontAwesome;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  content: "\f016";
  position: relative;
  top: -2px;
}
.file_icon:before.fa-pull-left {
  margin-right: .3em;
}
.file_icon:before.fa-pull-right {
  margin-left: .3em;
}
.file_icon:before.pull-left {
  margin-right: .3em;
}
.file_icon:before.pull-right {
  margin-left: .3em;
}
#notebook_toolbar .pull-right {
  padding-top: 0px;
  margin-right: -1px;
}
ul#new-menu {
  left: auto;
  right: 0;
}
#new-menu .dropdown-header {
  font-size: 10px;
  border-bottom: 1px solid #e5e5e5;
  padding: 0 0 3px;
  margin: -3px 20px 0;
}
.kernel-menu-icon {
  padding-right: 12px;
  width: 24px;
  content: "\f096";
}
.kernel-menu-icon:before {
  content: "\f096";
}
.kernel-menu-icon-current:before {
  content: "\f00c";
}
#tab_content {
  padding-top: 20px;
}
#running .panel-group .panel {
  margin-top: 3px;
  margin-bottom: 1em;
}
#running .panel-group .panel .panel-heading {
  background-color: #EEE;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 7px;
  padding-right: 7px;
  line-height: 22px;
}
#running .panel-group .panel .panel-heading a:focus,
#running .panel-group .panel .panel-heading a:hover {
  text-decoration: none;
}
#running .panel-group .panel .panel-body {
  padding: 0px;
}
#running .panel-group .panel .panel-body .list_container {
  margin-top: 0px;
  margin-bottom: 0px;
  border: 0px;
  border-radius: 0px;
}
#running .panel-group .panel .panel-body .list_container .list_item {
  border-bottom: 1px solid #ddd;
}
#running .panel-group .panel .panel-body .list_container .list_item:last-child {
  border-bottom: 0px;
}
.delete-button {
  display: none;
}
.duplicate-button {
  display: none;
}
.rename-button {
  display: none;
}
.move-button {
  display: none;
}
.download-button {
  display: none;
}
.shutdown-button {
  display: none;
}
.dynamic-instructions {
  display: inline-block;
  padding-top: 4px;
}
/*!
*
* IPython text editor webapp
*
*/
.selected-keymap i.fa {
  padding: 0px 5px;
}
.selected-keymap i.fa:before {
  content: "\f00c";
}
#mode-menu {
  overflow: auto;
  max-height: 20em;
}
.edit_app #header {
  -webkit-box-shadow: 0px 0px 12px 1px rgba(87, 87, 87, 0.2);
  box-shadow: 0px 0px 12px 1px rgba(87, 87, 87, 0.2);
}
.edit_app #menubar .navbar {
  /* Use a negative 1 bottom margin, so the border overlaps the border of the
    header */
  margin-bottom: -1px;
}
.dirty-indicator {
  display: inline-block;
  font: normal normal normal 14px/1 FontAwesome;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 20px;
}
.dirty-indicator.fa-pull-left {
  margin-right: .3em;
}
.dirty-indicator.fa-pull-right {
  margin-left: .3em;
}
.dirty-indicator.pull-left {
  margin-right: .3em;
}
.dirty-indicator.pull-right {
  margin-left: .3em;
}
.dirty-indicator-dirty {
  display: inline-block;
  font: normal normal normal 14px/1 FontAwesome;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 20px;
}
.dirty-indicator-dirty.fa-pull-left {
  margin-right: .3em;
}
.dirty-indicator-dirty.fa-pull-right {
  margin-left: .3em;
}
.dirty-indicator-dirty.pull-left {
  margin-right: .3em;
}
.dirty-indicator-dirty.pull-right {
  margin-left: .3em;
}
.dirty-indicator-clean {
  display: inline-block;
  font: normal normal normal 14px/1 FontAwesome;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 20px;
}
.dirty-indicator-clean.fa-pull-left {
  margin-right: .3em;
}
.dirty-indicator-clean.fa-pull-right {
  margin-left: .3em;
}
.dirty-indicator-clean.pull-left {
  margin-right: .3em;
}
.dirty-indicator-clean.pull-right {
  margin-left: .3em;
}
.dirty-indicator-clean:before {
  display: inline-block;
  font: normal normal normal 14px/1 FontAwesome;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  content: "\f00c";
}
.dirty-indicator-clean:before.fa-pull-left {
  margin-right: .3em;
}
.dirty-indicator-clean:before.fa-pull-right {
  margin-left: .3em;
}
.dirty-indicator-clean:before.pull-left {
  margin-right: .3em;
}
.dirty-indicator-clean:before.pull-right {
  margin-left: .3em;
}
#filename {
  font-size: 16pt;
  display: table;
  padding: 0px 5px;
}
#current-mode {
  padding-left: 5px;
  padding-right: 5px;
}
#texteditor-backdrop {
  padding-top: 20px;
  padding-bottom: 20px;
}
@media not print {
  #texteditor-backdrop {
    background-color: #EEE;
  }
}
@media print {
  #texteditor-backdrop #texteditor-container .CodeMirror-gutter,
  #texteditor-backdrop #texteditor-container .CodeMirror-gutters {
    background-color: #fff;
  }
}
@media not print {
  #texteditor-backdrop #texteditor-container .CodeMirror-gutter,
  #texteditor-backdrop #texteditor-container .CodeMirror-gutters {
    background-color: #fff;
  }
}
@media not print {
  #texteditor-backdrop #texteditor-container {
    padding: 0px;
    background-color: #fff;
    -webkit-box-shadow: 0px 0px 12px 1px rgba(87, 87, 87, 0.2);
    box-shadow: 0px 0px 12px 1px rgba(87, 87, 87, 0.2);
  }
}
.CodeMirror-dialog {
  background-color: #fff;
}
/*!
*
* IPython notebook
*
*/
/* CSS font colors for translated ANSI escape sequences */
/* The color values are a mix of
   http://www.xcolors.net/dl/baskerville-ivorylight and
   http://www.xcolors.net/dl/euphrasia */
.ansi-black-fg {
  color: #3E424D;
}
.ansi-black-bg {
  background-color: #3E424D;
}
.ansi-black-intense-fg {
  color: #282C36;
}
.ansi-black-intense-bg {
  background-color: #282C36;
}
.ansi-red-fg {
  color: #E75C58;
}
.ansi-red-bg {
  background-color: #E75C58;
}
.ansi-red-intense-fg {
  color: #B22B31;
}
.ansi-red-intense-bg {
  background-color: #B22B31;
}
.ansi-green-fg {
  color: #00A250;
}
.ansi-green-bg {
  background-color: #00A250;
}
.ansi-green-intense-fg {
  color: #007427;
}
.ansi-green-intense-bg {
  background-color: #007427;
}
.ansi-yellow-fg {
  color: #DDB62B;
}
.ansi-yellow-bg {
  background-color: #DDB62B;
}
.ansi-yellow-intense-fg {
  color: #B27D12;
}
.ansi-yellow-intense-bg {
  background-color: #B27D12;
}
.ansi-blue-fg {
  color: #208FFB;
}
.ansi-blue-bg {
  background-color: #208FFB;
}
.ansi-blue-intense-fg {
  color: #0065CA;
}
.ansi-blue-intense-bg {
  background-color: #0065CA;
}
.ansi-magenta-fg {
  color: #D160C4;
}
.ansi-magenta-bg {
  background-color: #D160C4;
}
.ansi-magenta-intense-fg {
  color: #A03196;
}
.ansi-magenta-intense-bg {
  background-color: #A03196;
}
.ansi-cyan-fg {
  color: #60C6C8;
}
.ansi-cyan-bg {
  background-color: #60C6C8;
}
.ansi-cyan-intense-fg {
  color: #258F8F;
}
.ansi-cyan-intense-bg {
  background-color: #258F8F;
}
.ansi-white-fg {
  color: #C5C1B4;
}
.ansi-white-bg {
  background-color: #C5C1B4;
}
.ansi-white-intense-fg {
  color: #A1A6B2;
}
.ansi-white-intense-bg {
  background-color: #A1A6B2;
}
.ansi-default-inverse-fg {
  color: #FFFFFF;
}
.ansi-default-inverse-bg {
  background-color: #000000;
}
.ansi-bold {
  font-weight: bold;
}
.ansi-underline {
  text-decoration: underline;
}
/* The following styles are deprecated an will be removed in a future version */
.ansibold {
  font-weight: bold;
}
.ansi-inverse {
  outline: 0.5px dotted;
}
/* use dark versions for foreground, to improve visibility */
.ansiblack {
  color: black;
}
.ansired {
  color: darkred;
}
.ansigreen {
  color: darkgreen;
}
.ansiyellow {
  color: #c4a000;
}
.ansiblue {
  color: darkblue;
}
.ansipurple {
  color: darkviolet;
}
.ansicyan {
  color: steelblue;
}
.ansigray {
  color: gray;
}
/* and light for background, for the same reason */
.ansibgblack {
  background-color: black;
}
.ansibgred {
  background-color: red;
}
.ansibggreen {
  background-color: green;
}
.ansibgyellow {
  background-color: yellow;
}
.ansibgblue {
  background-color: blue;
}
.ansibgpurple {
  background-color: magenta;
}
.ansibgcyan {
  background-color: cyan;
}
.ansibggray {
  background-color: gray;
}
div.cell {
  /* Old browsers */
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-box-align: stretch;
  display: -moz-box;
  -moz-box-orient: vertical;
  -moz-box-align: stretch;
  display: box;
  box-orient: vertical;
  box-align: stretch;
  /* Modern browsers */
  display: flex;
  flex-direction: column;
  align-items: stretch;
  border-radius: 2px;
  box-sizing: border-box;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  border-width: 1px;
  border-style: solid;
  border-color: transparent;
  width: 100%;
  padding: 5px;
  /* This acts as a spacer between cells, that is outside the border */
  margin: 0px;
  outline: none;
  position: relative;
  overflow: visible;
}
div.cell:before {
  position: absolute;
  display: block;
  top: -1px;
  left: -1px;
  width: 5px;
  height: calc(100% +  2px);
  content: '';
  background: transparent;
}
div.cell.jupyter-soft-selected {
  border-left-color: #E3F2FD;
  border-left-width: 1px;
  padding-left: 5px;
  border-right-color: #E3F2FD;
  border-right-width: 1px;
  background: #E3F2FD;
}
@media print {
  div.cell.jupyter-soft-selected {
    border-color: transparent;
  }
}
div.cell.selected,
div.cell.selected.jupyter-soft-selected {
  border-color: #ababab;
}
div.cell.selected:before,
div.cell.selected.jupyter-soft-selected:before {
  position: absolute;
  display: block;
  top: -1px;
  left: -1px;
  width: 5px;
  height: calc(100% +  2px);
  content: '';
  background: #42A5F5;
}
@media print {
  div.cell.selected,
  div.cell.selected.jupyter-soft-selected {
    border-color: transparent;
  }
}
.edit_mode div.cell.selected {
  border-color: #66BB6A;
}
.edit_mode div.cell.selected:before {
  position: absolute;
  display: block;
  top: -1px;
  left: -1px;
  width: 5px;
  height: calc(100% +  2px);
  content: '';
  background: #66BB6A;
}
@media print {
  .edit_mode div.cell.selected {
    border-color: transparent;
  }
}
.prompt {
  /* This needs to be wide enough for 3 digit prompt numbers: In[100]: */
  min-width: 14ex;
  /* This padding is tuned to match the padding on the CodeMirror editor. */
  padding: 0.4em;
  margin: 0px;
  font-family: monospace;
  text-align: right;
  /* This has to match that of the the CodeMirror class line-height below */
  line-height: 1.21429em;
  /* Don't highlight prompt number selection */
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -khtml-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  /* Use default cursor */
  cursor: default;
}
@media (max-width: 540px) {
  .prompt {
    text-align: left;
  }
}
div.inner_cell {
  min-width: 0;
  /* Old browsers */
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-box-align: stretch;
  display: -moz-box;
  -moz-box-orient: vertical;
  -moz-box-align: stretch;
  display: box;
  box-orient: vertical;
  box-align: stretch;
  /* Modern browsers */
  display: flex;
  flex-direction: column;
  align-items: stretch;
  /* Old browsers */
  -webkit-box-flex: 1;
  -moz-box-flex: 1;
  box-flex: 1;
  /* Modern browsers */
  flex: 1;
}
/* input_area and input_prompt must match in top border and margin for alignment */
div.input_area {
  border: 1px solid #cfcfcf;
  border-radius: 2px;
  background: #f7f7f7;
  line-height: 1.21429em;
}
/* This is needed so that empty prompt areas can collapse to zero height when there
   is no content in the output_subarea and the prompt. The main purpose of this is
   to make sure that empty JavaScript output_subareas have no height. */
div.prompt:empty {
  padding-top: 0;
  padding-bottom: 0;
}
div.unrecognized_cell {
  padding: 5px 5px 5px 0px;
  /* Old browsers */
  display: -webkit-box;
  -webkit-box-orient: horizontal;
  -webkit-box-align: stretch;
  display: -moz-box;
  -moz-box-orient: horizontal;
  -moz-box-align: stretch;
  display: box;
  box-orient: horizontal;
  box-align: stretch;
  /* Modern browsers */
  display: flex;
  flex-direction: row;
  align-items: stretch;
}
div.unrecognized_cell .inner_cell {
  border-radius: 2px;
  padding: 5px;
  font-weight: bold;
  color: red;
  border: 1px solid #cfcfcf;
  background: #eaeaea;
}
div.unrecognized_cell .inner_cell a {
  color: inherit;
  text-decoration: none;
}
div.unrecognized_cell .inner_cell a:hover {
  color: inherit;
  text-decoration: none;
}
@media (max-width: 540px) {
  div.unrecognized_cell > div.prompt {
    display: none;
  }
}
div.code_cell {
  /* avoid page breaking on code cells when printing */
}
@media print {
  div.code_cell {
    page-break-inside: avoid;
  }
}
/* any special styling for code cells that are currently running goes here */
div.input {
  page-break-inside: avoid;
  /* Old browsers */
  display: -webkit-box;
  -webkit-box-orient: horizontal;
  -webkit-box-align: stretch;
  display: -moz-box;
  -moz-box-orient: horizontal;
  -moz-box-align: stretch;
  display: box;
  box-orient: horizontal;
  box-align: stretch;
  /* Modern browsers */
  display: flex;
  flex-direction: row;
  align-items: stretch;
}
@media (max-width: 540px) {
  div.input {
    /* Old browsers */
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-box-align: stretch;
    display: -moz-box;
    -moz-box-orient: vertical;
    -moz-box-align: stretch;
    display: box;
    box-orient: vertical;
    box-align: stretch;
    /* Modern browsers */
    display: flex;
    flex-direction: column;
    align-items: stretch;
  }
}
/* input_area and input_prompt must match in top border and margin for alignment */
div.input_prompt {
  color: #303F9F;
  border-top: 1px solid transparent;
}
div.input_area > div.highlight {
  margin: 0.4em;
  border: none;
  padding: 0px;
  background-color: transparent;
}
div.input_area > div.highlight > pre {
  margin: 0px;
  border: none;
  padding: 0px;
  background-color: transparent;
}
/* The following gets added to the <head> if it is detected that the user has a
 * monospace font with inconsistent normal/bold/italic height.  See
 * notebookmain.js.  Such fonts will have keywords vertically offset with
 * respect to the rest of the text.  The user should select a better font.
 * See: https://github.com/ipython/ipython/issues/1503
 *
 * .CodeMirror span {
 *      vertical-align: bottom;
 * }
 */
.CodeMirror {
  line-height: 1.21429em;
  /* Changed from 1em to our global default */
  font-size: 14px;
  height: auto;
  /* Changed to auto to autogrow */
  background: none;
  /* Changed from white to allow our bg to show through */
}
.CodeMirror-scroll {
  /*  The CodeMirror docs are a bit fuzzy on if overflow-y should be hidden or visible.*/
  /*  We have found that if it is visible, vertical scrollbars appear with font size changes.*/
  overflow-y: hidden;
  overflow-x: auto;
}
.CodeMirror-lines {
  /* In CM2, this used to be 0.4em, but in CM3 it went to 4px. We need the em value because */
  /* we have set a different line-height and want this to scale with that. */
  /* Note that this should set vertical padding only, since CodeMirror assumes
       that horizontal padding will be set on CodeMirror pre */
  padding: 0.4em 0;
}
.CodeMirror-linenumber {
  padding: 0 8px 0 4px;
}
.CodeMirror-gutters {
  border-bottom-left-radius: 2px;
  border-top-left-radius: 2px;
}
.CodeMirror pre {
  /* In CM3 this went to 4px from 0 in CM2. This sets horizontal padding only,
    use .CodeMirror-lines for vertical */
  padding: 0 0.4em;
  border: 0;
  border-radius: 0;
}
.CodeMirror-cursor {
  border-left: 1.4px solid black;
}
@media screen and (min-width: 2138px) and (max-width: 4319px) {
  .CodeMirror-cursor {
    border-left: 2px solid black;
  }
}
@media screen and (min-width: 4320px) {
  .CodeMirror-cursor {
    border-left: 4px solid black;
  }
}
/*

Original style from softwaremaniacs.org (c) Ivan Sagalaev <Maniac@SoftwareManiacs.Org>
Adapted from GitHub theme

*/
.highlight-base {
  color: #000;
}
.highlight-variable {
  color: #000;
}
.highlight-variable-2 {
  color: #1a1a1a;
}
.highlight-variable-3 {
  color: #333333;
}
.highlight-string {
  color: #BA2121;
}
.highlight-comment {
  color: #408080;
  font-style: italic;
}
.highlight-number {
  color: #080;
}
.highlight-atom {
  color: #88F;
}
.highlight-keyword {
  color: #008000;
  font-weight: bold;
}
.highlight-builtin {
  color: #008000;
}
.highlight-error {
  color: #f00;
}
.highlight-operator {
  color: #AA22FF;
  font-weight: bold;
}
.highlight-meta {
  color: #AA22FF;
}
/* previously not defined, copying from default codemirror */
.highlight-def {
  color: #00f;
}
.highlight-string-2 {
  color: #f50;
}
.highlight-qualifier {
  color: #555;
}
.highlight-bracket {
  color: #997;
}
.highlight-tag {
  color: #170;
}
.highlight-attribute {
  color: #00c;
}
.highlight-header {
  color: blue;
}
.highlight-quote {
  color: #090;
}
.highlight-link {
  color: #00c;
}
/* apply the same style to codemirror */
.cm-s-ipython span.cm-keyword {
  color: #008000;
  font-weight: bold;
}
.cm-s-ipython span.cm-atom {
  color: #88F;
}
.cm-s-ipython span.cm-number {
  color: #080;
}
.cm-s-ipython span.cm-def {
  color: #00f;
}
.cm-s-ipython span.cm-variable {
  color: #000;
}
.cm-s-ipython span.cm-operator {
  color: #AA22FF;
  font-weight: bold;
}
.cm-s-ipython span.cm-variable-2 {
  color: #1a1a1a;
}
.cm-s-ipython span.cm-variable-3 {
  color: #333333;
}
.cm-s-ipython span.cm-comment {
  color: #408080;
  font-style: italic;
}
.cm-s-ipython span.cm-string {
  color: #BA2121;
}
.cm-s-ipython span.cm-string-2 {
  color: #f50;
}
.cm-s-ipython span.cm-meta {
  color: #AA22FF;
}
.cm-s-ipython span.cm-qualifier {
  color: #555;
}
.cm-s-ipython span.cm-builtin {
  color: #008000;
}
.cm-s-ipython span.cm-bracket {
  color: #997;
}
.cm-s-ipython span.cm-tag {
  color: #170;
}
.cm-s-ipython span.cm-attribute {
  color: #00c;
}
.cm-s-ipython span.cm-header {
  color: blue;
}
.cm-s-ipython span.cm-quote {
  color: #090;
}
.cm-s-ipython span.cm-link {
  color: #00c;
}
.cm-s-ipython span.cm-error {
  color: #f00;
}
.cm-s-ipython span.cm-tab {
  background: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADAAAAAMCAYAAAAkuj5RAAAAAXNSR0IArs4c6QAAAGFJREFUSMft1LsRQFAQheHPowAKoACx3IgEKtaEHujDjORSgWTH/ZOdnZOcM/sgk/kFFWY0qV8foQwS4MKBCS3qR6ixBJvElOobYAtivseIE120FaowJPN75GMu8j/LfMwNjh4HUpwg4LUAAAAASUVORK5CYII=);
  background-position: right;
  background-repeat: no-repeat;
}
div.output_wrapper {
  /* this position must be relative to enable descendents to be absolute within it */
  position: relative;
  /* Old browsers */
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-box-align: stretch;
  display: -moz-box;
  -moz-box-orient: vertical;
  -moz-box-align: stretch;
  display: box;
  box-orient: vertical;
  box-align: stretch;
  /* Modern browsers */
  display: flex;
  flex-direction: column;
  align-items: stretch;
  z-index: 1;
}
/* class for the output area when it should be height-limited */
div.output_scroll {
  /* ideally, this would be max-height, but FF barfs all over that */
  height: 24em;
  /* FF needs this *and the wrapper* to specify full width, or it will shrinkwrap */
  width: 100%;
  overflow: auto;
  border-radius: 2px;
  -webkit-box-shadow: inset 0 2px 8px rgba(0, 0, 0, 0.8);
  box-shadow: inset 0 2px 8px rgba(0, 0, 0, 0.8);
  display: block;
}
/* output div while it is collapsed */
div.output_collapsed {
  margin: 0px;
  padding: 0px;
  /* Old browsers */
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-box-align: stretch;
  display: -moz-box;
  -moz-box-orient: vertical;
  -moz-box-align: stretch;
  display: box;
  box-orient: vertical;
  box-align: stretch;
  /* Modern browsers */
  display: flex;
  flex-direction: column;
  align-items: stretch;
}
div.out_prompt_overlay {
  height: 100%;
  padding: 0px 0.4em;
  position: absolute;
  border-radius: 2px;
}
div.out_prompt_overlay:hover {
  /* use inner shadow to get border that is computed the same on WebKit/FF */
  -webkit-box-shadow: inset 0 0 1px #000;
  box-shadow: inset 0 0 1px #000;
  background: rgba(240, 240, 240, 0.5);
}
div.output_prompt {
  color: #D84315;
}
/* This class is the outer container of all output sections. */
div.output_area {
  padding: 0px;
  page-break-inside: avoid;
  /* Old browsers */
  display: -webkit-box;
  -webkit-box-orient: horizontal;
  -webkit-box-align: stretch;
  display: -moz-box;
  -moz-box-orient: horizontal;
  -moz-box-align: stretch;
  display: box;
  box-orient: horizontal;
  box-align: stretch;
  /* Modern browsers */
  display: flex;
  flex-direction: row;
  align-items: stretch;
}
div.output_area .MathJax_Display {
  text-align: left !important;
}
div.output_area .rendered_html table {
  margin-left: 0;
  margin-right: 0;
}
div.output_area .rendered_html img {
  margin-left: 0;
  margin-right: 0;
}
div.output_area img,
div.output_area svg {
  max-width: 100%;
  height: auto;
}
div.output_area img.unconfined,
div.output_area svg.unconfined {
  max-width: none;
}
div.output_area .mglyph > img {
  max-width: none;
}
/* This is needed to protect the pre formating from global settings such
   as that of bootstrap */
.output {
  /* Old browsers */
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-box-align: stretch;
  display: -moz-box;
  -moz-box-orient: vertical;
  -moz-box-align: stretch;
  display: box;
  box-orient: vertical;
  box-align: stretch;
  /* Modern browsers */
  display: flex;
  flex-direction: column;
  align-items: stretch;
}
@media (max-width: 540px) {
  div.output_area {
    /* Old browsers */
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-box-align: stretch;
    display: -moz-box;
    -moz-box-orient: vertical;
    -moz-box-align: stretch;
    display: box;
    box-orient: vertical;
    box-align: stretch;
    /* Modern browsers */
    display: flex;
    flex-direction: column;
    align-items: stretch;
  }
}
div.output_area pre {
  margin: 0;
  padding: 1px 0 1px 0;
  border: 0;
  vertical-align: baseline;
  color: black;
  background-color: transparent;
  border-radius: 0;
}
/* This class is for the output subarea inside the output_area and after
   the prompt div. */
div.output_subarea {
  overflow-x: auto;
  padding: 0.4em;
  /* Old browsers */
  -webkit-box-flex: 1;
  -moz-box-flex: 1;
  box-flex: 1;
  /* Modern browsers */
  flex: 1;
  max-width: calc(100% - 14ex);
}
div.output_scroll div.output_subarea {
  overflow-x: visible;
}
/* The rest of the output_* classes are for special styling of the different
   output types */
/* all text output has this class: */
div.output_text {
  text-align: left;
  color: #000;
  /* This has to match that of the the CodeMirror class line-height below */
  line-height: 1.21429em;
}
/* stdout/stderr are 'text' as well as 'stream', but execute_result/error are *not* streams */
div.output_stderr {
  background: #fdd;
  /* very light red background for stderr */
}
div.output_latex {
  text-align: left;
}
/* Empty output_javascript divs should have no height */
div.output_javascript:empty {
  padding: 0;
}
.js-error {
  color: darkred;
}
/* raw_input styles */
div.raw_input_container {
  line-height: 1.21429em;
  padding-top: 5px;
}
pre.raw_input_prompt {
  /* nothing needed here. */
}
input.raw_input {
  font-family: monospace;
  font-size: inherit;
  color: inherit;
  width: auto;
  /* make sure input baseline aligns with prompt */
  vertical-align: baseline;
  /* padding + margin = 0.5em between prompt and cursor */
  padding: 0em 0.25em;
  margin: 0em 0.25em;
}
input.raw_input:focus {
  box-shadow: none;
}
p.p-space {
  margin-bottom: 10px;
}
div.output_unrecognized {
  padding: 5px;
  font-weight: bold;
  color: red;
}
div.output_unrecognized a {
  color: inherit;
  text-decoration: none;
}
div.output_unrecognized a:hover {
  color: inherit;
  text-decoration: none;
}
.rendered_html {
  color: #000;
  /* any extras will just be numbers: */
}
.rendered_html em {
  font-style: italic;
}
.rendered_html strong {
  font-weight: bold;
}
.rendered_html u {
  text-decoration: underline;
}
.rendered_html :link {
  text-decoration: underline;
}
.rendered_html :visited {
  text-decoration: underline;
}
.rendered_html h1 {
  font-size: 185.7%;
  margin: 1.08em 0 0 0;
  font-weight: bold;
  line-height: 1.0;
}
.rendered_html h2 {
  font-size: 157.1%;
  margin: 1.27em 0 0 0;
  font-weight: bold;
  line-height: 1.0;
}
.rendered_html h3 {
  font-size: 128.6%;
  margin: 1.55em 0 0 0;
  font-weight: bold;
  line-height: 1.0;
}
.rendered_html h4 {
  font-size: 100%;
  margin: 2em 0 0 0;
  font-weight: bold;
  line-height: 1.0;
}
.rendered_html h5 {
  font-size: 100%;
  margin: 2em 0 0 0;
  font-weight: bold;
  line-height: 1.0;
  font-style: italic;
}
.rendered_html h6 {
  font-size: 100%;
  margin: 2em 0 0 0;
  font-weight: bold;
  line-height: 1.0;
  font-style: italic;
}
.rendered_html h1:first-child {
  margin-top: 0.538em;
}
.rendered_html h2:first-child {
  margin-top: 0.636em;
}
.rendered_html h3:first-child {
  margin-top: 0.777em;
}
.rendered_html h4:first-child {
  margin-top: 1em;
}
.rendered_html h5:first-child {
  margin-top: 1em;
}
.rendered_html h6:first-child {
  margin-top: 1em;
}
.rendered_html ul:not(.list-inline),
.rendered_html ol:not(.list-inline) {
  padding-left: 2em;
}
.rendered_html ul {
  list-style: disc;
}
.rendered_html ul ul {
  list-style: square;
  margin-top: 0;
}
.rendered_html ul ul ul {
  list-style: circle;
}
.rendered_html ol {
  list-style: decimal;
}
.rendered_html ol ol {
  list-style: upper-alpha;
  margin-top: 0;
}
.rendered_html ol ol ol {
  list-style: lower-alpha;
}
.rendered_html ol ol ol ol {
  list-style: lower-roman;
}
.rendered_html ol ol ol ol ol {
  list-style: decimal;
}
.rendered_html * + ul {
  margin-top: 1em;
}
.rendered_html * + ol {
  margin-top: 1em;
}
.rendered_html hr {
  color: black;
  background-color: black;
}
.rendered_html pre {
  margin: 1em 2em;
  padding: 0px;
  background-color: #fff;
}
.rendered_html code {
  background-color: #eff0f1;
}
.rendered_html p code {
  padding: 1px 5px;
}
.rendered_html pre code {
  background-color: #fff;
}
.rendered_html pre,
.rendered_html code {
  border: 0;
  color: #000;
  font-size: 100%;
}
.rendered_html blockquote {
  margin: 1em 2em;
}
.rendered_html table {
  margin-left: auto;
  margin-right: auto;
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
.rendered_html tr,
.rendered_html th,
.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
.rendered_html th {
  font-weight: bold;
}
.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}
.rendered_html * + table {
  margin-top: 1em;
}
.rendered_html p {
  text-align: left;
}
.rendered_html * + p {
  margin-top: 1em;
}
.rendered_html img {
  display: block;
  margin-left: auto;
  margin-right: auto;
}
.rendered_html * + img {
  margin-top: 1em;
}
.rendered_html img,
.rendered_html svg {
  max-width: 100%;
  height: auto;
}
.rendered_html img.unconfined,
.rendered_html svg.unconfined {
  max-width: none;
}
.rendered_html .alert {
  margin-bottom: initial;
}
.rendered_html * + .alert {
  margin-top: 1em;
}
[dir="rtl"] .rendered_html p {
  text-align: right;
}
div.text_cell {
  /* Old browsers */
  display: -webkit-box;
  -webkit-box-orient: horizontal;
  -webkit-box-align: stretch;
  display: -moz-box;
  -moz-box-orient: horizontal;
  -moz-box-align: stretch;
  display: box;
  box-orient: horizontal;
  box-align: stretch;
  /* Modern browsers */
  display: flex;
  flex-direction: row;
  align-items: stretch;
}
@media (max-width: 540px) {
  div.text_cell > div.prompt {
    display: none;
  }
}
div.text_cell_render {
  /*font-family: "Helvetica Neue", Arial, Helvetica, Geneva, sans-serif;*/
  outline: none;
  resize: none;
  width: inherit;
  border-style: none;
  padding: 0.5em 0.5em 0.5em 0.4em;
  color: #000;
  box-sizing: border-box;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
}
a.anchor-link:link {
  text-decoration: none;
  padding: 0px 20px;
  visibility: hidden;
}
h1:hover .anchor-link,
h2:hover .anchor-link,
h3:hover .anchor-link,
h4:hover .anchor-link,
h5:hover .anchor-link,
h6:hover .anchor-link {
  visibility: visible;
}
.text_cell.rendered .input_area {
  display: none;
}
.text_cell.rendered .rendered_html {
  overflow-x: auto;
  overflow-y: hidden;
}
.text_cell.rendered .rendered_html tr,
.text_cell.rendered .rendered_html th,
.text_cell.rendered .rendered_html td {
  max-width: none;
}
.text_cell.unrendered .text_cell_render {
  display: none;
}
.text_cell .dropzone .input_area {
  border: 2px dashed #bababa;
  margin: -1px;
}
.cm-header-1,
.cm-header-2,
.cm-header-3,
.cm-header-4,
.cm-header-5,
.cm-header-6 {
  font-weight: bold;
  font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
}
.cm-header-1 {
  font-size: 185.7%;
}
.cm-header-2 {
  font-size: 157.1%;
}
.cm-header-3 {
  font-size: 128.6%;
}
.cm-header-4 {
  font-size: 110%;
}
.cm-header-5 {
  font-size: 100%;
  font-style: italic;
}
.cm-header-6 {
  font-size: 100%;
  font-style: italic;
}
/*!
*
* IPython notebook webapp
*
*/
@media (max-width: 767px) {
  .notebook_app {
    padding-left: 0px;
    padding-right: 0px;
  }
}
#ipython-main-app {
  box-sizing: border-box;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  height: 100%;
}
div#notebook_panel {
  margin: 0px;
  padding: 0px;
  box-sizing: border-box;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  height: 100%;
}
div#notebook {
  font-size: 14px;
  line-height: 20px;
  overflow-y: hidden;
  overflow-x: auto;
  width: 100%;
  /* This spaces the page away from the edge of the notebook area */
  padding-top: 20px;
  margin: 0px;
  outline: none;
  box-sizing: border-box;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  min-height: 100%;
}
@media not print {
  #notebook-container {
    padding: 15px;
    background-color: #fff;
    min-height: 0;
    -webkit-box-shadow: 0px 0px 12px 1px rgba(87, 87, 87, 0.2);
    box-shadow: 0px 0px 12px 1px rgba(87, 87, 87, 0.2);
  }
}
@media print {
  #notebook-container {
    width: 100%;
  }
}
div.ui-widget-content {
  border: 1px solid #ababab;
  outline: none;
}
pre.dialog {
  background-color: #f7f7f7;
  border: 1px solid #ddd;
  border-radius: 2px;
  padding: 0.4em;
  padding-left: 2em;
}
p.dialog {
  padding: 0.2em;
}
/* Word-wrap output correctly.  This is the CSS3 spelling, though Firefox seems
   to not honor it correctly.  Webkit browsers (Chrome, rekonq, Safari) do.
 */
pre,
code,
kbd,
samp {
  white-space: pre-wrap;
}
#fonttest {
  font-family: monospace;
}
p {
  margin-bottom: 0;
}
.end_space {
  min-height: 100px;
  transition: height .2s ease;
}
.notebook_app > #header {
  -webkit-box-shadow: 0px 0px 12px 1px rgba(87, 87, 87, 0.2);
  box-shadow: 0px 0px 12px 1px rgba(87, 87, 87, 0.2);
}
@media not print {
  .notebook_app {
    background-color: #EEE;
  }
}
kbd {
  border-style: solid;
  border-width: 1px;
  box-shadow: none;
  margin: 2px;
  padding-left: 2px;
  padding-right: 2px;
  padding-top: 1px;
  padding-bottom: 1px;
}
.jupyter-keybindings {
  padding: 1px;
  line-height: 24px;
  border-bottom: 1px solid gray;
}
.jupyter-keybindings input {
  margin: 0;
  padding: 0;
  border: none;
}
.jupyter-keybindings i {
  padding: 6px;
}
.well code {
  background-color: #ffffff;
  border-color: #ababab;
  border-width: 1px;
  border-style: solid;
  padding: 2px;
  padding-top: 1px;
  padding-bottom: 1px;
}
/* CSS for the cell toolbar */
.celltoolbar {
  border: thin solid #CFCFCF;
  border-bottom: none;
  background: #EEE;
  border-radius: 2px 2px 0px 0px;
  width: 100%;
  height: 29px;
  padding-right: 4px;
  /* Old browsers */
  display: -webkit-box;
  -webkit-box-orient: horizontal;
  -webkit-box-align: stretch;
  display: -moz-box;
  -moz-box-orient: horizontal;
  -moz-box-align: stretch;
  display: box;
  box-orient: horizontal;
  box-align: stretch;
  /* Modern browsers */
  display: flex;
  flex-direction: row;
  align-items: stretch;
  /* Old browsers */
  -webkit-box-pack: end;
  -moz-box-pack: end;
  box-pack: end;
  /* Modern browsers */
  justify-content: flex-end;
  display: -webkit-flex;
}
@media print {
  .celltoolbar {
    display: none;
  }
}
.ctb_hideshow {
  display: none;
  vertical-align: bottom;
}
/* ctb_show is added to the ctb_hideshow div to show the cell toolbar.
   Cell toolbars are only shown when the ctb_global_show class is also set.
*/
.ctb_global_show .ctb_show.ctb_hideshow {
  display: block;
}
.ctb_global_show .ctb_show + .input_area,
.ctb_global_show .ctb_show + div.text_cell_input,
.ctb_global_show .ctb_show ~ div.text_cell_render {
  border-top-right-radius: 0px;
  border-top-left-radius: 0px;
}
.ctb_global_show .ctb_show ~ div.text_cell_render {
  border: 1px solid #cfcfcf;
}
.celltoolbar {
  font-size: 87%;
  padding-top: 3px;
}
.celltoolbar select {
  display: block;
  width: 100%;
  height: 32px;
  padding: 6px 12px;
  font-size: 13px;
  line-height: 1.42857143;
  color: #555555;
  background-color: #fff;
  background-image: none;
  border: 1px solid #ccc;
  border-radius: 2px;
  -webkit-box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  -webkit-transition: border-color ease-in-out .15s, box-shadow ease-in-out .15s;
  -o-transition: border-color ease-in-out .15s, box-shadow ease-in-out .15s;
  transition: border-color ease-in-out .15s, box-shadow ease-in-out .15s;
  height: 30px;
  padding: 5px 10px;
  font-size: 12px;
  line-height: 1.5;
  border-radius: 1px;
  width: inherit;
  font-size: inherit;
  height: 22px;
  padding: 0px;
  display: inline-block;
}
.celltoolbar select:focus {
  border-color: #66afe9;
  outline: 0;
  -webkit-box-shadow: inset 0 1px 1px rgba(0,0,0,.075), 0 0 8px rgba(102, 175, 233, 0.6);
  box-shadow: inset 0 1px 1px rgba(0,0,0,.075), 0 0 8px rgba(102, 175, 233, 0.6);
}
.celltoolbar select::-moz-placeholder {
  color: #999;
  opacity: 1;
}
.celltoolbar select:-ms-input-placeholder {
  color: #999;
}
.celltoolbar select::-webkit-input-placeholder {
  color: #999;
}
.celltoolbar select::-ms-expand {
  border: 0;
  background-color: transparent;
}
.celltoolbar select[disabled],
.celltoolbar select[readonly],
fieldset[disabled] .celltoolbar select {
  background-color: #eeeeee;
  opacity: 1;
}
.celltoolbar select[disabled],
fieldset[disabled] .celltoolbar select {
  cursor: not-allowed;
}
textarea.celltoolbar select {
  height: auto;
}
select.celltoolbar select {
  height: 30px;
  line-height: 30px;
}
textarea.celltoolbar select,
select[multiple].celltoolbar select {
  height: auto;
}
.celltoolbar label {
  margin-left: 5px;
  margin-right: 5px;
}
.tags_button_container {
  width: 100%;
  display: flex;
}
.tag-container {
  display: flex;
  flex-direction: row;
  flex-grow: 1;
  overflow: hidden;
  position: relative;
}
.tag-container > * {
  margin: 0 4px;
}
.remove-tag-btn {
  margin-left: 4px;
}
.tags-input {
  display: flex;
}
.cell-tag:last-child:after {
  content: "";
  position: absolute;
  right: 0;
  width: 40px;
  height: 100%;
  /* Fade to background color of cell toolbar */
  background: linear-gradient(to right, rgba(0, 0, 0, 0), #EEE);
}
.tags-input > * {
  margin-left: 4px;
}
.cell-tag,
.tags-input input,
.tags-input button {
  display: block;
  width: 100%;
  height: 32px;
  padding: 6px 12px;
  font-size: 13px;
  line-height: 1.42857143;
  color: #555555;
  background-color: #fff;
  background-image: none;
  border: 1px solid #ccc;
  border-radius: 2px;
  -webkit-box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  -webkit-transition: border-color ease-in-out .15s, box-shadow ease-in-out .15s;
  -o-transition: border-color ease-in-out .15s, box-shadow ease-in-out .15s;
  transition: border-color ease-in-out .15s, box-shadow ease-in-out .15s;
  height: 30px;
  padding: 5px 10px;
  font-size: 12px;
  line-height: 1.5;
  border-radius: 1px;
  box-shadow: none;
  width: inherit;
  font-size: inherit;
  height: 22px;
  line-height: 22px;
  padding: 0px 4px;
  display: inline-block;
}
.cell-tag:focus,
.tags-input input:focus,
.tags-input button:focus {
  border-color: #66afe9;
  outline: 0;
  -webkit-box-shadow: inset 0 1px 1px rgba(0,0,0,.075), 0 0 8px rgba(102, 175, 233, 0.6);
  box-shadow: inset 0 1px 1px rgba(0,0,0,.075), 0 0 8px rgba(102, 175, 233, 0.6);
}
.cell-tag::-moz-placeholder,
.tags-input input::-moz-placeholder,
.tags-input button::-moz-placeholder {
  color: #999;
  opacity: 1;
}
.cell-tag:-ms-input-placeholder,
.tags-input input:-ms-input-placeholder,
.tags-input button:-ms-input-placeholder {
  color: #999;
}
.cell-tag::-webkit-input-placeholder,
.tags-input input::-webkit-input-placeholder,
.tags-input button::-webkit-input-placeholder {
  color: #999;
}
.cell-tag::-ms-expand,
.tags-input input::-ms-expand,
.tags-input button::-ms-expand {
  border: 0;
  background-color: transparent;
}
.cell-tag[disabled],
.tags-input input[disabled],
.tags-input button[disabled],
.cell-tag[readonly],
.tags-input input[readonly],
.tags-input button[readonly],
fieldset[disabled] .cell-tag,
fieldset[disabled] .tags-input input,
fieldset[disabled] .tags-input button {
  background-color: #eeeeee;
  opacity: 1;
}
.cell-tag[disabled],
.tags-input input[disabled],
.tags-input button[disabled],
fieldset[disabled] .cell-tag,
fieldset[disabled] .tags-input input,
fieldset[disabled] .tags-input button {
  cursor: not-allowed;
}
textarea.cell-tag,
textarea.tags-input input,
textarea.tags-input button {
  height: auto;
}
select.cell-tag,
select.tags-input input,
select.tags-input button {
  height: 30px;
  line-height: 30px;
}
textarea.cell-tag,
textarea.tags-input input,
textarea.tags-input button,
select[multiple].cell-tag,
select[multiple].tags-input input,
select[multiple].tags-input button {
  height: auto;
}
.cell-tag,
.tags-input button {
  padding: 0px 4px;
}
.cell-tag {
  background-color: #fff;
  white-space: nowrap;
}
.tags-input input[type=text]:focus {
  outline: none;
  box-shadow: none;
  border-color: #ccc;
}
.completions {
  position: absolute;
  z-index: 110;
  overflow: hidden;
  border: 1px solid #ababab;
  border-radius: 2px;
  -webkit-box-shadow: 0px 6px 10px -1px #adadad;
  box-shadow: 0px 6px 10px -1px #adadad;
  line-height: 1;
}
.completions select {
  background: white;
  outline: none;
  border: none;
  padding: 0px;
  margin: 0px;
  overflow: auto;
  font-family: monospace;
  font-size: 110%;
  color: #000;
  width: auto;
}
.completions select option.context {
  color: #286090;
}
#kernel_logo_widget .current_kernel_logo {
  display: none;
  margin-top: -1px;
  margin-bottom: -1px;
  width: 32px;
  height: 32px;
}
[dir="rtl"] #kernel_logo_widget {
  float: left !important;
  float: left;
}
.modal .modal-body .move-path {
  display: flex;
  flex-direction: row;
  justify-content: space;
  align-items: center;
}
.modal .modal-body .move-path .server-root {
  padding-right: 20px;
}
.modal .modal-body .move-path .path-input {
  flex: 1;
}
#menubar {
  box-sizing: border-box;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  margin-top: 1px;
}
#menubar .navbar {
  border-top: 1px;
  border-radius: 0px 0px 2px 2px;
  margin-bottom: 0px;
}
#menubar .navbar-toggle {
  float: left;
  padding-top: 7px;
  padding-bottom: 7px;
  border: none;
}
#menubar .navbar-collapse {
  clear: left;
}
[dir="rtl"] #menubar .navbar-toggle {
  float: right;
}
[dir="rtl"] #menubar .navbar-collapse {
  clear: right;
}
[dir="rtl"] #menubar .navbar-nav {
  float: right;
}
[dir="rtl"] #menubar .nav {
  padding-right: 0px;
}
[dir="rtl"] #menubar .navbar-nav > li {
  float: right;
}
[dir="rtl"] #menubar .navbar-right {
  float: left !important;
}
[dir="rtl"] ul.dropdown-menu {
  text-align: right;
  left: auto;
}
[dir="rtl"] ul#new-menu.dropdown-menu {
  right: auto;
  left: 0;
}
.nav-wrapper {
  border-bottom: 1px solid #e7e7e7;
}
i.menu-icon {
  padding-top: 4px;
}
[dir="rtl"] i.menu-icon.pull-right {
  float: left !important;
  float: left;
}
ul#help_menu li a {
  overflow: hidden;
  padding-right: 2.2em;
}
ul#help_menu li a i {
  margin-right: -1.2em;
}
[dir="rtl"] ul#help_menu li a {
  padding-left: 2.2em;
}
[dir="rtl"] ul#help_menu li a i {
  margin-right: 0;
  margin-left: -1.2em;
}
[dir="rtl"] ul#help_menu li a i.pull-right {
  float: left !important;
  float: left;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu > .dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
}
[dir="rtl"] .dropdown-submenu > .dropdown-menu {
  right: 100%;
  margin-right: -1px;
}
.dropdown-submenu:hover > .dropdown-menu {
  display: block;
}
.dropdown-submenu > a:after {
  display: inline-block;
  font: normal normal normal 14px/1 FontAwesome;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  display: block;
  content: "\f0da";
  float: right;
  color: #333333;
  margin-top: 2px;
  margin-right: -10px;
}
.dropdown-submenu > a:after.fa-pull-left {
  margin-right: .3em;
}
.dropdown-submenu > a:after.fa-pull-right {
  margin-left: .3em;
}
.dropdown-submenu > a:after.pull-left {
  margin-right: .3em;
}
.dropdown-submenu > a:after.pull-right {
  margin-left: .3em;
}
[dir="rtl"] .dropdown-submenu > a:after {
  float: left;
  content: "\f0d9";
  margin-right: 0;
  margin-left: -10px;
}
.dropdown-submenu:hover > a:after {
  color: #262626;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left > .dropdown-menu {
  left: -100%;
  margin-left: 10px;
}
#notification_area {
  float: right !important;
  float: right;
  z-index: 10;
}
[dir="rtl"] #notification_area {
  float: left !important;
  float: left;
}
.indicator_area {
  float: right !important;
  float: right;
  color: #777;
  margin-left: 5px;
  margin-right: 5px;
  width: 11px;
  z-index: 10;
  text-align: center;
  width: auto;
}
[dir="rtl"] .indicator_area {
  float: left !important;
  float: left;
}
#kernel_indicator {
  float: right !important;
  float: right;
  color: #777;
  margin-left: 5px;
  margin-right: 5px;
  width: 11px;
  z-index: 10;
  text-align: center;
  width: auto;
  border-left: 1px solid;
}
#kernel_indicator .kernel_indicator_name {
  padding-left: 5px;
  padding-right: 5px;
}
[dir="rtl"] #kernel_indicator {
  float: left !important;
  float: left;
  border-left: 0;
  border-right: 1px solid;
}
#modal_indicator {
  float: right !important;
  float: right;
  color: #777;
  margin-left: 5px;
  margin-right: 5px;
  width: 11px;
  z-index: 10;
  text-align: center;
  width: auto;
}
[dir="rtl"] #modal_indicator {
  float: left !important;
  float: left;
}
#readonly-indicator {
  float: right !important;
  float: right;
  color: #777;
  margin-left: 5px;
  margin-right: 5px;
  width: 11px;
  z-index: 10;
  text-align: center;
  width: auto;
  margin-top: 2px;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 0px;
  display: none;
}
.modal_indicator:before {
  width: 1.28571429em;
  text-align: center;
}
.edit_mode .modal_indicator:before {
  display: inline-block;
  font: normal normal normal 14px/1 FontAwesome;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  content: "\f040";
}
.edit_mode .modal_indicator:before.fa-pull-left {
  margin-right: .3em;
}
.edit_mode .modal_indicator:before.fa-pull-right {
  margin-left: .3em;
}
.edit_mode .modal_indicator:before.pull-left {
  margin-right: .3em;
}
.edit_mode .modal_indicator:before.pull-right {
  margin-left: .3em;
}
.command_mode .modal_indicator:before {
  display: inline-block;
  font: normal normal normal 14px/1 FontAwesome;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  content: ' ';
}
.command_mode .modal_indicator:before.fa-pull-left {
  margin-right: .3em;
}
.command_mode .modal_indicator:before.fa-pull-right {
  margin-left: .3em;
}
.command_mode .modal_indicator:before.pull-left {
  margin-right: .3em;
}
.command_mode .modal_indicator:before.pull-right {
  margin-left: .3em;
}
.kernel_idle_icon:before {
  display: inline-block;
  font: normal normal normal 14px/1 FontAwesome;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  content: "\f10c";
}
.kernel_idle_icon:before.fa-pull-left {
  margin-right: .3em;
}
.kernel_idle_icon:before.fa-pull-right {
  margin-left: .3em;
}
.kernel_idle_icon:before.pull-left {
  margin-right: .3em;
}
.kernel_idle_icon:before.pull-right {
  margin-left: .3em;
}
.kernel_busy_icon:before {
  display: inline-block;
  font: normal normal normal 14px/1 FontAwesome;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  content: "\f111";
}
.kernel_busy_icon:before.fa-pull-left {
  margin-right: .3em;
}
.kernel_busy_icon:before.fa-pull-right {
  margin-left: .3em;
}
.kernel_busy_icon:before.pull-left {
  margin-right: .3em;
}
.kernel_busy_icon:before.pull-right {
  margin-left: .3em;
}
.kernel_dead_icon:before {
  display: inline-block;
  font: normal normal normal 14px/1 FontAwesome;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  content: "\f1e2";
}
.kernel_dead_icon:before.fa-pull-left {
  margin-right: .3em;
}
.kernel_dead_icon:before.fa-pull-right {
  margin-left: .3em;
}
.kernel_dead_icon:before.pull-left {
  margin-right: .3em;
}
.kernel_dead_icon:before.pull-right {
  margin-left: .3em;
}
.kernel_disconnected_icon:before {
  display: inline-block;
  font: normal normal normal 14px/1 FontAwesome;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  content: "\f127";
}
.kernel_disconnected_icon:before.fa-pull-left {
  margin-right: .3em;
}
.kernel_disconnected_icon:before.fa-pull-right {
  margin-left: .3em;
}
.kernel_disconnected_icon:before.pull-left {
  margin-right: .3em;
}
.kernel_disconnected_icon:before.pull-right {
  margin-left: .3em;
}
.notification_widget {
  color: #777;
  z-index: 10;
  background: rgba(240, 240, 240, 0.5);
  margin-right: 4px;
  color: #333;
  background-color: #fff;
  border-color: #ccc;
}
.notification_widget:focus,
.notification_widget.focus {
  color: #333;
  background-color: #e6e6e6;
  border-color: #8c8c8c;
}
.notification_widget:hover {
  color: #333;
  background-color: #e6e6e6;
  border-color: #adadad;
}
.notification_widget:active,
.notification_widget.active,
.open > .dropdown-toggle.notification_widget {
  color: #333;
  background-color: #e6e6e6;
  border-color: #adadad;
}
.notification_widget:active:hover,
.notification_widget.active:hover,
.open > .dropdown-toggle.notification_widget:hover,
.notification_widget:active:focus,
.notification_widget.active:focus,
.open > .dropdown-toggle.notification_widget:focus,
.notification_widget:active.focus,
.notification_widget.active.focus,
.open > .dropdown-toggle.notification_widget.focus {
  color: #333;
  background-color: #d4d4d4;
  border-color: #8c8c8c;
}
.notification_widget:active,
.notification_widget.active,
.open > .dropdown-toggle.notification_widget {
  background-image: none;
}
.notification_widget.disabled:hover,
.notification_widget[disabled]:hover,
fieldset[disabled] .notification_widget:hover,
.notification_widget.disabled:focus,
.notification_widget[disabled]:focus,
fieldset[disabled] .notification_widget:focus,
.notification_widget.disabled.focus,
.notification_widget[disabled].focus,
fieldset[disabled] .notification_widget.focus {
  background-color: #fff;
  border-color: #ccc;
}
.notification_widget .badge {
  color: #fff;
  background-color: #333;
}
.notification_widget.warning {
  color: #fff;
  background-color: #f0ad4e;
  border-color: #eea236;
}
.notification_widget.warning:focus,
.notification_widget.warning.focus {
  color: #fff;
  background-color: #ec971f;
  border-color: #985f0d;
}
.notification_widget.warning:hover {
  color: #fff;
  background-color: #ec971f;
  border-color: #d58512;
}
.notification_widget.warning:active,
.notification_widget.warning.active,
.open > .dropdown-toggle.notification_widget.warning {
  color: #fff;
  background-color: #ec971f;
  border-color: #d58512;
}
.notification_widget.warning:active:hover,
.notification_widget.warning.active:hover,
.open > .dropdown-toggle.notification_widget.warning:hover,
.notification_widget.warning:active:focus,
.notification_widget.warning.active:focus,
.open > .dropdown-toggle.notification_widget.warning:focus,
.notification_widget.warning:active.focus,
.notification_widget.warning.active.focus,
.open > .dropdown-toggle.notification_widget.warning.focus {
  color: #fff;
  background-color: #d58512;
  border-color: #985f0d;
}
.notification_widget.warning:active,
.notification_widget.warning.active,
.open > .dropdown-toggle.notification_widget.warning {
  background-image: none;
}
.notification_widget.warning.disabled:hover,
.notification_widget.warning[disabled]:hover,
fieldset[disabled] .notification_widget.warning:hover,
.notification_widget.warning.disabled:focus,
.notification_widget.warning[disabled]:focus,
fieldset[disabled] .notification_widget.warning:focus,
.notification_widget.warning.disabled.focus,
.notification_widget.warning[disabled].focus,
fieldset[disabled] .notification_widget.warning.focus {
  background-color: #f0ad4e;
  border-color: #eea236;
}
.notification_widget.warning .badge {
  color: #f0ad4e;
  background-color: #fff;
}
.notification_widget.success {
  color: #fff;
  background-color: #5cb85c;
  border-color: #4cae4c;
}
.notification_widget.success:focus,
.notification_widget.success.focus {
  color: #fff;
  background-color: #449d44;
  border-color: #255625;
}
.notification_widget.success:hover {
  color: #fff;
  background-color: #449d44;
  border-color: #398439;
}
.notification_widget.success:active,
.notification_widget.success.active,
.open > .dropdown-toggle.notification_widget.success {
  color: #fff;
  background-color: #449d44;
  border-color: #398439;
}
.notification_widget.success:active:hover,
.notification_widget.success.active:hover,
.open > .dropdown-toggle.notification_widget.success:hover,
.notification_widget.success:active:focus,
.notification_widget.success.active:focus,
.open > .dropdown-toggle.notification_widget.success:focus,
.notification_widget.success:active.focus,
.notification_widget.success.active.focus,
.open > .dropdown-toggle.notification_widget.success.focus {
  color: #fff;
  background-color: #398439;
  border-color: #255625;
}
.notification_widget.success:active,
.notification_widget.success.active,
.open > .dropdown-toggle.notification_widget.success {
  background-image: none;
}
.notification_widget.success.disabled:hover,
.notification_widget.success[disabled]:hover,
fieldset[disabled] .notification_widget.success:hover,
.notification_widget.success.disabled:focus,
.notification_widget.success[disabled]:focus,
fieldset[disabled] .notification_widget.success:focus,
.notification_widget.success.disabled.focus,
.notification_widget.success[disabled].focus,
fieldset[disabled] .notification_widget.success.focus {
  background-color: #5cb85c;
  border-color: #4cae4c;
}
.notification_widget.success .badge {
  color: #5cb85c;
  background-color: #fff;
}
.notification_widget.info {
  color: #fff;
  background-color: #5bc0de;
  border-color: #46b8da;
}
.notification_widget.info:focus,
.notification_widget.info.focus {
  color: #fff;
  background-color: #31b0d5;
  border-color: #1b6d85;
}
.notification_widget.info:hover {
  color: #fff;
  background-color: #31b0d5;
  border-color: #269abc;
}
.notification_widget.info:active,
.notification_widget.info.active,
.open > .dropdown-toggle.notification_widget.info {
  color: #fff;
  background-color: #31b0d5;
  border-color: #269abc;
}
.notification_widget.info:active:hover,
.notification_widget.info.active:hover,
.open > .dropdown-toggle.notification_widget.info:hover,
.notification_widget.info:active:focus,
.notification_widget.info.active:focus,
.open > .dropdown-toggle.notification_widget.info:focus,
.notification_widget.info:active.focus,
.notification_widget.info.active.focus,
.open > .dropdown-toggle.notification_widget.info.focus {
  color: #fff;
  background-color: #269abc;
  border-color: #1b6d85;
}
.notification_widget.info:active,
.notification_widget.info.active,
.open > .dropdown-toggle.notification_widget.info {
  background-image: none;
}
.notification_widget.info.disabled:hover,
.notification_widget.info[disabled]:hover,
fieldset[disabled] .notification_widget.info:hover,
.notification_widget.info.disabled:focus,
.notification_widget.info[disabled]:focus,
fieldset[disabled] .notification_widget.info:focus,
.notification_widget.info.disabled.focus,
.notification_widget.info[disabled].focus,
fieldset[disabled] .notification_widget.info.focus {
  background-color: #5bc0de;
  border-color: #46b8da;
}
.notification_widget.info .badge {
  color: #5bc0de;
  background-color: #fff;
}
.notification_widget.danger {
  color: #fff;
  background-color: #d9534f;
  border-color: #d43f3a;
}
.notification_widget.danger:focus,
.notification_widget.danger.focus {
  color: #fff;
  background-color: #c9302c;
  border-color: #761c19;
}
.notification_widget.danger:hover {
  color: #fff;
  background-color: #c9302c;
  border-color: #ac2925;
}
.notification_widget.danger:active,
.notification_widget.danger.active,
.open > .dropdown-toggle.notification_widget.danger {
  color: #fff;
  background-color: #c9302c;
  border-color: #ac2925;
}
.notification_widget.danger:active:hover,
.notification_widget.danger.active:hover,
.open > .dropdown-toggle.notification_widget.danger:hover,
.notification_widget.danger:active:focus,
.notification_widget.danger.active:focus,
.open > .dropdown-toggle.notification_widget.danger:focus,
.notification_widget.danger:active.focus,
.notification_widget.danger.active.focus,
.open > .dropdown-toggle.notification_widget.danger.focus {
  color: #fff;
  background-color: #ac2925;
  border-color: #761c19;
}
.notification_widget.danger:active,
.notification_widget.danger.active,
.open > .dropdown-toggle.notification_widget.danger {
  background-image: none;
}
.notification_widget.danger.disabled:hover,
.notification_widget.danger[disabled]:hover,
fieldset[disabled] .notification_widget.danger:hover,
.notification_widget.danger.disabled:focus,
.notification_widget.danger[disabled]:focus,
fieldset[disabled] .notification_widget.danger:focus,
.notification_widget.danger.disabled.focus,
.notification_widget.danger[disabled].focus,
fieldset[disabled] .notification_widget.danger.focus {
  background-color: #d9534f;
  border-color: #d43f3a;
}
.notification_widget.danger .badge {
  color: #d9534f;
  background-color: #fff;
}
div#pager {
  background-color: #fff;
  font-size: 14px;
  line-height: 20px;
  overflow: hidden;
  display: none;
  position: fixed;
  bottom: 0px;
  width: 100%;
  max-height: 50%;
  padding-top: 8px;
  -webkit-box-shadow: 0px 0px 12px 1px rgba(87, 87, 87, 0.2);
  box-shadow: 0px 0px 12px 1px rgba(87, 87, 87, 0.2);
  /* Display over codemirror */
  z-index: 100;
  /* Hack which prevents jquery ui resizable from changing top. */
  top: auto !important;
}
div#pager pre {
  line-height: 1.21429em;
  color: #000;
  background-color: #f7f7f7;
  padding: 0.4em;
}
div#pager #pager-button-area {
  position: absolute;
  top: 8px;
  right: 20px;
}
div#pager #pager-contents {
  position: relative;
  overflow: auto;
  width: 100%;
  height: 100%;
}
div#pager #pager-contents #pager-container {
  position: relative;
  padding: 15px 0px;
  box-sizing: border-box;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
}
div#pager .ui-resizable-handle {
  top: 0px;
  height: 8px;
  background: #f7f7f7;
  border-top: 1px solid #cfcfcf;
  border-bottom: 1px solid #cfcfcf;
  /* This injects handle bars (a short, wide = symbol) for 
        the resize handle. */
}
div#pager .ui-resizable-handle::after {
  content: '';
  top: 2px;
  left: 50%;
  height: 3px;
  width: 30px;
  margin-left: -15px;
  position: absolute;
  border-top: 1px solid #cfcfcf;
}
.quickhelp {
  /* Old browsers */
  display: -webkit-box;
  -webkit-box-orient: horizontal;
  -webkit-box-align: stretch;
  display: -moz-box;
  -moz-box-orient: horizontal;
  -moz-box-align: stretch;
  display: box;
  box-orient: horizontal;
  box-align: stretch;
  /* Modern browsers */
  display: flex;
  flex-direction: row;
  align-items: stretch;
  line-height: 1.8em;
}
.shortcut_key {
  display: inline-block;
  width: 21ex;
  text-align: right;
  font-family: monospace;
}
.shortcut_descr {
  display: inline-block;
  /* Old browsers */
  -webkit-box-flex: 1;
  -moz-box-flex: 1;
  box-flex: 1;
  /* Modern browsers */
  flex: 1;
}
span.save_widget {
  height: 30px;
  margin-top: 4px;
  display: flex;
  justify-content: flex-start;
  align-items: baseline;
  width: 50%;
  flex: 1;
}
span.save_widget span.filename {
  height: 100%;
  line-height: 1em;
  margin-left: 16px;
  border: none;
  font-size: 146.5%;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  border-radius: 2px;
}
span.save_widget span.filename:hover {
  background-color: #e6e6e6;
}
[dir="rtl"] span.save_widget.pull-left {
  float: right !important;
  float: right;
}
[dir="rtl"] span.save_widget span.filename {
  margin-left: 0;
  margin-right: 16px;
}
span.checkpoint_status,
span.autosave_status {
  font-size: small;
  white-space: nowrap;
  padding: 0 5px;
}
@media (max-width: 767px) {
  span.save_widget {
    font-size: small;
    padding: 0 0 0 5px;
  }
  span.checkpoint_status,
  span.autosave_status {
    display: none;
  }
}
@media (min-width: 768px) and (max-width: 991px) {
  span.checkpoint_status {
    display: none;
  }
  span.autosave_status {
    font-size: x-small;
  }
}
.toolbar {
  padding: 0px;
  margin-left: -5px;
  margin-top: 2px;
  margin-bottom: 5px;
  box-sizing: border-box;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
}
.toolbar select,
.toolbar label {
  width: auto;
  vertical-align: middle;
  margin-right: 2px;
  margin-bottom: 0px;
  display: inline;
  font-size: 92%;
  margin-left: 0.3em;
  margin-right: 0.3em;
  padding: 0px;
  padding-top: 3px;
}
.toolbar .btn {
  padding: 2px 8px;
}
.toolbar .btn-group {
  margin-top: 0px;
  margin-left: 5px;
}
.toolbar-btn-label {
  margin-left: 6px;
}
#maintoolbar {
  margin-bottom: -3px;
  margin-top: -8px;
  border: 0px;
  min-height: 27px;
  margin-left: 0px;
  padding-top: 11px;
  padding-bottom: 3px;
}
#maintoolbar .navbar-text {
  float: none;
  vertical-align: middle;
  text-align: right;
  margin-left: 5px;
  margin-right: 0px;
  margin-top: 0px;
}
.select-xs {
  height: 24px;
}
[dir="rtl"] .btn-group > .btn,
.btn-group-vertical > .btn {
  float: right;
}
.pulse,
.dropdown-menu > li > a.pulse,
li.pulse > a.dropdown-toggle,
li.pulse.open > a.dropdown-toggle {
  background-color: #F37626;
  color: white;
}
/**
 * Primary styles
 *
 * Author: Jupyter Development Team
 */
/** WARNING IF YOU ARE EDITTING THIS FILE, if this is a .css file, It has a lot
 * of chance of beeing generated from the ../less/[samename].less file, you can
 * try to get back the less file by reverting somme commit in history
 **/
/*
 * We'll try to get something pretty, so we
 * have some strange css to have the scroll bar on
 * the left with fix button on the top right of the tooltip
 */
@-moz-keyframes fadeOut {
  from {
    opacity: 1;
  }
  to {
    opacity: 0;
  }
}
@-webkit-keyframes fadeOut {
  from {
    opacity: 1;
  }
  to {
    opacity: 0;
  }
}
@-moz-keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}
@-webkit-keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}
/*properties of tooltip after "expand"*/
.bigtooltip {
  overflow: auto;
  height: 200px;
  -webkit-transition-property: height;
  -webkit-transition-duration: 500ms;
  -moz-transition-property: height;
  -moz-transition-duration: 500ms;
  transition-property: height;
  transition-duration: 500ms;
}
/*properties of tooltip before "expand"*/
.smalltooltip {
  -webkit-transition-property: height;
  -webkit-transition-duration: 500ms;
  -moz-transition-property: height;
  -moz-transition-duration: 500ms;
  transition-property: height;
  transition-duration: 500ms;
  text-overflow: ellipsis;
  overflow: hidden;
  height: 80px;
}
.tooltipbuttons {
  position: absolute;
  padding-right: 15px;
  top: 0px;
  right: 0px;
}
.tooltiptext {
  /*avoid the button to overlap on some docstring*/
  padding-right: 30px;
}
.ipython_tooltip {
  max-width: 700px;
  /*fade-in animation when inserted*/
  -webkit-animation: fadeOut 400ms;
  -moz-animation: fadeOut 400ms;
  animation: fadeOut 400ms;
  -webkit-animation: fadeIn 400ms;
  -moz-animation: fadeIn 400ms;
  animation: fadeIn 400ms;
  vertical-align: middle;
  background-color: #f7f7f7;
  overflow: visible;
  border: #ababab 1px solid;
  outline: none;
  padding: 3px;
  margin: 0px;
  padding-left: 7px;
  font-family: monospace;
  min-height: 50px;
  -moz-box-shadow: 0px 6px 10px -1px #adadad;
  -webkit-box-shadow: 0px 6px 10px -1px #adadad;
  box-shadow: 0px 6px 10px -1px #adadad;
  border-radius: 2px;
  position: absolute;
  z-index: 1000;
}
.ipython_tooltip a {
  float: right;
}
.ipython_tooltip .tooltiptext pre {
  border: 0;
  border-radius: 0;
  font-size: 100%;
  background-color: #f7f7f7;
}
.pretooltiparrow {
  left: 0px;
  margin: 0px;
  top: -16px;
  width: 40px;
  height: 16px;
  overflow: hidden;
  position: absolute;
}
.pretooltiparrow:before {
  background-color: #f7f7f7;
  border: 1px #ababab solid;
  z-index: 11;
  content: "";
  position: absolute;
  left: 15px;
  top: 10px;
  width: 25px;
  height: 25px;
  -webkit-transform: rotate(45deg);
  -moz-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  -o-transform: rotate(45deg);
}
ul.typeahead-list i {
  margin-left: -10px;
  width: 18px;
}
[dir="rtl"] ul.typeahead-list i {
  margin-left: 0;
  margin-right: -10px;
}
ul.typeahead-list {
  max-height: 80vh;
  overflow: auto;
}
ul.typeahead-list > li > a {
  /** Firefox bug **/
  /* see https://github.com/jupyter/notebook/issues/559 */
  white-space: normal;
}
ul.typeahead-list  > li > a.pull-right {
  float: left !important;
  float: left;
}
[dir="rtl"] .typeahead-list {
  text-align: right;
}
.cmd-palette .modal-body {
  padding: 7px;
}
.cmd-palette form {
  background: white;
}
.cmd-palette input {
  outline: none;
}
.no-shortcut {
  min-width: 20px;
  color: transparent;
}
[dir="rtl"] .no-shortcut.pull-right {
  float: left !important;
  float: left;
}
[dir="rtl"] .command-shortcut.pull-right {
  float: left !important;
  float: left;
}
.command-shortcut:before {
  content: "(command mode)";
  padding-right: 3px;
  color: #777777;
}
.edit-shortcut:before {
  content: "(edit)";
  padding-right: 3px;
  color: #777777;
}
[dir="rtl"] .edit-shortcut.pull-right {
  float: left !important;
  float: left;
}
#find-and-replace #replace-preview .match,
#find-and-replace #replace-preview .insert {
  background-color: #BBDEFB;
  border-color: #90CAF9;
  border-style: solid;
  border-width: 1px;
  border-radius: 0px;
}
[dir="ltr"] #find-and-replace .input-group-btn + .form-control {
  border-left: none;
}
[dir="rtl"] #find-and-replace .input-group-btn + .form-control {
  border-right: none;
}
#find-and-replace #replace-preview .replace .match {
  background-color: #FFCDD2;
  border-color: #EF9A9A;
  border-radius: 0px;
}
#find-and-replace #replace-preview .replace .insert {
  background-color: #C8E6C9;
  border-color: #A5D6A7;
  border-radius: 0px;
}
#find-and-replace #replace-preview {
  max-height: 60vh;
  overflow: auto;
}
#find-and-replace #replace-preview pre {
  padding: 5px 10px;
}
.terminal-app {
  background: #EEE;
}
.terminal-app #header {
  background: #fff;
  -webkit-box-shadow: 0px 0px 12px 1px rgba(87, 87, 87, 0.2);
  box-shadow: 0px 0px 12px 1px rgba(87, 87, 87, 0.2);
}
.terminal-app .terminal {
  width: 100%;
  float: left;
  font-family: monospace;
  color: white;
  background: black;
  padding: 0.4em;
  border-radius: 2px;
  -webkit-box-shadow: 0px 0px 12px 1px rgba(87, 87, 87, 0.4);
  box-shadow: 0px 0px 12px 1px rgba(87, 87, 87, 0.4);
}
.terminal-app .terminal,
.terminal-app .terminal dummy-screen {
  line-height: 1em;
  font-size: 14px;
}
.terminal-app .terminal .xterm-rows {
  padding: 10px;
}
.terminal-app .terminal-cursor {
  color: black;
  background: white;
}
.terminal-app #terminado-container {
  margin-top: 20px;
}
/*# sourceMappingURL=style.min.css.map */
    </style>
<style type="text/css">
    pre { line-height: 125%; }
td.linenos .normal { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }
span.linenos { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }
td.linenos .special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }
span.linenos.special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }
.highlight .hll { background-color: #ffffcc }
.highlight { background: #f8f8f8; }
.highlight .c { color: #408080; font-style: italic } /* Comment */
.highlight .err { border: 1px solid #FF0000 } /* Error */
.highlight .k { color: #008000; font-weight: bold } /* Keyword */
.highlight .o { color: #666666 } /* Operator */
.highlight .ch { color: #408080; font-style: italic } /* Comment.Hashbang */
.highlight .cm { color: #408080; font-style: italic } /* Comment.Multiline */
.highlight .cp { color: #BC7A00 } /* Comment.Preproc */
.highlight .cpf { color: #408080; font-style: italic } /* Comment.PreprocFile */
.highlight .c1 { color: #408080; font-style: italic } /* Comment.Single */
.highlight .cs { color: #408080; font-style: italic } /* Comment.Special */
.highlight .gd { color: #A00000 } /* Generic.Deleted */
.highlight .ge { font-style: italic } /* Generic.Emph */
.highlight .gr { color: #FF0000 } /* Generic.Error */
.highlight .gh { color: #000080; font-weight: bold } /* Generic.Heading */
.highlight .gi { color: #00A000 } /* Generic.Inserted */
.highlight .go { color: #888888 } /* Generic.Output */
.highlight .gp { color: #000080; font-weight: bold } /* Generic.Prompt */
.highlight .gs { font-weight: bold } /* Generic.Strong */
.highlight .gu { color: #800080; font-weight: bold } /* Generic.Subheading */
.highlight .gt { color: #0044DD } /* Generic.Traceback */
.highlight .kc { color: #008000; font-weight: bold } /* Keyword.Constant */
.highlight .kd { color: #008000; font-weight: bold } /* Keyword.Declaration */
.highlight .kn { color: #008000; font-weight: bold } /* Keyword.Namespace */
.highlight .kp { color: #008000 } /* Keyword.Pseudo */
.highlight .kr { color: #008000; font-weight: bold } /* Keyword.Reserved */
.highlight .kt { color: #B00040 } /* Keyword.Type */
.highlight .m { color: #666666 } /* Literal.Number */
.highlight .s { color: #BA2121 } /* Literal.String */
.highlight .na { color: #7D9029 } /* Name.Attribute */
.highlight .nb { color: #008000 } /* Name.Builtin */
.highlight .nc { color: #0000FF; font-weight: bold } /* Name.Class */
.highlight .no { color: #880000 } /* Name.Constant */
.highlight .nd { color: #AA22FF } /* Name.Decorator */
.highlight .ni { color: #999999; font-weight: bold } /* Name.Entity */
.highlight .ne { color: #D2413A; font-weight: bold } /* Name.Exception */
.highlight .nf { color: #0000FF } /* Name.Function */
.highlight .nl { color: #A0A000 } /* Name.Label */
.highlight .nn { color: #0000FF; font-weight: bold } /* Name.Namespace */
.highlight .nt { color: #008000; font-weight: bold } /* Name.Tag */
.highlight .nv { color: #19177C } /* Name.Variable */
.highlight .ow { color: #AA22FF; font-weight: bold } /* Operator.Word */
.highlight .w { color: #bbbbbb } /* Text.Whitespace */
.highlight .mb { color: #666666 } /* Literal.Number.Bin */
.highlight .mf { color: #666666 } /* Literal.Number.Float */
.highlight .mh { color: #666666 } /* Literal.Number.Hex */
.highlight .mi { color: #666666 } /* Literal.Number.Integer */
.highlight .mo { color: #666666 } /* Literal.Number.Oct */
.highlight .sa { color: #BA2121 } /* Literal.String.Affix */
.highlight .sb { color: #BA2121 } /* Literal.String.Backtick */
.highlight .sc { color: #BA2121 } /* Literal.String.Char */
.highlight .dl { color: #BA2121 } /* Literal.String.Delimiter */
.highlight .sd { color: #BA2121; font-style: italic } /* Literal.String.Doc */
.highlight .s2 { color: #BA2121 } /* Literal.String.Double */
.highlight .se { color: #BB6622; font-weight: bold } /* Literal.String.Escape */
.highlight .sh { color: #BA2121 } /* Literal.String.Heredoc */
.highlight .si { color: #BB6688; font-weight: bold } /* Literal.String.Interpol */
.highlight .sx { color: #008000 } /* Literal.String.Other */
.highlight .sr { color: #BB6688 } /* Literal.String.Regex */
.highlight .s1 { color: #BA2121 } /* Literal.String.Single */
.highlight .ss { color: #19177C } /* Literal.String.Symbol */
.highlight .bp { color: #008000 } /* Name.Builtin.Pseudo */
.highlight .fm { color: #0000FF } /* Name.Function.Magic */
.highlight .vc { color: #19177C } /* Name.Variable.Class */
.highlight .vg { color: #19177C } /* Name.Variable.Global */
.highlight .vi { color: #19177C } /* Name.Variable.Instance */
.highlight .vm { color: #19177C } /* Name.Variable.Magic */
.highlight .il { color: #666666 } /* Literal.Number.Integer.Long */
    </style>


<style type="text/css">
/* Overrides of notebook CSS for static HTML export */
body {
  overflow: visible;
  padding: 8px;
}

div#notebook {
  overflow: visible;
  border-top: none;
}@media print {
  div.cell {
    display: block;
    page-break-inside: avoid;
  } 
  div.output_wrapper { 
    display: block;
    page-break-inside: avoid; 
  }
  div.output { 
    display: block;
    page-break-inside: avoid; 
  }
}
</style>

<!-- Custom stylesheet, it must be in the same directory as the html file -->
<link rel="stylesheet" href="custom.css">

<!-- Loading mathjax macro -->
<!-- Load mathjax -->
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS_HTML"></script>
    <!-- MathJax configuration -->
    <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        tex2jax: {
            inlineMath: [ ['$','$'], ["\\(","\\)"] ],
            displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
            processEscapes: true,
            processEnvironments: true
        },
        // Center justify equations in code and markdown cells. Elsewhere
        // we use CSS to left justify single line equations in code cells.
        displayAlign: 'center',
        "HTML-CSS": {
            styles: {'.MathJax_Display': {"margin": 0}},
            linebreaks: { automatic: true }
        }
    });
    </script>
    <!-- End of mathjax configuration --></head>

 <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/jqueryui/1.12.1/jquery-ui.css">

<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>

<link rel="stylesheet" type="text/css" href="https://rawgit.com/ipython-contrib/jupyter_contrib_nbextensions/master/src/jupyter_contrib_nbextensions/nbextensions/toc2/main.css">

<link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

<script src="https://rawgit.com/ipython-contrib/jupyter_contrib_nbextensions/master/src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.js"></script>

<script>
$( document ).ready(function(){

            var cfg = {"base_numbering": 1, "nav_menu": {}, "number_sections": true, "sideBar": true, "skip_h1_title": false, "title_cell": "Table of Contents", "title_sidebar": "Contents", "toc_cell": true, "toc_position": {}, "toc_section_display": true, "toc_window_display": true};

            // fire the main function with these parameters
            require(['nbextensions/toc2/toc2'], function (toc2) {
                toc2.table_of_contents(cfg);
            });
    });
</script>
<body>
  <div tabindex="-1" id="notebook" class="border-box-sizing">
    <div class="container" id="notebook-container">

<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p><h1>Table of Contents<span class="tocSkip"></span></h1></p>
<div class="toc"><ul class="toc-item"><li><span><a href="#Structural-variants-in-Lewy-body-dementias-and-frontotemporal-dementia/amyotrophic-lateral-sclerosis" data-toc-modified-id="Structural-variants-in-Lewy-body-dementias-and-frontotemporal-dementia/amyotrophic-lateral-sclerosis-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Structural variants in Lewy body dementias and frontotemporal dementia/amyotrophic lateral sclerosis</a></span></li><li><span><a href="#Make-vcf-file-of-the-high-quality-subset-of-structural-variants-(clean-samples-and-filtered-structural-variants)" data-toc-modified-id="Make-vcf-file-of-the-high-quality-subset-of-structural-variants-(clean-samples-and-filtered-structural-variants)-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Make vcf file of the high-quality subset of structural variants (clean samples and filtered structural variants)</a></span></li><li><span><a href="#Descriptive-statistics-of-filtered-high-quality-subset-data" data-toc-modified-id="Descriptive-statistics-of-filtered-high-quality-subset-data-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Descriptive statistics of filtered high quality subset data</a></span><ul class="toc-item"><li><span><a href="#Mean-number-of-structural-variants-per-sample-(autosomes-only)" data-toc-modified-id="Mean-number-of-structural-variants-per-sample-(autosomes-only)-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Mean number of structural variants per sample (autosomes only)</a></span></li><li><span><a href="#Structural-variant-descriptives-(allele-frequency,-length,-type)" data-toc-modified-id="Structural-variant-descriptives-(allele-frequency,-length,-type)-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Structural variant descriptives (allele frequency, length, type)</a></span><ul class="toc-item"><li><span><a href="#Data-wrangling" data-toc-modified-id="Data-wrangling-3.2.1"><span class="toc-item-num">3.2.1&nbsp;&nbsp;</span>Data wrangling</a></span></li><li><span><a href="#Build-R-data-frame-and-make-Figure-2-subplots" data-toc-modified-id="Build-R-data-frame-and-make-Figure-2-subplots-3.2.2"><span class="toc-item-num">3.2.2&nbsp;&nbsp;</span>Build R data frame and make Figure 2 subplots</a></span></li><li><span><a href="#Get-descriptives" data-toc-modified-id="Get-descriptives-3.2.3"><span class="toc-item-num">3.2.3&nbsp;&nbsp;</span>Get descriptives</a></span></li><li><span><a href="#Get-descriptives-by-case-control-status" data-toc-modified-id="Get-descriptives-by-case-control-status-3.2.4"><span class="toc-item-num">3.2.4&nbsp;&nbsp;</span>Get descriptives by case-control status</a></span></li></ul></li></ul></li><li><span><a href="#Structural-variant-validation-vs.-Nanopore" data-toc-modified-id="Structural-variant-validation-vs.-Nanopore-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Structural variant validation vs. Nanopore</a></span><ul class="toc-item"><li><span><a href="#Unfiltered-data" data-toc-modified-id="Unfiltered-data-4.1"><span class="toc-item-num">4.1&nbsp;&nbsp;</span>Unfiltered data</a></span><ul class="toc-item"><li><span><a href="#Lewy-body-dementia-case/control-samples" data-toc-modified-id="Lewy-body-dementia-case/control-samples-4.1.1"><span class="toc-item-num">4.1.1&nbsp;&nbsp;</span>Lewy body dementia case/control samples</a></span></li><li><span><a href="#Frontotemporal-dementia/amyotrophic-lateral-sclerosis-samples" data-toc-modified-id="Frontotemporal-dementia/amyotrophic-lateral-sclerosis-samples-4.1.2"><span class="toc-item-num">4.1.2&nbsp;&nbsp;</span>Frontotemporal dementia/amyotrophic lateral sclerosis samples</a></span></li><li><span><a href="#Summary-of-results" data-toc-modified-id="Summary-of-results-4.1.3"><span class="toc-item-num">4.1.3&nbsp;&nbsp;</span>Summary of results</a></span></li></ul></li><li><span><a href="#Filtered-structural-variants" data-toc-modified-id="Filtered-structural-variants-4.2"><span class="toc-item-num">4.2&nbsp;&nbsp;</span>Filtered structural variants</a></span><ul class="toc-item"><li><span><a href="#Lewy-body-dementia-and-frontotemporal-dementia/amyotrophic-lateral-sclerosis-samples" data-toc-modified-id="Lewy-body-dementia-and-frontotemporal-dementia/amyotrophic-lateral-sclerosis-samples-4.2.1"><span class="toc-item-num">4.2.1&nbsp;&nbsp;</span>Lewy body dementia and frontotemporal dementia/amyotrophic lateral sclerosis samples</a></span></li><li><span><a href="#Summary-of-results" data-toc-modified-id="Summary-of-results-4.2.2"><span class="toc-item-num">4.2.2&nbsp;&nbsp;</span>Summary of results</a></span></li></ul></li><li><span><a href="#Filtered-structural-variants---allele-frequency-<1-%-vs->=1-%" data-toc-modified-id="Filtered-structural-variants---allele-frequency-<1-%-vs->=1-%-4.3"><span class="toc-item-num">4.3&nbsp;&nbsp;</span>Filtered structural variants - allele frequency &lt;1 % vs &gt;=1 %</a></span></li></ul></li><li><span><a href="#TPCN1-replication-in-BBseq/DementiaSeq2" data-toc-modified-id="TPCN1-replication-in-BBseq/DementiaSeq2-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>TPCN1 replication in BBseq/DementiaSeq2</a></span><ul class="toc-item"><li><span><a href="#Generate-principal-components" data-toc-modified-id="Generate-principal-components-5.1"><span class="toc-item-num">5.1&nbsp;&nbsp;</span>Generate principal components</a></span></li><li><span><a href="#Run-manta-to-genotype-TPCN1-deletion" data-toc-modified-id="Run-manta-to-genotype-TPCN1-deletion-5.2"><span class="toc-item-num">5.2&nbsp;&nbsp;</span>Run manta to genotype TPCN1 deletion</a></span><ul class="toc-item"><li><span><a href="#Run-samplot-to-visually-check-SV-calls" data-toc-modified-id="Run-samplot-to-visually-check-SV-calls-5.2.1"><span class="toc-item-num">5.2.1&nbsp;&nbsp;</span>Run samplot to visually check SV calls</a></span></li></ul></li><li><span><a href="#Lewy-body-dementias-clinical-and-pathological" data-toc-modified-id="Lewy-body-dementias-clinical-and-pathological-5.3"><span class="toc-item-num">5.3&nbsp;&nbsp;</span>Lewy body dementias clinical and pathological</a></span><ul class="toc-item"><li><span><a href="#Make-phenotype-file-of-all-Lewy-body-dementia-samples" data-toc-modified-id="Make-phenotype-file-of-all-Lewy-body-dementia-samples-5.3.1"><span class="toc-item-num">5.3.1&nbsp;&nbsp;</span>Make phenotype file of all Lewy body dementia samples</a></span></li><li><span><a href="#Make-covariate-file-and-select-covariate-with-step()" data-toc-modified-id="Make-covariate-file-and-select-covariate-with-step()-5.3.2"><span class="toc-item-num">5.3.2&nbsp;&nbsp;</span>Make covariate file and select covariate with step()</a></span></li><li><span><a href="#Run-association-analysis-on-SNPs/Indels-of-TPCN1-block" data-toc-modified-id="Run-association-analysis-on-SNPs/Indels-of-TPCN1-block-5.3.3"><span class="toc-item-num">5.3.3&nbsp;&nbsp;</span>Run association analysis on SNPs/Indels of TPCN1 block</a></span></li><li><span><a href="#Run-association-analysis-with-structural-variant-calls" data-toc-modified-id="Run-association-analysis-with-structural-variant-calls-5.3.4"><span class="toc-item-num">5.3.4&nbsp;&nbsp;</span>Run association analysis with structural variant calls</a></span></li><li><span><a href="#Meta-analysis-of-structural-variant-results" data-toc-modified-id="Meta-analysis-of-structural-variant-results-5.3.5"><span class="toc-item-num">5.3.5&nbsp;&nbsp;</span>Meta-analysis of structural variant results</a></span></li></ul></li></ul></li><li><span><a href="#50-neurodegenerative-disease-genes-gene-set-analyses" data-toc-modified-id="50-neurodegenerative-disease-genes-gene-set-analyses-6"><span class="toc-item-num">6&nbsp;&nbsp;</span>50 neurodegenerative disease genes gene-set analyses</a></span><ul class="toc-item"><li><span><a href="#File-preparations" data-toc-modified-id="File-preparations-6.1"><span class="toc-item-num">6.1&nbsp;&nbsp;</span>File preparations</a></span><ul class="toc-item"><li><span><a href="#Subset-50-neurodegenerative-disease-gene-regions" data-toc-modified-id="Subset-50-neurodegenerative-disease-gene-regions-6.1.1"><span class="toc-item-num">6.1.1&nbsp;&nbsp;</span>Subset 50 neurodegenerative disease gene regions</a></span></li><li><span><a href="#Create-vcfs" data-toc-modified-id="Create-vcfs-6.1.2"><span class="toc-item-num">6.1.2&nbsp;&nbsp;</span>Create vcfs</a></span></li><li><span><a href="#Case-control-allele-frequencies" data-toc-modified-id="Case-control-allele-frequencies-6.1.3"><span class="toc-item-num">6.1.3&nbsp;&nbsp;</span>Case-control allele frequencies</a></span></li><li><span><a href="#CADD-SV-scores" data-toc-modified-id="CADD-SV-scores-6.1.4"><span class="toc-item-num">6.1.4&nbsp;&nbsp;</span>CADD-SV scores</a></span></li><li><span><a href="#GeneHancer-regulatory-elements" data-toc-modified-id="GeneHancer-regulatory-elements-6.1.5"><span class="toc-item-num">6.1.5&nbsp;&nbsp;</span>GeneHancer regulatory elements</a></span></li></ul></li><li><span><a href="#Lewy-body-dementias" data-toc-modified-id="Lewy-body-dementias-6.2"><span class="toc-item-num">6.2&nbsp;&nbsp;</span>Lewy body dementias</a></span><ul class="toc-item"><li><span><a href="#Import-vcfs-into-pandas-as-the-main-data-frame" data-toc-modified-id="Import-vcfs-into-pandas-as-the-main-data-frame-6.2.1"><span class="toc-item-num">6.2.1&nbsp;&nbsp;</span>Import vcfs into pandas as the main data frame</a></span></li><li><span><a href="#Import-case-control-allele-frequencies-and-add-them-to-data-frame" data-toc-modified-id="Import-case-control-allele-frequencies-and-add-them-to-data-frame-6.2.2"><span class="toc-item-num">6.2.2&nbsp;&nbsp;</span>Import case-control allele frequencies and add them to data frame</a></span></li><li><span><a href="#List-only-sample-IDs-that-have-minor-(mostly-alternative)-allele" data-toc-modified-id="List-only-sample-IDs-that-have-minor-(mostly-alternative)-allele-6.2.3"><span class="toc-item-num">6.2.3&nbsp;&nbsp;</span>List only sample IDs that have minor (mostly alternative) allele</a></span></li><li><span><a href="#Gene-region,-promoter,-exon,-non-coding,-GeneHancer-and-CADD-SV-overlap" data-toc-modified-id="Gene-region,-promoter,-exon,-non-coding,-GeneHancer-and-CADD-SV-overlap-6.2.4"><span class="toc-item-num">6.2.4&nbsp;&nbsp;</span>Gene region, promoter, exon, non-coding, GeneHancer and CADD-SV overlap</a></span></li><li><span><a href="#Tidy-main-data-frame" data-toc-modified-id="Tidy-main-data-frame-6.2.5"><span class="toc-item-num">6.2.5&nbsp;&nbsp;</span>Tidy main data frame</a></span></li></ul></li><li><span><a href="#Frontotemporal-dementia/amyotrophic-lateral-sclerosis" data-toc-modified-id="Frontotemporal-dementia/amyotrophic-lateral-sclerosis-6.3"><span class="toc-item-num">6.3&nbsp;&nbsp;</span>Frontotemporal dementia/amyotrophic lateral sclerosis</a></span><ul class="toc-item"><li><span><a href="#Import-vcfs-into-pandas-as-the-main-data-frame" data-toc-modified-id="Import-vcfs-into-pandas-as-the-main-data-frame-6.3.1"><span class="toc-item-num">6.3.1&nbsp;&nbsp;</span>Import vcfs into pandas as the main data frame</a></span></li><li><span><a href="#Import-case-control-allele-frequencies-and-add-them-to-data-frame" data-toc-modified-id="Import-case-control-allele-frequencies-and-add-them-to-data-frame-6.3.2"><span class="toc-item-num">6.3.2&nbsp;&nbsp;</span>Import case-control allele frequencies and add them to data frame</a></span></li><li><span><a href="#List-only-sample-IDs-that-have-minor-(mostly-alternative)-allele" data-toc-modified-id="List-only-sample-IDs-that-have-minor-(mostly-alternative)-allele-6.3.3"><span class="toc-item-num">6.3.3&nbsp;&nbsp;</span>List only sample IDs that have minor (mostly alternative) allele</a></span></li><li><span><a href="#Gene-region,-promoter,-exon,-non-coding,-GeneHancer-and-CADD-SV-overlap" data-toc-modified-id="Gene-region,-promoter,-exon,-non-coding,-GeneHancer-and-CADD-SV-overlap-6.3.4"><span class="toc-item-num">6.3.4&nbsp;&nbsp;</span>Gene region, promoter, exon, non-coding, GeneHancer and CADD-SV overlap</a></span></li><li><span><a href="#Tidy-main-data-frame" data-toc-modified-id="Tidy-main-data-frame-6.3.5"><span class="toc-item-num">6.3.5&nbsp;&nbsp;</span>Tidy main data frame</a></span></li></ul></li></ul></li><li><span><a href="#Shiny-app" data-toc-modified-id="Shiny-app-7"><span class="toc-item-num">7&nbsp;&nbsp;</span>Shiny app</a></span><ul class="toc-item"><li><span><a href="#50-Neurodegenerative-disease-gene-images" data-toc-modified-id="50-Neurodegenerative-disease-gene-images-7.1"><span class="toc-item-num">7.1&nbsp;&nbsp;</span>50 Neurodegenerative disease gene images</a></span></li><li><span><a href="#Data-preparation" data-toc-modified-id="Data-preparation-7.2"><span class="toc-item-num">7.2&nbsp;&nbsp;</span>Data preparation</a></span></li><li><span><a href="#Shiny-app-code" data-toc-modified-id="Shiny-app-code-7.3"><span class="toc-item-num">7.3&nbsp;&nbsp;</span>Shiny app code</a></span></li></ul></li><li><span><a href="#Miscellaneous" data-toc-modified-id="Miscellaneous-8"><span class="toc-item-num">8&nbsp;&nbsp;</span>Miscellaneous</a></span><ul class="toc-item"><li><span><a href="#Manhattan-plots" data-toc-modified-id="Manhattan-plots-8.1"><span class="toc-item-num">8.1&nbsp;&nbsp;</span>Manhattan plots</a></span><ul class="toc-item"><li><span><a href="#Lewy-body-dementias-plot" data-toc-modified-id="Lewy-body-dementias-plot-8.1.1"><span class="toc-item-num">8.1.1&nbsp;&nbsp;</span>Lewy body dementias plot</a></span></li><li><span><a href="#Frontotemporal-dementia/amyotrophic-lateral-sclerosis-plot" data-toc-modified-id="Frontotemporal-dementia/amyotrophic-lateral-sclerosis-plot-8.1.2"><span class="toc-item-num">8.1.2&nbsp;&nbsp;</span>Frontotemporal dementia/amyotrophic lateral sclerosis plot</a></span></li></ul></li><li><span><a href="#95-%-condifence-intervals-for-genome-wide-association-study-hits" data-toc-modified-id="95-%-condifence-intervals-for-genome-wide-association-study-hits-8.2"><span class="toc-item-num">8.2&nbsp;&nbsp;</span>95 % condifence intervals for genome-wide association study hits</a></span></li><li><span><a href="#TPCN1-plots" data-toc-modified-id="TPCN1-plots-8.3"><span class="toc-item-num">8.3&nbsp;&nbsp;</span>TPCN1 plots</a></span><ul class="toc-item"><li><span><a href="#Locus-zoom" data-toc-modified-id="Locus-zoom-8.3.1"><span class="toc-item-num">8.3.1&nbsp;&nbsp;</span>Locus zoom</a></span></li></ul></li><li><span><a href="#Beta-beta-plots" data-toc-modified-id="Beta-beta-plots-8.4"><span class="toc-item-num">8.4&nbsp;&nbsp;</span>Beta-beta plots</a></span><ul class="toc-item"><li><span><a href="#Beta-beta-plot-of-Belenguez-Alzheimer's-disease-TPCN1-and-our-TPCN1-loci" data-toc-modified-id="Beta-beta-plot-of-Belenguez-Alzheimer's-disease-TPCN1-and-our-TPCN1-loci-8.4.1"><span class="toc-item-num">8.4.1&nbsp;&nbsp;</span>Beta-beta plot of Belenguez Alzheimer's disease TPCN1 and our TPCN1 loci</a></span></li><li><span><a href="#Beta-beta-plot-of-Lewy-body-dementias-and-Parkinsons'-disease" data-toc-modified-id="Beta-beta-plot-of-Lewy-body-dementias-and-Parkinsons'-disease-8.4.2"><span class="toc-item-num">8.4.2&nbsp;&nbsp;</span>Beta-beta plot of Lewy body dementias and Parkinsons' disease</a></span></li></ul></li><li><span><a href="#Cohort-demographics" data-toc-modified-id="Cohort-demographics-8.5"><span class="toc-item-num">8.5&nbsp;&nbsp;</span>Cohort demographics</a></span></li><li><span><a href="#Bigbed-files-for-UCSC/Ensembl" data-toc-modified-id="Bigbed-files-for-UCSC/Ensembl-8.6"><span class="toc-item-num">8.6&nbsp;&nbsp;</span>Bigbed files for UCSC/Ensembl</a></span></li><li><span><a href="#Number-of-tested-filtered-clean-minor-allele-frequecy-<1%-and->=1%-structural-variants-tested-against-Nanopore" data-toc-modified-id="Number-of-tested-filtered-clean-minor-allele-frequecy-<1%-and->=1%-structural-variants-tested-against-Nanopore-8.7"><span class="toc-item-num">8.7&nbsp;&nbsp;</span>Number of tested filtered clean minor allele frequecy &lt;1% and &gt;=1% structural variants tested against Nanopore</a></span></li><li><span><a href="#Generate-spreadsheets-for-high-quality-structural-variants" data-toc-modified-id="Generate-spreadsheets-for-high-quality-structural-variants-8.8"><span class="toc-item-num">8.8&nbsp;&nbsp;</span>Generate spreadsheets for high-quality structural variants</a></span></li></ul></li></ul></div>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Structural-variants-in-Lewy-body-dementias-and-frontotemporal-dementia/amyotrophic-lateral-sclerosis">Structural variants in Lewy body dementias and frontotemporal dementia/amyotrophic lateral sclerosis<a class="anchor-link" href="#Structural-variants-in-Lewy-body-dementias-and-frontotemporal-dementia/amyotrophic-lateral-sclerosis">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Working directory</span>
<span class="n">WD</span><span class="o">=</span><span class="s2">&quot;$PATH2&quot;</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Common abbreviations in notebook:</p>
<ol>
<li>LBD = Lewy body dementias</li>
<li>FTD = Frontotemporal dementia (/amyotrophic lateral sclerosis)</li>
<li>SV = Structural variant</li>
<li>AF = allele frequency</li>
<li>df = data frame</li>
<li>SVLEN = structural variant lenght</li>
<li>SVTYPE = structural variant type</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Make-vcf-file-of-the-high-quality-subset-of-structural-variants-(clean-samples-and-filtered-structural-variants)">Make vcf file of the high-quality subset of structural variants (clean samples and filtered structural variants)<a class="anchor-link" href="#Make-vcf-file-of-the-high-quality-subset-of-structural-variants-(clean-samples-and-filtered-structural-variants)">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/clean_high_quality_vcfs
module load plink/2.3-alpha

<span class="c1">#Make a list of clean, analyzed samples (i.e. all covariates known)</span>
<span class="c1">##List all samples</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span> 
    awk <span class="s1">&#39;(($8 !=&quot;NA&quot;) &amp;&amp; NR &gt;1) {print $1}&#39;</span> <span class="nv">$PATH5</span>/COVARIATES.SNVindels.SV.freeze9.<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>.controls.UNRELATED_minGQ300.txt &gt; tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_analyzed_samples.txt
<span class="k">done</span>
<span class="c1">##List cases and controls separately</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span> 
    awk <span class="s1">&#39;($8 !=&quot;NA&quot; &amp;&amp; NR &gt;1 &amp;&amp; $4==1 ) {print $1}&#39;</span> <span class="nv">$PATH5</span>/COVARIATES.SNVindels.SV.freeze9.<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>.controls.UNRELATED_minGQ300.txt &gt; tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_analyzed_controls.txt
    awk <span class="s1">&#39;($8 !=&quot;NA&quot; &amp;&amp; NR &gt;1 &amp;&amp; $4==2 ) {print $1}&#39;</span> <span class="nv">$PATH5</span>/COVARIATES.SNVindels.SV.freeze9.<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>.controls.UNRELATED_minGQ300.txt &gt; tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_analyzed_cases.txt
<span class="k">done</span>

<span class="c1">#Make a list of clean SVs</span>
<span class="c1">##Calculate AFs of SVs so that AF=0 SVs that filtering has created can be excluded (N.B.! in these AF calculations are included samples that will be excluded! Need to recalculate AFs later!)</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    plink2 <span class="se">\</span>
    --pfile <span class="nv">$PATH5</span>/<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_minGQ300/SV.<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>.forMerging <span class="se">\</span>
    --freq <span class="se">\</span>
    --out tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_SV_afreq
<span class="k">done</span>

<span class="c1">##Make a list of SVs with AF &gt; 0 (exclude 0 and nan AFs)</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    awk <span class="s1">&#39;(NR !=1 &amp;&amp; $5 != 0 &amp;&amp; $5 !=&quot;nan&quot;) {print $2}&#39;</span> tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_SV_afreq.afreq &gt; tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_SVS_AF_over0.txt
<span class="k">done</span>

<span class="c1">##Extract all filtered SVs</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    awk <span class="s1">&#39;{print $3}&#39;</span> <span class="nv">$PATH5</span>/<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_minGQ300/SV.<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>.forMerging.pvar <span class="p">|</span> grep <span class="s2">&quot;</span><span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span><span class="s2">&quot;</span> &gt; tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_SVs.txt
<span class="k">done</span>

<span class="c1">##Get union of all filtered SVs and filtered SVs with AF &gt; 0</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    grep -wFf tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_SVS_AF_over0.txt tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_SVs.txt &gt; <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_clean_SVs_AF_over_0.txt
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/clean_high_quality_vcfs
<span class="c1">#Get SV numbers and sample counts (SV numbers will drop after excluding new AF=0 SVs)</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    <span class="nb">echo</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span><span class="s2"> number of clean SVs:&quot;</span>
    wc -l <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_clean_SVs_AF_over_0.txt
    <span class="nb">echo</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span><span class="s2"> number of samples:&quot;</span>
    wc -l tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_analyzed_samples.txt
    <span class="nb">echo</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span><span class="s2"> number of cases:&quot;</span>
    wc -l tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_analyzed_cases.txt
    <span class="nb">echo</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span><span class="s2"> number of controls:&quot;</span>
    wc -l tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_analyzed_controls.txt
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/clean_high_quality_vcfs
module load bcftools

<span class="c1">#Make clean vcfs by filtering clean samples and SVs from 1% FDR GATK-SV output vcf</span>
<span class="c1">#Set GQ&lt;300 to missing and index</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    bcftools view -S tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_analyzed_samples.txt -i <span class="s2">&quot;ID=@</span><span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span><span class="s2">_clean_SVs_AF_over_0.txt&quot;</span> -Oz -o tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>.vcf.gz /<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_gatksv_1perc_fdr_cleaned_filters_qual_recalibrated/<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_1perc_fdr.cleaned_filters_qual_recalibrated.LNGsampleID.vcf.gz
    bcftools +setGT -Oz -o <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>.vcf.gz -- -t q -i <span class="s1">&#39;GQ&lt;300&#39;</span> -n .
    tabix -f -p vcf <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/clean_high_quality_vcfs
module load bcftools

<span class="c1">#Split vcfs by case-control status</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> STATUS <span class="k">in</span> cases controls<span class="p">;</span> <span class="k">do</span>
        bcftools view -S tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_analyzed_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>.txt -Oz -o rm_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>.vcf.gz <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz
    <span class="k">done</span>
<span class="k">done</span>

<span class="c1">#Add AF tags to all vcfs</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
   bcftools +fill-tags -Oz -o rename_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz -- -t AF,AC,AN
<span class="k">done</span>

<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> STATUS <span class="k">in</span> cases controls<span class="p">;</span> <span class="k">do</span>
        bcftools +fill-tags -Oz -o <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz rm_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>.vcf.gz -- -t AF,AC,AN
    <span class="k">done</span>
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/clean_high_quality_vcfs
module load samtools

<span class="c1">#Remove rm files and rename rename files to overwrite older version</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    mv rename_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz
<span class="k">done</span>

rm rm_*

<span class="c1">#Redo tabix indexing on all vcfs</span>
parallel -j1 tabix -f -p vcf <span class="o">{}</span> ::: *.vcf.gz
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/clean_high_quality_vcfs
module load samtools
<span class="c1">#Exclude SVs with AF = 0 from all vcfs (after exclusion of samples, new AF=0 SVs have been created) and rename to overwrite old file</span>

<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    bcftools view -i <span class="s1">&#39;AF &gt; 0&#39;</span> -Oz -o temp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>.vcf.gz <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz
    mv temp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>.vcf.gz <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz
<span class="k">done</span>

<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> STATUS <span class="k">in</span> cases controls<span class="p">;</span> <span class="k">do</span>
        bcftools view -i <span class="s1">&#39;AF &gt; 0&#39;</span> -Oz -o temp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>.vcf.gz <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz
        mv temp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>.vcf.gz <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz
    <span class="k">done</span>
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/clean_high_quality_vcfs
module load samtools

<span class="c1">#Redo tabix indexing on all vcfs</span>
parallel -j1 tabix -f -p vcf <span class="o">{}</span> ::: *.vcf.gz
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/clean_high_quality_vcfs
module load samtools

<span class="c1">#Make autosome only vcfs, recalculate AF, AN and AC, filter out AF=0 SVs</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    bcftools view -r chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22 -Oz -o rm_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_autosomes.vcf.gz <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz
    bcftools +fill-tags -Oz -o rm2_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_autosomes.vcf.gz rm_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_autosomes.vcf.gz -- -t AF,AC,AN
    bcftools view -i <span class="s1">&#39;AF &gt; 0&#39;</span> -Oz -o <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_autosomes.vcf.gz rm2_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_autosomes.vcf.gz 
    rm rm*_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_autosomes.vcf.gz
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/clean_high_quality_vcfs
module load samtools

<span class="c1">#Redo tabix indexing on all vcfs</span>
parallel -j <span class="nv">$SLURM_CPUS_PER_TASK</span> tabix -f -p vcf <span class="o">{}</span> ::: *.vcf.gz
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/clean_high_quality_vcfs
module load samtools
<span class="c1">#Get variant counts</span>
<span class="nb">echo</span> <span class="s2">&quot;LBD all clean SVs:&quot;</span>
bcftools query -f <span class="s1">&#39;%ID\n&#39;</span> LBD_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz <span class="p">|</span> wc -l
<span class="nb">echo</span> <span class="s2">&quot;LBD cases all clean SVs:&quot;</span>
bcftools query -f <span class="s1">&#39;%ID\n&#39;</span> LBD_cases_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz <span class="p">|</span> wc -l
<span class="nb">echo</span> <span class="s2">&quot;LBD controls all clean SVs:&quot;</span>
bcftools query -f <span class="s1">&#39;%ID\n&#39;</span> LBD_controls_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz <span class="p">|</span> wc -l
<span class="nb">echo</span> <span class="s2">&quot;FTD all clean all SVs:&quot;</span>
bcftools query -f <span class="s1">&#39;%ID\n&#39;</span> FTD_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz <span class="p">|</span> wc -l
<span class="nb">echo</span> <span class="s2">&quot;FTD cases clean all SVs:&quot;</span>
bcftools query -f <span class="s1">&#39;%ID\n&#39;</span> FTD_cases_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz <span class="p">|</span> wc -l
<span class="nb">echo</span> <span class="s2">&quot;FTD controls clean all SVs:&quot;</span>
bcftools query -f <span class="s1">&#39;%ID\n&#39;</span> FTD_controls_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz <span class="p">|</span> wc -l

<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;LBD autosomal clean SVs:&quot;</span>
bcftools query -f <span class="s1">&#39;%ID\n&#39;</span> LBD_high_quality_subset_clean_GQ300missing_autosomes.vcf.gz <span class="p">|</span> wc -l
<span class="nb">echo</span> <span class="s2">&quot;FTD autosomal clean SVs:&quot;</span>
bcftools query -f <span class="s1">&#39;%ID\n&#39;</span> FTD_high_quality_subset_clean_GQ300missing_autosomes.vcf.gz <span class="p">|</span> wc -l
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Descriptive-statistics-of-filtered-high-quality-subset-data">Descriptive statistics of filtered high quality subset data<a class="anchor-link" href="#Descriptive-statistics-of-filtered-high-quality-subset-data">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Mean-number-of-structural-variants-per-sample-(autosomes-only)">Mean number of structural variants per sample (autosomes only)<a class="anchor-link" href="#Mean-number-of-structural-variants-per-sample-(autosomes-only)">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/clean_high_quality_vcfs
module load bcftools

<span class="c1">#Run bcftools stats to get SV counts per sample (autosomes, high quality subset SVs). Do separately for all, cases and controls</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    bcftools stats -S tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_analyzed_samples.txt <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_autosomes.vcf.gz &gt; <span class="nv">$PATH2</span>/descriptive_stats_and_fig2/<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_autosomes_stats.txt
    bcftools stats -S tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_analyzed_cases.txt <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_autosomes.vcf.gz &gt; <span class="nv">$PATH2</span>/descriptive_stats_and_fig2/<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_cases_high_quality_subset_clean_GQ300missing_autosomes_stats.txt
    bcftools stats -S tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_analyzed_controls.txt <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_autosomes.vcf.gz &gt; <span class="nv">$PATH2</span>/descriptive_stats_and_fig2/<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_controls_high_quality_subset_clean_GQ300missing_autosomes_stats.txt
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/descriptive_stats_and_fig2
<span class="c1">#Calculate average number of SVs in autosome per sample across filtered samples: at rows starting PCS, is number of reference hom variants and number of variants with missing genotype</span>
<span class="c1">#Total number of SVs with alt = total number of SVs in data set - nRefHom-nMissing</span>
<span class="nv">N_LBD</span><span class="o">=</span><span class="m">146027</span>
<span class="nv">N_FTD</span><span class="o">=</span><span class="m">153678</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD<span class="p">;</span> <span class="k">do</span>
    <span class="nb">echo</span> <span class="s2">&quot;Mean number of SVs in autosomes per sample in filtered FDR1% </span><span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span><span class="s2"> run, all samples&quot;</span>
    grep <span class="s1">&#39;PSC&#39;</span> <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_autosomes_stats.txt <span class="p">|</span> awk -v <span class="nv">N</span><span class="o">=</span><span class="nv">$N_LBD</span> <span class="s1">&#39;(NR&gt;2){print N-$4-$14}&#39;</span> <span class="p">|</span> awk <span class="s1">&#39;{ total += $1; count++ } END { print total/count }&#39;</span>
    <span class="nb">echo</span> <span class="s2">&quot;Mean number of SVs in autosomes per cases in filtered FDR1% </span><span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span><span class="s2"> run, all samples&quot;</span>
    grep <span class="s1">&#39;PSC&#39;</span> <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_cases_high_quality_subset_clean_GQ300missing_autosomes_stats.txt <span class="p">|</span> awk -v <span class="nv">N</span><span class="o">=</span><span class="nv">$N_LBD</span> <span class="s1">&#39;(NR&gt;2){print N-$4-$14}&#39;</span> <span class="p">|</span> awk <span class="s1">&#39;{ total += $1; count++ } END { print total/count }&#39;</span>
    <span class="nb">echo</span> <span class="s2">&quot;Mean number of SVs in autosomes per controls in filtered FDR1% </span><span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span><span class="s2"> run, all samples&quot;</span>
    grep <span class="s1">&#39;PSC&#39;</span> <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_controls_high_quality_subset_clean_GQ300missing_autosomes_stats.txt <span class="p">|</span> awk -v <span class="nv">N</span><span class="o">=</span><span class="nv">$N_LBD</span> <span class="s1">&#39;(NR&gt;2){print N-$4-$14}&#39;</span> <span class="p">|</span> awk <span class="s1">&#39;{ total += $1; count++ } END { print total/count }&#39;</span>
<span class="k">done</span>

<span class="k">for</span> PHENO <span class="k">in</span> FTD<span class="p">;</span> <span class="k">do</span>
    <span class="nb">echo</span> <span class="s2">&quot;Mean number of SVs in autosomes per sample in filtered FDR1% </span><span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span><span class="s2"> run, all samples&quot;</span>
    grep <span class="s1">&#39;PSC&#39;</span> <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_autosomes_stats.txt <span class="p">|</span> awk -v <span class="nv">N</span><span class="o">=</span><span class="nv">$N_FTD</span> <span class="s1">&#39;(NR&gt;2){print N-$4-$14}&#39;</span> <span class="p">|</span> awk <span class="s1">&#39;{ total += $1; count++ } END { print total/count }&#39;</span>
    <span class="nb">echo</span> <span class="s2">&quot;Mean number of SVs in autosomes per cases in filtered FDR1% </span><span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span><span class="s2"> run, all samples&quot;</span>
    grep <span class="s1">&#39;PSC&#39;</span> <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_cases_high_quality_subset_clean_GQ300missing_autosomes_stats.txt <span class="p">|</span> awk -v <span class="nv">N</span><span class="o">=</span><span class="nv">$N_FTD</span> <span class="s1">&#39;(NR&gt;2){print N-$4-$14}&#39;</span> <span class="p">|</span> awk <span class="s1">&#39;{ total += $1; count++ } END { print total/count }&#39;</span>
    <span class="nb">echo</span> <span class="s2">&quot;Mean number of SVs in autosomes per controls in filtered FDR1% </span><span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span><span class="s2"> run, all samples&quot;</span>
    grep <span class="s1">&#39;PSC&#39;</span> <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_controls_high_quality_subset_clean_GQ300missing_autosomes_stats.txt <span class="p">|</span> awk -v <span class="nv">N</span><span class="o">=</span><span class="nv">$N_FTD</span> <span class="s1">&#39;(NR&gt;2){print N-$4-$14}&#39;</span> <span class="p">|</span> awk <span class="s1">&#39;{ total += $1; count++ } END { print total/count }&#39;</span>
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Structural-variant-descriptives-(allele-frequency,-length,-type)">Structural variant descriptives (allele frequency, length, type)<a class="anchor-link" href="#Structural-variant-descriptives-(allele-frequency,-length,-type)">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Data-wrangling">Data wrangling<a class="anchor-link" href="#Data-wrangling">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/descriptive_stats_and_fig2
module load bcftools

<span class="c1">#Extract AF,AC,SVLEN,SVTYPE,CHROM info per variant for LBD and FTD</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    bcftools query -f <span class="s1">&#39;%ID %SVTYPE %SVLEN %AF %AC %CHROM\n&#39;</span> ../clean_high_quality_vcfs/<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz &gt; <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr_INFO.txt
<span class="k">done</span>

<span class="c1">#Extract AF,AC,SVLEN,SVTYPE,CHROM info per variant for LBD and FTD cases and controls separately</span>
<span class="c1">#N.B.! Case and control vcfs include AF=0 variants!</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> STATUS <span class="k">in</span> cases controls<span class="p">;</span> <span class="k">do</span>
        bcftools query -f <span class="s1">&#39;%ID %SVTYPE %SVLEN %AF %AC %CHROM\n&#39;</span> ../clean_high_quality_vcfs/<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz &gt; <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr_INFO.txt
    <span class="k">done</span>
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Build-R-data-frame-and-make-Figure-2-subplots">Build R data frame and make Figure 2 subplots<a class="anchor-link" href="#Build-R-data-frame-and-make-Figure-2-subplots">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%</span><span class="k">load_ext</span> rpy2.ipython
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">####Filtered SVs GATK-SV style bar plots: LBD</span>

<span class="c1">#Set wd</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;$PATH2/descriptive_stats_and_fig2&quot;</span><span class="p">)</span>

<span class="c1">#Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">scales</span><span class="p">)</span>

<span class="c1">#Import files</span>

<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;LBD_high_quality_subset_clean_GQ300missing_all_chr_INFO.txt&quot;</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;ID&quot;</span><span class="p">,</span><span class="s">&quot;SVTYPE&quot;</span><span class="p">,</span><span class="s">&quot;SVLEN&quot;</span><span class="p">,</span><span class="s">&quot;AF&quot;</span><span class="p">,</span><span class="s">&quot;AC&quot;</span><span class="p">,</span><span class="s">&quot;CHROM&quot;</span><span class="p">)</span>

<span class="c1">#Rename BND and CTX to OTH</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="o">%in%</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;BND&quot;</span><span class="p">,</span><span class="s">&quot;CTX&quot;</span><span class="p">)]</span> <span class="o">&lt;-</span> <span class="s">&quot;OTH&quot;</span>

<span class="c1">#Make SVTYPEs into factors</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="p">,</span> <span class="n">levels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;DEL&quot;</span><span class="p">,</span> <span class="s">&quot;DUP&quot;</span><span class="p">,</span> <span class="s">&quot;CNV&quot;</span><span class="p">,</span> <span class="s">&quot;INS&quot;</span><span class="p">,</span> <span class="s">&quot;INV&quot;</span><span class="p">,</span> <span class="s">&quot;CPX&quot;</span><span class="p">,</span> <span class="s">&quot;OTH&quot;</span><span class="p">),</span> <span class="n">labels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;DEL&quot;</span><span class="p">,</span> <span class="s">&quot;DUP&quot;</span><span class="p">,</span> <span class="s">&quot;CNV&quot;</span><span class="p">,</span> <span class="s">&quot;INS&quot;</span><span class="p">,</span> <span class="s">&quot;INV&quot;</span><span class="p">,</span> <span class="s">&quot;CPX&quot;</span><span class="p">,</span> <span class="s">&quot;OTH&quot;</span><span class="p">))</span>


<span class="c1">#Add SVLEN and AF groups, add AC=1 separately since threshold varies accordig to number of samples</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVLEN_group</span> <span class="o">&lt;-</span> <span class="nf">cut</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">SVLEN</span><span class="p">,</span> <span class="n">breaks</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">-2</span><span class="p">,</span><span class="m">0</span><span class="p">,</span><span class="m">100</span><span class="p">,</span><span class="m">500</span><span class="p">,</span><span class="m">2500</span><span class="p">,</span><span class="m">10000</span><span class="p">,</span><span class="m">50000</span><span class="p">,</span><span class="m">10</span><span class="o">^</span><span class="m">10</span><span class="p">),</span> <span class="n">labels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;unresolved&quot;</span><span class="p">,</span> <span class="s">&quot;&lt;100bp&quot;</span><span class="p">,</span> <span class="s">&quot;100bp-500bp&quot;</span><span class="p">,</span> <span class="s">&quot;500bp-2.5kb&quot;</span><span class="p">,</span> <span class="s">&quot;2.5kb-10kb&quot;</span><span class="p">,</span> <span class="s">&quot;10kb-50kb&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50kb&quot;</span><span class="p">))</span>
<span class="n">df</span><span class="o">$</span><span class="n">AF_group</span> <span class="o">&lt;-</span> <span class="nf">cut</span><span class="p">(</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">AF</span><span class="p">),</span> <span class="n">breaks</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0.01</span><span class="p">,</span><span class="m">0.10</span><span class="p">,</span><span class="m">0.50</span><span class="p">,</span><span class="m">1</span><span class="p">),</span> <span class="n">labels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;AF&lt;1%&quot;</span><span class="p">,</span> <span class="s">&quot;1-10%&quot;</span><span class="p">,</span> <span class="s">&quot;10-50%&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50%&quot;</span><span class="p">))</span>
<span class="nf">levels</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">AF_group</span><span class="p">)[</span><span class="m">5</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s">&quot;AC=1&quot;</span>
<span class="n">df</span><span class="o">$</span><span class="n">AF_group</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">AC</span> <span class="o">==</span> <span class="m">1</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s">&quot;AC=1&quot;</span>

<span class="c1">#Make grouped data frames. N.B.! Exclude &quot;Other&quot; category from plots as per GATK-SV model image since SVLEN is more of a guess for them</span>
<span class="n">df_grouped_AF_SVTYPE</span> <span class="o">&lt;-</span> <span class="n">df</span> <span class="o">%&gt;%</span>
  <span class="nf">group_by</span><span class="p">(</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="n">AF_group</span><span class="p">)</span> <span class="o">%&gt;%</span>
  <span class="nf">summarize</span><span class="p">(</span><span class="n">count</span> <span class="o">=</span> <span class="nf">n</span><span class="p">())</span>

<span class="n">tmp</span> <span class="o">&lt;-</span> <span class="n">df_grouped_AF_SVTYPE</span>
<span class="n">tmp</span><span class="o">$</span><span class="n">AF_group</span> <span class="o">&lt;-</span> <span class="s">&quot;ALL&quot;</span>
<span class="n">df_grouped_AF_SVTYPE</span> <span class="o">&lt;-</span> <span class="nf">rbind</span><span class="p">(</span><span class="n">df_grouped_AF_SVTYPE</span><span class="p">,</span><span class="n">tmp</span><span class="p">)</span>
<span class="n">df_grouped_AF_SVTYPE</span><span class="o">$</span><span class="n">AF_group</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">df_grouped_AF_SVTYPE</span><span class="o">$</span><span class="n">AF_group</span><span class="p">,</span> <span class="n">ordered</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;ALL&quot;</span><span class="p">,</span> <span class="s">&quot;AC=1&quot;</span><span class="p">,</span> <span class="s">&quot;AF&lt;1%&quot;</span><span class="p">,</span> <span class="s">&quot;1-10%&quot;</span><span class="p">,</span> <span class="s">&quot;10-50%&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50%&quot;</span><span class="p">))</span>

<span class="n">df_grouped_SVLEN_SVTYPE</span> <span class="o">&lt;-</span> <span class="n">df</span> <span class="o">%&gt;%</span>
  <span class="nf">filter</span><span class="p">(</span><span class="n">SVTYPE</span> <span class="o">!=</span> <span class="s">&quot;OTH&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span>
  <span class="nf">group_by</span><span class="p">(</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="n">SVLEN_group</span><span class="p">)</span> <span class="o">%&gt;%</span>
  <span class="nf">summarize</span><span class="p">(</span><span class="n">count</span> <span class="o">=</span> <span class="nf">n</span><span class="p">())</span>

<span class="n">tmp</span> <span class="o">&lt;-</span> <span class="n">df_grouped_SVLEN_SVTYPE</span>
<span class="n">tmp</span><span class="o">$</span><span class="n">SVLEN_group</span> <span class="o">&lt;-</span> <span class="s">&quot;ALL&quot;</span>
<span class="n">df_grouped_SVLEN_SVTYPE</span> <span class="o">&lt;-</span> <span class="nf">rbind</span><span class="p">(</span><span class="n">df_grouped_SVLEN_SVTYPE</span><span class="p">,</span><span class="n">tmp</span><span class="p">)</span>
<span class="n">df_grouped_SVLEN_SVTYPE</span><span class="o">$</span><span class="n">SVLEN_group</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">df_grouped_SVLEN_SVTYPE</span><span class="o">$</span><span class="n">SVLEN_group</span><span class="p">,</span> <span class="n">ordered</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;ALL&quot;</span><span class="p">,</span> <span class="s">&quot;&lt;100bp&quot;</span><span class="p">,</span> <span class="s">&quot;100bp-500bp&quot;</span><span class="p">,</span> <span class="s">&quot;500bp-2.5kb&quot;</span><span class="p">,</span> <span class="s">&quot;2.5kb-10kb&quot;</span><span class="p">,</span> <span class="s">&quot;10kb-50kb&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50kb&quot;</span><span class="p">))</span>

<span class="c1">#Make a second data frame for scaled plots to include ALL group</span>
<span class="n">df2</span> <span class="o">&lt;-</span> <span class="n">df</span>
<span class="n">df2</span><span class="o">$</span><span class="n">AF_group</span> <span class="o">&lt;-</span> <span class="s">&quot;ALL&quot;</span>
<span class="n">df2</span><span class="o">$</span><span class="n">SVLEN_group</span> <span class="o">&lt;-</span> <span class="s">&quot;ALL&quot;</span>
<span class="n">df2</span> <span class="o">&lt;-</span> <span class="nf">rbind</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="n">df2</span><span class="p">)</span>
<span class="n">df2</span><span class="o">$</span><span class="n">SVLEN_group</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">df2</span><span class="o">$</span><span class="n">SVLEN_group</span><span class="p">,</span> <span class="n">ordered</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;ALL&quot;</span><span class="p">,</span> <span class="s">&quot;&lt;100bp&quot;</span><span class="p">,</span> <span class="s">&quot;100bp-500bp&quot;</span><span class="p">,</span> <span class="s">&quot;500bp-2.5kb&quot;</span><span class="p">,</span> <span class="s">&quot;2.5kb-10kb&quot;</span><span class="p">,</span> <span class="s">&quot;10kb-50kb&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50kb&quot;</span><span class="p">))</span>
<span class="n">df2</span><span class="o">$</span><span class="n">AF_group</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">df2</span><span class="o">$</span><span class="n">AF_group</span><span class="p">,</span> <span class="n">ordered</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;ALL&quot;</span><span class="p">,</span> <span class="s">&quot;AC=1&quot;</span><span class="p">,</span> <span class="s">&quot;AF&lt;1%&quot;</span><span class="p">,</span> <span class="s">&quot;1-10%&quot;</span><span class="p">,</span> <span class="s">&quot;10-50%&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50%&quot;</span><span class="p">))</span>


<span class="c1">#Plots</span>
<span class="c1">##Variant count by SVTYPE</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="n">SVTYPE</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">geom_bar</span><span class="p">(</span><span class="n">width</span><span class="o">=</span><span class="m">0.8</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="s">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="s">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="s">&quot;#F69A57&quot;</span><span class="p">,</span> <span class="s">&quot;#86E496&quot;</span><span class="p">,</span> <span class="s">&quot;#395946&quot;</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">geom_text</span><span class="p">(</span><span class="n">stat</span><span class="o">=</span><span class="s">&#39;count&#39;</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">label</span> <span class="o">=</span> <span class="nf">format</span><span class="p">(</span><span class="n">..count..</span><span class="p">,</span> <span class="n">big.mark</span> <span class="o">=</span> <span class="s">&quot;,&quot;</span><span class="p">,</span> <span class="n">scientific</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)),</span> <span class="n">size</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="s">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="s">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="s">&quot;#F69A57&quot;</span><span class="p">,</span> <span class="s">&quot;#86E496&quot;</span><span class="p">,</span> <span class="s">&quot;#395946&quot;</span><span class="p">),</span> <span class="n">vjust</span><span class="o">=</span><span class="m">-1</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;SV Count&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">scale_y_continuous</span><span class="p">(</span><span class="n">labels</span> <span class="o">=</span> <span class="n">comma</span><span class="p">,</span> <span class="n">expand</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">-0.001</span><span class="p">,</span><span class="m">0</span><span class="p">),</span> <span class="n">limits</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="nf">nrow</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="o">==</span> <span class="s">&quot;DEL&quot;</span><span class="p">])</span> <span class="o">+</span> <span class="m">5000</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">guides</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">theme</span><span class="p">(</span><span class="n">axis.title.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">16</span><span class="p">),</span> <span class="n">axis.ticks.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span> <span class="n">axis.text.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">12</span><span class="p">),</span> <span class="n">axis.text.x</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">14</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="m">90</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="s">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="s">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="s">&quot;#F69A57&quot;</span><span class="p">,</span> <span class="s">&quot;#86E496&quot;</span><span class="p">,</span> <span class="s">&quot;#395946&quot;</span><span class="p">)),</span> <span class="n">panel.background</span> <span class="o">=</span> <span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="s">&quot;white&quot;</span><span class="p">),</span> <span class="n">axis.line.y</span> <span class="o">=</span> <span class="nf">element_line</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> <span class="n">colour</span> <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">)))</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span> <span class="o">=</span> <span class="s">&quot;LBD_filtered_SV_variant_count_plot.png&quot;</span><span class="p">,</span> <span class="n">plot</span> <span class="o">=</span> <span class="nf">last_plot</span><span class="p">(),</span> <span class="n">device</span> <span class="o">=</span> <span class="s">&quot;png&quot;</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="m">7</span><span class="p">,</span> <span class="n">units</span> <span class="o">=</span> <span class="s">&quot;in&quot;</span> <span class="p">)</span>

<span class="c1">##SV count by AF</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">df_grouped_AF_SVTYPE</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">AF_group</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">count</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">geom_bar</span><span class="p">(</span><span class="n">width</span> <span class="o">=</span> <span class="m">0.8</span><span class="p">,</span> <span class="n">position</span><span class="o">=</span><span class="nf">position_stack</span><span class="p">(</span><span class="n">reverse</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">),</span> <span class="n">stat</span><span class="o">=</span><span class="s">&quot;identity&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="s">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="s">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="s">&quot;#F69A57&quot;</span><span class="p">,</span> <span class="s">&quot;#86E496&quot;</span><span class="p">,</span> <span class="s">&quot;#395946&quot;</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">scale_y_continuous</span><span class="p">(</span><span class="n">labels</span> <span class="o">=</span> <span class="n">comma</span><span class="p">,</span> <span class="n">expand</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">-0.0001</span><span class="p">,</span><span class="m">0</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">guides</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;SV Count by AF LBD&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">theme</span><span class="p">(</span><span class="n">axis.title.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">),</span> <span class="n">axis.ticks.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span> <span class="n">axis.text.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">),</span> <span class="n">axis.text.x</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="m">90</span><span class="p">),</span> <span class="n">panel.background</span> <span class="o">=</span> <span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="s">&quot;white&quot;</span><span class="p">),</span> <span class="n">axis.line.y</span> <span class="o">=</span> <span class="nf">element_line</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> <span class="n">colour</span> <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">)))</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span> <span class="o">=</span> <span class="s">&quot;LBD_filtered_SV_count_by_AF_plot.png&quot;</span><span class="p">,</span> <span class="n">plot</span> <span class="o">=</span> <span class="nf">last_plot</span><span class="p">(),</span> <span class="n">device</span> <span class="o">=</span> <span class="s">&quot;png&quot;</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="m">7</span><span class="p">,</span> <span class="n">units</span> <span class="o">=</span> <span class="s">&quot;in&quot;</span> <span class="p">)</span>

<span class="c1">##SV count by AF scaled</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">df2</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">AF_group</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">geom_bar</span><span class="p">(</span><span class="n">width</span> <span class="o">=</span> <span class="m">0.8</span><span class="p">,</span> <span class="n">position</span> <span class="o">=</span> <span class="nf">position_fill</span><span class="p">(</span><span class="n">reverse</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="s">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="s">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="s">&quot;#F69A57&quot;</span><span class="p">,</span> <span class="s">&quot;#86E496&quot;</span><span class="p">,</span> <span class="s">&quot;#395946&quot;</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">scale_y_continuous</span><span class="p">(</span><span class="n">breaks</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0.20</span><span class="p">,</span><span class="m">0.40</span><span class="p">,</span><span class="m">0.60</span><span class="p">,</span><span class="m">0.8</span><span class="p">,</span><span class="m">1</span><span class="p">),</span> <span class="n">labels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;0%&quot;</span><span class="p">,</span> <span class="s">&quot;20%&quot;</span><span class="p">,</span> <span class="s">&quot;40%&quot;</span><span class="p">,</span> <span class="s">&quot;60%&quot;</span><span class="p">,</span> <span class="s">&#39;80%&#39;</span><span class="p">,</span> <span class="s">&quot;100%&quot;</span><span class="p">),</span> <span class="n">expand</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">-0.001</span><span class="p">,</span><span class="m">0.001</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">guides</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;SV Count by AF LBD scaled&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">theme</span><span class="p">(</span><span class="n">axis.title.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">),</span> <span class="n">axis.ticks.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span> <span class="n">axis.text.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">),</span> <span class="n">axis.text.x</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="m">90</span><span class="p">),</span> <span class="n">panel.background</span> <span class="o">=</span> <span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="s">&quot;white&quot;</span><span class="p">),</span> <span class="n">axis.line.y</span> <span class="o">=</span> <span class="nf">element_line</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> <span class="n">colour</span> <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">)))</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span> <span class="o">=</span> <span class="s">&quot;LBD_filtered_SV_count_by_AF_plot_scaled.png&quot;</span><span class="p">,</span> <span class="n">plot</span> <span class="o">=</span> <span class="nf">last_plot</span><span class="p">(),</span> <span class="n">device</span> <span class="o">=</span> <span class="s">&quot;png&quot;</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="m">7</span><span class="p">,</span> <span class="n">units</span> <span class="o">=</span> <span class="s">&quot;in&quot;</span> <span class="p">)</span>

<span class="c1">##SV count by SVLEN</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">df_grouped_SVLEN_SVTYPE</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">SVLEN_group</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">count</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">geom_bar</span><span class="p">(</span><span class="n">width</span> <span class="o">=</span> <span class="m">0.8</span><span class="p">,</span> <span class="n">position</span><span class="o">=</span><span class="nf">position_stack</span><span class="p">(</span><span class="n">reverse</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">),</span> <span class="n">stat</span><span class="o">=</span><span class="s">&quot;identity&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="s">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="s">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="s">&quot;#F69A57&quot;</span><span class="p">,</span> <span class="s">&quot;#86E496&quot;</span><span class="p">,</span> <span class="s">&quot;#395946&quot;</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">scale_y_continuous</span><span class="p">(</span><span class="n">labels</span> <span class="o">=</span> <span class="n">comma</span><span class="p">,</span> <span class="n">expand</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">-0.001</span><span class="p">,</span><span class="m">0</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">guides</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;SV Count by SVLEN LBD&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">theme</span><span class="p">(</span><span class="n">axis.title.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">),</span> <span class="n">axis.ticks.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span> <span class="n">axis.text.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">),</span> <span class="n">axis.text.x</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="m">90</span><span class="p">),</span> <span class="n">panel.background</span> <span class="o">=</span> <span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="s">&quot;white&quot;</span><span class="p">),</span> <span class="n">axis.line.y</span> <span class="o">=</span> <span class="nf">element_line</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> <span class="n">colour</span> <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">)))</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span> <span class="o">=</span> <span class="s">&quot;LBD_filtered_SV_count_by_SVLEN_plot.png&quot;</span><span class="p">,</span> <span class="n">plot</span> <span class="o">=</span> <span class="nf">last_plot</span><span class="p">(),</span> <span class="n">device</span> <span class="o">=</span> <span class="s">&quot;png&quot;</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="m">7</span><span class="p">,</span> <span class="n">units</span> <span class="o">=</span> <span class="s">&quot;in&quot;</span> <span class="p">)</span>

<span class="c1">##SV count by SVLEN scaled</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">ggplot</span><span class="p">(</span><span class="nf">subset</span><span class="p">(</span><span class="n">df2</span><span class="p">,</span> <span class="n">SVTYPE</span> <span class="o">!=</span> <span class="s">&quot;OTH&quot;</span><span class="p">),</span> <span class="nf">aes</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">SVLEN_group</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">geom_bar</span><span class="p">(</span><span class="n">width</span> <span class="o">=</span> <span class="m">0.8</span><span class="p">,,</span> <span class="n">position</span> <span class="o">=</span> <span class="nf">position_fill</span><span class="p">(</span><span class="n">reverse</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="s">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="s">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="s">&quot;#F69A57&quot;</span><span class="p">,</span> <span class="s">&quot;#86E496&quot;</span><span class="p">,</span> <span class="s">&quot;#395946&quot;</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">scale_y_continuous</span><span class="p">(</span><span class="n">breaks</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0.20</span><span class="p">,</span><span class="m">0.40</span><span class="p">,</span><span class="m">0.60</span><span class="p">,</span><span class="m">0.8</span><span class="p">,</span><span class="m">1</span><span class="p">),</span> <span class="n">labels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;0%&quot;</span><span class="p">,</span> <span class="s">&quot;20%&quot;</span><span class="p">,</span> <span class="s">&quot;40%&quot;</span><span class="p">,</span> <span class="s">&quot;60%&quot;</span><span class="p">,</span> <span class="s">&#39;80%&#39;</span><span class="p">,</span> <span class="s">&quot;100%&quot;</span><span class="p">),</span> <span class="n">expand</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">-0.001</span><span class="p">,</span><span class="m">0.001</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">guides</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;SV Count by SVLEN LBD scaled&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">theme</span><span class="p">(</span><span class="n">axis.title.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">),</span> <span class="n">axis.ticks.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span> <span class="n">axis.text.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">),</span> <span class="n">axis.text.x</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="m">90</span><span class="p">),</span> <span class="n">panel.background</span> <span class="o">=</span> <span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="s">&quot;white&quot;</span><span class="p">),</span> <span class="n">axis.line.y</span> <span class="o">=</span> <span class="nf">element_line</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> <span class="n">colour</span> <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">)))</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span> <span class="o">=</span> <span class="s">&quot;LBD_filtered_SV_count_by_SVLEN_plot_scaled.png&quot;</span><span class="p">,</span> <span class="n">plot</span> <span class="o">=</span> <span class="nf">last_plot</span><span class="p">(),</span> <span class="n">device</span> <span class="o">=</span> <span class="s">&quot;png&quot;</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="m">7</span><span class="p">,</span> <span class="n">units</span> <span class="o">=</span> <span class="s">&quot;in&quot;</span> <span class="p">)</span>


<span class="c1">#Save df</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="s">&quot;LBD_descriptives_df.txt&quot;</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">####Filtered SVs GATK-SV style bar plots: FTD</span>

<span class="c1">#Set wd</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;$PATH2/descriptive_stats_and_fig2&quot;</span><span class="p">)</span>

<span class="c1">#Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">scales</span><span class="p">)</span>

<span class="c1">#Import files</span>

<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;FTD_high_quality_subset_clean_GQ300missing_all_chr_INFO.txt&quot;</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;ID&quot;</span><span class="p">,</span><span class="s">&quot;SVTYPE&quot;</span><span class="p">,</span><span class="s">&quot;SVLEN&quot;</span><span class="p">,</span><span class="s">&quot;AF&quot;</span><span class="p">,</span><span class="s">&quot;AC&quot;</span><span class="p">,</span><span class="s">&quot;CHROM&quot;</span><span class="p">)</span>

<span class="c1">#Rename BND and CTX to OTH</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="o">%in%</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;BND&quot;</span><span class="p">,</span><span class="s">&quot;CTX&quot;</span><span class="p">)]</span> <span class="o">&lt;-</span> <span class="s">&quot;OTH&quot;</span>

<span class="c1">#Make SVTYPEs into factors</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="p">,</span> <span class="n">levels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;DEL&quot;</span><span class="p">,</span> <span class="s">&quot;DUP&quot;</span><span class="p">,</span> <span class="s">&quot;CNV&quot;</span><span class="p">,</span> <span class="s">&quot;INS&quot;</span><span class="p">,</span> <span class="s">&quot;INV&quot;</span><span class="p">,</span> <span class="s">&quot;CPX&quot;</span><span class="p">,</span> <span class="s">&quot;OTH&quot;</span><span class="p">),</span> <span class="n">labels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;DEL&quot;</span><span class="p">,</span> <span class="s">&quot;DUP&quot;</span><span class="p">,</span> <span class="s">&quot;CNV&quot;</span><span class="p">,</span> <span class="s">&quot;INS&quot;</span><span class="p">,</span> <span class="s">&quot;INV&quot;</span><span class="p">,</span> <span class="s">&quot;CPX&quot;</span><span class="p">,</span> <span class="s">&quot;OTH&quot;</span><span class="p">))</span>


<span class="c1">#Add SVLEN and AF groups, add AC=1 separately since threshold varies accordig to number of samples</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVLEN_group</span> <span class="o">&lt;-</span> <span class="nf">cut</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">SVLEN</span><span class="p">,</span> <span class="n">breaks</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">-2</span><span class="p">,</span><span class="m">0</span><span class="p">,</span><span class="m">100</span><span class="p">,</span><span class="m">500</span><span class="p">,</span><span class="m">2500</span><span class="p">,</span><span class="m">10000</span><span class="p">,</span><span class="m">50000</span><span class="p">,</span><span class="m">10</span><span class="o">^</span><span class="m">10</span><span class="p">),</span> <span class="n">labels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;unresolved&quot;</span><span class="p">,</span> <span class="s">&quot;&lt;100bp&quot;</span><span class="p">,</span> <span class="s">&quot;100bp-500bp&quot;</span><span class="p">,</span> <span class="s">&quot;500bp-2.5kb&quot;</span><span class="p">,</span> <span class="s">&quot;2.5kb-10kb&quot;</span><span class="p">,</span> <span class="s">&quot;10kb-50kb&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50kb&quot;</span><span class="p">))</span>
<span class="n">df</span><span class="o">$</span><span class="n">AF_group</span> <span class="o">&lt;-</span> <span class="nf">cut</span><span class="p">(</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">AF</span><span class="p">),</span> <span class="n">breaks</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0.01</span><span class="p">,</span><span class="m">0.10</span><span class="p">,</span><span class="m">0.50</span><span class="p">,</span><span class="m">1</span><span class="p">),</span> <span class="n">labels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;AF&lt;1%&quot;</span><span class="p">,</span> <span class="s">&quot;1-10%&quot;</span><span class="p">,</span> <span class="s">&quot;10-50%&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50%&quot;</span><span class="p">))</span>
<span class="nf">levels</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">AF_group</span><span class="p">)[</span><span class="m">5</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s">&quot;AC=1&quot;</span>
<span class="n">df</span><span class="o">$</span><span class="n">AF_group</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">AC</span> <span class="o">==</span> <span class="m">1</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s">&quot;AC=1&quot;</span>

<span class="c1">#Make grouped data frames. N.B.! Exclude &quot;Other&quot; category from plots as per GATK-SV model image since SVLEN is more of a guess for them</span>
<span class="n">df_grouped_AF_SVTYPE</span> <span class="o">&lt;-</span> <span class="n">df</span> <span class="o">%&gt;%</span>
  <span class="nf">group_by</span><span class="p">(</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="n">AF_group</span><span class="p">)</span> <span class="o">%&gt;%</span>
  <span class="nf">summarize</span><span class="p">(</span><span class="n">count</span> <span class="o">=</span> <span class="nf">n</span><span class="p">())</span>

<span class="n">tmp</span> <span class="o">&lt;-</span> <span class="n">df_grouped_AF_SVTYPE</span>
<span class="n">tmp</span><span class="o">$</span><span class="n">AF_group</span> <span class="o">&lt;-</span> <span class="s">&quot;ALL&quot;</span>
<span class="n">df_grouped_AF_SVTYPE</span> <span class="o">&lt;-</span> <span class="nf">rbind</span><span class="p">(</span><span class="n">df_grouped_AF_SVTYPE</span><span class="p">,</span><span class="n">tmp</span><span class="p">)</span>
<span class="n">df_grouped_AF_SVTYPE</span><span class="o">$</span><span class="n">AF_group</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">df_grouped_AF_SVTYPE</span><span class="o">$</span><span class="n">AF_group</span><span class="p">,</span> <span class="n">ordered</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;ALL&quot;</span><span class="p">,</span> <span class="s">&quot;AC=1&quot;</span><span class="p">,</span> <span class="s">&quot;AF&lt;1%&quot;</span><span class="p">,</span> <span class="s">&quot;1-10%&quot;</span><span class="p">,</span> <span class="s">&quot;10-50%&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50%&quot;</span><span class="p">))</span>

<span class="n">df_grouped_SVLEN_SVTYPE</span> <span class="o">&lt;-</span> <span class="n">df</span> <span class="o">%&gt;%</span>
  <span class="nf">filter</span><span class="p">(</span><span class="n">SVTYPE</span> <span class="o">!=</span> <span class="s">&quot;OTH&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span>
  <span class="nf">group_by</span><span class="p">(</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="n">SVLEN_group</span><span class="p">)</span> <span class="o">%&gt;%</span>
  <span class="nf">summarize</span><span class="p">(</span><span class="n">count</span> <span class="o">=</span> <span class="nf">n</span><span class="p">())</span>

<span class="n">tmp</span> <span class="o">&lt;-</span> <span class="n">df_grouped_SVLEN_SVTYPE</span>
<span class="n">tmp</span><span class="o">$</span><span class="n">SVLEN_group</span> <span class="o">&lt;-</span> <span class="s">&quot;ALL&quot;</span>
<span class="n">df_grouped_SVLEN_SVTYPE</span> <span class="o">&lt;-</span> <span class="nf">rbind</span><span class="p">(</span><span class="n">df_grouped_SVLEN_SVTYPE</span><span class="p">,</span><span class="n">tmp</span><span class="p">)</span>
<span class="n">df_grouped_SVLEN_SVTYPE</span><span class="o">$</span><span class="n">SVLEN_group</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">df_grouped_SVLEN_SVTYPE</span><span class="o">$</span><span class="n">SVLEN_group</span><span class="p">,</span> <span class="n">ordered</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;ALL&quot;</span><span class="p">,</span> <span class="s">&quot;&lt;100bp&quot;</span><span class="p">,</span> <span class="s">&quot;100bp-500bp&quot;</span><span class="p">,</span> <span class="s">&quot;500bp-2.5kb&quot;</span><span class="p">,</span> <span class="s">&quot;2.5kb-10kb&quot;</span><span class="p">,</span> <span class="s">&quot;10kb-50kb&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50kb&quot;</span><span class="p">))</span>

<span class="c1">#Make a second data frame for scaled plots to include ALL group</span>
<span class="n">df2</span> <span class="o">&lt;-</span> <span class="n">df</span>
<span class="n">df2</span><span class="o">$</span><span class="n">AF_group</span> <span class="o">&lt;-</span> <span class="s">&quot;ALL&quot;</span>
<span class="n">df2</span><span class="o">$</span><span class="n">SVLEN_group</span> <span class="o">&lt;-</span> <span class="s">&quot;ALL&quot;</span>
<span class="n">df2</span> <span class="o">&lt;-</span> <span class="nf">rbind</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="n">df2</span><span class="p">)</span>
<span class="n">df2</span><span class="o">$</span><span class="n">SVLEN_group</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">df2</span><span class="o">$</span><span class="n">SVLEN_group</span><span class="p">,</span> <span class="n">ordered</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;ALL&quot;</span><span class="p">,</span> <span class="s">&quot;&lt;100bp&quot;</span><span class="p">,</span> <span class="s">&quot;100bp-500bp&quot;</span><span class="p">,</span> <span class="s">&quot;500bp-2.5kb&quot;</span><span class="p">,</span> <span class="s">&quot;2.5kb-10kb&quot;</span><span class="p">,</span> <span class="s">&quot;10kb-50kb&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50kb&quot;</span><span class="p">))</span>
<span class="n">df2</span><span class="o">$</span><span class="n">AF_group</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">df2</span><span class="o">$</span><span class="n">AF_group</span><span class="p">,</span> <span class="n">ordered</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;ALL&quot;</span><span class="p">,</span> <span class="s">&quot;AC=1&quot;</span><span class="p">,</span> <span class="s">&quot;AF&lt;1%&quot;</span><span class="p">,</span> <span class="s">&quot;1-10%&quot;</span><span class="p">,</span> <span class="s">&quot;10-50%&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50%&quot;</span><span class="p">))</span>


<span class="c1">#Plots</span>
<span class="c1">##Variant count by SVTYPE</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="n">SVTYPE</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">geom_bar</span><span class="p">(</span><span class="n">width</span><span class="o">=</span><span class="m">0.8</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="s">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="s">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="s">&quot;#F69A57&quot;</span><span class="p">,</span> <span class="s">&quot;#86E496&quot;</span><span class="p">,</span> <span class="s">&quot;#395946&quot;</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">geom_text</span><span class="p">(</span><span class="n">stat</span><span class="o">=</span><span class="s">&#39;count&#39;</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">label</span> <span class="o">=</span> <span class="nf">format</span><span class="p">(</span><span class="n">..count..</span><span class="p">,</span> <span class="n">big.mark</span> <span class="o">=</span> <span class="s">&quot;,&quot;</span><span class="p">,</span> <span class="n">scientific</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)),</span> <span class="n">size</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="s">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="s">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="s">&quot;#F69A57&quot;</span><span class="p">,</span> <span class="s">&quot;#86E496&quot;</span><span class="p">,</span> <span class="s">&quot;#395946&quot;</span><span class="p">),</span> <span class="n">vjust</span><span class="o">=</span><span class="m">-1</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;SV Count&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">scale_y_continuous</span><span class="p">(</span><span class="n">labels</span> <span class="o">=</span> <span class="n">comma</span><span class="p">,</span> <span class="n">expand</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">-0.001</span><span class="p">,</span><span class="m">0</span><span class="p">),</span> <span class="n">limits</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="nf">nrow</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="o">==</span> <span class="s">&quot;DEL&quot;</span><span class="p">])</span> <span class="o">+</span> <span class="m">5000</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">guides</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">theme</span><span class="p">(</span><span class="n">axis.title.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">16</span><span class="p">),</span> <span class="n">axis.ticks.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span> <span class="n">axis.text.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">12</span><span class="p">),</span> <span class="n">axis.text.x</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">14</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="m">90</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="s">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="s">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="s">&quot;#F69A57&quot;</span><span class="p">,</span> <span class="s">&quot;#86E496&quot;</span><span class="p">,</span> <span class="s">&quot;#395946&quot;</span><span class="p">)),</span> <span class="n">panel.background</span> <span class="o">=</span> <span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="s">&quot;white&quot;</span><span class="p">),</span> <span class="n">axis.line.y</span> <span class="o">=</span> <span class="nf">element_line</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> <span class="n">colour</span> <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">)))</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span> <span class="o">=</span> <span class="s">&quot;FTD_filtered_SV_variant_count_plot.png&quot;</span><span class="p">,</span> <span class="n">plot</span> <span class="o">=</span> <span class="nf">last_plot</span><span class="p">(),</span> <span class="n">device</span> <span class="o">=</span> <span class="s">&quot;png&quot;</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="m">7</span><span class="p">,</span> <span class="n">units</span> <span class="o">=</span> <span class="s">&quot;in&quot;</span> <span class="p">)</span>

<span class="c1">##SV count by AF</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">df_grouped_AF_SVTYPE</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">AF_group</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">count</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">geom_bar</span><span class="p">(</span><span class="n">width</span> <span class="o">=</span> <span class="m">0.8</span><span class="p">,</span> <span class="n">position</span><span class="o">=</span><span class="nf">position_stack</span><span class="p">(</span><span class="n">reverse</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">),</span> <span class="n">stat</span><span class="o">=</span><span class="s">&quot;identity&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="s">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="s">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="s">&quot;#F69A57&quot;</span><span class="p">,</span> <span class="s">&quot;#86E496&quot;</span><span class="p">,</span> <span class="s">&quot;#395946&quot;</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">scale_y_continuous</span><span class="p">(</span><span class="n">labels</span> <span class="o">=</span> <span class="n">comma</span><span class="p">,</span> <span class="n">expand</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">-0.0001</span><span class="p">,</span><span class="m">0</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">guides</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;SV Count by AF FTD&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">theme</span><span class="p">(</span><span class="n">axis.title.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">),</span> <span class="n">axis.ticks.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span> <span class="n">axis.text.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">),</span> <span class="n">axis.text.x</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="m">90</span><span class="p">),</span> <span class="n">panel.background</span> <span class="o">=</span> <span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="s">&quot;white&quot;</span><span class="p">),</span> <span class="n">axis.line.y</span> <span class="o">=</span> <span class="nf">element_line</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> <span class="n">colour</span> <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">)))</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span> <span class="o">=</span> <span class="s">&quot;FTD_filtered_SV_count_by_AF_plot.png&quot;</span><span class="p">,</span> <span class="n">plot</span> <span class="o">=</span> <span class="nf">last_plot</span><span class="p">(),</span> <span class="n">device</span> <span class="o">=</span> <span class="s">&quot;png&quot;</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="m">7</span><span class="p">,</span> <span class="n">units</span> <span class="o">=</span> <span class="s">&quot;in&quot;</span> <span class="p">)</span>

<span class="c1">##SV count by AF scaled</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">df2</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">AF_group</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">geom_bar</span><span class="p">(</span><span class="n">width</span> <span class="o">=</span> <span class="m">0.8</span><span class="p">,</span> <span class="n">position</span> <span class="o">=</span> <span class="nf">position_fill</span><span class="p">(</span><span class="n">reverse</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="s">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="s">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="s">&quot;#F69A57&quot;</span><span class="p">,</span> <span class="s">&quot;#86E496&quot;</span><span class="p">,</span> <span class="s">&quot;#395946&quot;</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">scale_y_continuous</span><span class="p">(</span><span class="n">breaks</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0.20</span><span class="p">,</span><span class="m">0.40</span><span class="p">,</span><span class="m">0.60</span><span class="p">,</span><span class="m">0.8</span><span class="p">,</span><span class="m">1</span><span class="p">),</span> <span class="n">labels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;0%&quot;</span><span class="p">,</span> <span class="s">&quot;20%&quot;</span><span class="p">,</span> <span class="s">&quot;40%&quot;</span><span class="p">,</span> <span class="s">&quot;60%&quot;</span><span class="p">,</span> <span class="s">&#39;80%&#39;</span><span class="p">,</span> <span class="s">&quot;100%&quot;</span><span class="p">),</span> <span class="n">expand</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">-0.001</span><span class="p">,</span><span class="m">0.001</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">guides</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;SV Count by AF FTD scaled&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">theme</span><span class="p">(</span><span class="n">axis.title.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">),</span> <span class="n">axis.ticks.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span> <span class="n">axis.text.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">),</span> <span class="n">axis.text.x</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="m">90</span><span class="p">),</span> <span class="n">panel.background</span> <span class="o">=</span> <span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="s">&quot;white&quot;</span><span class="p">),</span> <span class="n">axis.line.y</span> <span class="o">=</span> <span class="nf">element_line</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> <span class="n">colour</span> <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">)))</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span> <span class="o">=</span> <span class="s">&quot;FTD_filtered_SV_count_by_AF_plot_scaled.png&quot;</span><span class="p">,</span> <span class="n">plot</span> <span class="o">=</span> <span class="nf">last_plot</span><span class="p">(),</span> <span class="n">device</span> <span class="o">=</span> <span class="s">&quot;png&quot;</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="m">7</span><span class="p">,</span> <span class="n">units</span> <span class="o">=</span> <span class="s">&quot;in&quot;</span> <span class="p">)</span>

<span class="c1">##SV count by SVLEN</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">df_grouped_SVLEN_SVTYPE</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">SVLEN_group</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">count</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">geom_bar</span><span class="p">(</span><span class="n">width</span> <span class="o">=</span> <span class="m">0.8</span><span class="p">,</span> <span class="n">position</span><span class="o">=</span><span class="nf">position_stack</span><span class="p">(</span><span class="n">reverse</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">),</span> <span class="n">stat</span><span class="o">=</span><span class="s">&quot;identity&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="s">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="s">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="s">&quot;#F69A57&quot;</span><span class="p">,</span> <span class="s">&quot;#86E496&quot;</span><span class="p">,</span> <span class="s">&quot;#395946&quot;</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">scale_y_continuous</span><span class="p">(</span><span class="n">labels</span> <span class="o">=</span> <span class="n">comma</span><span class="p">,</span> <span class="n">expand</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">-0.001</span><span class="p">,</span><span class="m">0</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">guides</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;SV Count by SVLEN FTD&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">theme</span><span class="p">(</span><span class="n">axis.title.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">),</span> <span class="n">axis.ticks.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span> <span class="n">axis.text.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">),</span> <span class="n">axis.text.x</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="m">90</span><span class="p">),</span> <span class="n">panel.background</span> <span class="o">=</span> <span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="s">&quot;white&quot;</span><span class="p">),</span> <span class="n">axis.line.y</span> <span class="o">=</span> <span class="nf">element_line</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> <span class="n">colour</span> <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">)))</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span> <span class="o">=</span> <span class="s">&quot;FTD_filtered_SV_count_by_SVLEN_plot.png&quot;</span><span class="p">,</span> <span class="n">plot</span> <span class="o">=</span> <span class="nf">last_plot</span><span class="p">(),</span> <span class="n">device</span> <span class="o">=</span> <span class="s">&quot;png&quot;</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="m">7</span><span class="p">,</span> <span class="n">units</span> <span class="o">=</span> <span class="s">&quot;in&quot;</span> <span class="p">)</span>

<span class="c1">##SV count by SVLEN scaled</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">ggplot</span><span class="p">(</span><span class="nf">subset</span><span class="p">(</span><span class="n">df2</span><span class="p">,</span> <span class="n">SVTYPE</span> <span class="o">!=</span> <span class="s">&quot;OTH&quot;</span><span class="p">),</span> <span class="nf">aes</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">SVLEN_group</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">geom_bar</span><span class="p">(</span><span class="n">width</span> <span class="o">=</span> <span class="m">0.8</span><span class="p">,,</span> <span class="n">position</span> <span class="o">=</span> <span class="nf">position_fill</span><span class="p">(</span><span class="n">reverse</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="s">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="s">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="s">&quot;#F69A57&quot;</span><span class="p">,</span> <span class="s">&quot;#86E496&quot;</span><span class="p">,</span> <span class="s">&quot;#395946&quot;</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">scale_y_continuous</span><span class="p">(</span><span class="n">breaks</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0.20</span><span class="p">,</span><span class="m">0.40</span><span class="p">,</span><span class="m">0.60</span><span class="p">,</span><span class="m">0.8</span><span class="p">,</span><span class="m">1</span><span class="p">),</span> <span class="n">labels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;0%&quot;</span><span class="p">,</span> <span class="s">&quot;20%&quot;</span><span class="p">,</span> <span class="s">&quot;40%&quot;</span><span class="p">,</span> <span class="s">&quot;60%&quot;</span><span class="p">,</span> <span class="s">&#39;80%&#39;</span><span class="p">,</span> <span class="s">&quot;100%&quot;</span><span class="p">),</span> <span class="n">expand</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">-0.001</span><span class="p">,</span><span class="m">0.001</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">guides</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;SV Count by SVLEN FTD scaled&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">theme</span><span class="p">(</span><span class="n">axis.title.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">),</span> <span class="n">axis.ticks.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span> <span class="n">axis.text.y</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">),</span> <span class="n">axis.text.x</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">20</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="m">90</span><span class="p">),</span> <span class="n">panel.background</span> <span class="o">=</span> <span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="s">&quot;white&quot;</span><span class="p">),</span> <span class="n">axis.line.y</span> <span class="o">=</span> <span class="nf">element_line</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> <span class="n">colour</span> <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">)))</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span> <span class="o">=</span> <span class="s">&quot;FTD_filtered_SV_count_by_SVLEN_plot_scaled.png&quot;</span><span class="p">,</span> <span class="n">plot</span> <span class="o">=</span> <span class="nf">last_plot</span><span class="p">(),</span> <span class="n">device</span> <span class="o">=</span> <span class="s">&quot;png&quot;</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="m">7</span><span class="p">,</span> <span class="n">units</span> <span class="o">=</span> <span class="s">&quot;in&quot;</span> <span class="p">)</span>

<span class="c1">#Save df</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="s">&quot;FTD_descriptives_df.txt&quot;</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Get-descriptives">Get descriptives<a class="anchor-link" href="#Get-descriptives">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Get</p>
<ol>
<li>Median structural variant lenght of filtered data (BND, CTX excluded)</li>
<li>Structural variant type frequencies</li>
<li>Proportion of SVs with allele frequency &lt;1 (including AC=1)</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">#Use dfs from GATK-SV style barplot section </span>

<span class="c1">#Set wd</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;$PATH2/descriptive_stats_and_fig2&quot;</span><span class="p">)</span>

<span class="c1">#Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">gmodels</span><span class="p">)</span>

<span class="c1">#Import dfs</span>
<span class="n">df_LBD</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;LBD_descriptives_df.txt&quot;</span><span class="p">)</span>
<span class="n">df_FTD</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;FTD_descriptives_df.txt&quot;</span><span class="p">)</span>

<span class="c1">#Get median SVLENs</span>
<span class="n">tmp_SVLEN_LBD</span> <span class="o">&lt;-</span> <span class="n">df_LBD</span> <span class="o">%&gt;%</span>
  <span class="nf">filter</span><span class="p">(</span><span class="n">SVTYPE</span> <span class="o">!=</span><span class="s">&quot;OTH&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span>
  <span class="nf">select</span><span class="p">(</span><span class="n">SVLEN</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;LBD median: &quot;</span><span class="p">,</span><span class="nf">median</span><span class="p">(</span><span class="n">tmp_SVLEN_LBD</span><span class="o">$</span><span class="n">SVLEN</span><span class="p">)))</span>

<span class="n">tmp_SVLEN_FTD</span> <span class="o">&lt;-</span> <span class="n">df_FTD</span> <span class="o">%&gt;%</span>
  <span class="nf">filter</span><span class="p">(</span><span class="n">SVTYPE</span> <span class="o">!=</span><span class="s">&quot;OTH&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span>
  <span class="nf">select</span><span class="p">(</span><span class="n">SVLEN</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;FTD median: &quot;</span><span class="p">,</span><span class="nf">median</span><span class="p">(</span><span class="n">tmp_SVLEN_FTD</span><span class="o">$</span><span class="n">SVLEN</span><span class="p">)))</span>


<span class="c1">#Get SVTYPE frequencies</span>
<span class="c1">###Get SVTTYPE frequencies</span>
<span class="nf">print</span><span class="p">(</span><span class="s">&quot;LBD SVTYPES&quot;</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">CrossTable</span><span class="p">(</span><span class="n">df_LBD</span><span class="o">$</span><span class="n">SVTYPE</span><span class="p">))</span>

<span class="nf">print</span><span class="p">(</span><span class="s">&quot;FTD SVTYPES&quot;</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">CrossTable</span><span class="p">(</span><span class="n">df_FTD</span><span class="o">$</span><span class="n">SVTYPE</span><span class="p">))</span>

<span class="c1">#Get proportion of MAF&lt;1 % SVs</span>
<span class="nf">print</span><span class="p">(</span><span class="s">&quot;LBD AF groups&quot;</span><span class="p">)</span>
<span class="nf">CrossTable</span><span class="p">(</span><span class="n">df_LBD</span><span class="o">$</span><span class="n">AF_group</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="s">&quot;FTD AF groups&quot;</span><span class="p">)</span>
<span class="nf">CrossTable</span><span class="p">(</span><span class="n">df_FTD</span><span class="o">$</span><span class="n">AF_group</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Get-descriptives-by-case-control-status">Get descriptives by case-control status<a class="anchor-link" href="#Get-descriptives-by-case-control-status">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%</span><span class="k">load_ext</span> rpy2.ipython
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">#Make R dfs for LBD and FTD case-control status</span>

<span class="c1">#Set wd</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;$PATH2/descriptive_stats_and_fig2&quot;</span><span class="p">)</span>

<span class="c1">#Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">scales</span><span class="p">)</span>

<span class="c1">########LBD cases</span>

<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;LBD_cases_high_quality_subset_clean_GQ300missing_all_chr_INFO.txt&quot;</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;ID&quot;</span><span class="p">,</span><span class="s">&quot;SVTYPE&quot;</span><span class="p">,</span><span class="s">&quot;SVLEN&quot;</span><span class="p">,</span><span class="s">&quot;AF&quot;</span><span class="p">,</span><span class="s">&quot;AC&quot;</span><span class="p">,</span><span class="s">&quot;CHROM&quot;</span><span class="p">)</span>

<span class="c1">#Exclude AF=0 SVs</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="n">df</span> <span class="o">%&gt;%</span>
    <span class="nf">filter</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">AF</span> <span class="o">&gt;</span> <span class="m">0</span><span class="p">)</span>

<span class="c1">#Rename BND and CTX to OTH</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="o">%in%</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;BND&quot;</span><span class="p">,</span><span class="s">&quot;CTX&quot;</span><span class="p">)]</span> <span class="o">&lt;-</span> <span class="s">&quot;OTH&quot;</span>

<span class="c1">#Make SVTYPEs into factors</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="p">,</span> <span class="n">levels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;DEL&quot;</span><span class="p">,</span> <span class="s">&quot;DUP&quot;</span><span class="p">,</span> <span class="s">&quot;CNV&quot;</span><span class="p">,</span> <span class="s">&quot;INS&quot;</span><span class="p">,</span> <span class="s">&quot;INV&quot;</span><span class="p">,</span> <span class="s">&quot;CPX&quot;</span><span class="p">,</span> <span class="s">&quot;OTH&quot;</span><span class="p">),</span> <span class="n">labels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;DEL&quot;</span><span class="p">,</span> <span class="s">&quot;DUP&quot;</span><span class="p">,</span> <span class="s">&quot;CNV&quot;</span><span class="p">,</span> <span class="s">&quot;INS&quot;</span><span class="p">,</span> <span class="s">&quot;INV&quot;</span><span class="p">,</span> <span class="s">&quot;CPX&quot;</span><span class="p">,</span> <span class="s">&quot;OTH&quot;</span><span class="p">))</span>


<span class="c1">#Add SVLEN and AF groups, add AC=1 separately since threshold varies accordig to number of samples</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVLEN_group</span> <span class="o">&lt;-</span> <span class="nf">cut</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">SVLEN</span><span class="p">,</span> <span class="n">breaks</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">-2</span><span class="p">,</span><span class="m">0</span><span class="p">,</span><span class="m">100</span><span class="p">,</span><span class="m">500</span><span class="p">,</span><span class="m">2500</span><span class="p">,</span><span class="m">10000</span><span class="p">,</span><span class="m">50000</span><span class="p">,</span><span class="m">10</span><span class="o">^</span><span class="m">10</span><span class="p">),</span> <span class="n">labels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;unresolved&quot;</span><span class="p">,</span> <span class="s">&quot;&lt;100bp&quot;</span><span class="p">,</span> <span class="s">&quot;100bp-500bp&quot;</span><span class="p">,</span> <span class="s">&quot;500bp-2.5kb&quot;</span><span class="p">,</span> <span class="s">&quot;2.5kb-10kb&quot;</span><span class="p">,</span> <span class="s">&quot;10kb-50kb&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50kb&quot;</span><span class="p">))</span>
<span class="n">df</span><span class="o">$</span><span class="n">AF_group</span> <span class="o">&lt;-</span> <span class="nf">cut</span><span class="p">(</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">AF</span><span class="p">),</span> <span class="n">breaks</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0.01</span><span class="p">,</span><span class="m">0.10</span><span class="p">,</span><span class="m">0.50</span><span class="p">,</span><span class="m">1</span><span class="p">),</span> <span class="n">labels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;AF&lt;1%&quot;</span><span class="p">,</span> <span class="s">&quot;1-10%&quot;</span><span class="p">,</span> <span class="s">&quot;10-50%&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50%&quot;</span><span class="p">))</span>
<span class="nf">levels</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">AF_group</span><span class="p">)[</span><span class="m">5</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s">&quot;AC=1&quot;</span>
<span class="n">df</span><span class="o">$</span><span class="n">AF_group</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">AC</span> <span class="o">==</span> <span class="m">1</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s">&quot;AC=1&quot;</span>

<span class="c1">#Save</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="s">&quot;LBD_cases_descriptives_df.txt&quot;</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>

<span class="nf">rm</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>

<span class="c1">########LBD controls</span>

<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;LBD_controls_high_quality_subset_clean_GQ300missing_all_chr_INFO.txt&quot;</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;ID&quot;</span><span class="p">,</span><span class="s">&quot;SVTYPE&quot;</span><span class="p">,</span><span class="s">&quot;SVLEN&quot;</span><span class="p">,</span><span class="s">&quot;AF&quot;</span><span class="p">,</span><span class="s">&quot;AC&quot;</span><span class="p">,</span><span class="s">&quot;CHROM&quot;</span><span class="p">)</span>

<span class="c1">#Exclude AF=0 SVs</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="n">df</span> <span class="o">%&gt;%</span>
    <span class="nf">filter</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">AF</span> <span class="o">&gt;</span> <span class="m">0</span><span class="p">)</span>

<span class="c1">#Rename BND and CTX to OTH</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="o">%in%</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;BND&quot;</span><span class="p">,</span><span class="s">&quot;CTX&quot;</span><span class="p">)]</span> <span class="o">&lt;-</span> <span class="s">&quot;OTH&quot;</span>

<span class="c1">#Make SVTYPEs into factors</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="p">,</span> <span class="n">levels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;DEL&quot;</span><span class="p">,</span> <span class="s">&quot;DUP&quot;</span><span class="p">,</span> <span class="s">&quot;CNV&quot;</span><span class="p">,</span> <span class="s">&quot;INS&quot;</span><span class="p">,</span> <span class="s">&quot;INV&quot;</span><span class="p">,</span> <span class="s">&quot;CPX&quot;</span><span class="p">,</span> <span class="s">&quot;OTH&quot;</span><span class="p">),</span> <span class="n">labels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;DEL&quot;</span><span class="p">,</span> <span class="s">&quot;DUP&quot;</span><span class="p">,</span> <span class="s">&quot;CNV&quot;</span><span class="p">,</span> <span class="s">&quot;INS&quot;</span><span class="p">,</span> <span class="s">&quot;INV&quot;</span><span class="p">,</span> <span class="s">&quot;CPX&quot;</span><span class="p">,</span> <span class="s">&quot;OTH&quot;</span><span class="p">))</span>


<span class="c1">#Add SVLEN and AF groups, add AC=1 separately since threshold varies accordig to number of samples</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVLEN_group</span> <span class="o">&lt;-</span> <span class="nf">cut</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">SVLEN</span><span class="p">,</span> <span class="n">breaks</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">-2</span><span class="p">,</span><span class="m">0</span><span class="p">,</span><span class="m">100</span><span class="p">,</span><span class="m">500</span><span class="p">,</span><span class="m">2500</span><span class="p">,</span><span class="m">10000</span><span class="p">,</span><span class="m">50000</span><span class="p">,</span><span class="m">10</span><span class="o">^</span><span class="m">10</span><span class="p">),</span> <span class="n">labels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;unresolved&quot;</span><span class="p">,</span> <span class="s">&quot;&lt;100bp&quot;</span><span class="p">,</span> <span class="s">&quot;100bp-500bp&quot;</span><span class="p">,</span> <span class="s">&quot;500bp-2.5kb&quot;</span><span class="p">,</span> <span class="s">&quot;2.5kb-10kb&quot;</span><span class="p">,</span> <span class="s">&quot;10kb-50kb&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50kb&quot;</span><span class="p">))</span>
<span class="n">df</span><span class="o">$</span><span class="n">AF_group</span> <span class="o">&lt;-</span> <span class="nf">cut</span><span class="p">(</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">AF</span><span class="p">),</span> <span class="n">breaks</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0.01</span><span class="p">,</span><span class="m">0.10</span><span class="p">,</span><span class="m">0.50</span><span class="p">,</span><span class="m">1</span><span class="p">),</span> <span class="n">labels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;AF&lt;1%&quot;</span><span class="p">,</span> <span class="s">&quot;1-10%&quot;</span><span class="p">,</span> <span class="s">&quot;10-50%&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50%&quot;</span><span class="p">))</span>
<span class="nf">levels</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">AF_group</span><span class="p">)[</span><span class="m">5</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s">&quot;AC=1&quot;</span>
<span class="n">df</span><span class="o">$</span><span class="n">AF_group</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">AC</span> <span class="o">==</span> <span class="m">1</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s">&quot;AC=1&quot;</span>

<span class="c1">#Save</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="s">&quot;LBD_controls_descriptives_df.txt&quot;</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>

<span class="nf">rm</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>

<span class="c1">########FTD cases</span>

<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;FTD_cases_high_quality_subset_clean_GQ300missing_all_chr_INFO.txt&quot;</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;ID&quot;</span><span class="p">,</span><span class="s">&quot;SVTYPE&quot;</span><span class="p">,</span><span class="s">&quot;SVLEN&quot;</span><span class="p">,</span><span class="s">&quot;AF&quot;</span><span class="p">,</span><span class="s">&quot;AC&quot;</span><span class="p">,</span><span class="s">&quot;CHROM&quot;</span><span class="p">)</span>

<span class="c1">#Exclude AF=0 SVs</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="n">df</span> <span class="o">%&gt;%</span>
    <span class="nf">filter</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">AF</span> <span class="o">&gt;</span> <span class="m">0</span><span class="p">)</span>

<span class="c1">#Rename BND and CTX to OTH</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="o">%in%</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;BND&quot;</span><span class="p">,</span><span class="s">&quot;CTX&quot;</span><span class="p">)]</span> <span class="o">&lt;-</span> <span class="s">&quot;OTH&quot;</span>

<span class="c1">#Make SVTYPEs into factors</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="p">,</span> <span class="n">levels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;DEL&quot;</span><span class="p">,</span> <span class="s">&quot;DUP&quot;</span><span class="p">,</span> <span class="s">&quot;CNV&quot;</span><span class="p">,</span> <span class="s">&quot;INS&quot;</span><span class="p">,</span> <span class="s">&quot;INV&quot;</span><span class="p">,</span> <span class="s">&quot;CPX&quot;</span><span class="p">,</span> <span class="s">&quot;OTH&quot;</span><span class="p">),</span> <span class="n">labels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;DEL&quot;</span><span class="p">,</span> <span class="s">&quot;DUP&quot;</span><span class="p">,</span> <span class="s">&quot;CNV&quot;</span><span class="p">,</span> <span class="s">&quot;INS&quot;</span><span class="p">,</span> <span class="s">&quot;INV&quot;</span><span class="p">,</span> <span class="s">&quot;CPX&quot;</span><span class="p">,</span> <span class="s">&quot;OTH&quot;</span><span class="p">))</span>


<span class="c1">#Add SVLEN and AF groups, add AC=1 separately since threshold varies accordig to number of samples</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVLEN_group</span> <span class="o">&lt;-</span> <span class="nf">cut</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">SVLEN</span><span class="p">,</span> <span class="n">breaks</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">-2</span><span class="p">,</span><span class="m">0</span><span class="p">,</span><span class="m">100</span><span class="p">,</span><span class="m">500</span><span class="p">,</span><span class="m">2500</span><span class="p">,</span><span class="m">10000</span><span class="p">,</span><span class="m">50000</span><span class="p">,</span><span class="m">10</span><span class="o">^</span><span class="m">10</span><span class="p">),</span> <span class="n">labels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;unresolved&quot;</span><span class="p">,</span> <span class="s">&quot;&lt;100bp&quot;</span><span class="p">,</span> <span class="s">&quot;100bp-500bp&quot;</span><span class="p">,</span> <span class="s">&quot;500bp-2.5kb&quot;</span><span class="p">,</span> <span class="s">&quot;2.5kb-10kb&quot;</span><span class="p">,</span> <span class="s">&quot;10kb-50kb&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50kb&quot;</span><span class="p">))</span>
<span class="n">df</span><span class="o">$</span><span class="n">AF_group</span> <span class="o">&lt;-</span> <span class="nf">cut</span><span class="p">(</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">AF</span><span class="p">),</span> <span class="n">breaks</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0.01</span><span class="p">,</span><span class="m">0.10</span><span class="p">,</span><span class="m">0.50</span><span class="p">,</span><span class="m">1</span><span class="p">),</span> <span class="n">labels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;AF&lt;1%&quot;</span><span class="p">,</span> <span class="s">&quot;1-10%&quot;</span><span class="p">,</span> <span class="s">&quot;10-50%&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50%&quot;</span><span class="p">))</span>
<span class="nf">levels</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">AF_group</span><span class="p">)[</span><span class="m">5</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s">&quot;AC=1&quot;</span>
<span class="n">df</span><span class="o">$</span><span class="n">AF_group</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">AC</span> <span class="o">==</span> <span class="m">1</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s">&quot;AC=1&quot;</span>

<span class="c1">#Save</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="s">&quot;FTD_cases_descriptives_df.txt&quot;</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>

<span class="nf">rm</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>

<span class="c1">########FTD controls</span>

<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;FTD_controls_high_quality_subset_clean_GQ300missing_all_chr_INFO.txt&quot;</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;ID&quot;</span><span class="p">,</span><span class="s">&quot;SVTYPE&quot;</span><span class="p">,</span><span class="s">&quot;SVLEN&quot;</span><span class="p">,</span><span class="s">&quot;AF&quot;</span><span class="p">,</span><span class="s">&quot;AC&quot;</span><span class="p">,</span><span class="s">&quot;CHROM&quot;</span><span class="p">)</span>

<span class="c1">#Exclude AF=0 SVs</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="n">df</span> <span class="o">%&gt;%</span>
    <span class="nf">filter</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">AF</span> <span class="o">&gt;</span> <span class="m">0</span><span class="p">)</span>

<span class="c1">#Rename BND and CTX to OTH</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="o">%in%</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;BND&quot;</span><span class="p">,</span><span class="s">&quot;CTX&quot;</span><span class="p">)]</span> <span class="o">&lt;-</span> <span class="s">&quot;OTH&quot;</span>

<span class="c1">#Make SVTYPEs into factors</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span> <span class="p">,</span> <span class="n">levels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;DEL&quot;</span><span class="p">,</span> <span class="s">&quot;DUP&quot;</span><span class="p">,</span> <span class="s">&quot;CNV&quot;</span><span class="p">,</span> <span class="s">&quot;INS&quot;</span><span class="p">,</span> <span class="s">&quot;INV&quot;</span><span class="p">,</span> <span class="s">&quot;CPX&quot;</span><span class="p">,</span> <span class="s">&quot;OTH&quot;</span><span class="p">),</span> <span class="n">labels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;DEL&quot;</span><span class="p">,</span> <span class="s">&quot;DUP&quot;</span><span class="p">,</span> <span class="s">&quot;CNV&quot;</span><span class="p">,</span> <span class="s">&quot;INS&quot;</span><span class="p">,</span> <span class="s">&quot;INV&quot;</span><span class="p">,</span> <span class="s">&quot;CPX&quot;</span><span class="p">,</span> <span class="s">&quot;OTH&quot;</span><span class="p">))</span>


<span class="c1">#Add SVLEN and AF groups, add AC=1 separately since threshold varies accordig to number of samples</span>
<span class="n">df</span><span class="o">$</span><span class="n">SVLEN_group</span> <span class="o">&lt;-</span> <span class="nf">cut</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">SVLEN</span><span class="p">,</span> <span class="n">breaks</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">-2</span><span class="p">,</span><span class="m">0</span><span class="p">,</span><span class="m">100</span><span class="p">,</span><span class="m">500</span><span class="p">,</span><span class="m">2500</span><span class="p">,</span><span class="m">10000</span><span class="p">,</span><span class="m">50000</span><span class="p">,</span><span class="m">10</span><span class="o">^</span><span class="m">10</span><span class="p">),</span> <span class="n">labels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;unresolved&quot;</span><span class="p">,</span> <span class="s">&quot;&lt;100bp&quot;</span><span class="p">,</span> <span class="s">&quot;100bp-500bp&quot;</span><span class="p">,</span> <span class="s">&quot;500bp-2.5kb&quot;</span><span class="p">,</span> <span class="s">&quot;2.5kb-10kb&quot;</span><span class="p">,</span> <span class="s">&quot;10kb-50kb&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50kb&quot;</span><span class="p">))</span>
<span class="n">df</span><span class="o">$</span><span class="n">AF_group</span> <span class="o">&lt;-</span> <span class="nf">cut</span><span class="p">(</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">AF</span><span class="p">),</span> <span class="n">breaks</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0.01</span><span class="p">,</span><span class="m">0.10</span><span class="p">,</span><span class="m">0.50</span><span class="p">,</span><span class="m">1</span><span class="p">),</span> <span class="n">labels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;AF&lt;1%&quot;</span><span class="p">,</span> <span class="s">&quot;1-10%&quot;</span><span class="p">,</span> <span class="s">&quot;10-50%&quot;</span><span class="p">,</span> <span class="s">&quot;&gt;50%&quot;</span><span class="p">))</span>
<span class="nf">levels</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">AF_group</span><span class="p">)[</span><span class="m">5</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s">&quot;AC=1&quot;</span>
<span class="n">df</span><span class="o">$</span><span class="n">AF_group</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">AC</span> <span class="o">==</span> <span class="m">1</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s">&quot;AC=1&quot;</span>

<span class="c1">#Save</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="s">&quot;FTD_controls_descriptives_df.txt&quot;</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>

<span class="nf">rm</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">#####LBD case and control descriptives</span>
<span class="c1">#Set wd</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;$PATH2/descriptive_stats_and_fig2&quot;</span><span class="p">)</span>

<span class="c1">#Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">gmodels</span><span class="p">)</span>

<span class="c1">#Import dfs</span>
<span class="n">df_LBD_cases</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;LBD_cases_descriptives_df.txt&quot;</span><span class="p">)</span>
<span class="n">df_LBD_controls</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;LBD_controls_descriptives_df.txt&quot;</span><span class="p">)</span>

<span class="c1">#Add group info</span>
<span class="n">df_LBD_controls</span><span class="o">$</span><span class="n">V5</span> <span class="o">&lt;-</span> <span class="s">&quot;Control&quot;</span>
<span class="n">df_LBD_cases</span><span class="o">$</span><span class="n">V5</span> <span class="o">&lt;-</span> <span class="s">&quot;Case&quot;</span>

<span class="c1">#Merge</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">rbind</span><span class="p">(</span><span class="n">df_LBD_cases</span><span class="p">,</span> <span class="n">df_LBD_controls</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)[</span><span class="m">9</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s">&quot;GROUP&quot;</span>

<span class="c1">#Get median SVLENs</span>
<span class="n">tmp_SVLEN_LBD_cases</span> <span class="o">&lt;-</span> <span class="n">df_LBD_cases</span> <span class="o">%&gt;%</span>
  <span class="nf">filter</span><span class="p">(</span><span class="n">SVTYPE</span> <span class="o">!=</span><span class="s">&quot;OTH&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span>
  <span class="nf">select</span><span class="p">(</span><span class="n">SVLEN</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;LBD_cases median: &quot;</span><span class="p">,</span><span class="nf">median</span><span class="p">(</span><span class="n">tmp_SVLEN_LBD_cases</span><span class="o">$</span><span class="n">SVLEN</span><span class="p">)))</span>

<span class="n">tmp_SVLEN_LBD_controls</span> <span class="o">&lt;-</span> <span class="n">df_LBD_controls</span> <span class="o">%&gt;%</span>
  <span class="nf">filter</span><span class="p">(</span><span class="n">SVTYPE</span> <span class="o">!=</span><span class="s">&quot;OTH&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span>
  <span class="nf">select</span><span class="p">(</span><span class="n">SVLEN</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;LBD_controls median: &quot;</span><span class="p">,</span><span class="nf">median</span><span class="p">(</span><span class="n">tmp_SVLEN_LBD_controls</span><span class="o">$</span><span class="n">SVLEN</span><span class="p">)))</span>


<span class="c1">#Get SVTYPE frequencies</span>
<span class="c1">###Get SVTTYPE frequencies</span>
<span class="nf">print</span><span class="p">(</span><span class="s">&quot;LBD_cases SVTYPES&quot;</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">CrossTable</span><span class="p">(</span><span class="n">df_LBD_cases</span><span class="o">$</span><span class="n">SVTYPE</span><span class="p">))</span>

<span class="nf">print</span><span class="p">(</span><span class="s">&quot;LBD_controls SVTYPES&quot;</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">CrossTable</span><span class="p">(</span><span class="n">df_LBD_controls</span><span class="o">$</span><span class="n">SVTYPE</span><span class="p">))</span>

<span class="c1">###Statistical test: compare case-control SVTYPES per SVTYPE</span>
<span class="n">m</span> <span class="o">&lt;-</span> <span class="nf">as.matrix</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">GROUP</span><span class="p">,</span> <span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="n">m</span><span class="p">)</span>

<span class="nf">for </span><span class="p">(</span><span class="n">i</span> <span class="n">in</span> <span class="m">1</span><span class="o">:</span><span class="m">6</span><span class="p">)</span> <span class="p">{</span>
    <span class="n">m_tmp</span> <span class="o">&lt;-</span> <span class="nf">matrix</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">m</span><span class="p">[</span><span class="m">1</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="nf">rowSums</span><span class="p">(</span><span class="n">m</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span><span class="o">-</span><span class="n">m</span><span class="p">[</span><span class="m">1</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="n">m</span><span class="p">[</span><span class="m">2</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="nf">rowSums</span><span class="p">(</span><span class="n">m</span><span class="p">)[</span><span class="m">2</span><span class="p">]</span><span class="o">-</span><span class="n">m</span><span class="p">[</span><span class="m">2</span><span class="p">,</span><span class="n">i</span><span class="p">]),</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">,</span> <span class="n">byrow</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
    <span class="nf">print</span><span class="p">(</span><span class="n">m_tmp</span><span class="p">)</span>
    <span class="nf">print</span><span class="p">(</span><span class="nf">fisher.test</span><span class="p">(</span><span class="n">m_tmp</span><span class="p">))</span>
<span class="p">}</span>

<span class="c1">#Get SVLEn group frequencies</span>
<span class="nf">print</span><span class="p">(</span><span class="s">&quot;LBD cases SVLENs&quot;</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">CrossTable</span><span class="p">(</span><span class="n">df_LBD_cases</span><span class="o">$</span><span class="n">SVLEN_group</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="s">&quot;LBD controls SVLENs&quot;</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">CrossTable</span><span class="p">(</span><span class="n">df_LBD_controls</span><span class="o">$</span><span class="n">SVLEN_group</span><span class="p">))</span>

<span class="n">m</span> <span class="o">&lt;-</span> <span class="nf">as.matrix</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">GROUP</span><span class="p">,</span> <span class="n">df</span><span class="o">$</span><span class="n">SVLEN_group</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="n">m</span><span class="p">)</span>

<span class="nf">for </span><span class="p">(</span><span class="n">i</span> <span class="n">in</span> <span class="m">1</span><span class="o">:</span><span class="m">7</span><span class="p">)</span> <span class="p">{</span>
    <span class="n">m_tmp</span> <span class="o">&lt;-</span> <span class="nf">matrix</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">m</span><span class="p">[</span><span class="m">1</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="nf">rowSums</span><span class="p">(</span><span class="n">m</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span><span class="o">-</span><span class="n">m</span><span class="p">[</span><span class="m">1</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="n">m</span><span class="p">[</span><span class="m">2</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="nf">rowSums</span><span class="p">(</span><span class="n">m</span><span class="p">)[</span><span class="m">2</span><span class="p">]</span><span class="o">-</span><span class="n">m</span><span class="p">[</span><span class="m">2</span><span class="p">,</span><span class="n">i</span><span class="p">]),</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">,</span> <span class="n">byrow</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
    <span class="nf">print</span><span class="p">(</span><span class="n">m_tmp</span><span class="p">)</span>
    <span class="nf">print</span><span class="p">(</span><span class="nf">fisher.test</span><span class="p">(</span><span class="n">m_tmp</span><span class="p">))</span>
<span class="p">}</span>

<span class="c1">#Get proportion of MAF&lt;1 % SVs</span>
<span class="nf">print</span><span class="p">(</span><span class="s">&quot;LBD_cases AF groups&quot;</span><span class="p">)</span>
<span class="nf">CrossTable</span><span class="p">(</span><span class="n">df_LBD_cases</span><span class="o">$</span><span class="n">AF_group</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="s">&quot;LBD_controls AF groups&quot;</span><span class="p">)</span>
<span class="nf">CrossTable</span><span class="p">(</span><span class="n">df_LBD_controls</span><span class="o">$</span><span class="n">AF_group</span><span class="p">)</span>

<span class="n">m</span> <span class="o">&lt;-</span> <span class="nf">as.matrix</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">GROUP</span><span class="p">,</span> <span class="n">df</span><span class="o">$</span><span class="n">AF_group</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="n">m</span><span class="p">)</span>

<span class="nf">for </span><span class="p">(</span><span class="n">i</span> <span class="n">in</span> <span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">)</span> <span class="p">{</span>
    <span class="n">m_tmp</span> <span class="o">&lt;-</span> <span class="nf">matrix</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">m</span><span class="p">[</span><span class="m">1</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="nf">rowSums</span><span class="p">(</span><span class="n">m</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span><span class="o">-</span><span class="n">m</span><span class="p">[</span><span class="m">1</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="n">m</span><span class="p">[</span><span class="m">2</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="nf">rowSums</span><span class="p">(</span><span class="n">m</span><span class="p">)[</span><span class="m">2</span><span class="p">]</span><span class="o">-</span><span class="n">m</span><span class="p">[</span><span class="m">2</span><span class="p">,</span><span class="n">i</span><span class="p">]),</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">,</span> <span class="n">byrow</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
    <span class="nf">print</span><span class="p">(</span><span class="n">m_tmp</span><span class="p">)</span>
    <span class="nf">print</span><span class="p">(</span><span class="nf">fisher.test</span><span class="p">(</span><span class="n">m_tmp</span><span class="p">))</span>
<span class="p">}</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">#####FTD case and control descriptives</span>
<span class="c1">#Set wd</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;$PATH2/descriptive_stats_and_fig2&quot;</span><span class="p">)</span>

<span class="c1">#Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">gmodels</span><span class="p">)</span>

<span class="c1">#Import dfs</span>
<span class="n">df_FTD_cases</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;FTD_cases_descriptives_df.txt&quot;</span><span class="p">)</span>
<span class="n">df_FTD_controls</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;FTD_controls_descriptives_df.txt&quot;</span><span class="p">)</span>

<span class="c1">#Add group info</span>
<span class="n">df_FTD_controls</span><span class="o">$</span><span class="n">V5</span> <span class="o">&lt;-</span> <span class="s">&quot;Control&quot;</span>
<span class="n">df_FTD_cases</span><span class="o">$</span><span class="n">V5</span> <span class="o">&lt;-</span> <span class="s">&quot;Case&quot;</span>

<span class="c1">#Merge</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">rbind</span><span class="p">(</span><span class="n">df_FTD_cases</span><span class="p">,</span> <span class="n">df_FTD_controls</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)[</span><span class="m">9</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s">&quot;GROUP&quot;</span>

<span class="c1">#Get median SVLENs</span>
<span class="n">tmp_SVLEN_FTD_cases</span> <span class="o">&lt;-</span> <span class="n">df_FTD_cases</span> <span class="o">%&gt;%</span>
  <span class="nf">filter</span><span class="p">(</span><span class="n">SVTYPE</span> <span class="o">!=</span><span class="s">&quot;OTH&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span>
  <span class="nf">select</span><span class="p">(</span><span class="n">SVLEN</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;FTD_cases median: &quot;</span><span class="p">,</span><span class="nf">median</span><span class="p">(</span><span class="n">tmp_SVLEN_FTD_cases</span><span class="o">$</span><span class="n">SVLEN</span><span class="p">)))</span>

<span class="n">tmp_SVLEN_FTD_controls</span> <span class="o">&lt;-</span> <span class="n">df_FTD_controls</span> <span class="o">%&gt;%</span>
  <span class="nf">filter</span><span class="p">(</span><span class="n">SVTYPE</span> <span class="o">!=</span><span class="s">&quot;OTH&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span>
  <span class="nf">select</span><span class="p">(</span><span class="n">SVLEN</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;FTD_controls median: &quot;</span><span class="p">,</span><span class="nf">median</span><span class="p">(</span><span class="n">tmp_SVLEN_FTD_controls</span><span class="o">$</span><span class="n">SVLEN</span><span class="p">)))</span>


<span class="c1">#Get SVTYPE frequencies</span>
<span class="c1">###Get SVTTYPE frequencies</span>
<span class="nf">print</span><span class="p">(</span><span class="s">&quot;FTD_cases SVTYPES&quot;</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">CrossTable</span><span class="p">(</span><span class="n">df_FTD_cases</span><span class="o">$</span><span class="n">SVTYPE</span><span class="p">))</span>

<span class="nf">print</span><span class="p">(</span><span class="s">&quot;FTD_controls SVTYPES&quot;</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">CrossTable</span><span class="p">(</span><span class="n">df_FTD_controls</span><span class="o">$</span><span class="n">SVTYPE</span><span class="p">))</span>

<span class="n">m</span> <span class="o">&lt;-</span> <span class="nf">as.matrix</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">GROUP</span><span class="p">,</span> <span class="n">df</span><span class="o">$</span><span class="n">SVTYPE</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="n">m</span><span class="p">)</span>

<span class="nf">for </span><span class="p">(</span><span class="n">i</span> <span class="n">in</span> <span class="m">1</span><span class="o">:</span><span class="m">6</span><span class="p">)</span> <span class="p">{</span>
    <span class="n">m_tmp</span> <span class="o">&lt;-</span> <span class="nf">matrix</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">m</span><span class="p">[</span><span class="m">1</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="nf">rowSums</span><span class="p">(</span><span class="n">m</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span><span class="o">-</span><span class="n">m</span><span class="p">[</span><span class="m">1</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="n">m</span><span class="p">[</span><span class="m">2</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="nf">rowSums</span><span class="p">(</span><span class="n">m</span><span class="p">)[</span><span class="m">2</span><span class="p">]</span><span class="o">-</span><span class="n">m</span><span class="p">[</span><span class="m">2</span><span class="p">,</span><span class="n">i</span><span class="p">]),</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">,</span> <span class="n">byrow</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
    <span class="nf">print</span><span class="p">(</span><span class="n">m_tmp</span><span class="p">)</span>
    <span class="nf">print</span><span class="p">(</span><span class="nf">fisher.test</span><span class="p">(</span><span class="n">m_tmp</span><span class="p">))</span>
<span class="p">}</span>

<span class="c1">#Get SVLEn group frequencies</span>
<span class="nf">print</span><span class="p">(</span><span class="s">&quot;FTD cases SVLENs&quot;</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">CrossTable</span><span class="p">(</span><span class="n">df_FTD_cases</span><span class="o">$</span><span class="n">SVLEN_group</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="s">&quot;FTD controls SVLENs&quot;</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">CrossTable</span><span class="p">(</span><span class="n">df_FTD_controls</span><span class="o">$</span><span class="n">SVLEN_group</span><span class="p">))</span>

<span class="n">m</span> <span class="o">&lt;-</span> <span class="nf">as.matrix</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">GROUP</span><span class="p">,</span> <span class="n">df</span><span class="o">$</span><span class="n">SVLEN_group</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="n">m</span><span class="p">)</span>

<span class="nf">for </span><span class="p">(</span><span class="n">i</span> <span class="n">in</span> <span class="m">1</span><span class="o">:</span><span class="m">7</span><span class="p">)</span> <span class="p">{</span>
    <span class="n">m_tmp</span> <span class="o">&lt;-</span> <span class="nf">matrix</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">m</span><span class="p">[</span><span class="m">1</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="nf">rowSums</span><span class="p">(</span><span class="n">m</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span><span class="o">-</span><span class="n">m</span><span class="p">[</span><span class="m">1</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="n">m</span><span class="p">[</span><span class="m">2</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="nf">rowSums</span><span class="p">(</span><span class="n">m</span><span class="p">)[</span><span class="m">2</span><span class="p">]</span><span class="o">-</span><span class="n">m</span><span class="p">[</span><span class="m">2</span><span class="p">,</span><span class="n">i</span><span class="p">]),</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">,</span> <span class="n">byrow</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
    <span class="nf">print</span><span class="p">(</span><span class="n">m_tmp</span><span class="p">)</span>
    <span class="nf">print</span><span class="p">(</span><span class="nf">fisher.test</span><span class="p">(</span><span class="n">m_tmp</span><span class="p">))</span>
<span class="p">}</span>

<span class="c1">#Get proportion of MAF&lt;1 % SVs</span>
<span class="nf">print</span><span class="p">(</span><span class="s">&quot;FTD_cases AF groups&quot;</span><span class="p">)</span>
<span class="nf">CrossTable</span><span class="p">(</span><span class="n">df_FTD_cases</span><span class="o">$</span><span class="n">AF_group</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="s">&quot;FTD_controls AF groups&quot;</span><span class="p">)</span>
<span class="nf">CrossTable</span><span class="p">(</span><span class="n">df_FTD_controls</span><span class="o">$</span><span class="n">AF_group</span><span class="p">)</span>

<span class="n">m</span> <span class="o">&lt;-</span> <span class="nf">as.matrix</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">GROUP</span><span class="p">,</span> <span class="n">df</span><span class="o">$</span><span class="n">AF_group</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="n">m</span><span class="p">)</span>

<span class="nf">for </span><span class="p">(</span><span class="n">i</span> <span class="n">in</span> <span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">)</span> <span class="p">{</span>
    <span class="n">m_tmp</span> <span class="o">&lt;-</span> <span class="nf">matrix</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">m</span><span class="p">[</span><span class="m">1</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="nf">rowSums</span><span class="p">(</span><span class="n">m</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span><span class="o">-</span><span class="n">m</span><span class="p">[</span><span class="m">1</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="n">m</span><span class="p">[</span><span class="m">2</span><span class="p">,</span><span class="n">i</span><span class="p">],</span><span class="nf">rowSums</span><span class="p">(</span><span class="n">m</span><span class="p">)[</span><span class="m">2</span><span class="p">]</span><span class="o">-</span><span class="n">m</span><span class="p">[</span><span class="m">2</span><span class="p">,</span><span class="n">i</span><span class="p">]),</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">,</span> <span class="n">byrow</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
    <span class="nf">print</span><span class="p">(</span><span class="n">m_tmp</span><span class="p">)</span>
    <span class="nf">print</span><span class="p">(</span><span class="nf">fisher.test</span><span class="p">(</span><span class="n">m_tmp</span><span class="p">))</span>
<span class="p">}</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Structural-variant-validation-vs.-Nanopore">Structural variant validation vs. Nanopore<a class="anchor-link" href="#Structural-variant-validation-vs.-Nanopore">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>We have 20 samples that have been also run on Nanopore and SV called with Sniffles2.</p>
<p>Samples:<br>
UMARY634 = RES08390<br>
UMARY1544 = RES08094<br>
UMARY4542 = RES08098<br>
UMARY5077 = RES08391<br>
UMARY1571 = RES08095<br>
UMIC1205 = RES00064<br>
UMIC1527 = RES00275<br>
UMIC1601 = RES00071<br>
UMIC959 = RES08227<br>
BARCS1192 = RES01897<br>
BARCS1429 = RES01877<br>
BARCS1720 = RES01827<br>
BARCS1777 = RES01859<br>
INDU20004038 = RES02284<br>
OREG2142 = RES00032<br>
OREG2595 = RES00048</p>
<p>X4904 = RES05724<br>
X4948 = RES05722<br>
X4980 = RES05720<br>
X5324 = RES05717</p>
<p>Goals:</p>
<ol>
<li>Structural variant validation per-sample per structural variant type</li>
<li>Genotype concordance of top GWAS hits</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Unfiltered-data">Unfiltered data<a class="anchor-link" href="#Unfiltered-data">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Lewy-body-dementia-case/control-samples">Lewy body dementia case/control samples<a class="anchor-link" href="#Lewy-body-dementia-case/control-samples">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Extract each LBD sample of interest separately from 1% GATK-SV (all variants on autosomes), include only SVs with an alternative allele (=exclude 0/0s)</span>
module load samtools
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation
mkdir SR_vcfs
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/SR_vcfs

<span class="nv">SR_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    bcftools view <span class="se">\</span>
    -r chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22 <span class="se">\</span>
    --threads <span class="nv">$SLURM_CPUS_PER_TASK</span> <span class="se">\</span>
    -s <span class="nv">$SAMPLE</span> <span class="se">\</span>
    -Ou <span class="se">\</span>
    /LBD_gatksv_1perc_fdr_cleaned_filters_qual_recalibrated/LBD_1perc_fdr.cleaned_filters_qual_recalibrated.LNGsampleID.vcf.gz <span class="se">\</span>
    <span class="p">|</span> <span class="se">\</span>
    bcftools view -i <span class="s1">&#39;GT[*]=&quot;alt&quot;&#39;</span> <span class="se">\</span>
    --threads <span class="nv">$SLURM_CPUS_PER_TASK</span> <span class="se">\</span>
    -Oz <span class="se">\</span>
    -o <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_altgtonly_LBD_1perc_fdr.cleaned_filters_qual_recalibrated.LNGsampleID.vcf.gz 
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Index vcfs</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>//nanopore_vs_sr_validation/SR_vcfs
parallel -j1 tabix -p vcf <span class="o">{}</span> ::: *_altgtonly_LBD_1perc_fdr.cleaned_filters_qual_recalibrated.LNGsampleID.vcf.gz
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Split by SV type into separate vcfs</span>
module load samtools

<span class="nb">cd</span> <span class="nv">$PATH3</span>//nanopore_vs_sr_validation/SR_vcfs
mkdir split_SR_vcfs

<span class="nv">SR_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> svtype <span class="k">in</span> <span class="s2">&quot;DEL&quot;</span> <span class="s2">&quot;INS&quot;</span> <span class="s2">&quot;DUP&quot;</span> <span class="s2">&quot;INV&quot;</span> <span class="s2">&quot;CPX&quot;</span><span class="p">;</span> <span class="k">do</span> 
        <span class="nv">filt</span><span class="o">=</span><span class="s2">&quot;SVTYPE=\&quot;</span><span class="nv">$svtype</span><span class="s2">\&quot;&quot;</span>
        bcftools view -i <span class="nv">$filt</span> <span class="se">\</span>
        --threads <span class="nv">$SLURM_CPUS_PER_TASK</span> <span class="se">\</span>
        -O z <span class="se">\</span>
        -o split_SR_vcfs/<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_altgtonly_LBD_1perc_fdr.cleaned_filters_qual_recalibrated.LNGsampleID.vcf.gz 
    <span class="k">done</span>
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Rename DUP SVtype to INS since that is what sniffles2 for Nanopore (mostly) uses. But keep DUP file separate from INS to be able to differ DUP from INS</span>
module load samtools
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/SR_vcfs/split_SR_vcfs

<span class="nv">SR_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    zcat <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_DUP_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="p">|</span> <span class="se">\</span>
    sed <span class="s1">&#39;s/SVTYPE=DUP/SVTYPE=INS/g&#39;</span> &gt; <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_DUP_as_INS_1percfdr_pass_multialleic_alt_SVs.vcf
    bgzip <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_DUP_as_INS_1percfdr_pass_multialleic_alt_SVs.vcf
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Index vcfs </span>
module load samtools
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/SR_vcfs/split_SR_vcfs

parallel -j <span class="nv">$SLURM_CPUS_PER_TASK</span> tabix -f -p vcf <span class="o">{}</span> ::: *vcf.gz
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#To match Nanopore format, for INS SR vcfs, make %END = %POS</span>
module load bcftools
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/SR_vcfs/split_SR_vcfs

<span class="c1">##Extract needed info</span>
<span class="nv">SR_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    bcftools query -f <span class="s1">&#39;%CHROM\t%POS\t%END\t%POS\n&#39;</span> <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_INS_1percfdr_pass_multialleic_alt_SVs.vcf.gz &gt;  <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_end_annotation.txt
    bgzip -f <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_end_annotation.txt
    tabix -f -s1 -b2 -e3 <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_end_annotation.txt.gz
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">##Update END flag: but make it STOP named since END cannot be directly altered.</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/SR_vcfs/split_SR_vcfs
module load bcftools

<span class="nb">echo</span> <span class="s2">&quot;##INFO=&lt;ID=STOP,Number=1,Type=Integer,Description=&quot;&quot;End position of the structural variant&quot;&quot;&gt;&quot;</span>  &gt; tmp_stop_header.txt

<span class="c1">##Add STOP flag and remove END flag, then rename STOP to END</span>
<span class="nv">SR_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    bcftools annotate -a <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_end_annotation.txt.gz -c CHROM,FROM,TO,STOP -x INFO/END -h tmp_stop_header.txt -Oz -o <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_tmp.vcf.gz <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_INS_1percfdr_pass_multialleic_alt_SVs.vcf.gz
    zcat <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_tmp.vcf.gz <span class="p">|</span> sed <span class="s1">&#39;s/STOP/END/g&#39;</span> &gt; <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_INS_end_is_pos_1percfdr_pass_multialleic_alt_SVs.vcf
    bgzip <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_INS_end_is_pos_1percfdr_pass_multialleic_alt_SVs.vcf
    tabix -p vcf <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_INS_end_is_pos_1percfdr_pass_multialleic_alt_SVs.vcf.gz
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Split LBD Nanopore vcfs by SV type</span>
module load bcftools
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/vcfs_sniffles2
mkdir split_vcfs

<span class="c1">#Split by SV type into separate vcfs</span>
module load samtools

<span class="nv">SAMPLES_NANOPORE</span><span class="o">=</span><span class="s2">&quot;UMARY634 UMARY1544 UMARY4542 UMARY5077 UMARY1571 UMIC1205 UMIC1527 UMIC1601 UMIC959 BARCS1192 BARCS1429 BARCS1720 BARCS1777 INDU2004038 OREG2142 OREG2595&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SAMPLES_NANOPORE</span><span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> svtype <span class="k">in</span> <span class="s2">&quot;DEL&quot;</span> <span class="s2">&quot;INS&quot;</span> <span class="s2">&quot;DUP&quot;</span> <span class="s2">&quot;INV&quot;</span><span class="p">;</span> <span class="k">do</span> 
        <span class="nv">filt</span><span class="o">=</span><span class="s2">&quot;SVTYPE=\&quot;</span><span class="nv">$svtype</span><span class="s2">\&quot;&quot;</span>
        bcftools view -i <span class="nv">$filt</span> <span class="se">\</span>
        --threads <span class="nv">$SLURM_CPUS_PER_TASK</span> <span class="se">\</span>
        -O z <span class="se">\</span>
        -o split_vcfs/<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_fastq_pass.wf_sv.vcf.gz <span class="se">\</span>
        <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_fastq_pass.wf_sv.vcf.gz 
    <span class="k">done</span>
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Index split Nanopore vcfs by SV type</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/vcfs_sniffles2/split_vcfs

module load samtools

parallel -j <span class="nv">$SLURM_CPUS_PER_TASK</span> tabix -p vcf <span class="o">{}</span> ::: *vcf.gz
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Run truvari bench per SV type per sample</span>
<span class="c1">##Validation criterion are: refdist:200bp, pctsize=0.70, pctsim=0,typeignore=FALSE</span>
<span class="c1">##So differences in start and ends sites max 200bp (default 500), 70% size overlap (default), no sequence identity since not resolved (default 0.7), SV type must match (default)</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation
mkdir truvari_bench

<span class="nb">source</span> /data/<span class="nv">$USER</span>/conda/etc/profile.d/conda.sh <span class="o">&amp;&amp;</span> <span class="nb">source</span> /data/<span class="nv">$USER</span>/conda/etc/profile.d/mamba.sh
conda activate truvari

<span class="c1">###Nanopore samples in same order now than short read samples</span>
<span class="nv">SR_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048&quot;</span>
<span class="nv">SAMPLES_NANOPORE</span><span class="o">=</span><span class="s2">&quot;BARCS1720 BARCS1192 INDU2004038 UMARY5077 UMIC959 UMARY1571 UMARY634 BARCS1777 BARCS1429 UMARY1544 OREG2142 UMIC1205 UMIC1527 UMARY4542 UMIC1601 OREG2595&quot;</span>


<span class="c1">##DEL,INV</span>
<span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="nv">array</span><span class="o">=(</span>BARCS1720 BARCS1192 INDU2004038 UMARY5077 UMIC959 UMARY1571 UMARY634 BARCS1777 BARCS1429 UMARY1544 OREG2142 UMIC1205 UMIC1527 UMARY4542 UMIC1601 OREG2595<span class="o">)</span>
    <span class="k">for</span> svtype <span class="k">in</span> DEL INV<span class="p">;</span> <span class="k">do</span>
        truvari bench <span class="se">\</span>
        --refdist<span class="o">=</span><span class="m">200</span> <span class="se">\</span>
        --pctsize<span class="o">=</span><span class="m">0</span>.70 <span class="se">\</span>
        --pctsim<span class="o">=</span><span class="m">0</span> <span class="se">\</span>
        --sizemax<span class="o">=</span><span class="m">999999999</span> <span class="se">\</span>
        -b vcfs_sniffles2/split_vcfs/<span class="si">${</span><span class="nv">array</span><span class="p">[</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="p">]</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_fastq_pass.wf_sv.vcf.gz <span class="se">\</span>
        -c SR_vcfs/split_SR_vcfs/<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        -f <span class="nv">$PATH1</span>/Homo_sapiens_assembly38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
        -o truvari_bench/truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span> 
    <span class="k">done</span>
    <span class="nv">i</span><span class="o">=</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>+1
<span class="k">done</span>

<span class="c1">##INS, separately since SR path name is different</span>
<span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="nv">array</span><span class="o">=(</span><span class="nv">$SAMPLES_NANOPORE</span><span class="o">)</span>
    <span class="k">for</span> svtype <span class="k">in</span> INS<span class="p">;</span> <span class="k">do</span>
        truvari bench <span class="se">\</span>
        --refdist<span class="o">=</span><span class="m">200</span> <span class="se">\</span>
        --pctsize<span class="o">=</span><span class="m">0</span>.70 <span class="se">\</span>
        --pctsim<span class="o">=</span><span class="m">0</span> <span class="se">\</span>
        --sizemax<span class="o">=</span><span class="m">999999999</span> <span class="se">\</span>
        -b vcfs_sniffles2/split_vcfs/<span class="si">${</span><span class="nv">array</span><span class="p">[</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="p">]</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_fastq_pass.wf_sv.vcf.gz <span class="se">\</span>
        -c SR_vcfs/split_SR_vcfs/<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_INS_end_is_pos_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        -f <span class="nv">$PATH1</span>/Homo_sapiens_assembly38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
        -o truvari_bench/truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span> 
    <span class="k">done</span>
    <span class="nv">i</span><span class="o">=</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>+1
<span class="k">done</span>


<span class="c1">##Run SR DUPs as INS</span>
<span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="nv">array</span><span class="o">=(</span><span class="nv">$SAMPLES_NANOPORE</span><span class="o">)</span>
    <span class="k">for</span> svtype <span class="k">in</span> DUP<span class="p">;</span> <span class="k">do</span>
        truvari bench <span class="se">\</span>
        --refdist<span class="o">=</span><span class="m">200</span> <span class="se">\</span>
        --pctsize<span class="o">=</span><span class="m">0</span>.70 <span class="se">\</span>
        --pctsim<span class="o">=</span><span class="m">0</span> <span class="se">\</span>
        --sizemax<span class="o">=</span><span class="m">999999999</span> <span class="se">\</span>
        -b vcfs_sniffles2/split_vcfs/<span class="si">${</span><span class="nv">array</span><span class="p">[</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="p">]</span><span class="si">}</span>_INS_fastq_pass.wf_sv.vcf.gz <span class="se">\</span>
        -c SR_vcfs/split_SR_vcfs/<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_DUP_as_INS_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        -f <span class="nv">$PATH1</span>/Homo_sapiens_assembly38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
        -o truvari_bench/truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span> 
    <span class="k">done</span>
    <span class="nv">i</span><span class="o">=</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>+1
<span class="k">done</span>

conda deactivate
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Frontotemporal-dementia/amyotrophic-lateral-sclerosis-samples">Frontotemporal dementia/amyotrophic lateral sclerosis samples<a class="anchor-link" href="#Frontotemporal-dementia/amyotrophic-lateral-sclerosis-samples">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Extract each FTD sample of interest separately, include only SVs with an alternative allele (=exclude 0/0s) </span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/SR_vcfs
module load samtools

<span class="nv">FTD_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES05717 RES05720 RES05722 RES05724&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$FTD_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    bcftools view <span class="se">\</span>
    -r chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22 <span class="se">\</span>
    -s <span class="nv">$SAMPLE</span> <span class="se">\</span>
    -Ou <span class="se">\</span>
    /FTD_gatksv_1perc_fdr_cleaned_filters_qual_recalibrated/FTD_1perc_fdr.cleaned_filters_qual_recalibrated.LNGsampleID.vcf.gz <span class="se">\</span>
    <span class="p">|</span> <span class="se">\</span>
    bcftools view -i <span class="s1">&#39;GT[*]=&quot;alt&quot;&#39;</span> <span class="se">\</span>
    -Oz <span class="se">\</span>
    -o <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_altgtonly_FTD_1perc_fdr.cleaned_filters_qual_recalibrated.LNGsampleID.vcf.gz 
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Split by SV type into separate vcfs</span>
module load samtools
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/SR_vcfs

<span class="nv">FTD_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES05717 RES05720 RES05722 RES05724&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$FTD_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> svtype <span class="k">in</span> <span class="s2">&quot;DEL&quot;</span> <span class="s2">&quot;INS&quot;</span> <span class="s2">&quot;DUP&quot;</span> <span class="s2">&quot;INV&quot;</span><span class="p">;</span> <span class="k">do</span> 
        <span class="nv">filt</span><span class="o">=</span><span class="s2">&quot;SVTYPE=\&quot;</span><span class="nv">$svtype</span><span class="s2">\&quot;&quot;</span>
        bcftools view -i <span class="nv">$filt</span> <span class="se">\</span>
        -O z <span class="se">\</span>
        -o split_SR_vcfs/<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_altgtonly_FTD_1perc_fdr.cleaned_filters_qual_recalibrated.LNGsampleID.vcf.gz 
    <span class="k">done</span>
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Rename DUP SVtype to INS since that is what cuteSV for Nanopore (mostly) uses. But keep DUP file separate from INS to be able to differ DUP from INS</span>
module load samtools
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/SR_vcfs/split_SR_vcfs

<span class="nv">FTD_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES05717 RES05720 RES05722 RES05724&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$FTD_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    zcat <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_DUP_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="p">|</span> <span class="se">\</span>
    sed <span class="s1">&#39;s/SVTYPE=DUP/SVTYPE=INS/g&#39;</span> &gt; <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_DUP_as_INS_1percfdr_pass_multialleic_alt_SVs.vcf
    bgzip <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_DUP_as_INS_1percfdr_pass_multialleic_alt_SVs.vcf
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#To match Nanopore format, for INS SR vcfs, make %END = %POS</span>
module load bcftools
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/SR_vcfs/split_SR_vcfs

<span class="c1">##Extract needed info</span>
<span class="nv">FTD_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES05717 RES05720 RES05722 RES05724&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$FTD_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    bcftools query -f <span class="s1">&#39;%CHROM\t%POS\t%END\t%POS\n&#39;</span> <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_INS_1percfdr_pass_multialleic_alt_SVs.vcf.gz &gt;  <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_end_annotation.txt
    bgzip -f <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_end_annotation.txt
    tabix -f -s1 -b2 -e3 <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_end_annotation.txt.gz
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">##Update END flag for INS: but make it STOP named since END cannot be directly altered, then rename to END.</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/SR_vcfs/split_SR_vcfs
module load bcftools

<span class="nb">echo</span> <span class="s2">&quot;##INFO=&lt;ID=STOP,Number=1,Type=Integer,Description=&quot;&quot;End position of the structural variant&quot;&quot;&gt;&quot;</span>  &gt; tmp_stop_header.txt

<span class="c1">##Add STOP flag and remove END flag</span>
<span class="nv">FTD_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES05717 RES05720 RES05722 RES05724&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$FTD_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    bcftools annotate -a <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_end_annotation.txt.gz -c CHROM,FROM,TO,STOP -x INFO/END -h tmp_stop_header.txt -Oz -o <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_tmp.vcf.gz <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_INS_1percfdr_pass_multialleic_alt_SVs.vcf.gz
    zcat <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_tmp.vcf.gz <span class="p">|</span> sed <span class="s1">&#39;s/STOP/END/g&#39;</span> &gt; <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_INS_end_is_pos_1percfdr_pass_multialleic_alt_SVs.vcf
    bgzip <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_INS_end_is_pos_1percfdr_pass_multialleic_alt_SVs.vcf
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Index vcfs </span>
module load samtools
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/SR_vcfs/split_SR_vcfs


ls -1 *vcf.gz <span class="p">|</span> parallel -j <span class="nv">$SLURM_CPUS_PER_TASK</span> tabix -f -p vcf <span class="o">{}</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Split FTD Nanopore vcfs by SV type</span>
module load bcftools
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/vcfs_sniffles2

<span class="nv">SAMPLES_NANOPORE</span><span class="o">=</span><span class="s2">&quot;X5324 X4980 X4948 X4904&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SAMPLES_NANOPORE</span><span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> svtype <span class="k">in</span> <span class="s2">&quot;DEL&quot;</span> <span class="s2">&quot;INS&quot;</span> <span class="s2">&quot;DUP&quot;</span> <span class="s2">&quot;INV&quot;</span><span class="p">;</span> <span class="k">do</span> 
        <span class="nv">filt</span><span class="o">=</span><span class="s2">&quot;SVTYPE=\&quot;</span><span class="nv">$svtype</span><span class="s2">\&quot;&quot;</span>
        bcftools view -i <span class="nv">$filt</span> <span class="se">\</span>
        --threads <span class="nv">$SLURM_CPUS_PER_TASK</span> <span class="se">\</span>
        -O z <span class="se">\</span>
        -o split_vcfs/<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_fastq_pass.wf_sv.vcf.gz <span class="se">\</span>
        <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_fastq_pass.wf_sv.vcf.gz 
    <span class="k">done</span>
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Index split Nanopore vcfs by SV type</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/vcfs_sniffles2/split_vcfs
module load samtools

parallel -j1 tabix -f -p vcf <span class="o">{}</span> ::: *vcf.gz
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Run truvari bench per SV type per sample</span>
<span class="c1">##Validation criteria are: refdist:200bp, pctsize=0.70, pctsim=0,typeignore=FALSE</span>
<span class="c1">##So differences in start and ends sites max 200bp (default 500), 70% size overlap (default), no sequence identity since not resolved (default 0.7), SV type must match (default)</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation

<span class="nb">source</span> /data/<span class="nv">$USER</span>/conda/etc/profile.d/conda.sh <span class="o">&amp;&amp;</span> <span class="nb">source</span> /data/<span class="nv">$USER</span>/conda/etc/profile.d/mamba.sh
conda activate truvari

<span class="c1">###Nanopore samples in same order than short read samples!</span>
<span class="nv">FTD_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES05717 RES05720 RES05722 RES05724&quot;</span>
<span class="nv">SAMPLES_NANOPORE</span><span class="o">=</span><span class="s2">&quot;X5324 X4980 X4948 X4904&quot;</span>


<span class="c1">##DEL,INV</span>
<span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$FTD_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="nv">array</span><span class="o">=(</span><span class="nv">$SAMPLES_NANOPORE</span><span class="o">)</span>
    <span class="k">for</span> svtype <span class="k">in</span> DEL INV<span class="p">;</span> <span class="k">do</span>
        truvari bench <span class="se">\</span>
        --refdist<span class="o">=</span><span class="m">200</span> <span class="se">\</span>
        --pctsize<span class="o">=</span><span class="m">0</span>.70 <span class="se">\</span>
        --pctsim<span class="o">=</span><span class="m">0</span> <span class="se">\</span>
        --sizemax<span class="o">=</span><span class="m">999999999</span> <span class="se">\</span>
        -b vcfs_sniffles2/split_vcfs/<span class="si">${</span><span class="nv">array</span><span class="p">[</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="p">]</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_fastq_pass.wf_sv.vcf.gz <span class="se">\</span>
        -c SR_vcfs/split_SR_vcfs/<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        -f <span class="nv">$PATH1</span>/Homo_sapiens_assembly38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
        -o truvari_bench/truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span> 
    <span class="k">done</span>
    <span class="nv">i</span><span class="o">=</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>+1
<span class="k">done</span>

<span class="c1">##INS, separately since SR path name is different</span>
<span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$FTD_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="nv">array</span><span class="o">=(</span><span class="nv">$SAMPLES_NANOPORE</span><span class="o">)</span>
    <span class="k">for</span> svtype <span class="k">in</span> INS<span class="p">;</span> <span class="k">do</span>
        truvari bench <span class="se">\</span>
        --refdist<span class="o">=</span><span class="m">200</span> <span class="se">\</span>
        --pctsize<span class="o">=</span><span class="m">0</span>.70 <span class="se">\</span>
        --pctsim<span class="o">=</span><span class="m">0</span> <span class="se">\</span>
        --sizemax<span class="o">=</span><span class="m">999999999</span> <span class="se">\</span>
        -b vcfs_sniffles2/split_vcfs/<span class="si">${</span><span class="nv">array</span><span class="p">[</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="p">]</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_fastq_pass.wf_sv.vcf.gz <span class="se">\</span>
        -c SR_vcfs/split_SR_vcfs/<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_INS_end_is_pos_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        -f <span class="nv">$PATH1</span>/Homo_sapiens_assembly38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
        -o truvari_bench/truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span> 
    <span class="k">done</span>
    <span class="nv">i</span><span class="o">=</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>+1
<span class="k">done</span>


<span class="c1">##Run SR DUPs as INS</span>
<span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$FTD_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="nv">array</span><span class="o">=(</span><span class="nv">$SAMPLES_NANOPORE</span><span class="o">)</span>
    <span class="k">for</span> svtype <span class="k">in</span> DUP<span class="p">;</span> <span class="k">do</span>
        truvari bench <span class="se">\</span>
        --refdist<span class="o">=</span><span class="m">200</span> <span class="se">\</span>
        --pctsize<span class="o">=</span><span class="m">0</span>.70 <span class="se">\</span>
        --pctsim<span class="o">=</span><span class="m">0</span> <span class="se">\</span>
        --sizemax<span class="o">=</span><span class="m">999999999</span> <span class="se">\</span>
        -b vcfs_sniffles2/split_vcfs/<span class="si">${</span><span class="nv">array</span><span class="p">[</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="p">]</span><span class="si">}</span>_INS_fastq_pass.wf_sv.vcf.gz <span class="se">\</span>
        -c SR_vcfs/split_SR_vcfs/<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_DUP_as_INS_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        -f <span class="nv">$PATH1</span>/Homo_sapiens_assembly38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
        -o truvari_bench/truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span> 
    <span class="k">done</span>
    <span class="nv">i</span><span class="o">=</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>+1
<span class="k">done</span>

conda deactivate
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Summary-of-results">Summary of results<a class="anchor-link" href="#Summary-of-results">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Output precision and genotype concordance per sample per SVtype and outoput all to same file</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/truvari_bench

<span class="nv">ALL_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048 RES05717 RES05720 RES05722 RES05724&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$ALL_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> svtype <span class="k">in</span> DEL INV INS DUP<span class="p">;</span> <span class="k">do</span>
        <span class="nb">echo</span> <span class="nv">$SAMPLE</span> <span class="nv">$svtype</span> &gt; tmp.txt
        grep -e <span class="s1">&#39;precision&#39;</span> -e <span class="s1">&#39;gt_concordance&#39;</span> truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>/summary.txt &gt;&gt; tmp.txt
        tr <span class="s1">&#39;\n&#39;</span> <span class="s1">&#39; &#39;</span> &lt; tmp.txt <span class="p">|</span> sed <span class="s1">&#39;$s/ $/\n/&#39;</span> &gt;&gt; <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/results_unfiltered_SR_vs_nanopore_precision_gt_goncordance.txt
    <span class="k">done</span>
<span class="k">done</span>

cat <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/results_unfiltered_SR_vs_nanopore_precision_gt_goncordance.txt
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Filtered-structural-variants">Filtered structural variants<a class="anchor-link" href="#Filtered-structural-variants">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Lewy-body-dementia-and-frontotemporal-dementia/amyotrophic-lateral-sclerosis-samples">Lewy body dementia and frontotemporal dementia/amyotrophic lateral sclerosis samples<a class="anchor-link" href="#Lewy-body-dementia-and-frontotemporal-dementia/amyotrophic-lateral-sclerosis-samples">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/nanopore_validation
mkdir SR_vcfs
<span class="nb">cd</span> SR_vcfs

<span class="c1">#Extract each LBD sample of interest separately from clean high-quality subset vcfs (autosomes), include only SVs with an alternative allele (=exclude 0/0s)</span>

<span class="nv">SR_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
   <span class="nb">echo</span> <span class="s2">&quot;bcftools view \</span>
<span class="s2">    -s </span><span class="nv">$SAMPLE</span><span class="s2"> \</span>
<span class="s2">    -Ou \</span>
<span class="s2">    </span><span class="nv">$PATH2</span><span class="s2">/clean_high_quality_vcfs/LBD_high_quality_subset_clean_GQ300missing_autosomes.vcf.gz \</span>
<span class="s2">    | \</span>
<span class="s2">    bcftools view -i &#39;GT[*]=\&quot;alt\&quot;&#39; \</span>
<span class="s2">    -Oz \</span>
<span class="s2">    -o </span><span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span><span class="s2">_altgtonly_high_quality_subset_clean_GQ300missing_autosomes.vcf.gz&quot;</span> &gt;&gt; swarm_extract_samples.swarm
<span class="k">done</span>

<span class="c1">#Run swarm with swarm -t 8 -g 16 --module samtools swarm_extract_samples.swarm</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/nanopore_validation/SR_vcfs

<span class="c1">#Extract each FTD sample of interest separately from clean high-quality subset vcfs (autosomes), include only SVs with an alternative allele (=exclude 0/0s)</span>
<span class="c1">#Add to swarm file</span>
<span class="nv">SR_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES05717 RES05720 RES05722 RES05724&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
   <span class="nb">echo</span> <span class="s2">&quot;bcftools view \</span>
<span class="s2">    -s </span><span class="nv">$SAMPLE</span><span class="s2"> \</span>
<span class="s2">    -Ou \</span>
<span class="s2">    </span><span class="nv">$PATH2</span><span class="s2">/clean_high_quality_vcfs/FTD_high_quality_subset_clean_GQ300missing_autosomes.vcf.gz \</span>
<span class="s2">    | \</span>
<span class="s2">    bcftools view -i &#39;GT[*]=\&quot;alt\&quot;&#39; \</span>
<span class="s2">    -Oz \</span>
<span class="s2">    -o </span><span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span><span class="s2">_altgtonly_high_quality_subset_clean_GQ300missing_autosomes.vcf.gz&quot;</span> &gt;&gt; swarm_extract_samples.swarm
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/nanopore_validation/SR_vcfs
<span class="c1">#Run swarm of sample extraction</span>
swarm --partition quick --logdir swarm_logs --time<span class="o">=</span><span class="m">02</span>:00:00 -t <span class="m">8</span> -g <span class="m">16</span> --module bcftools swarm_extract_samples.swarm
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/nanopore_validation/SR_vcfs
module load samtools

<span class="c1">#Index vcfs</span>
parallel -j <span class="nv">$SLURM_CPUS_PER_TASK</span> tabix -p vcf <span class="o">{}</span> ::: *_altgtonly_high_quality_subset_clean_GQ300missing_autosomes.vcf.gz
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/nanopore_validation/SR_vcfs
mkdir split_SR_vcfs/
module load samtools

<span class="c1">#Split by SV type into separate vcfs</span>
<span class="nv">SR_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048 RES05717 RES05720 RES05722 RES05724&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> svtype <span class="k">in</span> <span class="s2">&quot;DEL&quot;</span> <span class="s2">&quot;INS&quot;</span> <span class="s2">&quot;DUP&quot;</span> <span class="s2">&quot;INV&quot;</span> <span class="s2">&quot;CPX&quot;</span><span class="p">;</span> <span class="k">do</span> 
        <span class="nv">filt</span><span class="o">=</span><span class="s2">&quot;SVTYPE=\&quot;</span><span class="nv">$svtype</span><span class="s2">\&quot;&quot;</span>
        bcftools view -i <span class="nv">$filt</span> <span class="se">\</span>
        --threads <span class="nv">$SLURM_CPUS_PER_TASK</span> <span class="se">\</span>
        -O z <span class="se">\</span>
        -o split_SR_vcfs/clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_altgtonly_high_quality_subset_clean_GQ300missing_autosomes.vcf.gz 
    <span class="k">done</span>
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/nanopore_validation/SR_vcfs/split_SR_vcfs/
module load samtools

<span class="c1">#Rename DUP SVtype to INS since that is what sniffles2 for Nanopore (mostly) uses. But keep DUP file separate from INS to be able to differ DUP from INS</span>
<span class="nv">SR_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048 RES05717 RES05720 RES05722 RES05724&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    zcat clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_DUP_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="p">|</span> <span class="se">\</span>
    sed <span class="s1">&#39;s/SVTYPE=DUP/SVTYPE=INS/g&#39;</span> &gt; clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_DUP_as_INS_1percfdr_pass_multialleic_alt_SVs.vcf
    bgzip clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_DUP_as_INS_1percfdr_pass_multialleic_alt_SVs.vcf
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Index vcfs </span>
module load samtools
<span class="nb">cd</span> <span class="nv">$PATH2</span>/nanopore_validation/SR_vcfs/split_SR_vcfs

parallel -j <span class="nv">$SLURM_CPUS_PER_TASK</span> tabix -f -p vcf <span class="o">{}</span> ::: *vcf.gz
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#To match Nanopore format, for INS SR vcfs, make %END = %POS</span>
module load bcftools
<span class="nb">cd</span> <span class="nv">$PATH2</span>/nanopore_validation/SR_vcfs/split_SR_vcfs

<span class="c1">##Extract needed info</span>
<span class="nv">SR_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048 RES05717 RES05720 RES05722 RES05724&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    bcftools query -f <span class="s1">&#39;%CHROM\t%POS\t%END\t%POS\n&#39;</span> clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_INS_1percfdr_pass_multialleic_alt_SVs.vcf.gz &gt;  clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_end_annotation.txt
    bgzip -f clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_end_annotation.txt
    tabix -f -s1 -b2 -e3 clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_end_annotation.txt.gz
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">##Update END flag: but make it STOP named since END cannot be directly altered.</span>
<span class="nb">cd</span> <span class="nv">$PATH2</span>/nanopore_validation/SR_vcfs/split_SR_vcfs
module load bcftools

<span class="nb">echo</span> <span class="s2">&quot;##INFO=&lt;ID=STOP,Number=1,Type=Integer,Description=&quot;&quot;End position of the structural variant&quot;&quot;&gt;&quot;</span>  &gt; tmp_stop_header.txt

<span class="c1">##Add STOP flag and remove END flag, then rename STOP to END</span>
<span class="nv">SR_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048 RES05717 RES05720 RES05722 RES05724&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    bcftools annotate -a clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_end_annotation.txt.gz -c CHROM,FROM,TO,STOP -x INFO/END -h tmp_stop_header.txt -Oz -o clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_tmp.vcf.gz clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_INS_1percfdr_pass_multialleic_alt_SVs.vcf.gz
    zcat clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_tmp.vcf.gz <span class="p">|</span> sed <span class="s1">&#39;s/STOP/END/g&#39;</span> &gt; clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_INS_end_is_pos_1percfdr_pass_multialleic_alt_SVs.vcf
    bgzip clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_INS_end_is_pos_1percfdr_pass_multialleic_alt_SVs.vcf
    tabix -f -p vcf clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_INS_end_is_pos_1percfdr_pass_multialleic_alt_SVs.vcf.gz
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Run truvari bench per SV type per sample for all samples</span>
<span class="c1">##Validation criterion are: refdist:200bp, pctsize=0.70, pctsim=0,typeignore=FALSE</span>
<span class="c1">##So differences in start and ends sites max 200bp (default 500), 70% size overlap (default), no sequence identity since not resolved (default 0.7), SV type must match (default)</span>
<span class="nb">cd</span> <span class="nv">$PATH2</span>/nanopore_validation/SR_vcfs/split_SR_vcfs
mkdir truvari_bench

<span class="nb">source</span> /data/<span class="nv">$USER</span>/conda/etc/profile.d/conda.sh <span class="o">&amp;&amp;</span> <span class="nb">source</span> /data/<span class="nv">$USER</span>/conda/etc/profile.d/mamba.sh
conda activate truvari

<span class="c1">###Nanopore samples in same order now than short read samples</span>
<span class="nv">SR_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048 RES05717 RES05720 RES05722 RES05724&quot;</span>
<span class="nv">SAMPLES_NANOPORE</span><span class="o">=</span><span class="s2">&quot;BARCS1720 BARCS1192 INDU2004038 UMARY5077 UMIC959 UMARY1571 UMARY634 BARCS1777 BARCS1429 UMARY1544 OREG2142 UMIC1205 UMIC1527 UMARY4542 UMIC1601 OREG2595 X5324 X4980 X4948 X4904&quot;</span>


<span class="c1">##DEL,INV</span>
<span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="nv">array</span><span class="o">=(</span><span class="nv">$SAMPLES_NANOPORE</span><span class="o">)</span>
    <span class="k">for</span> svtype <span class="k">in</span> DEL INV<span class="p">;</span> <span class="k">do</span>
        truvari bench <span class="se">\</span>
        --refdist<span class="o">=</span><span class="m">200</span> <span class="se">\</span>
        --pctsize<span class="o">=</span><span class="m">0</span>.70 <span class="se">\</span>
        --pctsim<span class="o">=</span><span class="m">0</span> <span class="se">\</span>
        --sizemax<span class="o">=</span><span class="m">999999999</span> <span class="se">\</span>
        -b <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/vcfs_sniffles2/split_vcfs/<span class="si">${</span><span class="nv">array</span><span class="p">[</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="p">]</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_fastq_pass.wf_sv.vcf.gz <span class="se">\</span>
        -c clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        -f <span class="nv">$PATH1</span>/Homo_sapiens_assembly38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
        -o truvari_bench/truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span> 
    <span class="k">done</span>
    <span class="nv">i</span><span class="o">=</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>+1
<span class="k">done</span>

<span class="c1">##INS, separately since SR path name is different</span>
<span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="nv">array</span><span class="o">=(</span><span class="nv">$SAMPLES_NANOPORE</span><span class="o">)</span>
    <span class="k">for</span> svtype <span class="k">in</span> INS<span class="p">;</span> <span class="k">do</span>
        truvari bench <span class="se">\</span>
        --refdist<span class="o">=</span><span class="m">200</span> <span class="se">\</span>
        --pctsize<span class="o">=</span><span class="m">0</span>.70 <span class="se">\</span>
        --pctsim<span class="o">=</span><span class="m">0</span> <span class="se">\</span>
        --sizemax<span class="o">=</span><span class="m">999999999</span> <span class="se">\</span>
        -b <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/vcfs_sniffles2/split_vcfs/<span class="si">${</span><span class="nv">array</span><span class="p">[</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="p">]</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_fastq_pass.wf_sv.vcf.gz <span class="se">\</span>
        -c clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_INS_end_is_pos_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        -f <span class="nv">$PATH1</span>/Homo_sapiens_assembly38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
        -o truvari_bench/truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span> 
    <span class="k">done</span>
    <span class="nv">i</span><span class="o">=</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>+1
<span class="k">done</span>


<span class="c1">##Run SR DUPs as INS</span>
<span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="nv">array</span><span class="o">=(</span><span class="nv">$SAMPLES_NANOPORE</span><span class="o">)</span>
    <span class="k">for</span> svtype <span class="k">in</span> DUP<span class="p">;</span> <span class="k">do</span>
        truvari bench <span class="se">\</span>
        --refdist<span class="o">=</span><span class="m">200</span> <span class="se">\</span>
        --pctsize<span class="o">=</span><span class="m">0</span>.70 <span class="se">\</span>
        --pctsim<span class="o">=</span><span class="m">0</span> <span class="se">\</span>
        --sizemax<span class="o">=</span><span class="m">999999999</span> <span class="se">\</span>
        -b <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/vcfs_sniffles2/split_vcfs/<span class="si">${</span><span class="nv">array</span><span class="p">[</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="p">]</span><span class="si">}</span>_INS_fastq_pass.wf_sv.vcf.gz <span class="se">\</span>
        -c clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_DUP_as_INS_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        -f <span class="nv">$PATH1</span>/Homo_sapiens_assembly38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
        -o truvari_bench/truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span> 
    <span class="k">done</span>
    <span class="nv">i</span><span class="o">=</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>+1
<span class="k">done</span>

conda deactivate
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Summary-of-results">Summary of results<a class="anchor-link" href="#Summary-of-results">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Output precision and genotype concordance per sample per SVtype and outoput all to same file</span>
<span class="nb">cd</span> <span class="nv">$PATH2</span>/nanopore_validation/SR_vcfs/split_SR_vcfs/truvari_bench

<span class="nv">ALL_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048 RES05717 RES05720 RES05722 RES05724&quot;</span>

<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$ALL_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> svtype <span class="k">in</span> DEL INV INS DUP<span class="p">;</span> <span class="k">do</span>
        <span class="nb">echo</span> <span class="nv">$SAMPLE</span> <span class="nv">$svtype</span> &gt; tmp.txt
        grep -e <span class="s1">&#39;precision&#39;</span> -e <span class="s1">&#39;gt_concordance&#39;</span> truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>/summary.txt &gt;&gt; tmp.txt
        tr <span class="s1">&#39;\n&#39;</span> <span class="s1">&#39; &#39;</span> &lt; tmp.txt <span class="p">|</span> sed <span class="s1">&#39;$s/ $/\n/&#39;</span> &gt;&gt; <span class="nv">$PATH2</span>/nanopore_validation/results_filtered_SR_vs_nanopore_precision_gt_goncordance.txt
    <span class="k">done</span>
<span class="k">done</span>

cat <span class="nv">$PATH2</span>/nanopore_validation/results_filtered_SR_vs_nanopore_precision_gt_goncordance.txt
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Filtered-structural-variants---allele-frequency-&lt;1-%-vs-&gt;=1-%">Filtered structural variants - allele frequency &lt;1 % vs &gt;=1 %<a class="anchor-link" href="#Filtered-structural-variants---allele-frequency-&lt;1-%-vs-&gt;=1-%">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/clean_high_quality_vcfs
module load bcftools

<span class="c1">#Make a list of MAF &lt;1 % and MAF &gt;=1% SVs in LBD and FTD clean autosomal dataset</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    bcftools query -f <span class="s1">&#39;%ID %AF\n&#39;</span> <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_autosomes.vcf.gz <span class="p">|</span> awk <span class="s1">&#39;($2&lt;0.01) {print $1}&#39;</span> &gt; <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_autosomal_clean_AF_under_0.01.txt
    bcftools query -f <span class="s1">&#39;%ID %AF\n&#39;</span> <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_autosomes.vcf.gz <span class="p">|</span> awk <span class="s1">&#39;($2&gt;=0.01) {print $1}&#39;</span> &gt; <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_autosomal_clean_AF_over_0.01.txt
<span class="k">done</span>

<span class="c1">#Cat phenotype LBD and FTD ID files per AF group</span>
cat LBD_autosomal_clean_AF_under_0.01.txt FTD_autosomal_clean_AF_under_0.01.txt &gt; LBD_FTD_autosomal_clean_AF_under_0.01.txt
cat LBD_autosomal_clean_AF_over_0.01.txt FTD_autosomal_clean_AF_over_0.01.txt &gt; LBD_FTD_autosomal_clean_AF_over_0.01.txt
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/nanopore_validation/SR_vcfs/split_SR_vcfs
module load bcftools
<span class="c1">#Use previously made properly split and formatted SVTYPE vcfs to split ny MAF</span>
<span class="nv">ALL_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048 RES05717 RES05720 RES05722 RES05724&quot;</span>

<span class="c1">##MAF&lt;1%</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$ALL_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> svtype <span class="k">in</span> DEL INS_end_is_pos DUP_as_INS INV<span class="p">;</span> <span class="k">do</span>
        bcftools view -i <span class="s1">&#39;ID=@$PATH2/clean_high_quality_vcfs/LBD_FTD_autosomal_clean_AF_under_0.01.txt&#39;</span> -Oz -o <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_MAF_under_0.01_filtered_1percfdr_pass_multialleic_alt_SVs.vcf.gz clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_1percfdr_pass_multialleic_alt_SVs.vcf.gz
        tabix -f -p vcf <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_MAF_under_0.01_filtered_1percfdr_pass_multialleic_alt_SVs.vcf.gz
    <span class="k">done</span>
<span class="k">done</span>

<span class="c1">##MAF&gt;=1%</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$ALL_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> svtype <span class="k">in</span> DEL INS_end_is_pos DUP_as_INS INV<span class="p">;</span> <span class="k">do</span>
        bcftools view -i <span class="s1">&#39;ID=@$PATH2/clean_high_quality_vcfs/LBD_FTD_autosomal_clean_AF_over_0.01.txt&#39;</span> -Oz -o <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_MAF_over_0.01_filtered_1percfdr_pass_multialleic_alt_SVs.vcf.gz clean_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_1percfdr_pass_multialleic_alt_SVs.vcf.gz
        tabix -f -p vcf <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_MAF_over_0.01_filtered_1percfdr_pass_multialleic_alt_SVs.vcf.gz
    <span class="k">done</span>
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Run truvari bench per SV type per sample for all samples</span>
<span class="c1">##Validation criterion are: refdist:200bp, pctsize=0.70, pctsim=0,typeignore=FALSE</span>
<span class="c1">##So differences in start and ends sites max 200bp (default 500), 70% size overlap (default), no sequence identity since not resolved (default 0.7), SV type must match (default)</span>
<span class="nb">cd</span> <span class="nv">$PATH2</span>/nanopore_validation/SR_vcfs/split_SR_vcfs

<span class="nb">source</span> /data/<span class="nv">$USER</span>/conda/etc/profile.d/conda.sh <span class="o">&amp;&amp;</span> <span class="nb">source</span> /data/<span class="nv">$USER</span>/conda/etc/profile.d/mamba.sh
conda activate truvari

<span class="c1">###Nanopore samples in same order now than short read samples</span>
<span class="nv">SR_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048 RES05717 RES05720 RES05722 RES05724&quot;</span>
<span class="nv">SAMPLES_NANOPORE</span><span class="o">=</span><span class="s2">&quot;BARCS1720 BARCS1192 INDU2004038 UMARY5077 UMIC959 UMARY1571 UMARY634 BARCS1777 BARCS1429 UMARY1544 OREG2142 UMIC1205 UMIC1527 UMARY4542 UMIC1601 OREG2595 X5324 X4980 X4948 X4904&quot;</span>

<span class="c1">##Under 1% MAF</span>
<span class="c1">###DEL,INV</span>
<span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="nv">array</span><span class="o">=(</span><span class="nv">$SAMPLES_NANOPORE</span><span class="o">)</span>
    <span class="k">for</span> svtype <span class="k">in</span> DEL INV<span class="p">;</span> <span class="k">do</span>
        truvari bench <span class="se">\</span>
        --refdist<span class="o">=</span><span class="m">200</span> <span class="se">\</span>
        --pctsize<span class="o">=</span><span class="m">0</span>.70 <span class="se">\</span>
        --pctsim<span class="o">=</span><span class="m">0</span> <span class="se">\</span>
        --sizemax<span class="o">=</span><span class="m">999999999</span> <span class="se">\</span>
        -b <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/vcfs_sniffles2/split_vcfs/<span class="si">${</span><span class="nv">array</span><span class="p">[</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="p">]</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_fastq_pass.wf_sv.vcf.gz <span class="se">\</span>
        -c <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_MAF_under_0.01_filtered_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        -f <span class="nv">$PATH1</span>/Homo_sapiens_assembly38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
        -o truvari_bench/truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_under_0.01
    <span class="k">done</span>
    <span class="nv">i</span><span class="o">=</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>+1
<span class="k">done</span>

<span class="c1">###INS, separately since SR path name is different</span>
<span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="nv">array</span><span class="o">=(</span><span class="nv">$SAMPLES_NANOPORE</span><span class="o">)</span>
    <span class="k">for</span> svtype <span class="k">in</span> INS<span class="p">;</span> <span class="k">do</span>
        truvari bench <span class="se">\</span>
        --refdist<span class="o">=</span><span class="m">200</span> <span class="se">\</span>
        --pctsize<span class="o">=</span><span class="m">0</span>.70 <span class="se">\</span>
        --pctsim<span class="o">=</span><span class="m">0</span> <span class="se">\</span>
        --sizemax<span class="o">=</span><span class="m">999999999</span> <span class="se">\</span>
        -b <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/vcfs_sniffles2/split_vcfs/<span class="si">${</span><span class="nv">array</span><span class="p">[</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="p">]</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_fastq_pass.wf_sv.vcf.gz <span class="se">\</span>
        -c <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_end_is_pos_MAF_under_0.01_filtered_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        -f <span class="nv">$PATH1</span>/Homo_sapiens_assembly38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
        -o truvari_bench/truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_under_0.01 
    <span class="k">done</span>
    <span class="nv">i</span><span class="o">=</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>+1
<span class="k">done</span>


<span class="c1">###Run SR DUPs as INS</span>
<span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="nv">array</span><span class="o">=(</span><span class="nv">$SAMPLES_NANOPORE</span><span class="o">)</span>
    <span class="k">for</span> svtype <span class="k">in</span> DUP<span class="p">;</span> <span class="k">do</span>
        truvari bench <span class="se">\</span>
        --refdist<span class="o">=</span><span class="m">200</span> <span class="se">\</span>
        --pctsize<span class="o">=</span><span class="m">0</span>.70 <span class="se">\</span>
        --pctsim<span class="o">=</span><span class="m">0</span> <span class="se">\</span>
        --sizemax<span class="o">=</span><span class="m">999999999</span> <span class="se">\</span>
        -b <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/vcfs_sniffles2/split_vcfs/<span class="si">${</span><span class="nv">array</span><span class="p">[</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="p">]</span><span class="si">}</span>_INS_fastq_pass.wf_sv.vcf.gz <span class="se">\</span>
        -c <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_as_INS_MAF_under_0.01_filtered_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        -f <span class="nv">$PATH1</span>/Homo_sapiens_assembly38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
        -o truvari_bench/truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_under_0.01
    <span class="k">done</span>
    <span class="nv">i</span><span class="o">=</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>+1
<span class="k">done</span>


<span class="c1">##Over 1% MAF</span>
<span class="c1">###DEL,INV</span>
<span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="nv">array</span><span class="o">=(</span><span class="nv">$SAMPLES_NANOPORE</span><span class="o">)</span>
    <span class="k">for</span> svtype <span class="k">in</span> DEL INV<span class="p">;</span> <span class="k">do</span>
        truvari bench <span class="se">\</span>
        --refdist<span class="o">=</span><span class="m">200</span> <span class="se">\</span>
        --pctsize<span class="o">=</span><span class="m">0</span>.70 <span class="se">\</span>
        --pctsim<span class="o">=</span><span class="m">0</span> <span class="se">\</span>
        --sizemax<span class="o">=</span><span class="m">999999999</span> <span class="se">\</span>
        -b <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/vcfs_sniffles2/split_vcfs/<span class="si">${</span><span class="nv">array</span><span class="p">[</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="p">]</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_fastq_pass.wf_sv.vcf.gz <span class="se">\</span>
        -c <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_MAF_over_0.01_filtered_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        -f <span class="nv">$PATH1</span>/Homo_sapiens_assembly38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
        -o truvari_bench/truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_over_0.01 
    <span class="k">done</span>
    <span class="nv">i</span><span class="o">=</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>+1
<span class="k">done</span>

<span class="c1">###INS, separately since SR path name is different</span>
<span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="nv">array</span><span class="o">=(</span><span class="nv">$SAMPLES_NANOPORE</span><span class="o">)</span>
    <span class="k">for</span> svtype <span class="k">in</span> INS<span class="p">;</span> <span class="k">do</span>
        truvari bench <span class="se">\</span>
        --refdist<span class="o">=</span><span class="m">200</span> <span class="se">\</span>
        --pctsize<span class="o">=</span><span class="m">0</span>.70 <span class="se">\</span>
        --pctsim<span class="o">=</span><span class="m">0</span> <span class="se">\</span>
        --sizemax<span class="o">=</span><span class="m">999999999</span> <span class="se">\</span>
        -b <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/vcfs_sniffles2/split_vcfs/<span class="si">${</span><span class="nv">array</span><span class="p">[</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="p">]</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_fastq_pass.wf_sv.vcf.gz <span class="se">\</span>
        -c <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_end_is_pos_MAF_over_0.01_filtered_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        -f <span class="nv">$PATH1</span>/Homo_sapiens_assembly38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
        -o truvari_bench/truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_over_0.01
    <span class="k">done</span>
    <span class="nv">i</span><span class="o">=</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>+1
<span class="k">done</span>


<span class="c1">###Run SR DUPs as INS</span>
<span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SR_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="nv">array</span><span class="o">=(</span><span class="nv">$SAMPLES_NANOPORE</span><span class="o">)</span>
    <span class="k">for</span> svtype <span class="k">in</span> DUP<span class="p">;</span> <span class="k">do</span>
        truvari bench <span class="se">\</span>
        --refdist<span class="o">=</span><span class="m">200</span> <span class="se">\</span>
        --pctsize<span class="o">=</span><span class="m">0</span>.70 <span class="se">\</span>
        --pctsim<span class="o">=</span><span class="m">0</span> <span class="se">\</span>
        --sizemax<span class="o">=</span><span class="m">999999999</span> <span class="se">\</span>
        -b <span class="nv">$PATH3</span>/nanopore_vs_sr_validation/vcfs_sniffles2/split_vcfs/<span class="si">${</span><span class="nv">array</span><span class="p">[</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="p">]</span><span class="si">}</span>_INS_fastq_pass.wf_sv.vcf.gz <span class="se">\</span>
        -c <span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_as_INS_MAF_over_0.01_filtered_1percfdr_pass_multialleic_alt_SVs.vcf.gz <span class="se">\</span>
        -f <span class="nv">$PATH1</span>/Homo_sapiens_assembly38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
        -o truvari_bench/truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_over_0.01 
    <span class="k">done</span>
    <span class="nv">i</span><span class="o">=</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>+1
<span class="k">done</span>

conda deactivate
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Output precision and genotype concordance per sample per SVtype and outoput all to same file</span>
<span class="nb">cd</span> <span class="nv">$PATH2</span>/nanopore_validation/SR_vcfs/split_SR_vcfs/truvari_bench

<span class="nv">ALL_SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048 RES05717 RES05720 RES05722 RES05724&quot;</span>

<span class="c1">#MAF under 1%</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$ALL_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> svtype <span class="k">in</span> DEL INV INS DUP<span class="p">;</span> <span class="k">do</span>
        <span class="nb">echo</span> <span class="nv">$SAMPLE</span> <span class="nv">$svtype</span> &gt; tmp.txt
        grep -e <span class="s1">&#39;precision&#39;</span> -e <span class="s1">&#39;gt_concordance&#39;</span> truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_under_0.01/summary.txt &gt;&gt; tmp.txt
        tr <span class="s1">&#39;\n&#39;</span> <span class="s1">&#39; &#39;</span> &lt; tmp.txt <span class="p">|</span> sed <span class="s1">&#39;$s/ $/\n/&#39;</span> &gt;&gt; <span class="nv">$PATH2</span>/nanopore_validation/results_filtered_maf_under_0.01_SR_vs_nanopore_precision_gt_goncordance.txt
    <span class="k">done</span>
<span class="k">done</span>

<span class="nb">echo</span> <span class="s2">&quot;Under 1% MAF&quot;</span>
cat <span class="nv">$PATH2</span>/nanopore_validation/results_filtered_maf_under_0.01_SR_vs_nanopore_precision_gt_goncordance.txt


<span class="c1">#MAF over 1%</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$ALL_SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> svtype <span class="k">in</span> DEL INV INS DUP<span class="p">;</span> <span class="k">do</span>
        <span class="nb">echo</span> <span class="nv">$SAMPLE</span> <span class="nv">$svtype</span> &gt; tmp.txt
        grep -e <span class="s1">&#39;precision&#39;</span> -e <span class="s1">&#39;gt_concordance&#39;</span> truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">svtype</span><span class="si">}</span>_over_0.01/summary.txt &gt;&gt; tmp.txt
        tr <span class="s1">&#39;\n&#39;</span> <span class="s1">&#39; &#39;</span> &lt; tmp.txt <span class="p">|</span> sed <span class="s1">&#39;$s/ $/\n/&#39;</span> &gt;&gt; <span class="nv">$PATH2</span>/nanopore_validation/results_filtered_maf_over_0.01_SR_vs_nanopore_precision_gt_goncordance.txt
    <span class="k">done</span>
<span class="k">done</span>

<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;Over 1% MAF&quot;</span>
cat <span class="nv">$PATH2</span>/nanopore_validation/results_filtered_maf_over_0.01_SR_vs_nanopore_precision_gt_goncordance.txt
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/nanopore_validation/SR_vcfs/split_SR_vcfs/truvari_bench

<span class="c1">#Check that MAF&lt;1% and MAF &gt;=1% SV counts match (one sample and one SVTYPE)</span>
<span class="nb">echo</span> <span class="s2">&quot;all&quot;</span>
cat truvari_bench_UMARY-1571_DEL/summary.txt
<span class="nb">echo</span> <span class="s2">&quot;under 1%&quot;</span>
cat truvari_bench_UMARY-1571_DEL_under_0.01/summary.txt
<span class="nb">echo</span> <span class="s2">&quot;over 1%&quot;</span>
cat truvari_bench_UMARY-1571_DEL_over_0.01/summary.txt
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Match!</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/nanopore_validation/SR_vcfs/split_SR_vcfs/truvari_bench
<span class="c1">#Check if there are INV MAF&lt;1% since all are 0 replication</span>
cat truvari_bench_*_INV_under_0.01/summary.txt
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>For all samples, there is 1 INV MAF&lt;1% to be tested.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="TPCN1-replication-in-BBseq/DementiaSeq2">TPCN1 replication in BBseq/DementiaSeq2<a class="anchor-link" href="#TPCN1-replication-in-BBseq/DementiaSeq2">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Generate-principal-components">Generate principal components<a class="anchor-link" href="#Generate-principal-components">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Generate PCAs for covariates (method copied from Ruth&#39;s notebook)</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2
mkdir -p PCAcovs

module load plink/2.3-alpha
module load flashpca/2.0

<span class="nv">cohort</span><span class="o">=</span><span class="s2">&quot;BBSeq_DemSeq2&quot;</span>

<span class="c1">##Prune clean plink files</span>
<span class="k">for</span> CHNUM <span class="k">in</span> <span class="o">{</span><span class="m">1</span>..22<span class="o">}</span><span class="p">;</span> <span class="k">do</span>
    plink2 <span class="se">\</span>
    --pfile <span class="nv">$PATH6</span>/FILTERED.BBS_RESv2.controls_chr<span class="si">${</span><span class="nv">CHNUM</span><span class="si">}</span> <span class="se">\</span>
    --allow-no-sex <span class="se">\</span>
    --maf <span class="m">0</span>.01 <span class="se">\</span>
    --geno <span class="m">0</span>.01 <span class="se">\</span>
    --hwe 5e-6 <span class="se">\</span>
    --autosome <span class="se">\</span>
    --exclude range <span class="nv">$PATH7</span>/GenomicRangeExlusion.forPCA.txt <span class="se">\</span>
    --make-bed <span class="se">\</span>
    --out PCAcovs/temp_chr<span class="si">${</span><span class="nv">CHNUM</span><span class="si">}</span> 
<span class="k">done</span>

<span class="k">for</span> CHNUM <span class="k">in</span> <span class="o">{</span><span class="m">1</span>..22<span class="o">}</span><span class="p">;</span> <span class="k">do</span>
    plink2 <span class="se">\</span>
    --bfile PCAcovs/temp_chr<span class="si">${</span><span class="nv">CHNUM</span><span class="si">}</span> <span class="se">\</span>
    --remove-nosex <span class="se">\</span>
    --geno <span class="m">0</span>.01 <span class="se">\</span>
    --maf <span class="m">0</span>.05 <span class="se">\</span>
    --indep-pairwise <span class="m">1000</span> <span class="m">10</span> <span class="m">0</span>.02 <span class="se">\</span>
    --out PCAcovs/temp_pruning_chr<span class="si">${</span><span class="nv">CHNUM</span><span class="si">}</span> 
<span class="k">done</span>

<span class="k">for</span> CHNUM <span class="k">in</span> <span class="o">{</span><span class="m">1</span>..22<span class="o">}</span><span class="p">;</span> <span class="k">do</span>
    plink2 <span class="se">\</span>
    --bfile PCAcovs/temp_chr<span class="si">${</span><span class="nv">CHNUM</span><span class="si">}</span> <span class="se">\</span>
    --remove-nosex <span class="se">\</span>
    --extract PCAcovs/temp_pruning_chr<span class="si">${</span><span class="nv">CHNUM</span><span class="si">}</span>.prune.in <span class="se">\</span>
    --keep-allele-order <span class="se">\</span>
    --make-bed <span class="se">\</span>
    --out PCAcovs/pruned.<span class="si">${</span><span class="nv">cohort</span><span class="si">}</span>.controls.UNRELATED_chr<span class="si">${</span><span class="nv">CHNUM</span><span class="si">}</span> 
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2
module load flashpca/2.0

<span class="nv">cohort</span><span class="o">=</span><span class="s2">&quot;BBSeq_DemSeq2&quot;</span>

<span class="c1">##Merge pruned binaries to single file with chr 1-22</span>
cat PCAcovs/pruned.<span class="si">${</span><span class="nv">cohort</span><span class="si">}</span>.controls.UNRELATED_chr1.fam &gt; PCAcovs/pruned.<span class="si">${</span><span class="nv">cohort</span><span class="si">}</span>.controls.UNRELATED_ALLchr_forPCA.fam
<span class="k">for</span> chr <span class="k">in</span> <span class="o">{</span><span class="m">1</span>..22<span class="o">}</span><span class="p">;</span> <span class="k">do</span> cat PCAcovs/pruned.<span class="si">${</span><span class="nv">cohort</span><span class="si">}</span>.controls.UNRELATED_chr<span class="si">${</span><span class="nv">chr</span><span class="si">}</span>.bim<span class="p">;</span> <span class="k">done</span> &gt; PCAcovs/pruned.<span class="si">${</span><span class="nv">cohort</span><span class="si">}</span>.controls.UNRELATED_ALLchr_forPCA.bim
<span class="o">(</span><span class="nb">echo</span> -en <span class="s2">&quot;\x6C\x1B\x01&quot;</span><span class="p">;</span> <span class="k">for</span> chr <span class="k">in</span> <span class="o">{</span><span class="m">1</span>..22<span class="o">}</span><span class="p">;</span> <span class="k">do</span> tail -c +4 PCAcovs/pruned.<span class="si">${</span><span class="nv">cohort</span><span class="si">}</span>.controls.UNRELATED_chr<span class="si">${</span><span class="nv">chr</span><span class="si">}</span>.bed<span class="p">;</span> <span class="k">done</span><span class="o">)</span> &gt; PCAcovs/pruned.<span class="si">${</span><span class="nv">cohort</span><span class="si">}</span>.controls.UNRELATED_ALLchr_forPCA.bed

<span class="c1">##Calculate/generate PCs based on pruned data set</span>
<span class="nb">cd</span> PCAcovs
flashpca --bfile pruned.<span class="si">${</span><span class="nv">cohort</span><span class="si">}</span>.controls.UNRELATED_ALLchr_forPCA --suffix _FILTERED.<span class="si">${</span><span class="nv">cohort</span><span class="si">}</span>.controls.UNRELATED_ALLchr_forPCA --numthreads <span class="m">19</span>

<span class="c1">##Move all log files to a new folder</span>
mkdir logFiles
mv *.log logFiles

<span class="c1">##Remove intermediate files</span>
rm temp_*
rm pruned.<span class="si">${</span><span class="nv">cohort</span><span class="si">}</span>.controls.UNRELATED_chr*.*
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%</span><span class="k">load_ext</span> rpy2.ipython
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Run-manta-to-genotype-TPCN1-deletion">Run manta to genotype TPCN1 deletion<a class="anchor-link" href="#Run-manta-to-genotype-TPCN1-deletion">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Prepare manta swarm command </span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2/manta_run/crams

<span class="c1">##Make rest of command file</span>
<span class="nv">i</span><span class="o">=</span><span class="m">1</span>
<span class="k">for</span> FILE <span class="k">in</span> replication_TPCN1_crams/*.cram<span class="p">;</span> <span class="k">do</span>
    <span class="nb">echo</span> <span class="s2">&quot;configManta.py \</span>
<span class="s2">    --bam </span><span class="si">${</span><span class="nv">FILE</span><span class="si">}</span><span class="s2"> \</span>
<span class="s2">    --referenceFasta=</span><span class="nv">$PATH1</span><span class="s2">/Homo_sapiens_assembly38/Homo_sapiens_assembly38.fasta \</span>
<span class="s2">    --region=chr12:113244516-113246325 \</span>
<span class="s2">    --runDir ../run/run_</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">&quot;</span> &gt;&gt; swarm_manta_config.swarm
    <span class="nv">i</span><span class="o">=</span><span class="k">$((</span>i+1<span class="k">))</span>
<span class="k">done</span>

swarm --time<span class="o">=</span><span class="m">00</span>:10:00 --partition quick -t <span class="m">2</span> -g <span class="m">4</span> --maxrunning <span class="m">50</span> --logdir swarm_logs -m manta swarm_manta_config.swarm
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Make a swarm job out of generated commands</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2/manta_run/crams

cat swarm_logs/swarm*.o <span class="p">|</span> grep <span class="s1">&#39;runWorkflow.py&#39;</span> &gt; ../run/swarm_run_manta.swarm

<span class="nb">cd</span> <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2/manta_run/run

swarm --partition quick --time<span class="o">=</span><span class="m">00</span>:10:00 -t <span class="m">10</span> -g <span class="m">10</span> --maxrunning <span class="m">50</span> --logdir swarm_logs -m manta swarm_run_manta.swarm
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Merge all individual SVs into one</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2/manta_run/run
module load bcftools

<span class="c1">##Make a list of all SVs in numerical order</span>
<span class="k">for</span> i <span class="k">in</span> <span class="o">{</span><span class="m">1</span>..2353<span class="o">}</span><span class="p">;</span> <span class="k">do</span>
    <span class="nb">echo</span> <span class="s2">&quot;run_</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">/results/variants/diploidSV.vcf.gz&quot;</span> &gt;&gt; merge_vcf_file_list.txt
<span class="k">done</span>

<span class="c1">##Merge</span>
bcftools merge --force-samples --missing-to-ref --file-list merge_vcf_file_list.txt -m none -Oz -o ../replication_TPCN1_BBSeq_Demseq2.vcf.gz
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Check SVs on that region, should be just one</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2/manta_run/
module load bcftools

bcftools query -f <span class="s1">&#39;%ID %POS %END %SVLEN %QUAL\n&#39;</span> replication_TPCN1_BBSeq_Demseq2.vcf.gz

<span class="c1">#Check genotype counts</span>
bcftools query -f <span class="s1">&#39;[%GT\n]&#39;</span> replication_TPCN1_BBSeq_Demseq2.vcf.gz <span class="p">|</span> sort <span class="p">|</span> uniq -c

<span class="c1">#Make genotype file</span>
bcftools query -f <span class="s1">&#39;[%SAMPLE %GT\n]&#39;</span> replication_TPCN1_BBSeq_Demseq2.vcf.gz &gt; TPCN1_DEL_BBSeq_Demseq2_genotypes.txt
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Check GQ distribution</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2/manta_run/
module load bcftools

bcftools query -f <span class="s1">&#39;[%GQ\n]&#39;</span> replication_TPCN1_BBSeq_Demseq2.vcf.gz <span class="p">|</span> awk <span class="s1">&#39;($1&gt;30)&#39;</span> <span class="p">|</span> wc -l
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Just one SV found and it is the right one! AF= 7.1%</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Run-samplot-to-visually-check-SV-calls">Run samplot to visually check SV calls<a class="anchor-link" href="#Run-samplot-to-visually-check-SV-calls">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Make swarm file</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2/manta_run/crams

find replication_TPCN1_crams -name *.cram  -printf <span class="s2">&quot;%f\n&quot;</span> <span class="p">|</span> parallel -j1 <span class="nb">echo</span> <span class="s2">&quot;samplot plot -b replication_TPCN1_crams/{} -o plots/{}_replication_TPCN1_del_samplot.pdf -c chr12 -s 113245316 -e 113245625 -w 500 -t DEL -d 0 -q 20 -r </span><span class="nv">$PATH1</span><span class="s2">/Homo_sapiens_assembly38/Homo_sapiens_assembly38.fasta&quot;</span> &gt;&gt; swarm_samplot.swarm 

swarm --maxrunning <span class="m">100</span> --logdir swarm_logs --module samplot -c <span class="m">1</span> -g <span class="m">2</span> --partition quick --time<span class="o">=</span><span class="m">00</span>:05:00 swarm_samplot.swarm
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2/manta_run/crams

<span class="nb">source</span> swarm_samplot.swarm
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Lewy-body-dementias-clinical-and-pathological">Lewy body dementias clinical and pathological<a class="anchor-link" href="#Lewy-body-dementias-clinical-and-pathological">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Pathological diagnoses include also "Lewy body disease".</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Make-phenotype-file-of-all-Lewy-body-dementia-samples">Make phenotype file of all Lewy body dementia samples<a class="anchor-link" href="#Make-phenotype-file-of-all-Lewy-body-dementia-samples">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Make phenotype file: pathological LBD/LRP vs controls</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2

<span class="c1">##Extract LBD cases with a pathological diagnosis</span>
awk <span class="s1">&#39;($5==&quot;LBD&quot;) {print $1,$2}&#39;</span> <span class="nv">$PATH10</span>/LBD.FTD/Phenotype.BBS_DementiaSeq_v2_harmonized.txt &gt; tmp_LBD_cases.txt

<span class="c1">##Add BrainBank Seq path LBD/LRP cases</span>
awk <span class="s1">&#39;{print $1,$1}&#39;</span> <span class="nv">$PATH1</span>/BrainBankSeq2022/LBD_samples.txt &gt;&gt; tmp_LBD_cases.txt
sort tmp_LBD_cases.txt <span class="p">|</span> uniq <span class="p">|</span> awk <span class="s1">&#39;{if(NR==1) {print &quot;FID&quot;,&quot;IID&quot;,&quot;LBD&quot;} else{print $1,$2,2}}&#39;</span> &gt; pheno_LBD_cases.txt

<span class="c1">##Make phenotype file by adding Controls</span>
awk <span class="s1">&#39;{print $1,$1,1}&#39;</span> <span class="nv">$PATH1</span>/BrainBankSeq2022/control_samples.txt &gt;&gt; pheno_LBD_cases.txt
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Make-covariate-file-and-select-covariate-with-step()">Make covariate file and select covariate with step()<a class="anchor-link" href="#Make-covariate-file-and-select-covariate-with-step()">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%</span><span class="k">load_ext</span> rpy2.ipython
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">#Combine age and sex info of BBSeq and DemSeq2 (LBD cases) into one file</span>

<span class="c1">##Set wd</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;$PATH3/replication_TPCN1_BBSeq_Demseq2/&quot;</span><span class="p">)</span>

<span class="c1">##Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">readxl</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>

<span class="c1">##Import files</span>
<span class="n">Phenotypes_LBD_Resolution_v2_genomes</span> <span class="o">&lt;-</span> <span class="nf">read_excel</span><span class="p">(</span><span class="s">&quot;Phenotypes_LBD_Resolution_v2_genomes.xlsx&quot;</span><span class="p">)</span>

<span class="n">HBRC_BBSeq_ages</span> <span class="o">&lt;-</span> <span class="nf">read_excel</span><span class="p">(</span><span class="s">&quot;HBRC_BBSeq_ages.xlsx&quot;</span><span class="p">)</span>
<span class="n">BrainBankSeq_study_participants</span> <span class="o">&lt;-</span> <span class="nf">read_excel</span><span class="p">(</span><span class="s">&quot;BrainBankSeq_study_participants.xlsx&quot;</span><span class="p">)</span>
<span class="n">Banner_cohort_June232020</span> <span class="o">&lt;-</span> <span class="nf">read_excel</span><span class="p">(</span><span class="s">&quot;Banner_cohort_June232020.xlsx&quot;</span><span class="p">)</span>

<span class="n">sample_key</span> <span class="o">&lt;-</span> <span class="nf">read_excel</span><span class="p">(</span><span class="s">&quot;sample_key.xlsx&quot;</span><span class="p">)</span>

<span class="c1">##Add BBSEq ID to Banner info</span>
<span class="n">Banner_cohort_June232020</span> <span class="o">&lt;-</span> <span class="n">Banner_cohort_June232020</span><span class="p">[,</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">5</span><span class="p">)]</span>
<span class="n">Banner_cohort_June232020</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">Banner_cohort_June232020</span><span class="p">,</span> <span class="n">sample_key</span><span class="p">,</span> <span class="n">by.x</span> <span class="o">=</span> <span class="s">&quot;LNG_ID&quot;</span><span class="p">,</span> <span class="n">by.y</span> <span class="o">=</span> <span class="s">&quot;Unique Samples&quot;</span><span class="p">)</span>

<span class="c1">##Merge BBSeq age info with sex info</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">BrainBankSeq_study_participants</span><span class="p">,</span> <span class="n">HBRC_BBSeq_ages</span><span class="p">,</span> <span class="n">by.x</span> <span class="o">=</span> <span class="s">&quot;Sample_ID&quot;</span><span class="p">,</span> <span class="n">by.y</span> <span class="o">=</span> <span class="s">&quot;LNG_ID&quot;</span><span class="p">,</span> <span class="n">all</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">Banner_cohort_June232020</span><span class="p">,</span> <span class="n">by.x</span> <span class="o">=</span> <span class="s">&quot;Sample_ID&quot;</span><span class="p">,</span> <span class="n">by.y</span> <span class="o">=</span> <span class="s">&quot;LNG_ID.y&quot;</span><span class="p">,</span> <span class="n">all</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>

<span class="c1">##Make data frame neater</span>
<span class="n">df</span><span class="o">$</span><span class="n">Age</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">LNG_ID</span> <span class="o">%like%</span> <span class="s">&quot;BANN&quot;</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">df</span><span class="o">$</span><span class="n">expired_age</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">LNG_ID</span> <span class="o">%like%</span> <span class="s">&quot;BANN&quot;</span><span class="p">]</span>
<span class="n">df</span><span class="o">$</span><span class="n">Age</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">`Unique Samples`</span> <span class="o">%like%</span> <span class="s">&quot;BRC&quot;</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">df</span><span class="o">$</span><span class="n">Age_Death</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">`Unique Samples`</span> <span class="o">%like%</span> <span class="s">&quot;BRC&quot;</span><span class="p">]</span>

<span class="n">df</span> <span class="o">&lt;-</span> <span class="n">df</span><span class="p">[,</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">5</span><span class="p">,</span><span class="m">6</span><span class="p">,</span><span class="m">11</span><span class="p">)]</span>

<span class="c1">##Concat BBSeq and Demseq2 LBD data</span>
<span class="n">Phenotypes_LBD_Resolution_v2_genomes</span> <span class="o">&lt;-</span> <span class="n">Phenotypes_LBD_Resolution_v2_genomes</span><span class="p">[,</span><span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="m">6</span><span class="p">,</span><span class="m">3</span><span class="p">,</span><span class="m">5</span><span class="p">)]</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">Phenotypes_LBD_Resolution_v2_genomes</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>

<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">rbind</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="n">Phenotypes_LBD_Resolution_v2_genomes</span><span class="p">)</span>

<span class="c1">##Drop samples with empty values</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">drop_na</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>

<span class="c1">##Some samples have age as a range, e.g. 60-69. CHange to middle pointt of range e.g. 65.</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">Age</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">Age</span> <span class="o">%like%</span> <span class="s">&quot;-&quot;</span><span class="p">]))</span>
<span class="n">df</span><span class="o">$</span><span class="n">Age</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">Age</span> <span class="o">==</span> <span class="s">&quot;50 - 59&quot;</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="m">55</span>
<span class="n">df</span><span class="o">$</span><span class="n">Age</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">Age</span> <span class="o">==</span> <span class="s">&quot;60 - 69&quot;</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="m">65</span>
<span class="n">df</span><span class="o">$</span><span class="n">Age</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">Age</span> <span class="o">==</span> <span class="s">&quot;70 - 79&quot;</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="m">75</span>
<span class="n">df</span><span class="o">$</span><span class="n">Age</span><span class="p">[</span><span class="n">df</span><span class="o">$</span><span class="n">Age</span> <span class="o">==</span> <span class="s">&quot;80 - 89&quot;</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="m">85</span>
<span class="c1">##Save as a covar file</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="s">&quot;BBSeq_DEMSeq2LBD_age_sex_status.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">quote</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>

<span class="c1">##N.B! In covar file, 23 titled with &quot;control&quot; are marked as cases. They are properly controls in the phenotype file.</span>
<span class="c1">##Likely due to dicrepancy between BBSeq specification and file from Ruth/Sonja.</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">#Make covar file</span>
<span class="c1">#Visualize PCs/Find outliers: no need to exclude, ethnicity outliers previously excluded</span>
<span class="c1">#Also find covariates to adjust for</span>

<span class="c1">##Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span> 

<span class="c1">##Set wd</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;$PATH3/replication_TPCN1_BBSeq_Demseq2/PCAcovs&quot;</span><span class="p">)</span>

<span class="c1">##Import file PC and phenotype file</span>
<span class="n">pcs</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;pcs_FILTERED.BBSeq_DemSeq2.controls.UNRELATED_ALLchr_forPCA&quot;</span><span class="p">)</span>
<span class="n">pheno</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;$PATH10/LBD.FTD/Phenotype.BBS_DementiaSeq_v2_harmonized.txt&quot;</span><span class="p">)</span>
<span class="n">age</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;../BBSeq_DEMSeq2LBD_age_sex_status.txt&quot;</span><span class="p">)</span>
<span class="n">age</span> <span class="o">&lt;-</span> <span class="n">age</span><span class="p">[,</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">4</span><span class="p">)]</span>

<span class="c1">##Merge</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">pcs</span><span class="p">,</span><span class="n">pheno</span><span class="p">,</span> <span class="n">by</span><span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;FID&quot;</span><span class="p">,</span><span class="s">&quot;IID&quot;</span><span class="p">))</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">age</span><span class="p">,</span> <span class="n">by.x</span> <span class="o">=</span> <span class="s">&quot;IID&quot;</span><span class="p">,</span> <span class="n">by.y</span> <span class="o">=</span> <span class="s">&quot;Sample_ID&quot;</span> <span class="p">)</span>


<span class="c1">##Plot</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">PC1</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">PC2</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="nf">as.factor</span><span class="p">(</span><span class="n">AFFECTION_STATUS</span><span class="p">)))</span> <span class="o">+</span>
    <span class="nf">geom_point</span><span class="p">())</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">PC1</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">PC2</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="nf">as.factor</span><span class="p">(</span><span class="n">SEX</span><span class="p">)))</span> <span class="o">+</span>
    <span class="nf">geom_point</span><span class="p">())</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">PC1</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">PC2</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="n">COHORT</span><span class="p">))</span> <span class="o">+</span>
    <span class="nf">geom_point</span><span class="p">())</span>

<span class="c1">##Neatify df</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="n">df</span><span class="p">[,</span><span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">3</span><span class="o">:</span><span class="m">14</span><span class="p">,</span><span class="m">24</span><span class="p">,</span><span class="m">15</span><span class="p">)]</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">drop_na</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>

<span class="c1">##Save df as a covariate file</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="s">&quot;../BBSeq_Demseq2_covariates.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">quote</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">#Find covariates to adjust for</span>

<span class="c1">##Set wd</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;$PATH3/replication_TPCN1_BBSeq_Demseq2/&quot;</span><span class="p">)</span>

<span class="c1">##Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>

<span class="c1">##Import files and make pheno 0-1 not 1-2</span>
<span class="n">pheno</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;pheno_LBD_cases.txt&quot;</span><span class="p">)</span>
<span class="n">pheno</span><span class="o">$</span><span class="n">LBD</span> <span class="o">&lt;-</span> <span class="n">pheno</span><span class="o">$</span><span class="n">LBD</span> <span class="o">-</span> <span class="m">1</span>

<span class="n">covars</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;BBSeq_Demseq2_covariates.txt&quot;</span><span class="p">)</span>

<span class="c1">##Merge</span>
<span class="n">COVS</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">covars</span><span class="p">,</span> <span class="n">pheno</span><span class="p">,</span> <span class="n">by</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;FID&quot;</span><span class="p">,</span><span class="s">&quot;IID&quot;</span><span class="p">))</span>
<span class="n">COVS</span><span class="o">$</span><span class="n">Age</span> <span class="o">&lt;-</span> <span class="nf">as.numeric</span><span class="p">(</span><span class="n">COVS</span><span class="o">$</span><span class="n">Age</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">head</span><span class="p">(</span><span class="n">COVS</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">COVS</span><span class="o">$</span><span class="n">path_LBD</span><span class="p">))</span>

<span class="c1">## Run step() to determine which covariate is most important</span>
<span class="c1">## model using glm() and then use step() to determine the model that has the lowest AIC</span>
<span class="c1">## this tell us which covariates you included in the model are important to use/for adjustment in your analysis</span>
<span class="n">model</span> <span class="o">&lt;-</span> <span class="nf">glm</span><span class="p">(</span><span class="n">LBD</span> <span class="o">~</span> <span class="n">SEX</span> <span class="o">+</span> <span class="n">Age</span> <span class="o">+</span> <span class="n">PC1</span> <span class="o">+</span> <span class="n">PC2</span> <span class="o">+</span> <span class="n">PC3</span> <span class="o">+</span> <span class="n">PC4</span> <span class="o">+</span> <span class="n">PC5</span> <span class="o">+</span> <span class="n">PC6</span> <span class="o">+</span> <span class="n">PC7</span> <span class="o">+</span> <span class="n">PC8</span> <span class="o">+</span> <span class="n">PC9</span> <span class="o">+</span> <span class="n">PC10</span> <span class="p">,</span> <span class="n">data</span> <span class="o">=</span> <span class="n">COVS</span><span class="p">,</span> <span class="n">family</span> <span class="o">=</span> <span class="s">&quot;binomial&quot;</span><span class="p">(</span><span class="n">link</span> <span class="o">=</span> <span class="s">&quot;logit&quot;</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">summary</span><span class="p">(</span><span class="n">model</span><span class="p">))</span>
<span class="nf">step</span><span class="p">(</span><span class="n">model</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Run-association-analysis-on-SNPs/Indels-of-TPCN1-block">Run association analysis on SNPs/Indels of TPCN1 block<a class="anchor-link" href="#Run-association-analysis-on-SNPs/Indels-of-TPCN1-block">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Run glm on the TPCN1 LD block, use only clean variants from Ruth&#39;s analysis</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2
module load plink/2.3-alpha

<span class="c1">##Clean variants from Ruth&#39;s files</span>
awk <span class="s1">&#39;($4 &gt; 113150000 &amp;&amp; $4&lt;113350000){print $2}&#39;</span> <span class="nv">$PATH5</span>/LBD_minGQ300/SNVindels.LBD.UNRELATED.forMerging.chr12.bim &gt; clean_TPCN1_LD_block_SNV_indels.txt

plink2 <span class="se">\</span>
--pfile <span class="nv">$PATH6</span>/FILTERED.BBS_RESv2.controls_chr12 <span class="se">\</span>
--glm <span class="nv">cols</span><span class="o">=</span>+a1freqcc hide-covar <span class="se">\</span>
--ci <span class="m">0</span>.95 <span class="se">\</span>
--remove <span class="nv">$PATH6</span>/FILTERED.BBS_RESv2.to.remove.relatedSamples_FID_IID.txt <span class="se">\</span>
--pheno pheno_LBD_cases.txt <span class="se">\</span>
--pheno-name LBD <span class="se">\</span>
--extract clean_TPCN1_LD_block_SNV_indels.txt <span class="se">\</span>
--maf <span class="m">0</span>.01 <span class="se">\</span>
--covar BBSeq_Demseq2_covariates.txt <span class="se">\</span>
--covar-name Age,PC1,PC2,PC3,PC4,PC5 <span class="se">\</span>
--covar-variance-standardize <span class="se">\</span>
--out LBD_vs_controls <span class="se">\</span>
--write-samples

<span class="c1">#Grep rs6489896 from the results</span>
grep -e <span class="s1">&#39;CHROM&#39;</span> -e <span class="s1">&#39;113281983&#39;</span> LBD_vs_controls.LBD.glm.logistic.hybrid
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2/
<span class="c1">#Extract number of analyzed samples per case-control status</span>
<span class="c1">#Files contained 567 cases and 274 controls (=862), but plink has used data from 829</span>

<span class="c1">##Print samples with covariates</span>
awk <span class="s1">&#39;(NR&gt;1) {print $1}&#39;</span> BBSeq_Demseq2_covariates.txt &gt; tmp.txt

<span class="c1">##Union of samples with covariates and pheno after exclusions</span>
grep -wFf tmp.txt pheno_LBD_cases.txt <span class="p">|</span> awk <span class="s1">&#39;{print $1}&#39;</span> &gt; tmp2.txt


<span class="c1">##union of samples after exclusion etc and with covar and pheno data</span>
grep -wFf tmp2.txt LBD_vs_controls.id <span class="p">|</span> awk <span class="s1">&#39;{print $1}&#39;</span> &gt; tmp_analyzed_samples.txt
wc -l tmp_analyzed_samples.txt

<span class="c1">#Get number of cases and controls</span>
grep -wFf tmp_analyzed_samples.txt pheno_LBD_cases.txt <span class="p">|</span> awk <span class="s1">&#39;{print $3}&#39;</span> <span class="p">|</span> sort <span class="p">|</span> uniq -c
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Run-association-analysis-with-structural-variant-calls">Run association analysis with structural variant calls<a class="anchor-link" href="#Run-association-analysis-with-structural-variant-calls">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Run PLINK2 glm</span>
<span class="c1">##To get same samples as in SNP analysis, include samples in SNP analysis</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2/
module load plink/2.3-alpha

<span class="c1">##Samples in SNP data</span>
awk <span class="s1">&#39;{print $1,$2}&#39;</span> <span class="nv">$PATH6</span>/FILTERED.BBS_RESv2.controls_chr12.psam &gt; FILTERED.BBS_RESv2.controls_IDs.txt
plink2 <span class="se">\</span>
--vcf <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2/manta_run/replication_TPCN1_BBSeq_Demseq2.vcf.gz <span class="se">\</span>
--double-id <span class="se">\</span>
--glm <span class="nv">cols</span><span class="o">=</span>+a1freqcc hide-covar <span class="se">\</span>
--ci <span class="m">0</span>.95 <span class="se">\</span>
--keep FILTERED.BBS_RESv2.controls_IDs.txt <span class="se">\</span>
--remove <span class="nv">$PATH6</span>/FILTERED.BBS_RESv2.to.remove.relatedSamples_FID_IID.txt <span class="se">\</span>
--pheno pheno_LBD_cases.txt <span class="se">\</span>
--pheno-name LBD <span class="se">\</span>
--maf <span class="m">0</span>.01 <span class="se">\</span>
--covar BBSeq_Demseq2_covariates.txt <span class="se">\</span>
--covar-name Age,PC1,PC2,PC3,PC4,PC5 <span class="se">\</span>
--covar-variance-standardize <span class="se">\</span>
--out LBD_vs_controls_manta_SV_calls

cat LBD_vs_controls_manta_SV_calls.LBD.glm.logistic.hybrid
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Meta-analysis-of-structural-variant-results">Meta-analysis of structural variant results<a class="anchor-link" href="#Meta-analysis-of-structural-variant-results">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2/
<span class="c1">#Format files</span>

<span class="c1">##Replication data</span>
<span class="c1">###Change SV ID so it matches GATK-SV</span>
awk <span class="s1">&#39;{if(NR &gt;1) {$3=&quot;LBD_DEL_chr12_9172&quot;} print}&#39;</span> LBD_vs_controls_manta_SV_calls.LBD.glm.logistic.hybrid &gt; tmp_all_LBD_SV_replication_to_metal.txt

<span class="c1">###Add beta column, chnage REF to N and ALT to &lt;DEL&gt;</span>
awk <span class="s1">&#39;{if(NR==1) {$17=&quot;BETA&quot;} else{$17 = log($12)} print}&#39;</span> tmp_all_LBD_SV_replication_to_metal.txt &gt; tmp2_all_LBD_SV_replication_to_metal.txt
awk <span class="s1">&#39;{if(NR&gt;1) {$4 = &quot;N&quot;} print}&#39;</span> tmp2_all_LBD_SV_replication_to_metal.txt &gt; tmp3_all_LBD_SV_replication_to_metal.txt
awk <span class="s1">&#39;{if(NR&gt;1) {$5 = &quot;&lt;DEL&gt;&quot;} print}&#39;</span> tmp3_all_LBD_SV_replication_to_metal.txt &gt; all_LBD_SV_replication_to_metal.txt

rm tmp*_LBD_SV_replication_to_metal.txt

<span class="c1">##GATK-SV joint LBD data</span>
<span class="c1">###Extract SV of interest</span>
grep -e <span class="s1">&#39;CHROM&#39;</span> -e <span class="s1">&#39;LBD_DEL_chr12_9172&#39;</span> <span class="nv">$PATH5</span>/Analysis.GLM.hg38_minGQ300/LBD/LBD.controls.UNRELATED.SNVindels.SV.glm.hwe1e-6.maf005.ALLchr.txt &gt; tmp_LBD_SV_GATKSV_joint_to_metal.txt

<span class="c1">###Add BETA column</span>
awk <span class="s1">&#39;{if(NR==1) {$24=&quot;BETA&quot;} else{$24 = log($19)} print}&#39;</span> tmp_LBD_SV_GATKSV_joint_to_metal.txt &gt; LBD_SV_GATKSV_joint_to_metal.txt
rm tmp_LBD_SV_GATKSV_joint_to_metal.txt
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Commands-to-run-METAL">Commands to run METAL<a class="anchor-link" href="#Commands-to-run-METAL">&#182;</a></h1><p>module load metal
metal<br>
MARKER ID<br>
ALLELE ALT REF<br>
PVALUE P<br>
EFFECT BETA 
WEIGHTLABEL OBS_CT<br>
STDERRLABEL LOG(OR)_SE<br>
SCHEME STDERR<br>
OUTFILE TPCN1_DEL_LBDjoint_all_LBD_replication .TBL<br>
PROCESS all_LBD_SV_replication_to_metal.txt<br>
PROCESS LBD_SV_GATKSV_joint_to_metal.txt<br>
ANALYZE<br>
QUIT</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2/
<span class="c1">#Check meta result</span>
cat TPCN1_DEL_LBDjoint_all_LBD_replication1.TBL
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="50-neurodegenerative-disease-genes-gene-set-analyses">50 neurodegenerative disease genes gene-set analyses<a class="anchor-link" href="#50-neurodegenerative-disease-genes-gene-set-analyses">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="File-preparations">File preparations<a class="anchor-link" href="#File-preparations">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">glob</span>
<span class="kn">import</span> <span class="nn">os</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Subset-50-neurodegenerative-disease-gene-regions">Subset 50 neurodegenerative disease gene regions<a class="anchor-link" href="#Subset-50-neurodegenerative-disease-gene-regions">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Create candidate-gene files from 1%FDR GQ300 clean files, code from Ruth&#39;s notebook</span>

<span class="nb">cd</span> <span class="nv">$PATH8</span>/DataSubset.50NDD

module load plink/2.3-alpha

<span class="nv">DATA</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$PATH2</span><span class="s2">/clean_high_quality_vcfs&quot;</span>
<span class="nv">OUT</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$PATH1</span><span class="s2">/NIH_FALL_2021_SVs_1percFDR&quot;</span>

mkdir <span class="nv">$OUT</span>/LBD_NDD_plink_files/
mkdir <span class="nv">$OUT</span>/FTD_NDD_plink_files/

<span class="c1">## Extract regions specific to each gene from clean vcfs</span>

<span class="c1">### LBD</span>
<span class="nv">cohort</span><span class="o">=</span><span class="s2">&quot;LBD&quot;</span>
<span class="k">for</span> FILE <span class="k">in</span> CLEAN.knownGenes.plink/Gene_Locs/*.txt<span class="p">;</span>
<span class="k">do</span>
<span class="nv">GENE</span><span class="o">=</span><span class="sb">`</span>awk <span class="s1">&#39;NR==1{print $4}&#39;</span> <span class="nv">$FILE</span><span class="sb">`</span>
<span class="nv">GENE</span><span class="o">=</span><span class="k">$(</span><span class="nb">echo</span> <span class="s2">&quot;</span><span class="nv">$GENE</span><span class="s2">&quot;</span> <span class="p">|</span> tr -d <span class="s1">&#39;\r&#39;</span><span class="k">)</span>

plink2 <span class="se">\</span>
--vcf <span class="nv">$DATA</span>/<span class="si">${</span><span class="nv">cohort</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz <span class="se">\</span>
--extract range <span class="nv">$FILE</span> <span class="se">\</span>
--keep <span class="nv">$DATA</span>/tmp_<span class="si">${</span><span class="nv">cohort</span><span class="si">}</span>_analyzed_samples.txt <span class="se">\</span>
--allow-no-sex <span class="se">\</span>
--keep-allele-order <span class="se">\</span>
--make-pgen <span class="se">\</span>
--out <span class="nv">$OUT</span>/LBD_NDD_plink_files/SV.<span class="si">${</span><span class="nv">cohort</span><span class="si">}</span>.keepRelated.subsetKnownGenes.flank1Mb.<span class="nv">$GENE</span>
<span class="k">done</span>

<span class="c1">### FTD</span>
<span class="nv">cohort</span><span class="o">=</span><span class="s2">&quot;FTD&quot;</span>
<span class="k">for</span> FILE <span class="k">in</span> CLEAN.knownGenes.plink/Gene_Locs/*.txt<span class="p">;</span>
<span class="k">do</span>
<span class="nv">GENE</span><span class="o">=</span><span class="sb">`</span>awk <span class="s1">&#39;NR==1{print $4}&#39;</span> <span class="nv">$FILE</span><span class="sb">`</span>
<span class="nv">GENE</span><span class="o">=</span><span class="k">$(</span><span class="nb">echo</span> <span class="s2">&quot;</span><span class="nv">$GENE</span><span class="s2">&quot;</span> <span class="p">|</span> tr -d <span class="s1">&#39;\r&#39;</span><span class="k">)</span>

plink2 <span class="se">\</span>
--vcf <span class="nv">$DATA</span>/<span class="si">${</span><span class="nv">cohort</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz <span class="se">\</span>
--extract range <span class="nv">$FILE</span> <span class="se">\</span>
--keep <span class="nv">$DATA</span>/tmp_<span class="si">${</span><span class="nv">cohort</span><span class="si">}</span>_analyzed_samples.txt <span class="se">\</span>
--allow-no-sex <span class="se">\</span>
--keep-allele-order <span class="se">\</span>
--make-pgen <span class="se">\</span>
--out <span class="nv">$OUT</span>/FTD_NDD_plink_files/SV.<span class="si">${</span><span class="nv">cohort</span><span class="si">}</span>.keepRelated.subsetKnownGenes.flank1Mb.<span class="nv">$GENE</span>
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Create-vcfs">Create vcfs<a class="anchor-link" href="#Create-vcfs">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Make one vcf file per gene (create output folders first), for loop over gene names</span>
<span class="nv">HOME</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$PATH1</span><span class="s2">/NIH_FALL_2021_SVs_1percFDR&quot;</span>
<span class="nb">cd</span> <span class="nv">$HOME</span>

mkdir vcfs_LBD/
mkdir vcfs_FTD/

module load plink/2.3-alpha

<span class="k">for</span> pheno <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> gene <span class="k">in</span> <span class="k">$(</span>ls -1 <span class="nv">$PATH1</span>/NIH_FALL_2021_SVs/NIH_files/CLEAN.knownGenes.plink/Gene_Locs/ <span class="p">|</span> awk -F<span class="s2">&quot;.&quot;</span> <span class="s1">&#39;{print $1}&#39;</span><span class="k">)</span><span class="p">;</span> <span class="k">do</span>
        plink2 <span class="se">\</span>
        --pfile <span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>_NDD_plink_files/SV.<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>.keepRelated.subsetKnownGenes.flank1Mb.<span class="si">${</span><span class="nv">gene</span><span class="si">}</span> <span class="se">\</span>
        --export vcf id-paste<span class="o">=</span>iid <span class="se">\</span>
        --out vcfs_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>/SV.<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>.keepRelated.subsetKnownGenes.flank1Mb.<span class="si">${</span><span class="nv">gene</span><span class="si">}</span>
    <span class="k">done</span>
<span class="k">done</span>

<span class="c1">#Bgzip and index vcfs. Concatenate all vcfs into one master vcf for all SV case-control frequency calculations</span>
module load samtools

<span class="k">for</span> pheno <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    <span class="nb">cd</span> <span class="nv">$HOME</span>/vcfs_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>/
    ls *vcf -1  <span class="p">|</span> parallel -j <span class="m">20</span> bgzip <span class="o">{}</span>
    ls *vcf.gz -1 <span class="p">|</span> parallel -j <span class="m">20</span> tabix -f -p vcf <span class="o">{}</span>

    ls *vcf.gz -1 &gt; <span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>_vcfs.txt
    bcftools concat -a -f <span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>_vcfs.txt -O v -o NDD_all_vcfs.vcf
    bcftools sort NDD_all_vcfs.vcf -O v -o NDD_all_vcfs_sorted.vcf
    bgzip NDD_all_vcfs_sorted.vcf
    tabix -f -p vcf NDD_all_vcfs_sorted.vcf.gz
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Case-control-allele-frequencies">Case-control allele frequencies<a class="anchor-link" href="#Case-control-allele-frequencies">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nv">HOME</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$PATH1</span><span class="s2">/NIH_FALL_2021_SVs_1percFDR&quot;</span>
<span class="nb">cd</span> <span class="nv">$HOME</span>

<span class="c1">#Make plink2 files from master vcf (N.B. some variants are duplicates because overlap multiple genes)</span>
module load plink/2.3-alpha

<span class="k">for</span> pheno <span class="k">in</span> LBD FTD<span class="p">;</span><span class="k">do</span>
    plink2 <span class="se">\</span>
    --vcf vcfs_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>/NDD_all_vcfs_sorted.vcf.gz <span class="se">\</span>
    --double-id <span class="se">\</span>
    --make-pgen <span class="se">\</span>
    --out vcfs_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>/NDD_all_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash

<span class="nv">HOME</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$PATH1</span><span class="s2">/NIH_FALL_2021_SVs_1percFDR&quot;</span>
<span class="nb">cd</span> <span class="nv">$HOME</span>

<span class="c1">#Calculate case and control allele frequencies, first subset use any original psam that includes phenotype data based on samples to keep as new psam file</span>
module load plink/2.3-alpha

mkdir freqs_LBD/
mkdir freqs_FTD/

<span class="c1">##Create covar files for subsetting</span>

cut -f1,2,3,4,5,8,13,14,15,16,17,18,19,20,21,22,23,24 <span class="nv">$PATH5</span>/COVARIATES.SNVindels.SV.freeze9.LBD.controls.UNRELATED_minGQ300.txt &gt; <span class="nv">$HOME</span>/COVARIATES.SNVindels.SV.freeze9.LBD.controls.UNRELATED_minGQ300_relevant_columns.txt
cut -f1,2,3,4,5,8,13,14,15,16,17,18,19,20,21,22,23,24 <span class="nv">$PATH5</span>/COVARIATES.SNVindels.SV.freeze9.FTD.controls.UNRELATED_minGQ300.txt &gt; <span class="nv">$HOME</span>/COVARIATES.SNVindels.SV.freeze9.FTD.controls.UNRELATED_minGQ300_relevant_columns.txt

<span class="c1">##Calculate AFs</span>
<span class="k">for</span> pheno <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    plink2 <span class="se">\</span>
    --pfile vcfs_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>/NDD_all_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span> <span class="se">\</span>
    --freq <span class="se">\</span>
    --covar COVARIATES.SNVindels.SV.freeze9.<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>.controls.UNRELATED_minGQ300_relevant_columns.txt <span class="se">\</span>
    --keep-if <span class="nv">AFFECTION_STATUS</span><span class="o">=</span><span class="m">2</span> <span class="se">\</span>
    --out freqs_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>/freqs_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>_cases 
    
    plink2 <span class="se">\</span>
    --pfile vcfs_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>/NDD_all_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span> <span class="se">\</span>
    --freq <span class="se">\</span>
    --covar COVARIATES.SNVindels.SV.freeze9.<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>.controls.UNRELATED_minGQ300_relevant_columns.txt <span class="se">\</span>
    --keep-if <span class="nv">AFFECTION_STATUS</span><span class="o">=</span><span class="m">1</span> <span class="se">\</span>
    --out freqs_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>/freqs_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>_controls

<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="CADD-SV-scores">CADD-SV scores<a class="anchor-link" href="#CADD-SV-scores">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Prepare input file: make bed from vcf, add chr prefix and sort</span>
<span class="nb">source</span> /data/<span class="nv">$USER</span>/conda/etc/profile.d/conda.sh
conda activate prepBED

<span class="k">for</span> pheno <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    SURVIVOR vcftobed <span class="nv">$PATH1</span>/NIH_FALL_2021_SVs_1percFDR/vcfs_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>/NDD_all_vcfs.vcf <span class="m">0</span> -1 <span class="nv">$PATH1</span>/CADD-SV-1.1/beds/output_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>_1percFDR.bed
    awk <span class="s1">&#39;OFS=&quot;\t&quot;{print &quot;chr&quot;$1,$2,$6,$11,$7}&#39;</span> <span class="nv">$PATH1</span>/CADD-SV-1.1/beds/output_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>_1percFDR.bed &gt; <span class="nv">$PATH1</span>/CADD-SV-1.1/beds/tmp_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>_1percFDR.bed
    sort -k1,1 -k2,2n <span class="nv">$PATH1</span>/CADD-SV-1.1/beds/tmp_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>_1percFDR.bed &gt; <span class="nv">$PATH1</span>/CADD-SV-1.1/input/id_<span class="si">${</span><span class="nv">pheno</span><span class="si">}</span>_1percFDR_NDD_all_vcfs_sorted.bed
<span class="k">done</span>

conda deactivate
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p><strong>N.B.! Manually enter set id to config file (wirhout id_ prefix and bed suffix)</strong></p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">!</span>cat <span class="nv">$PATH1</span>/CADD-SV-1.1/config.yml
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Initialize conda environment to run CADD-SV for LBD. Needs enough memory, otherwise process will be killed.</span>
<span class="nb">source</span> /data/<span class="nv">$USER</span>/conda/etc/profile.d/conda.sh
conda activate run.caddsv

<span class="nb">cd</span> <span class="nv">$PATH1</span>/CADD-SV-1.1
module load R
snakemake  --use-conda --configfile config.yml -j <span class="nv">$SLURM_CPUS_PER_TASK</span> 

conda deactivate
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p><strong>N.B.! Manually enter set id to config file (wirhout id_ prefix and bed suffix)</strong></p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">!</span>cat <span class="nv">$PATH1</span>/CADD-SV-1.1/config_FTD.yml
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Initialize conda environment to run CADD-SV for FTD. </span>
<span class="nb">source</span> /data/<span class="nv">$USER</span>/conda/etc/profile.d/conda.sh
conda activate run.caddsv

<span class="nb">cd</span> <span class="nv">$PATH1</span>/CADD-SV-1.1
module load R
snakemake  --use-conda --configfile config_FTD.yml -j <span class="nv">$SLURM_CPUS_PER_TASK</span>

conda deactivate
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Check number of scored SVs (not all SV types are scoreable by CADD)</span>

wc -l <span class="nv">$PATH1</span>/CADD-SV-1.1/output/LBD_1percFDR_NDD_all_vcfs_sorted_score.bed
wc -l <span class="nv">$PATH1</span>/CADD-SV-1.1/output/FTD_1percFDR_NDD_all_vcfs_sorted_score.bed
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="GeneHancer-regulatory-elements">GeneHancer regulatory elements<a class="anchor-link" href="#GeneHancer-regulatory-elements">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Import to python</span>
<span class="n">genehancer</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;$PATH1/NIH_FALL_2021_SVs/genhancer_candidate_gene_regions_genecards.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="n">genehancer</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Select only relevant columns, i.e. geneHancerStart, geneHancerEnd, geneHancerIdentifier and geneName</span>
<span class="n">genehancer</span> <span class="o">=</span> <span class="n">genehancer</span><span class="p">[[</span><span class="s2">&quot;Start&quot;</span><span class="p">,</span> <span class="s2">&quot;End&quot;</span><span class="p">,</span> <span class="s2">&quot;GHIDs&quot;</span><span class="p">,</span> <span class="s2">&quot;Symbol&quot;</span><span class="p">]]</span>

<span class="c1">#Rename columns like they were in previous versions to ensure all works</span>
<span class="n">genehancer</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;Start&quot;</span><span class="p">:</span><span class="s2">&quot;geneHancerStart&quot;</span><span class="p">,</span> <span class="s2">&quot;End&quot;</span><span class="p">:</span><span class="s2">&quot;geneHancerEnd&quot;</span><span class="p">,</span> <span class="s2">&quot;GHIDs&quot;</span><span class="p">:</span><span class="s2">&quot;geneHancerIdentifier&quot;</span><span class="p">,</span> <span class="s2">&quot;Symbol&quot;</span><span class="p">:</span><span class="s2">&quot;geneName&quot;</span><span class="p">},</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">genehancer</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Test if all geneHancerEnd &gt; genehancerStart</span>
<span class="n">genehancer</span><span class="p">[(</span><span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneHancerEnd&quot;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneHancerStart&quot;</span><span class="p">])]</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Prior to checking overlaps, change in genehancer data frame PRKN to PARK2</span>
<span class="n">genehancer</span> <span class="o">=</span> <span class="n">genehancer</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="n">to_replace</span><span class="o">=</span><span class="s2">&quot;PRKN&quot;</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="s2">&quot;PARK2&quot;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Lewy-body-dementias">Lewy body dementias<a class="anchor-link" href="#Lewy-body-dementias">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">os</span><span class="o">.</span><span class="n">chdir</span><span class="p">(</span><span class="s2">&quot;$PATH1/NIH_FALL_2021_SVs_1percFDR&quot;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">os</span><span class="o">.</span><span class="n">getcwd</span><span class="p">()</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#For each vcf (one per gene), print only columns of interest with gene name added as one of the columns.</span>
<span class="c1">#All columns are separated by tab, sample names with alt allele are separated with single space and their genotype with &quot;:&quot;</span>

<span class="nb">cd</span> <span class="nv">$PATH1</span>/NIH_FALL_2021_SVs_1percFDR
mkdir SV_text_files_LBD

module load bcftools

<span class="k">for</span> gene <span class="k">in</span> <span class="k">$(</span>ls -1 <span class="nv">$PATH1</span>/NIH_FALL_2021_SVs/NIH_files/CLEAN.knownGenes.plink/Gene_Locs/ <span class="p">|</span> awk -F<span class="s2">&quot;.&quot;</span> <span class="s1">&#39;{print $1}&#39;</span><span class="k">)</span><span class="p">;</span> <span class="k">do</span>
    bcftools query -f <span class="s2">&quot;%CHROM\t%POS\t%END\t%ID\t%REF\t%ALT\t%QUAL\t</span><span class="si">${</span><span class="nv">gene</span><span class="si">}</span><span class="s2">\t%FILTER\t%SVLEN\t%ALGORITHMS\t%EVIDENCE\t[%SAMPLE:%GT ]\n&quot;</span> vcfs_LBD/SV.LBD.keepRelated.subsetKnownGenes.flank1Mb.<span class="si">${</span><span class="nv">gene</span><span class="si">}</span>.vcf.gz &gt; SV_text_files_LBD/LBD_<span class="si">${</span><span class="nv">gene</span><span class="si">}</span>_to_pandas.txt
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Import-vcfs-into-pandas-as-the-main-data-frame">Import vcfs into pandas as the main data frame<a class="anchor-link" href="#Import-vcfs-into-pandas-as-the-main-data-frame">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Create list of files to be concatenated</span>
<span class="n">csv_file_list</span> <span class="o">=</span> <span class="n">glob</span><span class="o">.</span><span class="n">glob</span><span class="p">(</span><span class="s2">&quot;$PATH1/NIH_FALL_2021_SVs_1percFDR/SV_text_files_LBD/*to_pandas.txt&quot;</span><span class="p">)</span>

<span class="c1">#Make a list of data frames</span>
<span class="n">list_of_dataframes</span> <span class="o">=</span> <span class="p">[]</span>

<span class="k">for</span> <span class="n">filename</span> <span class="ow">in</span> <span class="n">csv_file_list</span><span class="p">:</span>
    <span class="n">list_of_dataframes</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">filename</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">None</span><span class="p">))</span>
  

<span class="c1">#Merge individual data frames into one</span>
<span class="n">NDD_all_LBD</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">list_of_dataframes</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">join</span><span class="o">=</span><span class="s2">&quot;outer&quot;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Add column names, columns are in the same order as vcf was queried with bcftools</span>
<span class="n">NDD_all_LBD</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;CHROM&quot;</span><span class="p">,</span><span class="s2">&quot;POS&quot;</span><span class="p">,</span><span class="s2">&quot;END&quot;</span><span class="p">,</span><span class="s2">&quot;ID&quot;</span><span class="p">,</span><span class="s2">&quot;REF&quot;</span><span class="p">,</span><span class="s2">&quot;ALT&quot;</span><span class="p">,</span><span class="s2">&quot;QUAL&quot;</span><span class="p">,</span> <span class="s2">&quot;GENE&quot;</span><span class="p">,</span> <span class="s2">&quot;FILTER&quot;</span><span class="p">,</span><span class="s2">&quot;SVLEN&quot;</span><span class="p">,</span><span class="s2">&quot;ALGORITHMS&quot;</span><span class="p">,</span><span class="s2">&quot;EVIDENCE&quot;</span><span class="p">,</span><span class="s2">&quot;SAMPLES&quot;</span><span class="p">]</span>
<span class="n">NDD_all_LBD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Import-case-control-allele-frequencies-and-add-them-to-data-frame">Import case-control allele frequencies and add them to data frame<a class="anchor-link" href="#Import-case-control-allele-frequencies-and-add-them-to-data-frame">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Import case and control allele frequencies, rename columns and select only relevant columns. Drop duplicates before merging</span>

<span class="n">LBD_freqs</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;$PATH1/NIH_FALL_2021_SVs_1percFDR/freqs_LBD/freqs_LBD_cases.afreq&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;\s+&quot;</span><span class="p">)</span>
<span class="n">LBD_freqs</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;ALT_FREQS&#39;</span><span class="p">:</span> <span class="s1">&#39;LBD_ALT_FREQS&#39;</span><span class="p">},</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">LBD_freqs</span> <span class="o">=</span> <span class="n">LBD_freqs</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,[</span><span class="mi">1</span><span class="p">,</span><span class="mi">4</span><span class="p">]]</span>
<span class="n">LBD_freqs</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="n">LBD_controls_freqs</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;$PATH1/NIH_FALL_2021_SVs_1percFDR/freqs_LBD/freqs_LBD_controls.afreq&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;\s+&quot;</span><span class="p">)</span>
<span class="n">LBD_controls_freqs</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;ALT_FREQS&#39;</span><span class="p">:</span> <span class="s1">&#39;controls_ALT_FREQS&#39;</span><span class="p">},</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">LBD_controls_freqs</span> <span class="o">=</span> <span class="n">LBD_controls_freqs</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,[</span><span class="mi">1</span><span class="p">,</span><span class="mi">4</span><span class="p">]]</span>
<span class="n">LBD_controls_freqs</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="n">LBD_freqs</span> <span class="o">=</span> <span class="n">LBD_freqs</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">LBD_controls_freqs</span><span class="p">,</span> <span class="n">on</span> <span class="o">=</span> <span class="s2">&quot;ID&quot;</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="s2">&quot;outer&quot;</span><span class="p">)</span>
<span class="n">LBD_freqs</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Add case-control frequencies to data frame (some SVs are &gt;1 times because they overlap &gt;1 genes of interest)</span>
<span class="n">NDD_all_LBD</span> <span class="o">=</span> <span class="n">NDD_all_LBD</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">LBD_freqs</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">&quot;ID&quot;</span><span class="p">)</span>
<span class="n">NDD_all_LBD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="List-only-sample-IDs-that-have-minor-(mostly-alternative)-allele">List only sample IDs that have minor (mostly alternative) allele<a class="anchor-link" href="#List-only-sample-IDs-that-have-minor-(mostly-alternative)-allele">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Split sample id strings (separated by single space) into a list. List items are strings of format ID:Genotype</span>
<span class="c1">#If frequncy in controls &lt; 0.5, alt allele is minor allele and keep only sample names that contain /1 (0/1 or 1/1)</span>
<span class="c1">#If frequncy in controls &gt; 0.5, ref allele is minor allele and keep only sample names that contain 0/ (0/0 or 0/1)</span>

<span class="n">geno_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">()</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">)):</span>
    <span class="k">if</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;controls_ALT_FREQS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mf">0.5</span><span class="p">:</span>
        <span class="n">sample_list</span> <span class="o">=</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;SAMPLES&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot; &quot;</span><span class="p">)</span>
        <span class="n">minor_allele_samples</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">sample_list</span> <span class="k">if</span> <span class="s1">&#39;/1&#39;</span> <span class="ow">in</span> <span class="n">x</span><span class="p">]</span>
        <span class="n">b</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span><span class="s2">&quot;ID&quot;</span><span class="p">:</span><span class="n">NDD_all_LBD</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="mi">3</span><span class="p">],</span> <span class="s2">&quot;minor_allele_samples&quot;</span><span class="p">:[</span><span class="n">minor_allele_samples</span><span class="p">]})</span>
        <span class="n">geno_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">geno_df</span><span class="p">,</span><span class="n">b</span><span class="p">])</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">sample_list</span> <span class="o">=</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;SAMPLES&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot; &quot;</span><span class="p">)</span>
        <span class="n">minor_allele_samples</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">sample_list</span> <span class="k">if</span> <span class="s1">&#39;0/&#39;</span> <span class="ow">in</span> <span class="n">x</span><span class="p">]</span>
        <span class="n">b</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span><span class="s2">&quot;ID&quot;</span><span class="p">:</span><span class="n">NDD_all_LBD</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="mi">3</span><span class="p">],</span> <span class="s2">&quot;minor_allele_samples&quot;</span><span class="p">:[</span><span class="n">minor_allele_samples</span><span class="p">]})</span>
        <span class="n">geno_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">geno_df</span><span class="p">,</span><span class="n">b</span><span class="p">])</span>


<span class="c1">#Drop duplicate values</span>
<span class="n">geno_df</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">subset</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;ID&quot;</span><span class="p">])</span>
<span class="n">geno_df</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># Add non-ref sample IDs to main data frame and reset indeces</span>

<span class="n">NDD_all_LBD</span> <span class="o">=</span> <span class="n">NDD_all_LBD</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">geno_df</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">&quot;ID&quot;</span><span class="p">)</span>
<span class="n">NDD_all_LBD</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">NDD_all_LBD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Gene-region,-promoter,-exon,-non-coding,-GeneHancer-and-CADD-SV-overlap">Gene region, promoter, exon, non-coding, GeneHancer and CADD-SV overlap<a class="anchor-link" href="#Gene-region,-promoter,-exon,-non-coding,-GeneHancer-and-CADD-SV-overlap">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Import information on starting files section</span>
<span class="c1">#N.B! biomart doesn&#39;t recognize PARK2 but PRKN, that name was substituted manually to biomart and from biomart back to PARK2</span>

<span class="n">chr_strand</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;$PATH1/NIH_FALL_2021_SVs/NDD_ensembl_104_canonical_gene_regions.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;\s+&quot;</span><span class="p">)</span>
<span class="n">chr_strand</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#To include promoter regions, substract 1000bp (gnomAD SV pipeline uses this value) from transcription start site if gene is </span>
<span class="c1">#on plus strand or add 1000bp to transcrip end if gene is on minus strand.</span>

<span class="n">chr_strand</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">chr_strand</span><span class="p">[</span><span class="s1">&#39;Strand&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s2">&quot;+&quot;</span><span class="p">,</span> <span class="s1">&#39;start&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">chr_strand</span><span class="p">[</span><span class="s2">&quot;txStart&quot;</span><span class="p">]</span> <span class="o">-</span> <span class="mi">1000</span>
<span class="n">chr_strand</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">chr_strand</span><span class="p">[</span><span class="s1">&#39;Strand&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s2">&quot;+&quot;</span><span class="p">,</span> <span class="s1">&#39;stop&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">chr_strand</span><span class="p">[</span><span class="s2">&quot;txEnd&quot;</span><span class="p">]</span>

<span class="n">chr_strand</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">chr_strand</span><span class="p">[</span><span class="s1">&#39;Strand&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s2">&quot;-&quot;</span><span class="p">,</span> <span class="s1">&#39;start&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">chr_strand</span><span class="p">[</span><span class="s2">&quot;txStart&quot;</span><span class="p">]</span>
<span class="n">chr_strand</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">chr_strand</span><span class="p">[</span><span class="s1">&#39;Strand&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s2">&quot;-&quot;</span><span class="p">,</span> <span class="s1">&#39;stop&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">chr_strand</span><span class="p">[</span><span class="s2">&quot;txEnd&quot;</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1000</span>

<span class="n">chr_strand</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Add start and stop sites to main data frame and reset indeces</span>

<span class="n">NDD_all_LBD</span> <span class="o">=</span> <span class="n">NDD_all_LBD</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">chr_strand</span><span class="p">,</span> <span class="n">left_on</span><span class="o">=</span><span class="s2">&quot;GENE&quot;</span><span class="p">,</span> <span class="n">right_on</span><span class="o">=</span><span class="s2">&quot;Gene&quot;</span><span class="p">)</span>
<span class="n">NDD_all_LBD</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">NDD_all_LBD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Import exon range information</span>
<span class="c1">#N.B.biomart doens&#39;t recognize PARK2 but PRKN, that name was substituted manually</span>

<span class="n">exons</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;$PATH1/NIH_FALL_2021_SVs/NDD_ensembl_104_canonical_exon_ranges.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;\s+&quot;</span><span class="p">)</span>

<span class="c1"># Make exonStarts and Stops into lists</span>

<span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonStarts&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonStarts&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;,&quot;</span><span class="p">)</span>
<span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonEnds&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonEnds&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;,&quot;</span><span class="p">)</span>

<span class="c1"># Remove empty value at the end of each list introduced by splitting</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span> <span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">exons</span><span class="p">)):</span>
    <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonStarts&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonStarts&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="k">if</span> <span class="n">x</span> <span class="o">!=</span><span class="s1">&#39;&#39;</span><span class="p">]</span>
    <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonEnds&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonEnds&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="k">if</span> <span class="n">x</span> <span class="o">!=</span><span class="s1">&#39;&#39;</span><span class="p">]</span>
    
<span class="c1">#Convert list of strings into list of numbers so they can be used as interval ranges</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span> <span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">exons</span><span class="p">)):</span>
    <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonStarts&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="nb">int</span><span class="p">(</span><span class="n">val</span><span class="p">)</span> <span class="k">for</span> <span class="n">val</span> <span class="ow">in</span> <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonStarts&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]]</span>
    <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonEnds&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="nb">int</span><span class="p">(</span><span class="n">val</span><span class="p">)</span> <span class="k">for</span> <span class="n">val</span> <span class="ow">in</span> <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonEnds&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]]</span>

<span class="c1"># Make exon interval arrays</span>
<span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exon_intervals&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">exons</span><span class="p">)):</span>
     <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exon_intervals&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">=</span>  <span class="n">pd</span><span class="o">.</span><span class="n">IntervalIndex</span><span class="o">.</span><span class="n">from_arrays</span><span class="p">(</span><span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonStarts&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonEnds&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s1">&#39;both&#39;</span><span class="p">)</span>

<span class="n">exons</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Merge exon range information per gene ID</span>

<span class="n">NDD_all_LBD</span> <span class="o">=</span> <span class="n">NDD_all_LBD</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">exons</span><span class="p">,</span> <span class="n">right_on</span><span class="o">=</span><span class="s2">&quot;Gene&quot;</span><span class="p">,</span> <span class="n">left_on</span><span class="o">=</span><span class="s2">&quot;GENE&quot;</span><span class="p">)</span>
<span class="n">NDD_all_LBD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Make a list from information on exon overlap. Exon start and stop positions are given for every row (otherwise would be need to check exon overlap chromsome-wise)</span>
<span class="c1">#pandas intervals need to be start &lt; stop so if loop to test if SV POS &lt; END: interval = pos-end, if not: interval=end-pos</span>

<span class="n">exon_overlaps</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">)):</span>
        <span class="k">if</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">&lt;</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]:</span>
            <span class="n">overlap_truefalse</span> <span class="o">=</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;exon_intervals&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]))</span>
            <span class="k">if</span> <span class="kc">True</span> <span class="ow">in</span> <span class="n">overlap_truefalse</span><span class="p">:</span>
                <span class="n">exon_overlaps</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Exonic&quot;</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">exon_overlaps</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Not_exonic&quot;</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">overlap_truefalse</span> <span class="o">=</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;exon_intervals&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]))</span>
            <span class="k">if</span> <span class="kc">True</span> <span class="ow">in</span> <span class="n">overlap_truefalse</span><span class="p">:</span>
                <span class="n">exon_overlaps</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Exonic&quot;</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">exon_overlaps</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Not_exonic&quot;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Add exon overlap information to data frame and reset indeces</span>
<span class="nb">print</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">exon_overlaps</span><span class="p">)</span> <span class="o">==</span> <span class="nb">len</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">))</span>

<span class="n">NDD_all_LBD</span> <span class="o">=</span> <span class="n">NDD_all_LBD</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">candidate_gene_exon_overlap</span> <span class="o">=</span> <span class="n">exon_overlaps</span><span class="p">)</span>
<span class="n">NDD_all_LBD</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">NDD_all_LBD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># Make a list from information on gene overlap, i.e. if variant pos-end interval overlaps with start-stop interval</span>
<span class="c1"># take into account that END can be smaller than POS like above</span>

<span class="n">gene_overlap_LBD</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span> <span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">)):</span>
    <span class="k">if</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]:</span>
        <span class="n">i1</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
        <span class="n">i2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;start&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;stop&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">i1</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">i2</span><span class="p">):</span>
            <span class="n">gene_overlap_LBD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Candidate_gene_overlap&quot;</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">gene_overlap_LBD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;No_candidate_gene_overlap&quot;</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">i1</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
        <span class="n">i2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;start&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;stop&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">i1</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">i2</span><span class="p">):</span>
            <span class="n">gene_overlap_LBD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Candidate_gene_overlap&quot;</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">gene_overlap_LBD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;No_candidate_gene_overlap&quot;</span><span class="p">)</span> 
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># Add overlap gene information on main data frame and reset indeces</span>
<span class="n">NDD_all_LBD</span> <span class="o">=</span> <span class="n">NDD_all_LBD</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">Candidate_gene_overlap</span> <span class="o">=</span> <span class="n">gene_overlap_LBD</span><span class="p">)</span>
<span class="n">NDD_all_LBD</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">NDD_all_LBD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># Make a list from information on promoter overlap, i.e. if gene on +: start-transcription start site ; gene on -: transcription end-stop</span>
<span class="c1"># Also need take into account that END can be smaller than POS.</span>
<span class="c1"># Therefore two main loops: one tests if POS &lt; END other one tests if Strand is + or -</span>

<span class="n">promoter_overlap_LBD</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span> <span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">)):</span>
    <span class="k">if</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;Strand&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">==</span> <span class="s2">&quot;+&quot;</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]:</span>
            <span class="n">i1</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="n">i2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;start&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;txStart&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">i1</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">i2</span><span class="p">):</span>
                <span class="n">promoter_overlap_LBD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Promoter_overlap&quot;</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">promoter_overlap_LBD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;No_promoter_overlap&quot;</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">i1</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="n">i2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;start&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;txStart&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">i1</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">i2</span><span class="p">):</span>
                <span class="n">promoter_overlap_LBD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Promoter_overlap&quot;</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">promoter_overlap_LBD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;No_promoter_overlap&quot;</span><span class="p">)</span> 
    <span class="k">else</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]:</span>
            <span class="n">i1</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="n">i2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;txEnd&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;stop&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">i1</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">i2</span><span class="p">):</span>
                <span class="n">promoter_overlap_LBD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Promoter_overlap&quot;</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">promoter_overlap_LBD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;No_promoter_overlap&quot;</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">i1</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="n">i2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;txEnd&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s2">&quot;stop&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">i1</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">i2</span><span class="p">):</span>
                <span class="n">promoter_overlap_LBD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Promoter_overlap&quot;</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">promoter_overlap_LBD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;No_promoter_overlap&quot;</span><span class="p">)</span> 
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># Add promoter overlap information on main data frame and reset indeces</span>
<span class="n">NDD_all_LBD</span> <span class="o">=</span> <span class="n">NDD_all_LBD</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">Candidate_gene_promoter_overlap</span> <span class="o">=</span> <span class="n">promoter_overlap_LBD</span><span class="p">)</span>
<span class="n">NDD_all_LBD</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">NDD_all_LBD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Add information on non-coding variants. Variant overlaps non-coding regions of candidate-gene if it overlaps the gene but is not exonic.</span>
<span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s1">&#39;Candidate_gene_non_coding_overlap&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;No_non_coding_overlap&#39;</span>
<span class="n">NDD_all_LBD2</span> <span class="o">=</span> <span class="n">NDD_all_LBD</span>
<span class="n">NDD_all_LBD2</span><span class="o">.</span><span class="n">loc</span><span class="p">[(</span><span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s1">&#39;candidate_gene_exon_overlap&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;Not_exonic&#39;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">NDD_all_LBD</span><span class="p">[</span><span class="s1">&#39;Candidate_gene_overlap&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s2">&quot;Candidate_gene_overlap&quot;</span> <span class="p">),</span> <span class="p">[</span><span class="s1">&#39;Candidate_gene_non_coding_overlap&#39;</span><span class="p">]]</span> <span class="o">=</span> <span class="s2">&quot;Non-coding_overlap&quot;</span>
<span class="n">NDD_all_LBD2</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Query GeneHancer overlaps based on gene information and create data frame out of overlaps</span>
<span class="n">LBD</span> <span class="o">=</span> <span class="n">NDD_all_LBD2</span>
<span class="n">genehancer_overlap_LBD</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span><span class="s2">&quot;ID&quot;</span><span class="p">:[],</span><span class="s2">&quot;geneHancerIdentifier&quot;</span><span class="p">:[]})</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">LBD</span><span class="o">.</span><span class="n">index</span><span class="p">:</span>
    <span class="k">for</span> <span class="n">j</span> <span class="ow">in</span> <span class="n">genehancer</span><span class="o">.</span><span class="n">index</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">LBD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="n">LBD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]:</span>
            <span class="n">i1</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">LBD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">LBD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="n">i2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneHancerStart&quot;</span><span class="p">][</span><span class="n">j</span><span class="p">],</span> <span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneHancerEnd&quot;</span><span class="p">][</span><span class="n">j</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="k">if</span> <span class="p">((</span><span class="n">i1</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">i2</span><span class="p">))</span> <span class="ow">and</span> <span class="p">(</span><span class="n">LBD</span><span class="p">[</span><span class="s2">&quot;GENE&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">==</span> <span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneName&quot;</span><span class="p">][</span><span class="n">j</span><span class="p">])):</span>
                <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span><span class="s2">&quot;ID&quot;</span><span class="p">:</span><span class="n">LBD</span><span class="p">[</span><span class="s2">&quot;ID&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="s2">&quot;geneHancerIdentifier&quot;</span><span class="p">:</span><span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneHancerIdentifier&quot;</span><span class="p">][</span><span class="n">j</span><span class="p">]},</span> <span class="n">index</span><span class="o">=</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
                <span class="n">genehancer_overlap_LBD</span> <span class="o">=</span> <span class="n">genehancer_overlap_LBD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="k">continue</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">i1</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">LBD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">LBD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="n">i2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneHancerStart&quot;</span><span class="p">][</span><span class="n">j</span><span class="p">],</span> <span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneHancerEnd&quot;</span><span class="p">][</span><span class="n">j</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="k">if</span> <span class="p">((</span><span class="n">i1</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">i2</span><span class="p">))</span> <span class="ow">and</span> <span class="p">(</span><span class="n">LBD</span><span class="p">[</span><span class="s2">&quot;GENE&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">==</span> <span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneName&quot;</span><span class="p">][</span><span class="n">j</span><span class="p">])):</span>
                <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span><span class="s2">&quot;ID&quot;</span><span class="p">:</span><span class="n">LBD</span><span class="p">[</span><span class="s2">&quot;ID&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="s2">&quot;geneHancerIdentifier&quot;</span><span class="p">:</span><span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneHancerIdentifier&quot;</span><span class="p">][</span><span class="n">j</span><span class="p">]},</span> <span class="n">index</span><span class="o">=</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
                <span class="n">genehancer_overlap_LBD</span> <span class="o">=</span> <span class="n">genehancer_overlap_LBD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="k">continue</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Check overlap df and save it</span>
<span class="n">genehancer_overlap_LBD</span>
<span class="n">genehancer_overlap_LBD</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s2">&quot;genehancer_overlap_LBD_1percFDR.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">index</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Format GeneHancer df for merging</span>
<span class="n">genehancer_overlap_LBD</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;genehancer_overlap_LBD.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">genehancer_overlap_LBD</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s2">&quot;ID&quot;</span><span class="p">)[</span><span class="s2">&quot;geneHancerIdentifier&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="nb">list</span><span class="p">)</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;genehancer&quot;</span><span class="p">)</span>
<span class="n">df</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Merge genehancer info to main data frame</span>
<span class="n">LBD</span> <span class="o">=</span> <span class="n">LBD</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">&quot;ID&quot;</span><span class="p">,</span> <span class="n">how</span> <span class="o">=</span> <span class="s2">&quot;left&quot;</span><span class="p">)</span>

<span class="c1">#Add info of no GeneHancer overlap for SVs that did not overlap</span>
<span class="n">LBD</span><span class="p">[</span><span class="s2">&quot;genehancer&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="s2">&quot;No_genehancer_overlap&quot;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Import unique CADD-SV scores</span>
<span class="c1">##CADD-SV output has mixed separators of tabs and spaces, so first create a tsv with wanted columns tab separated and unique SVs only (128 shared SVs between overlapping 1Mb gene regions)</span>
<span class="o">!</span>awk <span class="s1">&#39;OFS=&quot;\t&quot; {print $1,$2,$3,$4,$5,$6,$7,$8}&#39;</span> <span class="nv">$PATH1</span>/CADD-SV-1.1/output/LBD_1percFDR_NDD_all_vcfs_sorted_score.bed &gt; <span class="nv">$PATH1</span>/CADD-SV-1.1/output/LBD_1percFDR_NDD_all_vcfs_sorted_score.txt
<span class="o">!</span>head -n1 <span class="nv">$PATH1</span>/CADD-SV-1.1/output/LBD_1percFDR_NDD_all_vcfs_sorted_score.txt &gt; <span class="nv">$PATH1</span>/CADD-SV-1.1/output/LBD_1percFDR_NDD_all_vcfs_sorted_unique_score.txt
<span class="o">!</span>sort <span class="nv">$PATH1</span>/CADD-SV-1.1/output/LBD_1percFDR_NDD_all_vcfs_sorted_score.txt <span class="p">|</span> uniq &gt;&gt; <span class="nv">$PATH1</span>/CADD-SV-1.1/output/LBD_1percFDR_NDD_all_vcfs_sorted_unique_score.txt
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">##Import CADD-SV scores</span>
<span class="n">caddsv_LBD</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;$PATH1/CADD-SV-1.1/output/LBD_1percFDR_NDD_all_vcfs_sorted_unique_score.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">caddsv_LBD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Merge CADD-SV info to main LBD data frame</span>
<span class="n">LBD2</span> <span class="o">=</span> <span class="n">LBD</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">caddsv_LBD</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,[</span><span class="mi">4</span><span class="p">,</span><span class="mi">5</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">7</span><span class="p">]],</span> <span class="n">left_on</span><span class="o">=</span><span class="s2">&quot;ID&quot;</span><span class="p">,</span> <span class="n">right_on</span><span class="o">=</span><span class="s2">&quot;name&quot;</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="s2">&quot;left&quot;</span><span class="p">)</span>
<span class="n">LBD2</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>                                  
<span class="n">LBD2</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Tidy-main-data-frame">Tidy main data frame<a class="anchor-link" href="#Tidy-main-data-frame">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Tidy data frame by excluding redundant name column and minor allele sample column as last (array ends are noninclusive so +1 to range end).</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span> <span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">LBD2</span><span class="o">.</span><span class="n">columns</span><span class="p">)):</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">i</span><span class="p">,</span> <span class="n">LBD2</span><span class="o">.</span><span class="n">columns</span><span class="p">[</span><span class="n">i</span><span class="p">])</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">LBD2</span> <span class="o">=</span> <span class="n">LBD2</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="n">np</span><span class="o">.</span><span class="n">r_</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">12</span><span class="p">,</span><span class="mi">13</span><span class="p">:</span><span class="mi">15</span><span class="p">,</span><span class="mi">26</span><span class="p">:</span><span class="mi">31</span><span class="p">,</span><span class="mi">32</span><span class="p">:</span><span class="mi">35</span><span class="p">,</span><span class="mi">15</span><span class="p">]]</span>
<span class="n">LBD2</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Tidy by removing SVs for which AF=0 in both cases and controls.</span>
<span class="n">LBD3</span> <span class="o">=</span> <span class="n">LBD2</span><span class="p">[</span><span class="o">~</span><span class="p">((</span><span class="n">LBD2</span><span class="p">[</span><span class="s2">&quot;controls_ALT_FREQS&quot;</span><span class="p">]</span> <span class="o">==</span> <span class="mi">0</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">LBD2</span><span class="p">[</span><span class="s2">&quot;LBD_ALT_FREQS&quot;</span><span class="p">]</span> <span class="o">==</span> <span class="mi">0</span><span class="p">))]</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Also remove &quot;&lt;&gt;&quot; from ALT naming</span>
<span class="n">LBD3</span><span class="p">[</span><span class="s2">&quot;ALT&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">LBD3</span><span class="p">[</span><span class="s2">&quot;ALT&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s2">&quot;[&lt;&gt;]&quot;</span><span class="p">,</span><span class="s2">&quot;&quot;</span><span class="p">,</span> <span class="n">regex</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">LBD3</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Make results directory</span>
mkdir -p LBD_results
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Export to text file</span>
<span class="n">LBD3</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s2">&quot;LBD_results/LBD_1percFDR_all_info_unfiltered_with_CADDSV_genehancer.txt&quot;</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Frontotemporal-dementia/amyotrophic-lateral-sclerosis">Frontotemporal dementia/amyotrophic lateral sclerosis<a class="anchor-link" href="#Frontotemporal-dementia/amyotrophic-lateral-sclerosis">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Import-vcfs-into-pandas-as-the-main-data-frame">Import vcfs into pandas as the main data frame<a class="anchor-link" href="#Import-vcfs-into-pandas-as-the-main-data-frame">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#For each vcf (one per gene), print only columns of interest with gene name added as one of the columns.</span>
<span class="c1">#All columns are separated by tab, sample names with alt allele are separated with single space and their genotype with &quot;:&quot;</span>

<span class="nb">cd</span> <span class="nv">$PATH1</span>/NIH_FALL_2021_SVs_1percFDR
mkdir SV_text_files_FTD

module load bcftools

<span class="k">for</span> gene <span class="k">in</span> <span class="k">$(</span>ls -1 <span class="nv">$PATH1</span>/NIH_FALL_2021_SVs/NIH_files/CLEAN.knownGenes.plink/Gene_Locs/ <span class="p">|</span> awk -F<span class="s2">&quot;.&quot;</span> <span class="s1">&#39;{print $1}&#39;</span><span class="k">)</span><span class="p">;</span> <span class="k">do</span>
    bcftools query -f <span class="s2">&quot;%CHROM\t%POS\t%END\t%ID\t%REF\t%ALT\t%QUAL\t</span><span class="si">${</span><span class="nv">gene</span><span class="si">}</span><span class="s2">\t%FILTER\t%SVLEN\t%ALGORITHMS\t%EVIDENCE\t[%SAMPLE:%GT ]\n&quot;</span> vcfs_FTD/SV.FTD.keepRelated.subsetKnownGenes.flank1Mb.<span class="si">${</span><span class="nv">gene</span><span class="si">}</span>.vcf.gz &gt; SV_text_files_FTD/FTD_<span class="si">${</span><span class="nv">gene</span><span class="si">}</span>_to_pandas.txt
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Create list of files to be concatenated</span>
<span class="n">csv_file_list</span> <span class="o">=</span> <span class="n">glob</span><span class="o">.</span><span class="n">glob</span><span class="p">(</span><span class="s2">&quot;$PATH1/NIH_FALL_2021_SVs_1percFDR/SV_text_files_FTD/*to_pandas.txt&quot;</span><span class="p">)</span>

<span class="c1">#Make a list of data frames</span>
<span class="n">list_of_dataframes</span> <span class="o">=</span> <span class="p">[]</span>

<span class="k">for</span> <span class="n">filename</span> <span class="ow">in</span> <span class="n">csv_file_list</span><span class="p">:</span>
    <span class="n">list_of_dataframes</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">filename</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">None</span><span class="p">))</span>
  

<span class="c1">#Merge individual data frames into one</span>
<span class="n">NDD_all_FTD</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">list_of_dataframes</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">join</span><span class="o">=</span><span class="s2">&quot;outer&quot;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Add column names, columns are in the same order as vcf was queried with bcftools</span>
<span class="n">NDD_all_FTD</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;CHROM&quot;</span><span class="p">,</span><span class="s2">&quot;POS&quot;</span><span class="p">,</span><span class="s2">&quot;END&quot;</span><span class="p">,</span><span class="s2">&quot;ID&quot;</span><span class="p">,</span><span class="s2">&quot;REF&quot;</span><span class="p">,</span><span class="s2">&quot;ALT&quot;</span><span class="p">,</span><span class="s2">&quot;QUAL&quot;</span><span class="p">,</span> <span class="s2">&quot;GENE&quot;</span><span class="p">,</span> <span class="s2">&quot;FILTER&quot;</span><span class="p">,</span><span class="s2">&quot;SVLEN&quot;</span><span class="p">,</span><span class="s2">&quot;ALGORITHMS&quot;</span><span class="p">,</span><span class="s2">&quot;EVIDENCE&quot;</span><span class="p">,</span><span class="s2">&quot;SAMPLES&quot;</span><span class="p">]</span>
<span class="n">NDD_all_FTD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Import-case-control-allele-frequencies-and-add-them-to-data-frame">Import case-control allele frequencies and add them to data frame<a class="anchor-link" href="#Import-case-control-allele-frequencies-and-add-them-to-data-frame">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Import case and control allele frequencies, rename columns and select only relevant columns. Drop duplicates before merging</span>

<span class="n">FTD_freqs</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;$PATH1/NIH_FALL_2021_SVs_1percFDR/freqs_FTD/freqs_FTD_cases.afreq&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;\s+&quot;</span><span class="p">)</span>
<span class="n">FTD_freqs</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;ALT_FREQS&#39;</span><span class="p">:</span> <span class="s1">&#39;FTD_ALT_FREQS&#39;</span><span class="p">},</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">FTD_freqs</span> <span class="o">=</span> <span class="n">FTD_freqs</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,[</span><span class="mi">1</span><span class="p">,</span><span class="mi">4</span><span class="p">]]</span>
<span class="n">FTD_freqs</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="n">FTD_controls_freqs</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;$PATH1/NIH_FALL_2021_SVs_1percFDR/freqs_FTD/freqs_FTD_controls.afreq&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;\s+&quot;</span><span class="p">)</span>
<span class="n">FTD_controls_freqs</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;ALT_FREQS&#39;</span><span class="p">:</span> <span class="s1">&#39;controls_ALT_FREQS&#39;</span><span class="p">},</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">FTD_controls_freqs</span> <span class="o">=</span> <span class="n">FTD_controls_freqs</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,[</span><span class="mi">1</span><span class="p">,</span><span class="mi">4</span><span class="p">]]</span>
<span class="n">FTD_controls_freqs</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="n">FTD_freqs</span> <span class="o">=</span> <span class="n">FTD_freqs</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">FTD_controls_freqs</span><span class="p">,</span> <span class="n">on</span> <span class="o">=</span> <span class="s2">&quot;ID&quot;</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="s2">&quot;outer&quot;</span><span class="p">)</span>
<span class="n">FTD_freqs</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Add case-control frequencies to data frame (some SVs are &gt;1 times because they overlap &gt;1 genes of interest)</span>
<span class="n">NDD_all_FTD</span> <span class="o">=</span> <span class="n">NDD_all_FTD</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">FTD_freqs</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">&quot;ID&quot;</span><span class="p">)</span>
<span class="n">NDD_all_FTD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="List-only-sample-IDs-that-have-minor-(mostly-alternative)-allele">List only sample IDs that have minor (mostly alternative) allele<a class="anchor-link" href="#List-only-sample-IDs-that-have-minor-(mostly-alternative)-allele">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Split sample id strings (separated by single space) into a list. List items are strings of format ID:Genotype</span>
<span class="c1">#If frequncy in controls &lt; 0.5, alt allele is minor allele and keep only sample names that contain /1 (0/1 or 1/1)</span>
<span class="c1">#If frequncy in controls &gt; 0.5, ref allele is minor allele and keep only sample names that contain 0/ (0/0 or 0/1)</span>

<span class="n">geno_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">()</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">)):</span>
    <span class="k">if</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;controls_ALT_FREQS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mf">0.5</span><span class="p">:</span>
        <span class="n">sample_list</span> <span class="o">=</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;SAMPLES&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot; &quot;</span><span class="p">)</span>
        <span class="n">minor_allele_samples</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">sample_list</span> <span class="k">if</span> <span class="s1">&#39;/1&#39;</span> <span class="ow">in</span> <span class="n">x</span><span class="p">]</span>
        <span class="n">b</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span><span class="s2">&quot;ID&quot;</span><span class="p">:</span><span class="n">NDD_all_FTD</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="mi">3</span><span class="p">],</span> <span class="s2">&quot;minor_allele_samples&quot;</span><span class="p">:[</span><span class="n">minor_allele_samples</span><span class="p">]})</span>
        <span class="n">geno_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">geno_df</span><span class="p">,</span><span class="n">b</span><span class="p">])</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">sample_list</span> <span class="o">=</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;SAMPLES&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot; &quot;</span><span class="p">)</span>
        <span class="n">minor_allele_samples</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">sample_list</span> <span class="k">if</span> <span class="s1">&#39;0/&#39;</span> <span class="ow">in</span> <span class="n">x</span><span class="p">]</span>
        <span class="n">b</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span><span class="s2">&quot;ID&quot;</span><span class="p">:</span><span class="n">NDD_all_FTD</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="mi">3</span><span class="p">],</span> <span class="s2">&quot;minor_allele_samples&quot;</span><span class="p">:[</span><span class="n">minor_allele_samples</span><span class="p">]})</span>
        <span class="n">geno_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">geno_df</span><span class="p">,</span><span class="n">b</span><span class="p">])</span>


<span class="c1">#Drop duplicate values</span>
<span class="n">geno_df</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">subset</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;ID&quot;</span><span class="p">])</span>
<span class="n">geno_df</span>



<span class="c1"># Add non-ref sample IDs to main data frame and reset indeces</span>

<span class="n">NDD_all_FTD</span> <span class="o">=</span> <span class="n">NDD_all_FTD</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">geno_df</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">&quot;ID&quot;</span><span class="p">)</span>
<span class="n">NDD_all_FTD</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">NDD_all_FTD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Gene-region,-promoter,-exon,-non-coding,-GeneHancer-and-CADD-SV-overlap">Gene region, promoter, exon, non-coding, GeneHancer and CADD-SV overlap<a class="anchor-link" href="#Gene-region,-promoter,-exon,-non-coding,-GeneHancer-and-CADD-SV-overlap">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Import information on starting files section</span>
<span class="c1">#N.B! biomart doesn&#39;t recognize PARK2 but PRKN, that name was substituted manually to biomart and from biomart back to PARK2</span>

<span class="n">chr_strand</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;$PATH1/NIH_FALL_2021_SVs/NDD_ensembl_104_canonical_gene_regions.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;\s+&quot;</span><span class="p">)</span>
<span class="n">chr_strand</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#To include promoter regions, substract 1000bp (gnomAD SV pipeline uses this value) from transcription start site if gene is </span>
<span class="c1">#on plus strand or add 1000bp to transcrip end if gene is on minus strand.</span>

<span class="n">chr_strand</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">chr_strand</span><span class="p">[</span><span class="s1">&#39;Strand&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s2">&quot;+&quot;</span><span class="p">,</span> <span class="s1">&#39;start&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">chr_strand</span><span class="p">[</span><span class="s2">&quot;txStart&quot;</span><span class="p">]</span> <span class="o">-</span> <span class="mi">1000</span>
<span class="n">chr_strand</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">chr_strand</span><span class="p">[</span><span class="s1">&#39;Strand&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s2">&quot;+&quot;</span><span class="p">,</span> <span class="s1">&#39;stop&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">chr_strand</span><span class="p">[</span><span class="s2">&quot;txEnd&quot;</span><span class="p">]</span>

<span class="n">chr_strand</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">chr_strand</span><span class="p">[</span><span class="s1">&#39;Strand&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s2">&quot;-&quot;</span><span class="p">,</span> <span class="s1">&#39;start&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">chr_strand</span><span class="p">[</span><span class="s2">&quot;txStart&quot;</span><span class="p">]</span>
<span class="n">chr_strand</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">chr_strand</span><span class="p">[</span><span class="s1">&#39;Strand&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s2">&quot;-&quot;</span><span class="p">,</span> <span class="s1">&#39;stop&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">chr_strand</span><span class="p">[</span><span class="s2">&quot;txEnd&quot;</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1000</span>

<span class="n">chr_strand</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Add start and stop sites to main data frame and reset indeces</span>

<span class="n">NDD_all_FTD</span> <span class="o">=</span> <span class="n">NDD_all_FTD</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">chr_strand</span><span class="p">,</span> <span class="n">left_on</span><span class="o">=</span><span class="s2">&quot;GENE&quot;</span><span class="p">,</span> <span class="n">right_on</span><span class="o">=</span><span class="s2">&quot;Gene&quot;</span><span class="p">)</span>
<span class="n">NDD_all_FTD</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">NDD_all_FTD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Import exon range information</span>
<span class="c1">#N.B.biomart doens&#39;t recognize PARK2 but PRKN, that name was substituted manually</span>

<span class="n">exons</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;$PATH1/NIH_FALL_2021_SVs/NDD_ensembl_104_canonical_exon_ranges.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;\s+&quot;</span><span class="p">)</span>

<span class="c1"># Make exonStarts and Stops into lists</span>

<span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonStarts&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonStarts&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;,&quot;</span><span class="p">)</span>
<span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonEnds&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonEnds&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;,&quot;</span><span class="p">)</span>

<span class="c1"># Remove empty value at the end of each list introduced by splitting</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span> <span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">exons</span><span class="p">)):</span>
    <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonStarts&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonStarts&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="k">if</span> <span class="n">x</span> <span class="o">!=</span><span class="s1">&#39;&#39;</span><span class="p">]</span>
    <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonEnds&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonEnds&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="k">if</span> <span class="n">x</span> <span class="o">!=</span><span class="s1">&#39;&#39;</span><span class="p">]</span>
    
<span class="c1">#Convert list of strings into list of numbers so they can be used as interval ranges</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span> <span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">exons</span><span class="p">)):</span>
    <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonStarts&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="nb">int</span><span class="p">(</span><span class="n">val</span><span class="p">)</span> <span class="k">for</span> <span class="n">val</span> <span class="ow">in</span> <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonStarts&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]]</span>
    <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonEnds&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="nb">int</span><span class="p">(</span><span class="n">val</span><span class="p">)</span> <span class="k">for</span> <span class="n">val</span> <span class="ow">in</span> <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonEnds&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]]</span>

<span class="c1"># Make exon interval arrays</span>
<span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exon_intervals&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">exons</span><span class="p">)):</span>
     <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exon_intervals&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">=</span>  <span class="n">pd</span><span class="o">.</span><span class="n">IntervalIndex</span><span class="o">.</span><span class="n">from_arrays</span><span class="p">(</span><span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonStarts&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">exons</span><span class="p">[</span><span class="s2">&quot;exonEnds&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s1">&#39;both&#39;</span><span class="p">)</span>

<span class="n">exons</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Merge exon range information per gene ID</span>

<span class="n">NDD_all_FTD</span> <span class="o">=</span> <span class="n">NDD_all_FTD</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">exons</span><span class="p">,</span> <span class="n">right_on</span><span class="o">=</span><span class="s2">&quot;Gene&quot;</span><span class="p">,</span> <span class="n">left_on</span><span class="o">=</span><span class="s2">&quot;GENE&quot;</span><span class="p">)</span>
<span class="n">NDD_all_FTD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Make a list from information on exon overlap. Exon start and stop positions are given for every row (otherwise would be need to check exon overlap chromsome-wise)</span>
<span class="c1">#pandas intervals need to be start &lt; stop so if loop to test if SV POS &lt; END: interval = pos-end, if not: interval=end-pos</span>

<span class="n">exon_overlaps</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">)):</span>
        <span class="k">if</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">&lt;</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]:</span>
            <span class="n">overlap_truefalse</span> <span class="o">=</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;exon_intervals&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]))</span>
            <span class="k">if</span> <span class="kc">True</span> <span class="ow">in</span> <span class="n">overlap_truefalse</span><span class="p">:</span>
                <span class="n">exon_overlaps</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Exonic&quot;</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">exon_overlaps</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Not_exonic&quot;</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">overlap_truefalse</span> <span class="o">=</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;exon_intervals&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]))</span>
            <span class="k">if</span> <span class="kc">True</span> <span class="ow">in</span> <span class="n">overlap_truefalse</span><span class="p">:</span>
                <span class="n">exon_overlaps</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Exonic&quot;</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">exon_overlaps</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Not_exonic&quot;</span><span class="p">)</span>
                
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Add exon overlap information to data frame and reset indeces</span>
<span class="nb">print</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">exon_overlaps</span><span class="p">)</span> <span class="o">==</span> <span class="nb">len</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">))</span>

<span class="n">NDD_all_FTD</span> <span class="o">=</span> <span class="n">NDD_all_FTD</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">candidate_gene_exon_overlap</span> <span class="o">=</span> <span class="n">exon_overlaps</span><span class="p">)</span>
<span class="n">NDD_all_FTD</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">NDD_all_FTD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># Make a list from information on gene overlap, i.e. if variant pos-end interval overlaps with start-stop interval</span>
<span class="c1"># take into account that END can be smaller than POS like above</span>

<span class="n">gene_overlap_FTD</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span> <span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">)):</span>
    <span class="k">if</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]:</span>
        <span class="n">i1</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
        <span class="n">i2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;start&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;stop&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">i1</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">i2</span><span class="p">):</span>
            <span class="n">gene_overlap_FTD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Candidate_gene_overlap&quot;</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">gene_overlap_FTD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;No_candidate_gene_overlap&quot;</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">i1</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
        <span class="n">i2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;start&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;stop&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">i1</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">i2</span><span class="p">):</span>
            <span class="n">gene_overlap_FTD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Candidate_gene_overlap&quot;</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">gene_overlap_FTD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;No_candidate_gene_overlap&quot;</span><span class="p">)</span> 
            
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># Add overlap gene information on main data frame and reset indeces</span>
<span class="n">NDD_all_FTD</span> <span class="o">=</span> <span class="n">NDD_all_FTD</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">Candidate_gene_overlap</span> <span class="o">=</span> <span class="n">gene_overlap_FTD</span><span class="p">)</span>
<span class="n">NDD_all_FTD</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">NDD_all_FTD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># Make a list from information on promoter overlap, i.e. if gene on +: start-transcription start site ; gene on -: transcription end-stop</span>
<span class="c1"># Also need take into account that END can be smaller than POS.</span>
<span class="c1"># Therefore two main loops: one tests if POS &lt; END other one tests if Strand is + or -</span>

<span class="n">promoter_overlap_FTD</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span> <span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">)):</span>
    <span class="k">if</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;Strand&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">==</span> <span class="s2">&quot;+&quot;</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]:</span>
            <span class="n">i1</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="n">i2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;start&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;txStart&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">i1</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">i2</span><span class="p">):</span>
                <span class="n">promoter_overlap_FTD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Promoter_overlap&quot;</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">promoter_overlap_FTD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;No_promoter_overlap&quot;</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">i1</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="n">i2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;start&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;txStart&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">i1</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">i2</span><span class="p">):</span>
                <span class="n">promoter_overlap_FTD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Promoter_overlap&quot;</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">promoter_overlap_FTD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;No_promoter_overlap&quot;</span><span class="p">)</span> 
    <span class="k">else</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]:</span>
            <span class="n">i1</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="n">i2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;txEnd&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;stop&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">i1</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">i2</span><span class="p">):</span>
                <span class="n">promoter_overlap_FTD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Promoter_overlap&quot;</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">promoter_overlap_FTD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;No_promoter_overlap&quot;</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">i1</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="n">i2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;txEnd&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s2">&quot;stop&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">i1</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">i2</span><span class="p">):</span>
                <span class="n">promoter_overlap_FTD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Promoter_overlap&quot;</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">promoter_overlap_FTD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;No_promoter_overlap&quot;</span><span class="p">)</span> 
                
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># Add promoter overlap information on main data frame and reset indeces</span>
<span class="n">NDD_all_FTD</span> <span class="o">=</span> <span class="n">NDD_all_FTD</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">Candidate_gene_promoter_overlap</span> <span class="o">=</span> <span class="n">promoter_overlap_FTD</span><span class="p">)</span>
<span class="n">NDD_all_FTD</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">NDD_all_FTD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Add information on non-coding variants. Variant overlaps non-coding regions of candidate-gene if it overlaps the gene but is not exonic.</span>
<span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s1">&#39;Candidate_gene_non_coding_overlap&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;No_non_coding_overlap&#39;</span>
<span class="n">NDD_all_FTD2</span> <span class="o">=</span> <span class="n">NDD_all_FTD</span>
<span class="n">NDD_all_FTD2</span><span class="o">.</span><span class="n">loc</span><span class="p">[(</span><span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s1">&#39;candidate_gene_exon_overlap&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;Not_exonic&#39;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">NDD_all_FTD</span><span class="p">[</span><span class="s1">&#39;Candidate_gene_overlap&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s2">&quot;Candidate_gene_overlap&quot;</span> <span class="p">),</span> <span class="p">[</span><span class="s1">&#39;Candidate_gene_non_coding_overlap&#39;</span><span class="p">]]</span> <span class="o">=</span> <span class="s2">&quot;Non-coding_overlap&quot;</span>
<span class="n">NDD_all_FTD2</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Query GeneHancer overlaps based on gene information and create data frame out of overlaps</span>
<span class="n">FTD</span> <span class="o">=</span> <span class="n">NDD_all_FTD2</span>
<span class="n">genehancer_overlap_FTD</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span><span class="s2">&quot;ID&quot;</span><span class="p">:[],</span><span class="s2">&quot;geneHancerIdentifier&quot;</span><span class="p">:[]})</span>

<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">FTD</span><span class="o">.</span><span class="n">index</span><span class="p">:</span>
    <span class="k">for</span> <span class="n">j</span> <span class="ow">in</span> <span class="n">genehancer</span><span class="o">.</span><span class="n">index</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">FTD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="n">FTD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]:</span>
            <span class="n">i1</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">FTD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">FTD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="n">i2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneHancerStart&quot;</span><span class="p">][</span><span class="n">j</span><span class="p">],</span> <span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneHancerEnd&quot;</span><span class="p">][</span><span class="n">j</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="k">if</span> <span class="p">((</span><span class="n">i1</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">i2</span><span class="p">))</span> <span class="ow">and</span> <span class="p">(</span><span class="n">FTD</span><span class="p">[</span><span class="s2">&quot;GENE&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">==</span> <span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneName&quot;</span><span class="p">][</span><span class="n">j</span><span class="p">])):</span>
                <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span><span class="s2">&quot;ID&quot;</span><span class="p">:</span><span class="n">FTD</span><span class="p">[</span><span class="s2">&quot;ID&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="s2">&quot;geneHancerIdentifier&quot;</span><span class="p">:</span><span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneHancerIdentifier&quot;</span><span class="p">][</span><span class="n">j</span><span class="p">]},</span> <span class="n">index</span><span class="o">=</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
                <span class="n">genehancer_overlap_FTD</span> <span class="o">=</span> <span class="n">genehancer_overlap_FTD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="k">continue</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">i1</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">FTD</span><span class="p">[</span><span class="s2">&quot;END&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">FTD</span><span class="p">[</span><span class="s2">&quot;POS&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="n">i2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Interval</span><span class="p">(</span><span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneHancerStart&quot;</span><span class="p">][</span><span class="n">j</span><span class="p">],</span> <span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneHancerEnd&quot;</span><span class="p">][</span><span class="n">j</span><span class="p">],</span> <span class="n">closed</span><span class="o">=</span><span class="s2">&quot;both&quot;</span><span class="p">)</span>
            <span class="k">if</span> <span class="p">((</span><span class="n">i1</span><span class="o">.</span><span class="n">overlaps</span><span class="p">(</span><span class="n">i2</span><span class="p">))</span> <span class="ow">and</span> <span class="p">(</span><span class="n">FTD</span><span class="p">[</span><span class="s2">&quot;GENE&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">]</span> <span class="o">==</span> <span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneName&quot;</span><span class="p">][</span><span class="n">j</span><span class="p">])):</span>
                <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span><span class="s2">&quot;ID&quot;</span><span class="p">:</span><span class="n">FTD</span><span class="p">[</span><span class="s2">&quot;ID&quot;</span><span class="p">][</span><span class="n">i</span><span class="p">],</span> <span class="s2">&quot;geneHancerIdentifier&quot;</span><span class="p">:</span><span class="n">genehancer</span><span class="p">[</span><span class="s2">&quot;geneHancerIdentifier&quot;</span><span class="p">][</span><span class="n">j</span><span class="p">]},</span> <span class="n">index</span><span class="o">=</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
                <span class="n">genehancer_overlap_FTD</span> <span class="o">=</span> <span class="n">genehancer_overlap_FTD</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="k">continue</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Check overlap df and save it</span>
<span class="n">genehancer_overlap_FTD</span>
<span class="n">genehancer_overlap_FTD</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s2">&quot;genehancer_overlap_FTD_1percFDR.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">index</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Format GeneHancer df for merging</span>
<span class="c1">#genehancer_overlap_FTD = pd.read_csv(&quot;genehancer_overlap_FTD.txt&quot;, sep=&quot;\t&quot;)</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">genehancer_overlap_FTD</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s2">&quot;ID&quot;</span><span class="p">)[</span><span class="s2">&quot;geneHancerIdentifier&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="nb">list</span><span class="p">)</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;genehancer&quot;</span><span class="p">)</span>
<span class="n">df</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Merge genehancer info to main data frame</span>
<span class="n">FTD</span> <span class="o">=</span> <span class="n">FTD</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">&quot;ID&quot;</span><span class="p">,</span> <span class="n">how</span> <span class="o">=</span> <span class="s2">&quot;left&quot;</span><span class="p">)</span>

<span class="c1">#Add info of no GeneHancer overlap for SVs that did not overlap</span>
<span class="n">FTD</span><span class="p">[</span><span class="s2">&quot;genehancer&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="s2">&quot;No_genehancer_overlap&quot;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Import unique CADD-SV scores</span>
<span class="c1">##CADD-SV output has mixed separators of tabs and spaces, so first create a tsv with wanted columns tab separated and unique SVs only (128 shared SVs between overlapping 1Mb gene regions)</span>
<span class="o">!</span>awk <span class="s1">&#39;OFS=&quot;\t&quot; {print $1,$2,$3,$4,$5,$6,$7,$8}&#39;</span> <span class="nv">$PATH1</span>/CADD-SV-1.1/output/FTD_1percFDR_NDD_all_vcfs_sorted_score.bed &gt; <span class="nv">$PATH1</span>/CADD-SV-1.1/output/FTD_1percFDR_NDD_all_vcfs_sorted_score.txt
<span class="o">!</span>head -n1 <span class="nv">$PATH1</span>/CADD-SV-1.1/output/FTD_1percFDR_NDD_all_vcfs_sorted_score.txt &gt; <span class="nv">$PATH1</span>/CADD-SV-1.1/output/FTD_1percFDR_NDD_all_vcfs_sorted_unique_score.txt
<span class="o">!</span>sort <span class="nv">$PATH1</span>/CADD-SV-1.1/output/FTD_1percFDR_NDD_all_vcfs_sorted_score.txt <span class="p">|</span> uniq &gt;&gt; <span class="nv">$PATH1</span>/CADD-SV-1.1/output/FTD_1percFDR_NDD_all_vcfs_sorted_unique_score.txt
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">##Import CADD-SV scores</span>
<span class="n">caddsv_FTD</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;$PATH1/CADD-SV-1.1/output/FTD_1percFDR_NDD_all_vcfs_sorted_unique_score.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">caddsv_FTD</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Merge CADD-SV info to main FTD data frame</span>
<span class="n">FTD2</span> <span class="o">=</span> <span class="n">FTD</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">caddsv_FTD</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,[</span><span class="mi">4</span><span class="p">,</span><span class="mi">5</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">7</span><span class="p">]],</span> <span class="n">left_on</span><span class="o">=</span><span class="s2">&quot;ID&quot;</span><span class="p">,</span> <span class="n">right_on</span><span class="o">=</span><span class="s2">&quot;name&quot;</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="s2">&quot;left&quot;</span><span class="p">)</span>
<span class="n">FTD2</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>                                  
<span class="n">FTD2</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Tidy-main-data-frame">Tidy main data frame<a class="anchor-link" href="#Tidy-main-data-frame">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Tidy data frame by excluding redundant name column and minor allele sample column as last (array ends are noninclusive so +1 to range end).</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span> <span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">FTD2</span><span class="o">.</span><span class="n">columns</span><span class="p">)):</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">i</span><span class="p">,</span> <span class="n">FTD2</span><span class="o">.</span><span class="n">columns</span><span class="p">[</span><span class="n">i</span><span class="p">])</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">FTD2</span> <span class="o">=</span> <span class="n">FTD2</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="n">np</span><span class="o">.</span><span class="n">r_</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">12</span><span class="p">,</span><span class="mi">13</span><span class="p">:</span><span class="mi">15</span><span class="p">,</span><span class="mi">26</span><span class="p">:</span><span class="mi">31</span><span class="p">,</span><span class="mi">32</span><span class="p">:</span><span class="mi">35</span><span class="p">,</span><span class="mi">15</span><span class="p">]]</span>
<span class="n">FTD2</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Tidy by removing SVs for which AF=0 in both cases and controls.</span>
<span class="n">FTD3</span> <span class="o">=</span> <span class="n">FTD2</span><span class="p">[</span><span class="o">~</span><span class="p">((</span><span class="n">FTD2</span><span class="p">[</span><span class="s2">&quot;controls_ALT_FREQS&quot;</span><span class="p">]</span> <span class="o">==</span> <span class="mi">0</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">FTD2</span><span class="p">[</span><span class="s2">&quot;FTD_ALT_FREQS&quot;</span><span class="p">]</span> <span class="o">==</span> <span class="mi">0</span><span class="p">))]</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Also remove &quot;&lt;&gt;&quot; from ALT naming</span>
<span class="n">FTD3</span><span class="p">[</span><span class="s2">&quot;ALT&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">FTD3</span><span class="p">[</span><span class="s2">&quot;ALT&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s2">&quot;[&lt;&gt;]&quot;</span><span class="p">,</span><span class="s2">&quot;&quot;</span><span class="p">,</span> <span class="n">regex</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">FTD3</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Make results directory</span>
mkdir -p FTD_results
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Export to text file</span>
<span class="n">FTD3</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s2">&quot;FTD_results/FTD_1percFDR_all_info_unfiltered_with_CADDSV_genehancer.txt&quot;</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Shiny-app">Shiny app<a class="anchor-link" href="#Shiny-app">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="50-Neurodegenerative-disease-gene-images">50 Neurodegenerative disease gene images<a class="anchor-link" href="#50-Neurodegenerative-disease-gene-images">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">####Image preparation for shiny app </span>

<span class="c1">#Load libraries</span>
<span class="n">library</span><span class="p">(</span><span class="s2">&quot;Gviz&quot;</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="s2">&quot;tidyverse&quot;</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">AnnotationHub</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">biomaRt</span><span class="p">)</span>

<span class="c1">#Import files 50NDD files</span>
<span class="n">LBD</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">delim</span><span class="p">(</span><span class="s2">&quot;LBD_1percFDR_all_info_unfiltered_with_CADDSV_genehancer.txt&quot;</span><span class="p">)</span>
<span class="n">FTD</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">delim</span><span class="p">(</span><span class="s2">&quot;FTD_1percFDR_all_info_unfiltered_with_CADDSV_genehancer.txt&quot;</span><span class="p">)</span>
<span class="n">genes</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">csv</span><span class="p">(</span><span class="s2">&quot;NDD_ensembl_104_canonical_gene_regions.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">,</span> <span class="n">stringsAsFactors</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span>

<span class="c1">#Format files</span>
<span class="c1">##Add chr prefix</span>
<span class="n">LBD</span><span class="err">$</span><span class="n">CHROM</span> <span class="o">&lt;-</span> <span class="n">paste0</span><span class="p">(</span><span class="s2">&quot;chr&quot;</span><span class="p">,</span><span class="n">LBD</span><span class="err">$</span><span class="n">CHROM</span><span class="p">)</span>
<span class="n">FTD</span><span class="err">$</span><span class="n">CHROM</span> <span class="o">&lt;-</span> <span class="n">paste0</span><span class="p">(</span><span class="s2">&quot;chr&quot;</span><span class="p">,</span> <span class="n">FTD</span><span class="err">$</span><span class="n">CHROM</span><span class="p">)</span>
<span class="c1">#Change : in SVTYPE to _</span>
<span class="n">LBD</span><span class="err">$</span><span class="n">ALT</span> <span class="o">&lt;-</span> <span class="n">gsub</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">,</span><span class="s2">&quot;_&quot;</span><span class="p">,</span><span class="n">LBD</span><span class="err">$</span><span class="n">ALT</span><span class="p">)</span>
<span class="n">FTD</span><span class="err">$</span><span class="n">ALT</span> <span class="o">&lt;-</span> <span class="n">gsub</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">,</span><span class="s2">&quot;_&quot;</span><span class="p">,</span><span class="n">FTD</span><span class="err">$</span><span class="n">ALT</span><span class="p">)</span>
<span class="c1">#Make AFs in percentages column, two decimal places</span>
<span class="n">LBD</span><span class="err">$</span><span class="n">LBD_ALT_FREQS_perc</span> <span class="o">&lt;-</span> <span class="n">paste0</span><span class="p">(</span><span class="nb">round</span><span class="p">(</span><span class="n">LBD</span><span class="err">$</span><span class="n">LBD_ALT_FREQS</span> <span class="o">*</span><span class="mi">100</span><span class="p">,</span><span class="n">digits</span><span class="o">=</span><span class="mi">2</span><span class="p">),</span> <span class="s2">&quot;%&quot;</span><span class="p">)</span>
<span class="n">LBD</span><span class="err">$</span><span class="n">controls_ALT_FREQS_perc</span> <span class="o">&lt;-</span> <span class="n">paste0</span><span class="p">(</span><span class="nb">round</span><span class="p">(</span><span class="n">LBD</span><span class="err">$</span><span class="n">controls_ALT_FREQS</span> <span class="o">*</span><span class="mi">100</span><span class="p">,</span><span class="n">digits</span><span class="o">=</span><span class="mi">2</span><span class="p">),</span> <span class="s2">&quot;%&quot;</span><span class="p">)</span>
<span class="n">FTD</span><span class="err">$</span><span class="n">FTD_ALT_FREQS_perc</span> <span class="o">&lt;-</span> <span class="n">paste0</span><span class="p">(</span><span class="nb">round</span><span class="p">(</span><span class="n">FTD</span><span class="err">$</span><span class="n">FTD_ALT_FREQS</span> <span class="o">*</span><span class="mi">100</span><span class="p">,</span><span class="n">digits</span><span class="o">=</span><span class="mi">2</span><span class="p">),</span> <span class="s2">&quot;%&quot;</span><span class="p">)</span>
<span class="n">FTD</span><span class="err">$</span><span class="n">controls_ALT_FREQS_perc</span> <span class="o">&lt;-</span> <span class="n">paste0</span><span class="p">(</span><span class="nb">round</span><span class="p">(</span><span class="n">FTD</span><span class="err">$</span><span class="n">controls_ALT_FREQS</span> <span class="o">*</span><span class="mi">100</span><span class="p">,</span><span class="n">digits</span><span class="o">=</span><span class="mi">2</span><span class="p">),</span> <span class="s2">&quot;%&quot;</span><span class="p">)</span>
<span class="c1">##Remove SV carrier column</span>
<span class="n">LBD</span> <span class="o">&lt;-</span> <span class="n">LBD</span><span class="p">[,</span><span class="o">-</span><span class="mi">23</span><span class="p">]</span>
<span class="n">FTD</span> <span class="o">&lt;-</span> <span class="n">FTD</span><span class="p">[,</span><span class="o">-</span><span class="mi">23</span><span class="p">]</span>
<span class="c1">##Make all END &gt; POS for visualisation (if any)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;LBD&quot;</span><span class="p">)</span>
<span class="k">for</span> <span class="p">(</span><span class="n">i</span> <span class="ow">in</span> <span class="mi">1</span><span class="p">:</span><span class="n">nrow</span><span class="p">(</span><span class="n">LBD</span><span class="p">))</span> <span class="p">{</span>
  <span class="k">if</span> <span class="p">(</span><span class="n">LBD</span><span class="err">$</span><span class="n">END</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">LBD</span><span class="err">$</span><span class="n">POS</span><span class="p">[</span><span class="n">i</span><span class="p">])</span> <span class="p">{</span>
    <span class="nb">next</span>
  <span class="p">}</span> <span class="k">else</span> <span class="p">{</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">paste</span><span class="p">(</span><span class="s2">&quot;Initial pos:&quot;</span><span class="p">,</span> <span class="n">LBD</span><span class="err">$</span><span class="n">POS</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="s2">&quot;and initial end:&quot;</span><span class="p">,</span> <span class="n">LBD</span><span class="err">$</span><span class="n">END</span><span class="p">[</span><span class="n">i</span><span class="p">]))</span>
    <span class="n">tmp</span> <span class="o">&lt;-</span> <span class="n">LBD</span><span class="err">$</span><span class="n">END</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
    <span class="n">LBD</span><span class="err">$</span><span class="n">END</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">LBD</span><span class="err">$</span><span class="n">POS</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
    <span class="n">LBD</span><span class="err">$</span><span class="n">POS</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">tmp</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">paste</span><span class="p">(</span><span class="s2">&quot;Updated pos:&quot;</span><span class="p">,</span> <span class="n">LBD</span><span class="err">$</span><span class="n">POS</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="s2">&quot;and updated end:&quot;</span><span class="p">,</span> <span class="n">LBD</span><span class="err">$</span><span class="n">END</span><span class="p">[</span><span class="n">i</span><span class="p">]))</span>
  <span class="p">}</span>
<span class="p">}</span>

<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;FTD&quot;</span><span class="p">)</span>
<span class="k">for</span> <span class="p">(</span><span class="n">i</span> <span class="ow">in</span> <span class="mi">1</span><span class="p">:</span><span class="n">nrow</span><span class="p">(</span><span class="n">FTD</span><span class="p">))</span> <span class="p">{</span>
  <span class="k">if</span> <span class="p">(</span><span class="n">FTD</span><span class="err">$</span><span class="n">END</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">FTD</span><span class="err">$</span><span class="n">POS</span><span class="p">[</span><span class="n">i</span><span class="p">])</span> <span class="p">{</span>
    <span class="nb">next</span>
  <span class="p">}</span> <span class="k">else</span> <span class="p">{</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">paste</span><span class="p">(</span><span class="s2">&quot;Initial pos:&quot;</span><span class="p">,</span> <span class="n">FTD</span><span class="err">$</span><span class="n">POS</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="s2">&quot;and initial end:&quot;</span><span class="p">,</span> <span class="n">FTD</span><span class="err">$</span><span class="n">END</span><span class="p">[</span><span class="n">i</span><span class="p">]))</span>
    <span class="n">tmp</span> <span class="o">&lt;-</span> <span class="n">FTD</span><span class="err">$</span><span class="n">END</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
    <span class="n">FTD</span><span class="err">$</span><span class="n">END</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">FTD</span><span class="err">$</span><span class="n">POS</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
    <span class="n">FTD</span><span class="err">$</span><span class="n">POS</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">tmp</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">paste</span><span class="p">(</span><span class="s2">&quot;Updated pos:&quot;</span><span class="p">,</span> <span class="n">FTD</span><span class="err">$</span><span class="n">POS</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="s2">&quot;and updated end:&quot;</span><span class="p">,</span> <span class="n">FTD</span><span class="err">$</span><span class="n">END</span><span class="p">[</span><span class="n">i</span><span class="p">]))</span>
  <span class="p">}</span>
<span class="p">}</span>



<span class="c1">#Plot</span>
<span class="c1">##Initialize biomart track</span>
<span class="n">bm</span> <span class="o">&lt;-</span> <span class="n">useEnsembl</span><span class="p">(</span><span class="n">biomart</span> <span class="o">=</span> <span class="s2">&quot;genes&quot;</span><span class="p">)</span>
<span class="n">bm</span> <span class="o">&lt;-</span> <span class="n">useDataset</span><span class="p">(</span><span class="n">dataset</span> <span class="o">=</span> <span class="s2">&quot;hsapiens_gene_ensembl&quot;</span><span class="p">,</span> <span class="n">mart</span> <span class="o">=</span> <span class="n">bm</span><span class="p">)</span>

<span class="c1">##Make a df of canonical protein coding genes</span>
<span class="n">gene_info</span> <span class="o">&lt;-</span> <span class="k">as</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">frame</span><span class="p">(</span><span class="n">getBM</span><span class="p">(</span><span class="n">attributes</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="s2">&quot;chromosome_name&quot;</span><span class="p">,</span><span class="s2">&quot;start_position&quot;</span><span class="p">,</span><span class="s2">&quot;end_position&quot;</span><span class="p">,</span><span class="s2">&quot;gene_biotype&quot;</span><span class="p">,</span><span class="s2">&quot;external_gene_name&quot;</span><span class="p">),</span> <span class="n">filters</span> <span class="o">=</span> <span class="s2">&quot;transcript_is_canonical&quot;</span><span class="p">,</span> <span class="n">values</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">mart</span> <span class="o">=</span> <span class="n">bm</span><span class="p">))</span>
<span class="c1">###Include only autosomal and X variants and genes with gene name </span>
<span class="n">gene_info</span> <span class="o">&lt;-</span> <span class="n">gene_info</span> <span class="o">%&gt;%</span>
  <span class="nb">filter</span><span class="p">(</span><span class="n">gene_biotype</span> <span class="o">==</span> <span class="s2">&quot;protein_coding&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span>
  <span class="nb">filter</span><span class="p">(</span><span class="n">chromosome_name</span> <span class="o">%</span><span class="k">in</span>% c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,&quot;X&quot;)) %&gt;%
  <span class="nb">filter</span><span class="p">(</span><span class="n">external_gene_name</span> <span class="o">!=</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
<span class="c1">###Remove gene biotype column to save space</span>
<span class="n">gene_info</span> <span class="o">&lt;-</span> <span class="n">gene_info</span><span class="p">[,</span><span class="n">c</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">5</span><span class="p">)]</span>
<span class="c1">###Add chr prefix to chr codes</span>
<span class="n">gene_info</span><span class="err">$</span><span class="n">chromosome_name</span> <span class="o">&lt;-</span> <span class="n">paste0</span><span class="p">(</span><span class="s2">&quot;chr&quot;</span><span class="p">,</span><span class="n">gene_info</span><span class="err">$</span><span class="n">chromosome_name</span><span class="p">)</span>
<span class="c1">###Save df</span>
<span class="n">write</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="n">gene_info</span><span class="p">,</span> <span class="n">file</span> <span class="o">=</span> <span class="s2">&quot;gene_start_stop_name.txt&quot;</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">row</span><span class="o">.</span><span class="n">names</span> <span class="o">=</span> <span class="n">FALSE</span><span class="p">)</span>
  
<span class="c1">##LBD gene region (including promoter) overlaps</span>
<span class="k">for</span> <span class="p">(</span><span class="n">i</span> <span class="ow">in</span> <span class="mi">1</span><span class="p">:</span><span class="n">nrow</span><span class="p">(</span><span class="n">genes</span><span class="p">))</span> <span class="p">{</span>
  <span class="n">gene</span> <span class="o">=</span> <span class="n">genes</span><span class="err">$</span><span class="n">Gene</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
  <span class="n">start</span> <span class="o">=</span> <span class="n">genes</span><span class="err">$</span><span class="n">txStart</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">-</span><span class="mi">1000</span>
  <span class="n">stop</span> <span class="o">=</span> <span class="n">genes</span><span class="err">$</span><span class="n">txEnd</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1000</span>
  
  <span class="n">LBD_cases</span> <span class="o">&lt;-</span> <span class="n">LBD</span> <span class="o">%&gt;%</span>
    <span class="nb">filter</span><span class="p">(</span><span class="n">GENE</span> <span class="o">==</span> <span class="n">gene</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nb">filter</span><span class="p">(</span><span class="n">LBD_ALT_FREQS</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nb">filter</span><span class="p">((</span><span class="n">Candidate_gene_overlap</span> <span class="o">==</span> <span class="s2">&quot;Candidate_gene_overlap&quot;</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">Candidate_gene_promoter_overlap</span> <span class="o">==</span> <span class="s2">&quot;Promoter_overlap&quot;</span><span class="p">))</span>
  
  <span class="n">LBD_controls</span> <span class="o">&lt;-</span> <span class="n">LBD</span> <span class="o">%&gt;%</span>
    <span class="nb">filter</span><span class="p">(</span><span class="n">GENE</span> <span class="o">==</span> <span class="n">gene</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nb">filter</span><span class="p">(</span><span class="n">controls_ALT_FREQS</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nb">filter</span><span class="p">((</span><span class="n">Candidate_gene_overlap</span> <span class="o">==</span> <span class="s2">&quot;Candidate_gene_overlap&quot;</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">Candidate_gene_promoter_overlap</span> <span class="o">==</span> <span class="s2">&quot;Promoter_overlap&quot;</span><span class="p">))</span>
  
  <span class="n">LBD_cases</span><span class="err">$</span><span class="n">POS</span><span class="p">[</span><span class="n">LBD_cases</span><span class="err">$</span><span class="n">POS</span> <span class="o">&lt;</span> <span class="n">start</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">start</span> <span class="o">-</span> <span class="p">(</span><span class="mf">0.2</span><span class="o">*</span><span class="p">(</span><span class="n">stop</span><span class="o">-</span><span class="n">start</span><span class="p">))</span>
  <span class="n">LBD_cases</span><span class="err">$</span><span class="n">END</span><span class="p">[</span><span class="n">LBD_cases</span><span class="err">$</span><span class="n">END</span> <span class="o">&gt;</span> <span class="n">stop</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">stop</span> <span class="o">+</span> <span class="p">(</span><span class="mf">0.2</span><span class="o">*</span><span class="p">(</span><span class="n">stop</span><span class="o">-</span><span class="n">start</span><span class="p">))</span>
  <span class="n">LBD_controls</span><span class="err">$</span><span class="n">POS</span><span class="p">[</span><span class="n">LBD_controls</span><span class="err">$</span><span class="n">POS</span> <span class="o">&lt;</span> <span class="n">start</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">start</span> <span class="o">-</span> <span class="p">(</span><span class="mf">0.2</span><span class="o">*</span><span class="p">(</span><span class="n">stop</span><span class="o">-</span><span class="n">start</span><span class="p">))</span>
  <span class="n">LBD_controls</span><span class="err">$</span><span class="n">END</span><span class="p">[</span><span class="n">LBD_controls</span><span class="err">$</span><span class="n">END</span> <span class="o">&gt;</span> <span class="n">stop</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">stop</span> <span class="o">+</span> <span class="p">(</span><span class="mf">0.2</span><span class="o">*</span><span class="p">(</span><span class="n">stop</span><span class="o">-</span><span class="n">start</span><span class="p">))</span>
  
  <span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="s2">&quot;NIH_LBD_FTD_SV_shinyapp/NIH_SV_images/LBD/&quot;</span><span class="p">,</span><span class="n">gene</span><span class="p">,</span><span class="s2">&quot;_LBD_candidate_gene_overlap.png&quot;</span><span class="p">),</span> <span class="n">height</span> <span class="o">=</span> <span class="mi">1180</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="mi">1180</span><span class="p">,</span> <span class="n">units</span> <span class="o">=</span> <span class="s2">&quot;px&quot;</span><span class="p">,</span> <span class="n">res</span><span class="o">=</span><span class="mi">100</span><span class="p">)</span>
  
  <span class="k">if</span> <span class="p">((</span><span class="n">nrow</span><span class="p">(</span><span class="n">LBD_cases</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">nrow</span><span class="p">(</span><span class="n">LBD_controls</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">))</span> <span class="p">{</span>
    <span class="n">gtrack</span> <span class="o">&lt;-</span> <span class="n">GenomeAxisTrack</span><span class="p">()</span>
    <span class="n">itrack</span> <span class="o">&lt;-</span> <span class="n">IdeogramTrack</span><span class="p">(</span><span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">LBD_cases</span><span class="err">$</span><span class="n">CHROM</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>
    <span class="n">biomTrack</span> <span class="o">&lt;-</span> <span class="n">BiomartGeneRegionTrack</span><span class="p">(</span><span class="n">transcriptAnnotation</span> <span class="o">=</span> <span class="s2">&quot;symbol&quot;</span><span class="p">,</span> <span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span><span class="n">chromosome</span> <span class="o">=</span> <span class="n">LBD_cases</span><span class="err">$</span><span class="n">CHROM</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">start</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="n">gene</span><span class="p">,</span> <span class="nb">filter</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">transcript_is_canonical</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),</span> <span class="n">biomart</span> <span class="o">=</span> <span class="n">bm</span><span class="p">,</span> <span class="n">protein_coding</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr3</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr5</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">)</span>
    <span class="n">atrack_LBD</span> <span class="o">&lt;-</span> <span class="n">AnnotationTrack</span><span class="p">(</span><span class="n">start</span><span class="o">=</span><span class="n">LBD_cases</span><span class="err">$</span><span class="n">POS</span><span class="p">,</span> <span class="n">end</span><span class="o">=</span><span class="n">LBD_cases</span><span class="err">$</span><span class="n">END</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">LBD_cases</span><span class="err">$</span><span class="n">CHROM</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">genome</span><span class="o">=</span><span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;LBD&quot;</span><span class="p">,</span> <span class="n">feature</span> <span class="o">=</span> <span class="n">LBD_cases</span><span class="err">$</span><span class="n">ALT</span><span class="p">,</span> <span class="nb">id</span><span class="o">=</span><span class="n">LBD_cases</span><span class="err">$</span><span class="n">LBD_ALT_FREQS_perc</span><span class="p">,</span> <span class="n">showFeatureId</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">item</span><span class="o">=</span><span class="s2">&quot;black&quot;</span><span class="p">,</span> <span class="nb">min</span><span class="o">.</span><span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
    <span class="n">atrack_controls</span> <span class="o">&lt;-</span> <span class="n">AnnotationTrack</span><span class="p">(</span><span class="n">start</span><span class="o">=</span><span class="n">LBD_controls</span><span class="err">$</span><span class="n">POS</span><span class="p">,</span> <span class="n">end</span><span class="o">=</span><span class="n">LBD_controls</span><span class="err">$</span><span class="n">END</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">LBD_controls</span><span class="err">$</span><span class="n">CHROM</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">genome</span><span class="o">=</span><span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;Controls&quot;</span><span class="p">,</span> <span class="n">feature</span> <span class="o">=</span> <span class="n">LBD_controls</span><span class="err">$</span><span class="n">ALT</span><span class="p">,</span> <span class="nb">id</span><span class="o">=</span><span class="n">LBD_controls</span><span class="err">$</span><span class="n">controls_ALT_FREQS_perc</span><span class="p">,</span> <span class="n">showFeatureId</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">item</span><span class="o">=</span><span class="s2">&quot;black&quot;</span><span class="p">,</span> <span class="nb">min</span><span class="o">.</span><span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
    <span class="n">plotTracks</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span><span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack_LBD</span><span class="p">,</span> <span class="n">atrack_controls</span><span class="p">,</span> <span class="n">biomTrack</span><span class="p">),</span> <span class="kn">from</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;#808080&quot;</span><span class="p">,</span> <span class="n">extend</span><span class="o">.</span><span class="n">right</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span> <span class="n">extend</span><span class="o">.</span><span class="n">left</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">group</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">BND</span><span class="o">=</span><span class="s2">&quot;#377246&quot;</span><span class="p">,</span> <span class="n">CPX</span><span class="o">=</span><span class="s2">&quot;#86E496&quot;</span><span class="p">,</span> <span class="n">DEL</span><span class="o">=</span><span class="s2">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="n">DUP</span><span class="o">=</span><span class="s2">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="n">INS</span><span class="o">=</span><span class="s2">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="n">INS_ME</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_ALU</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_LINE1</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_SVA</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INV</span><span class="o">=</span><span class="s2">&quot;#F69A57&quot;</span><span class="p">)</span>
  <span class="p">}</span> <span class="k">else</span> <span class="k">if</span> <span class="p">((</span><span class="n">nrow</span><span class="p">(</span><span class="n">LBD_cases</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">nrow</span><span class="p">(</span><span class="n">LBD_controls</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">))</span> <span class="p">{</span>
    <span class="n">gtrack</span> <span class="o">&lt;-</span> <span class="n">GenomeAxisTrack</span><span class="p">()</span>
    <span class="n">itrack</span> <span class="o">&lt;-</span> <span class="n">IdeogramTrack</span><span class="p">(</span><span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">LBD_cases</span><span class="err">$</span><span class="n">CHROM</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>
    <span class="n">biomTrack</span> <span class="o">&lt;-</span> <span class="n">BiomartGeneRegionTrack</span><span class="p">(</span><span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span><span class="n">chromosome</span> <span class="o">=</span> <span class="n">LBD_cases</span><span class="err">$</span><span class="n">CHROM</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">start</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;ENSEMBL&quot;</span><span class="p">,</span> <span class="nb">filter</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">transcript_is_canonical</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),</span> <span class="n">biomart</span> <span class="o">=</span> <span class="n">bm</span><span class="p">,</span> <span class="n">protein_coding</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr3</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr5</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">)</span>
    <span class="n">atrack_LBD</span> <span class="o">&lt;-</span> <span class="n">AnnotationTrack</span><span class="p">(</span><span class="n">start</span><span class="o">=</span><span class="n">LBD_cases</span><span class="err">$</span><span class="n">POS</span><span class="p">,</span> <span class="n">end</span><span class="o">=</span><span class="n">LBD_cases</span><span class="err">$</span><span class="n">END</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">LBD_cases</span><span class="err">$</span><span class="n">CHROM</span><span class="p">,</span> <span class="n">genome</span><span class="o">=</span><span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;LBD&quot;</span><span class="p">,</span> <span class="n">feature</span> <span class="o">=</span> <span class="n">LBD_cases</span><span class="err">$</span><span class="n">ALT</span><span class="p">,</span> <span class="nb">id</span><span class="o">=</span><span class="n">LBD_cases</span><span class="err">$</span><span class="n">LBD_ALT_FREQS_perc</span><span class="p">,</span> <span class="n">showFeatureId</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">item</span><span class="o">=</span><span class="s2">&quot;black&quot;</span><span class="p">,</span> <span class="n">col</span><span class="o">.</span><span class="n">line</span><span class="o">=</span><span class="s2">&quot;white&quot;</span><span class="p">,</span> <span class="nb">min</span><span class="o">.</span><span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
    <span class="n">plotTracks</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span><span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack_LBD</span><span class="p">,</span> <span class="n">biomTrack</span><span class="p">),</span> <span class="kn">from</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;#808080&quot;</span><span class="p">,</span> <span class="n">extend</span><span class="o">.</span><span class="n">right</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span> <span class="n">extend</span><span class="o">.</span><span class="n">left</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">group</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">BND</span><span class="o">=</span><span class="s2">&quot;#377246&quot;</span><span class="p">,</span> <span class="n">CPX</span><span class="o">=</span><span class="s2">&quot;#86E496&quot;</span><span class="p">,</span> <span class="n">DEL</span><span class="o">=</span><span class="s2">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="n">DUP</span><span class="o">=</span><span class="s2">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="n">INS</span><span class="o">=</span><span class="s2">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="n">INS_ME</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_ALU</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_LINE1</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_SVA</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INV</span><span class="o">=</span><span class="s2">&quot;#F69A57&quot;</span><span class="p">)</span>
  <span class="p">}</span> <span class="k">else</span> <span class="k">if</span> <span class="p">((</span><span class="n">nrow</span><span class="p">(</span><span class="n">LBD_cases</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">nrow</span><span class="p">(</span><span class="n">LBD_controls</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">))</span> <span class="p">{</span>
    <span class="n">gtrack</span> <span class="o">&lt;-</span> <span class="n">GenomeAxisTrack</span><span class="p">(</span><span class="n">showTitle</span><span class="o">=</span><span class="n">TRUE</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;No SVs&quot;</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;#808080&quot;</span><span class="p">,</span> <span class="n">rotation</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
    <span class="n">plotTracks</span><span class="p">(</span><span class="n">gtrack</span><span class="p">,</span> <span class="kn">from</span> <span class="o">=</span> <span class="mi">100</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="mi">110</span><span class="p">,</span> <span class="n">title</span><span class="o">.</span><span class="n">width</span> <span class="o">=</span> <span class="mi">5</span><span class="p">)</span>
  <span class="p">}</span> <span class="k">else</span> <span class="p">{</span>
    <span class="n">gtrack</span> <span class="o">&lt;-</span> <span class="n">GenomeAxisTrack</span><span class="p">()</span>
    <span class="n">itrack</span> <span class="o">&lt;-</span> <span class="n">IdeogramTrack</span><span class="p">(</span><span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">LBD_controls</span><span class="err">$</span><span class="n">CHROM</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>
    <span class="n">biomTrack</span> <span class="o">&lt;-</span> <span class="n">BiomartGeneRegionTrack</span><span class="p">(</span><span class="n">transcriptAnnotation</span> <span class="o">=</span> <span class="s2">&quot;symbol&quot;</span><span class="p">,</span> <span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span><span class="n">chromosome</span> <span class="o">=</span> <span class="n">LBD_controls</span><span class="err">$</span><span class="n">CHROM</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">start</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="n">gene</span><span class="p">,</span> <span class="nb">filter</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">transcript_is_canonical</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),</span> <span class="n">biomart</span> <span class="o">=</span> <span class="n">bm</span><span class="p">,</span> <span class="n">protein_coding</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr3</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr5</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">)</span>
    <span class="n">atrack_controls</span> <span class="o">&lt;-</span> <span class="n">AnnotationTrack</span><span class="p">(</span><span class="n">start</span><span class="o">=</span><span class="n">LBD_controls</span><span class="err">$</span><span class="n">POS</span><span class="p">,</span> <span class="n">end</span><span class="o">=</span><span class="n">LBD_controls</span><span class="err">$</span><span class="n">END</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">LBD_controls</span><span class="err">$</span><span class="n">CHROM</span><span class="p">,</span> <span class="n">genome</span><span class="o">=</span><span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;Controls&quot;</span><span class="p">,</span> <span class="n">feature</span> <span class="o">=</span> <span class="n">LBD_controls</span><span class="err">$</span><span class="n">ALT</span><span class="p">,</span> <span class="nb">id</span><span class="o">=</span><span class="n">LBD_controls</span><span class="err">$</span><span class="n">controls_ALT_FREQS_perc</span><span class="p">,</span> <span class="n">showFeatureId</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">item</span><span class="o">=</span><span class="s2">&quot;black&quot;</span><span class="p">,</span> <span class="nb">min</span><span class="o">.</span><span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
    <span class="n">plotTracks</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span><span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack_controls</span><span class="p">,</span> <span class="n">biomTrack</span><span class="p">),</span> <span class="kn">from</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;#808080&quot;</span><span class="p">,</span> <span class="n">extend</span><span class="o">.</span><span class="n">right</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span> <span class="n">extend</span><span class="o">.</span><span class="n">left</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">group</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">BND</span><span class="o">=</span><span class="s2">&quot;#377246&quot;</span><span class="p">,</span> <span class="n">CPX</span><span class="o">=</span><span class="s2">&quot;#86E496&quot;</span><span class="p">,</span> <span class="n">DEL</span><span class="o">=</span><span class="s2">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="n">DUP</span><span class="o">=</span><span class="s2">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="n">INS</span><span class="o">=</span><span class="s2">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="n">INS_ME</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_ALU</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_LINE1</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_SVA</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INV</span><span class="o">=</span><span class="s2">&quot;#F69A57&quot;</span><span class="p">)</span>
  <span class="p">}</span>
  <span class="n">dev</span><span class="o">.</span><span class="n">off</span><span class="p">()</span>
<span class="p">}</span>

<span class="c1">##FTD gene region (including promoter) overlaps</span>
<span class="k">for</span> <span class="p">(</span><span class="n">i</span> <span class="ow">in</span> <span class="mi">1</span><span class="p">:</span><span class="n">nrow</span><span class="p">(</span><span class="n">genes</span><span class="p">))</span> <span class="p">{</span>
  <span class="n">gene</span> <span class="o">=</span> <span class="n">genes</span><span class="err">$</span><span class="n">Gene</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
  <span class="n">start</span> <span class="o">=</span> <span class="n">genes</span><span class="err">$</span><span class="n">txStart</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">-</span><span class="mi">1000</span>
  <span class="n">stop</span> <span class="o">=</span> <span class="n">genes</span><span class="err">$</span><span class="n">txEnd</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1000</span>
  
  <span class="n">FTD_cases</span> <span class="o">&lt;-</span> <span class="n">FTD</span> <span class="o">%&gt;%</span>
    <span class="nb">filter</span><span class="p">(</span><span class="n">GENE</span> <span class="o">==</span> <span class="n">gene</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nb">filter</span><span class="p">(</span><span class="n">FTD_ALT_FREQS</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nb">filter</span><span class="p">((</span><span class="n">Candidate_gene_overlap</span> <span class="o">==</span> <span class="s2">&quot;Candidate_gene_overlap&quot;</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">Candidate_gene_promoter_overlap</span> <span class="o">==</span> <span class="s2">&quot;Promoter_overlap&quot;</span><span class="p">))</span>
  
  <span class="n">FTD_controls</span> <span class="o">&lt;-</span> <span class="n">FTD</span> <span class="o">%&gt;%</span>
    <span class="nb">filter</span><span class="p">(</span><span class="n">GENE</span> <span class="o">==</span> <span class="n">gene</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nb">filter</span><span class="p">(</span><span class="n">controls_ALT_FREQS</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nb">filter</span><span class="p">((</span><span class="n">Candidate_gene_overlap</span> <span class="o">==</span> <span class="s2">&quot;Candidate_gene_overlap&quot;</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">Candidate_gene_promoter_overlap</span> <span class="o">==</span> <span class="s2">&quot;Promoter_overlap&quot;</span><span class="p">))</span>
  
  <span class="n">FTD_cases</span><span class="err">$</span><span class="n">POS</span><span class="p">[</span><span class="n">FTD_cases</span><span class="err">$</span><span class="n">POS</span> <span class="o">&lt;</span> <span class="n">start</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">start</span> <span class="o">-</span> <span class="p">(</span><span class="mf">0.2</span><span class="o">*</span><span class="p">(</span><span class="n">stop</span><span class="o">-</span><span class="n">start</span><span class="p">))</span>
  <span class="n">FTD_cases</span><span class="err">$</span><span class="n">END</span><span class="p">[</span><span class="n">FTD_cases</span><span class="err">$</span><span class="n">END</span> <span class="o">&gt;</span> <span class="n">stop</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">stop</span> <span class="o">+</span> <span class="p">(</span><span class="mf">0.2</span><span class="o">*</span><span class="p">(</span><span class="n">stop</span><span class="o">-</span><span class="n">start</span><span class="p">))</span>
  <span class="n">FTD_controls</span><span class="err">$</span><span class="n">POS</span><span class="p">[</span><span class="n">FTD_controls</span><span class="err">$</span><span class="n">POS</span> <span class="o">&lt;</span> <span class="n">start</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">start</span> <span class="o">-</span> <span class="p">(</span><span class="mf">0.2</span><span class="o">*</span><span class="p">(</span><span class="n">stop</span><span class="o">-</span><span class="n">start</span><span class="p">))</span>
  <span class="n">FTD_controls</span><span class="err">$</span><span class="n">END</span><span class="p">[</span><span class="n">FTD_controls</span><span class="err">$</span><span class="n">END</span> <span class="o">&gt;</span> <span class="n">stop</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">stop</span> <span class="o">+</span> <span class="p">(</span><span class="mf">0.2</span><span class="o">*</span><span class="p">(</span><span class="n">stop</span><span class="o">-</span><span class="n">start</span><span class="p">))</span>
  
  <span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="s2">&quot;NIH_LBD_FTD_SV_shinyapp/NIH_SV_images/FTD/&quot;</span><span class="p">,</span><span class="n">gene</span><span class="p">,</span><span class="s2">&quot;_FTD_candidate_gene_overlap.png&quot;</span><span class="p">),</span> <span class="n">height</span> <span class="o">=</span> <span class="mi">1180</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="mi">1180</span><span class="p">,</span> <span class="n">units</span> <span class="o">=</span> <span class="s2">&quot;px&quot;</span><span class="p">,</span> <span class="n">res</span><span class="o">=</span><span class="mi">100</span><span class="p">)</span>
  
  <span class="k">if</span> <span class="p">((</span><span class="n">nrow</span><span class="p">(</span><span class="n">FTD_cases</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">nrow</span><span class="p">(</span><span class="n">FTD_controls</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">))</span> <span class="p">{</span>
    <span class="n">gtrack</span> <span class="o">&lt;-</span> <span class="n">GenomeAxisTrack</span><span class="p">()</span>
    <span class="n">itrack</span> <span class="o">&lt;-</span> <span class="n">IdeogramTrack</span><span class="p">(</span><span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">FTD_cases</span><span class="err">$</span><span class="n">CHROM</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>
    <span class="n">biomTrack</span> <span class="o">&lt;-</span> <span class="n">BiomartGeneRegionTrack</span><span class="p">(</span><span class="n">transcriptAnnotation</span> <span class="o">=</span> <span class="s2">&quot;symbol&quot;</span><span class="p">,</span> <span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span><span class="n">chromosome</span> <span class="o">=</span> <span class="n">FTD_cases</span><span class="err">$</span><span class="n">CHROM</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">start</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="n">gene</span><span class="p">,</span> <span class="nb">filter</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">transcript_is_canonical</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),</span> <span class="n">biomart</span> <span class="o">=</span> <span class="n">bm</span><span class="p">,</span> <span class="n">protein_coding</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr3</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr5</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">)</span>
    <span class="n">atrack_FTD</span> <span class="o">&lt;-</span> <span class="n">AnnotationTrack</span><span class="p">(</span><span class="n">start</span><span class="o">=</span><span class="n">FTD_cases</span><span class="err">$</span><span class="n">POS</span><span class="p">,</span> <span class="n">end</span><span class="o">=</span><span class="n">FTD_cases</span><span class="err">$</span><span class="n">END</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">FTD_cases</span><span class="err">$</span><span class="n">CHROM</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">genome</span><span class="o">=</span><span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;FTD&quot;</span><span class="p">,</span> <span class="n">feature</span> <span class="o">=</span> <span class="n">FTD_cases</span><span class="err">$</span><span class="n">ALT</span><span class="p">,</span> <span class="nb">id</span><span class="o">=</span><span class="n">FTD_cases</span><span class="err">$</span><span class="n">FTD_ALT_FREQS_perc</span><span class="p">,</span> <span class="n">showFeatureId</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">item</span><span class="o">=</span><span class="s2">&quot;black&quot;</span><span class="p">,</span> <span class="nb">min</span><span class="o">.</span><span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
    <span class="n">atrack_controls</span> <span class="o">&lt;-</span> <span class="n">AnnotationTrack</span><span class="p">(</span><span class="n">start</span><span class="o">=</span><span class="n">FTD_controls</span><span class="err">$</span><span class="n">POS</span><span class="p">,</span> <span class="n">end</span><span class="o">=</span><span class="n">FTD_controls</span><span class="err">$</span><span class="n">END</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">FTD_controls</span><span class="err">$</span><span class="n">CHROM</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">genome</span><span class="o">=</span><span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;Controls&quot;</span><span class="p">,</span> <span class="n">feature</span> <span class="o">=</span> <span class="n">FTD_controls</span><span class="err">$</span><span class="n">ALT</span><span class="p">,</span> <span class="nb">id</span><span class="o">=</span><span class="n">FTD_controls</span><span class="err">$</span><span class="n">controls_ALT_FREQS_perc</span><span class="p">,</span> <span class="n">showFeatureId</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">item</span><span class="o">=</span><span class="s2">&quot;black&quot;</span><span class="p">,</span> <span class="n">col</span><span class="o">.</span><span class="n">line</span><span class="o">=</span><span class="s2">&quot;white&quot;</span><span class="p">,</span> <span class="nb">min</span><span class="o">.</span><span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
    <span class="n">plotTracks</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span><span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack_FTD</span><span class="p">,</span> <span class="n">atrack_controls</span><span class="p">,</span> <span class="n">biomTrack</span><span class="p">),</span> <span class="kn">from</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;#808080&quot;</span><span class="p">,</span> <span class="n">extend</span><span class="o">.</span><span class="n">right</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span> <span class="n">extend</span><span class="o">.</span><span class="n">left</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">group</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">BND</span><span class="o">=</span><span class="s2">&quot;#377246&quot;</span><span class="p">,</span> <span class="n">CPX</span><span class="o">=</span><span class="s2">&quot;#86E496&quot;</span><span class="p">,</span> <span class="n">DEL</span><span class="o">=</span><span class="s2">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="n">DUP</span><span class="o">=</span><span class="s2">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="n">INS</span><span class="o">=</span><span class="s2">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="n">INS_ME</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_ALU</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_LINE1</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_SVA</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INV</span><span class="o">=</span><span class="s2">&quot;#F69A57&quot;</span><span class="p">)</span>
  <span class="p">}</span> <span class="k">else</span> <span class="k">if</span> <span class="p">((</span><span class="n">nrow</span><span class="p">(</span><span class="n">FTD_cases</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">nrow</span><span class="p">(</span><span class="n">FTD_controls</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">))</span> <span class="p">{</span>
    <span class="n">gtrack</span> <span class="o">&lt;-</span> <span class="n">GenomeAxisTrack</span><span class="p">()</span>
    <span class="n">itrack</span> <span class="o">&lt;-</span> <span class="n">IdeogramTrack</span><span class="p">(</span><span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">FTD_cases</span><span class="err">$</span><span class="n">CHROM</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>
    <span class="n">biomTrack</span> <span class="o">&lt;-</span> <span class="n">BiomartGeneRegionTrack</span><span class="p">(</span><span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span><span class="n">chromosome</span> <span class="o">=</span> <span class="n">FTD_cases</span><span class="err">$</span><span class="n">CHROM</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">start</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;ENSEMBL&quot;</span><span class="p">,</span> <span class="nb">filter</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">transcript_is_canonical</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),</span> <span class="n">biomart</span> <span class="o">=</span> <span class="n">bm</span><span class="p">,</span> <span class="n">protein_coding</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr3</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr5</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">)</span>
    <span class="n">atrack_FTD</span> <span class="o">&lt;-</span> <span class="n">AnnotationTrack</span><span class="p">(</span><span class="n">start</span><span class="o">=</span><span class="n">FTD_cases</span><span class="err">$</span><span class="n">POS</span><span class="p">,</span> <span class="n">end</span><span class="o">=</span><span class="n">FTD_cases</span><span class="err">$</span><span class="n">END</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">FTD_cases</span><span class="err">$</span><span class="n">CHROM</span><span class="p">,</span> <span class="n">genome</span><span class="o">=</span><span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;FTD&quot;</span><span class="p">,</span> <span class="n">feature</span> <span class="o">=</span> <span class="n">FTD_cases</span><span class="err">$</span><span class="n">ALT</span><span class="p">,</span> <span class="nb">id</span><span class="o">=</span><span class="n">FTD_cases</span><span class="err">$</span><span class="n">FTD_ALT_FREQS_perc</span><span class="p">,</span> <span class="n">showFeatureId</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">item</span><span class="o">=</span><span class="s2">&quot;black&quot;</span><span class="p">,</span> <span class="nb">min</span><span class="o">.</span><span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
    <span class="n">plotTracks</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span><span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack_FTD</span><span class="p">,</span> <span class="n">biomTrack</span><span class="p">),</span> <span class="kn">from</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;#808080&quot;</span><span class="p">,</span> <span class="n">extend</span><span class="o">.</span><span class="n">right</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span> <span class="n">extend</span><span class="o">.</span><span class="n">left</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">group</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">BND</span><span class="o">=</span><span class="s2">&quot;#377246&quot;</span><span class="p">,</span> <span class="n">CPX</span><span class="o">=</span><span class="s2">&quot;#86E496&quot;</span><span class="p">,</span> <span class="n">DEL</span><span class="o">=</span><span class="s2">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="n">DUP</span><span class="o">=</span><span class="s2">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="n">INS</span><span class="o">=</span><span class="s2">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="n">INS_ME</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_ALU</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_LINE1</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_SVA</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INV</span><span class="o">=</span><span class="s2">&quot;#F69A57&quot;</span><span class="p">)</span>
  <span class="p">}</span> <span class="k">else</span> <span class="k">if</span> <span class="p">((</span><span class="n">nrow</span><span class="p">(</span><span class="n">FTD_cases</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">nrow</span><span class="p">(</span><span class="n">FTD_controls</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">))</span> <span class="p">{</span>
    <span class="n">gtrack</span> <span class="o">&lt;-</span> <span class="n">GenomeAxisTrack</span><span class="p">(</span><span class="n">showTitle</span><span class="o">=</span><span class="n">TRUE</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;No SVs&quot;</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;#808080&quot;</span><span class="p">,</span> <span class="n">rotation</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
    <span class="n">plotTracks</span><span class="p">(</span><span class="n">gtrack</span><span class="p">,</span> <span class="kn">from</span> <span class="o">=</span> <span class="mi">100</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="mi">110</span><span class="p">,</span> <span class="n">title</span><span class="o">.</span><span class="n">width</span> <span class="o">=</span> <span class="mi">5</span><span class="p">)</span>
  <span class="p">}</span> <span class="k">else</span> <span class="p">{</span>
    <span class="n">gtrack</span> <span class="o">&lt;-</span> <span class="n">GenomeAxisTrack</span><span class="p">()</span>
    <span class="n">itrack</span> <span class="o">&lt;-</span> <span class="n">IdeogramTrack</span><span class="p">(</span><span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">FTD_controls</span><span class="err">$</span><span class="n">CHROM</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>
    <span class="n">biomTrack</span> <span class="o">&lt;-</span> <span class="n">BiomartGeneRegionTrack</span><span class="p">(</span><span class="n">transcriptAnnotation</span> <span class="o">=</span> <span class="s2">&quot;symbol&quot;</span><span class="p">,</span> <span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span><span class="n">chromosome</span> <span class="o">=</span> <span class="n">FTD_controls</span><span class="err">$</span><span class="n">CHROM</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">start</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="n">gene</span><span class="p">,</span> <span class="nb">filter</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">transcript_is_canonical</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),</span> <span class="n">biomart</span> <span class="o">=</span> <span class="n">bm</span><span class="p">,</span> <span class="n">protein_coding</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr3</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr5</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">)</span>
    <span class="n">atrack_controls</span> <span class="o">&lt;-</span> <span class="n">AnnotationTrack</span><span class="p">(</span><span class="n">start</span><span class="o">=</span><span class="n">FTD_controls</span><span class="err">$</span><span class="n">POS</span><span class="p">,</span> <span class="n">end</span><span class="o">=</span><span class="n">FTD_controls</span><span class="err">$</span><span class="n">END</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">FTD_controls</span><span class="err">$</span><span class="n">CHROM</span><span class="p">,</span> <span class="n">genome</span><span class="o">=</span><span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;Controls&quot;</span><span class="p">,</span> <span class="n">feature</span> <span class="o">=</span> <span class="n">FTD_controls</span><span class="err">$</span><span class="n">ALT</span><span class="p">,</span> <span class="nb">id</span><span class="o">=</span><span class="n">FTD_controls</span><span class="err">$</span><span class="n">controls_ALT_FREQS_perc</span><span class="p">,</span> <span class="n">showFeatureId</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">item</span><span class="o">=</span><span class="s2">&quot;black&quot;</span><span class="p">,</span> <span class="nb">min</span><span class="o">.</span><span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
    <span class="n">plotTracks</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span><span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack_controls</span><span class="p">,</span> <span class="n">biomTrack</span><span class="p">),</span> <span class="kn">from</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;#808080&quot;</span><span class="p">,</span> <span class="n">extend</span><span class="o">.</span><span class="n">right</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span> <span class="n">extend</span><span class="o">.</span><span class="n">left</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">group</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">BND</span><span class="o">=</span><span class="s2">&quot;#377246&quot;</span><span class="p">,</span> <span class="n">CPX</span><span class="o">=</span><span class="s2">&quot;#86E496&quot;</span><span class="p">,</span> <span class="n">DEL</span><span class="o">=</span><span class="s2">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="n">DUP</span><span class="o">=</span><span class="s2">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="n">INS</span><span class="o">=</span><span class="s2">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="n">INS_ME</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_ALU</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_LINE1</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_SVA</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INV</span><span class="o">=</span><span class="s2">&quot;#F69A57&quot;</span><span class="p">)</span>
  <span class="p">}</span>
  <span class="n">dev</span><span class="o">.</span><span class="n">off</span><span class="p">()</span>
<span class="p">}</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>R version 4.1.3 (2022-03-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7</p>
<p>Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</p>
<p>locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</p>
<p>attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base</p>
<p>other attached packages:
 [1] biomaRt_2.50.3       AnnotationHub_3.2.2  BiocFileCache_2.2.1  dbplyr_2.2.1         forcats_0.5.2        stringr_1.4.1<br>
 [7] dplyr_1.0.10         purrr_0.3.5          readr_2.1.3          tidyr_1.2.1          tibble_3.1.8         ggplot2_3.4.0<br>
[13] tidyverse_1.3.2      Gviz_1.38.4          GenomicRanges_1.46.1 GenomeInfoDb_1.30.1  IRanges_2.32.0       S4Vectors_0.36.0<br>
[19] BiocGenerics_0.40.0</p>
<p>loaded via a namespace (and not attached):
  [1] readxl_1.4.1                  backports_1.4.1               Hmisc_4.7-1                   lazyeval_0.2.2<br>
  [5] splines_4.1.3                 BiocParallel_1.28.3           digest_0.6.30                 ensembldb_2.18.4<br>
  [9] htmltools_0.5.3               fansi_1.0.3                   magrittr_2.0.3                checkmate_2.1.0<br>
 [13] memoise_2.0.1                 BSgenome_1.62.0               googlesheets4_1.0.1           cluster_2.1.2<br>
 [17] tzdb_0.3.0                    Biostrings_2.62.0             modelr_0.1.10                 matrixStats_0.62.0<br>
 [21] timechange_0.1.1              prettyunits_1.1.1             jpeg_0.1-9                    colorspace_2.0-3<br>
 [25] blob_1.2.3                    rvest_1.0.3                   rappdirs_0.3.3                haven_2.5.1<br>
 [29] xfun_0.34                     crayon_1.5.2                  RCurl_1.98-1.9                jsonlite_1.8.3<br>
 [33] survival_3.2-13               VariantAnnotation_1.40.0      glue_1.6.2                    gtable_0.3.1<br>
 [37] gargle_1.2.1                  zlibbioc_1.40.0               XVector_0.34.0                DelayedArray_0.20.0<br>
 [41] scales_1.2.1                  DBI_1.1.3                     Rcpp_1.0.9                    xtable_1.8-4<br>
 [45] progress_1.2.2                htmlTable_2.4.1               foreign_0.8-82                bit_4.0.4<br>
 [49] Formula_1.2-4                 DT_0.26                       htmlwidgets_1.5.4             httr_1.4.4<br>
 [53] RColorBrewer_1.1-3            ellipsis_0.3.2                pkgconfig_2.0.3               XML_3.99-0.12<br>
 [57] nnet_7.3-17                   deldir_1.0-6                  utf8_1.2.2                    tidyselect_1.2.0<br>
 [61] rlang_1.0.6                   later_1.3.0                   AnnotationDbi_1.56.2          munsell_0.5.0<br>
 [65] BiocVersion_3.14.0            cellranger_1.1.0              tools_4.1.3                   cachem_1.0.6<br>
 [69] cli_3.4.1                     generics_0.1.3                RSQLite_2.2.18                broom_1.0.1<br>
 [73] fastmap_1.1.0                 yaml_2.3.6                    knitr_1.40                    bit64_4.0.5<br>
 [77] fs_1.5.2                      KEGGREST_1.34.0               AnnotationFilter_1.18.0       mime_0.12<br>
 [81] xml2_1.3.3                    compiler_4.1.3                rstudioapi_0.14               filelock_1.0.2<br>
 [85] curl_4.3.3                    png_0.1-7                     interactiveDisplayBase_1.32.0 reprex_2.0.2<br>
 [89] stringi_1.7.8                 GenomicFeatures_1.46.5        lattice_0.20-45               ProtGenerics_1.26.0<br>
 [93] Matrix_1.4-0                  vctrs_0.5.0                   pillar_1.8.1                  lifecycle_1.0.3<br>
 [97] BiocManager_1.30.19           data.table_1.14.4             bitops_1.0-7                  httpuv_1.6.6<br>
[101] rtracklayer_1.54.0            R6_2.5.1                      BiocIO_1.4.0                  latticeExtra_0.6-30<br>
[105] promises_1.2.0.1              gridExtra_2.3                 dichromat_2.0-0.1             assertthat_0.2.1<br>
[109] SummarizedExperiment_1.24.0   rjson_0.2.21                  withr_2.5.0                   GenomicAlignments_1.30.0<br>
[113] Rsamtools_2.10.0              GenomeInfoDbData_1.2.7        parallel_4.1.3                hms_1.1.2<br>
[117] rpart_4.1.16                  MatrixGenerics_1.6.0          googledrive_2.0.0             biovizBase_1.42.0<br>
[121] Biobase_2.54.0                shiny_1.7.3                   lubridate_1.9.0               base64enc_0.1-3</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Data-preparation">Data preparation<a class="anchor-link" href="#Data-preparation">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">#Data preparation for shiny app (run locally)</span>

<span class="c1">#Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="s">&quot;Gviz&quot;</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">AnnotationHub</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="s">&quot;tidyverse&quot;</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">biomaRt</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>

<span class="c1">#Import files</span>
<span class="n">LBD</span> <span class="o">&lt;-</span> <span class="nf">read.delim</span><span class="p">(</span><span class="s">&quot;LBD_high_quality_subset_clean_GQ300missing_all_chr.txt&quot;</span><span class="p">)</span>
<span class="n">FTD</span> <span class="o">&lt;-</span> <span class="nf">read.delim</span><span class="p">(</span><span class="s">&quot;FTD_high_quality_subset_clean_GQ300missing_all_chr.txt&quot;</span><span class="p">)</span>

<span class="c1">#Split into case and control dfs</span>
<span class="n">LBD_cases</span> <span class="o">&lt;-</span> <span class="n">LBD</span> <span class="o">%&gt;%</span>
  <span class="nf">filter</span><span class="p">(</span><span class="n">AF_cases</span> <span class="o">&gt;</span> <span class="m">0</span><span class="p">)</span>
<span class="n">LBD_controls</span> <span class="o">&lt;-</span> <span class="n">LBD</span> <span class="o">%&gt;%</span>
  <span class="nf">filter</span><span class="p">(</span><span class="n">AF_controls</span> <span class="o">&gt;</span> <span class="m">0</span><span class="p">)</span>

<span class="n">FTD_cases</span> <span class="o">&lt;-</span> <span class="n">FTD</span> <span class="o">%&gt;%</span>
  <span class="nf">filter</span><span class="p">(</span><span class="n">AF_cases</span> <span class="o">&gt;</span> <span class="m">0</span><span class="p">)</span>
<span class="n">FTD_controls</span> <span class="o">&lt;-</span> <span class="n">FTD</span> <span class="o">%&gt;%</span>
  <span class="nf">filter</span><span class="p">(</span><span class="n">AF_controls</span> <span class="o">&gt;</span> <span class="m">0</span><span class="p">)</span>

<span class="c1">#Change AFs to percentages, two decimal places</span>
<span class="n">LBD_cases</span><span class="o">$</span><span class="n">AF_cases</span> <span class="o">&lt;-</span> <span class="nf">paste0</span><span class="p">(</span><span class="nf">round</span><span class="p">(</span><span class="n">LBD_cases</span><span class="o">$</span><span class="n">AF_cases</span><span class="o">*</span><span class="m">100</span><span class="p">,</span><span class="n">digits</span><span class="o">=</span><span class="m">2</span><span class="p">),</span> <span class="s">&quot;%&quot;</span><span class="p">)</span>
<span class="n">LBD_controls</span><span class="o">$</span><span class="n">AF_controls</span> <span class="o">&lt;-</span> <span class="nf">paste0</span><span class="p">(</span><span class="nf">round</span><span class="p">(</span><span class="n">LBD_controls</span><span class="o">$</span><span class="n">AF_controls</span><span class="o">*</span><span class="m">100</span><span class="p">,</span><span class="n">digits</span><span class="o">=</span><span class="m">2</span><span class="p">),</span> <span class="s">&quot;%&quot;</span><span class="p">)</span>

<span class="n">FTD_cases</span><span class="o">$</span><span class="n">AF_cases</span> <span class="o">&lt;-</span> <span class="nf">paste0</span><span class="p">(</span><span class="nf">round</span><span class="p">(</span><span class="n">FTD_cases</span><span class="o">$</span><span class="n">AF_cases</span><span class="o">*</span><span class="m">100</span><span class="p">,</span><span class="n">digits</span><span class="o">=</span><span class="m">2</span><span class="p">),</span> <span class="s">&quot;%&quot;</span><span class="p">)</span>
<span class="n">FTD_controls</span><span class="o">$</span><span class="n">AF_controls</span> <span class="o">&lt;-</span> <span class="nf">paste0</span><span class="p">(</span><span class="nf">round</span><span class="p">(</span><span class="n">FTD_controls</span><span class="o">$</span><span class="n">AF_controls</span><span class="o">*</span><span class="m">100</span><span class="p">,</span><span class="n">digits</span><span class="o">=</span><span class="m">2</span><span class="p">),</span> <span class="s">&quot;%&quot;</span><span class="p">)</span>

<span class="c1">#Save dfs</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">LBD_cases</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="s">&quot;LBD_cases_shinyapp.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">quote</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">LBD_controls</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="s">&quot;LBD_controls_shinyapp.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">quote</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>

<span class="nf">write.table</span><span class="p">(</span><span class="n">FTD_cases</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="s">&quot;FTD_cases_shinyapp.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">quote</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">FTD_controls</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="s">&quot;FTD_controls_shinyapp.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">quote</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>

<span class="c1">#Import candidate gene files</span>
<span class="n">LBD</span> <span class="o">&lt;-</span> <span class="nf">read.delim</span><span class="p">(</span><span class="s">&quot;LBD_1percFDR_all_info_unfiltered_with_CADDSV_genehancer.txt&quot;</span><span class="p">,</span> <span class="n">stringsAsFactors</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="n">FTD</span> <span class="o">&lt;-</span> <span class="nf">read.delim</span><span class="p">(</span><span class="s">&quot;FTD_1percFDR_all_info_unfiltered_with_CADDSV_genehancer.txt&quot;</span><span class="p">,</span> <span class="n">stringsAsFactors</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="c1">#Remove SV carrier ID column</span>
<span class="n">LBD</span> <span class="o">&lt;-</span> <span class="n">LBD</span><span class="p">[,</span><span class="m">-23</span><span class="p">]</span>
<span class="n">FTD</span> <span class="o">&lt;-</span> <span class="n">FTD</span><span class="p">[,</span><span class="m">-23</span><span class="p">]</span>
<span class="c1">#Save dfs</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">LBD</span><span class="p">,</span><span class="n">file</span><span class="o">=</span><span class="s">&quot;LBD_1percFDR_all_info_unfiltered_with_CADDSV_genehancer_shinyapp.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">quote</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">FTD</span><span class="p">,</span><span class="n">file</span><span class="o">=</span><span class="s">&quot;FTD_1percFDR_all_info_unfiltered_with_CADDSV_genehancer_shinyapp.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">quote</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Shiny-app-code">Shiny app code<a class="anchor-link" href="#Shiny-app-code">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">####Shiny app to visualize SVs</span>

<span class="c1">#Set working directory where files are/put files to your working directory</span>
<span class="c1">#Load libraries</span>
<span class="n">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">Gviz</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">shiny</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">shinyWidgets</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">datamods</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">biomaRt</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">shinythemes</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">shinysky</span><span class="p">)</span>


<span class="c1">#Import files</span>
<span class="c1">##50 NDD genes</span>
<span class="n">LBD</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">delim</span><span class="p">(</span><span class="s2">&quot;LBD_1percFDR_all_info_unfiltered_with_CADDSV_genehancer_shinyapp.txt&quot;</span><span class="p">,</span> <span class="n">stringsAsFactors</span> <span class="o">=</span> <span class="n">FALSE</span><span class="p">)</span>
<span class="n">FTD</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">delim</span><span class="p">(</span><span class="s2">&quot;FTD_1percFDR_all_info_unfiltered_with_CADDSV_genehancer_shinyapp.txt&quot;</span><span class="p">,</span> <span class="n">stringsAsFactors</span> <span class="o">=</span> <span class="n">FALSE</span><span class="p">)</span>

<span class="c1">##Make chr ordered factor, the sort dfs by factor order</span>
<span class="n">LBD</span><span class="err">$</span><span class="n">CHROM</span> <span class="o">&lt;-</span> <span class="n">factor</span><span class="p">(</span><span class="n">LBD</span><span class="err">$</span><span class="n">CHROM</span><span class="p">,</span> <span class="n">ordered</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="n">seq</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">22</span><span class="p">),</span><span class="s2">&quot;X&quot;</span><span class="p">))</span>
<span class="n">FTD</span><span class="err">$</span><span class="n">CHROM</span> <span class="o">&lt;-</span> <span class="n">factor</span><span class="p">(</span><span class="n">FTD</span><span class="err">$</span><span class="n">CHROM</span><span class="p">,</span> <span class="n">ordered</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="n">seq</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">22</span><span class="p">),</span><span class="s2">&quot;X&quot;</span><span class="p">))</span>

<span class="n">LBD</span> <span class="o">&lt;-</span> <span class="n">arrange</span><span class="p">(</span><span class="n">LBD</span><span class="p">,</span> <span class="n">CHROM</span><span class="p">)</span>
<span class="n">FTD</span> <span class="o">&lt;-</span> <span class="n">arrange</span><span class="p">(</span><span class="n">FTD</span><span class="p">,</span> <span class="n">CHROM</span><span class="p">)</span>
<span class="c1">##High-quality SVs whole genome</span>
<span class="n">LBD_gw</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">delim</span><span class="p">(</span><span class="s2">&quot;LBD_high_quality_subset_clean_GQ300missing_all_chr.txt&quot;</span><span class="p">)</span>
<span class="n">FTD_gw</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">delim</span><span class="p">(</span><span class="s2">&quot;FTD_high_quality_subset_clean_GQ300missing_all_chr.txt&quot;</span><span class="p">)</span>

<span class="n">LBD_cases</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">csv</span><span class="p">(</span><span class="s2">&quot;LBD_cases_shinyapp.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">)</span>
<span class="n">LBD_controls</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">csv</span><span class="p">(</span><span class="s2">&quot;LBD_controls_shinyapp.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">)</span>
<span class="n">FTD_cases</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">csv</span><span class="p">(</span><span class="s2">&quot;FTD_cases_shinyapp.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">)</span>
<span class="n">FTD_controls</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">csv</span><span class="p">(</span><span class="s2">&quot;FTD_controls_shinyapp.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">)</span>
<span class="c1">##Make chrom ordered facror and sort by it so chr are ordered in shiny app</span>
<span class="n">LBD_gw</span><span class="err">$</span><span class="n">CHROM</span> <span class="o">&lt;-</span> <span class="n">factor</span><span class="p">(</span><span class="n">LBD_gw</span><span class="err">$</span><span class="n">CHROM</span><span class="p">,</span> <span class="n">ordered</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">rep</span><span class="p">(</span><span class="s2">&quot;chr&quot;</span><span class="p">,</span><span class="mi">22</span><span class="p">),</span><span class="n">seq</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">22</span><span class="p">)),</span><span class="s2">&quot;chrX&quot;</span><span class="p">))</span>
<span class="n">FTD_gw</span><span class="err">$</span><span class="n">CHROM</span> <span class="o">&lt;-</span> <span class="n">factor</span><span class="p">(</span><span class="n">FTD_gw</span><span class="err">$</span><span class="n">CHROM</span><span class="p">,</span> <span class="n">ordered</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">rep</span><span class="p">(</span><span class="s2">&quot;chr&quot;</span><span class="p">,</span><span class="mi">22</span><span class="p">),</span><span class="n">seq</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">22</span><span class="p">)),</span><span class="s2">&quot;chrX&quot;</span><span class="p">))</span>

<span class="n">LBD_gw</span> <span class="o">&lt;-</span> <span class="n">arrange</span><span class="p">(</span><span class="n">LBD_gw</span><span class="p">,</span> <span class="n">CHROM</span><span class="p">)</span>
<span class="n">FTD_gw</span> <span class="o">&lt;-</span> <span class="n">arrange</span><span class="p">(</span><span class="n">FTD_gw</span><span class="p">,</span> <span class="n">CHROM</span><span class="p">)</span>


<span class="c1">##Gene info file</span>
<span class="n">gene_info</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">csv</span><span class="p">(</span><span class="s2">&quot;gene_start_stop_name.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">)</span>

<span class="c1">#Initialize biomart track</span>
<span class="n">bm</span> <span class="o">&lt;-</span> <span class="n">useEnsembl</span><span class="p">(</span><span class="n">biomart</span> <span class="o">=</span> <span class="s2">&quot;genes&quot;</span><span class="p">)</span>
<span class="n">bm</span> <span class="o">&lt;-</span> <span class="n">useDataset</span><span class="p">(</span><span class="n">dataset</span> <span class="o">=</span> <span class="s2">&quot;hsapiens_gene_ensembl&quot;</span><span class="p">,</span> <span class="n">mart</span> <span class="o">=</span> <span class="n">bm</span><span class="p">)</span>

<span class="c1">#Define user interface</span>
<span class="n">ui</span> <span class="o">&lt;-</span> <span class="n">navbarPage</span><span class="p">(</span><span class="s2">&quot;LBD and FTD/ALS structural variants&quot;</span><span class="p">,</span> <span class="n">theme</span> <span class="o">=</span> <span class="n">shinytheme</span><span class="p">(</span><span class="s2">&quot;lumen&quot;</span><span class="p">),</span>
                 <span class="n">tabPanel</span><span class="p">(</span><span class="s2">&quot;Neurodegenerative gene region visualisation&quot;</span><span class="p">,</span> <span class="n">icon</span> <span class="o">=</span> <span class="n">icon</span><span class="p">(</span><span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;image&quot;</span><span class="p">,</span> <span class="n">lib</span> <span class="o">=</span> <span class="s2">&quot;font-awesome&quot;</span><span class="p">),</span>
                          <span class="n">sidebarLayout</span><span class="p">(</span>
                            <span class="n">sidebarPanel</span><span class="p">(</span>
                              <span class="n">width</span> <span class="o">=</span> <span class="mi">3</span><span class="p">,</span>
                              <span class="n">radioButtons</span><span class="p">(</span><span class="n">inputId</span> <span class="o">=</span> <span class="s2">&quot;pheno2&quot;</span><span class="p">,</span> <span class="n">label</span> <span class="o">=</span> <span class="s2">&quot;Phenotype&quot;</span><span class="p">,</span> <span class="n">choiceValues</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="s2">&quot;LBD&quot;</span><span class="p">,</span> <span class="s2">&quot;FTD&quot;</span><span class="p">),</span> <span class="n">choiceNames</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="s2">&quot;LBD&quot;</span><span class="p">,</span> <span class="s2">&quot;FTD/ALS&quot;</span><span class="p">),</span> <span class="n">inline</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),</span>
                              <span class="n">pickerInput</span><span class="p">(</span><span class="n">inputId</span> <span class="o">=</span> <span class="s2">&quot;gene2&quot;</span><span class="p">,</span> <span class="n">label</span> <span class="o">=</span> <span class="s2">&quot;Gene&quot;</span><span class="p">,</span> <span class="n">choices</span> <span class="o">=</span> <span class="n">unique</span><span class="p">(</span><span class="n">sort</span><span class="p">(</span><span class="n">LBD</span><span class="err">$</span><span class="n">GENE</span><span class="p">)),</span> <span class="n">multiple</span> <span class="o">=</span> <span class="n">FALSE</span><span class="p">),</span>
                              <span class="n">textOutput</span><span class="p">(</span><span class="n">outputId</span> <span class="o">=</span> <span class="s2">&quot;SV_info2&quot;</span><span class="p">),</span>
                              <span class="n">imageOutput</span><span class="p">(</span><span class="n">outputId</span> <span class="o">=</span> <span class="s2">&quot;color_legend&quot;</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="s2">&quot;100%&quot;</span><span class="p">)</span>
                            <span class="p">),</span>
                            <span class="n">mainPanel</span><span class="p">(</span>
                              <span class="n">imageOutput</span><span class="p">(</span><span class="n">outputId</span> <span class="o">=</span>  <span class="s2">&quot;img2&quot;</span><span class="p">)</span>
                            <span class="p">)</span>
                          <span class="p">)</span>
                 <span class="p">),</span>
                 <span class="n">tabPanel</span><span class="p">(</span><span class="s2">&quot;Neurodegenerative gene data frame&quot;</span><span class="p">,</span> <span class="n">icon</span> <span class="o">=</span> <span class="n">icon</span><span class="p">(</span><span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;table&quot;</span><span class="p">,</span> <span class="n">lib</span> <span class="o">=</span> <span class="s2">&quot;font-awesome&quot;</span><span class="p">),</span>
                          <span class="n">sidebarLayout</span><span class="p">(</span>
                            <span class="n">sidebarPanel</span><span class="p">(</span>
                              <span class="n">width</span> <span class="o">=</span> <span class="mi">3</span><span class="p">,</span>
                              <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">a</span><span class="p">(</span><span class="n">href</span><span class="o">=</span><span class="s2">&quot;https://github.com/kkaivola/LBD_FTD_Structural_variants&quot;</span><span class="p">,</span> 
                                     <span class="s2">&quot;Download the data frames from here&quot;</span><span class="p">),</span>
                              <span class="n">radioButtons</span><span class="p">(</span><span class="n">inputId</span> <span class="o">=</span> <span class="s2">&quot;dataset&quot;</span><span class="p">,</span> <span class="n">label</span> <span class="o">=</span> <span class="s2">&quot;Phenotype&quot;</span><span class="p">,</span> <span class="n">choiceValues</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="s2">&quot;LBD&quot;</span><span class="p">,</span> <span class="s2">&quot;FTD&quot;</span><span class="p">),</span> <span class="n">choiceNames</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="s2">&quot;LBD&quot;</span><span class="p">,</span> <span class="s2">&quot;FTD/ALS&quot;</span><span class="p">)),</span>
                              <span class="n">filter_data_ui</span><span class="p">(</span><span class="s2">&quot;filtering&quot;</span><span class="p">),</span>
                            <span class="p">),</span>
                            <span class="n">mainPanel</span><span class="p">(</span>
                              <span class="n">DT</span><span class="p">::</span><span class="n">dataTableOutput</span><span class="p">(</span><span class="n">outputId</span> <span class="o">=</span> <span class="s2">&quot;table&quot;</span><span class="p">)</span>
                            <span class="p">)</span>
                          <span class="p">)</span>
                 <span class="p">),</span>
                 <span class="n">tabPanel</span><span class="p">(</span><span class="s2">&quot;Genome-wide visualisation&quot;</span><span class="p">,</span> <span class="n">icon</span> <span class="o">=</span> <span class="n">icon</span><span class="p">(</span><span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;images&quot;</span><span class="p">,</span> <span class="n">lib</span> <span class="o">=</span> <span class="s2">&quot;font-awesome&quot;</span><span class="p">),</span>
                          <span class="n">sidebarLayout</span><span class="p">(</span>
                            <span class="n">sidebarPanel</span><span class="p">(</span>
                              <span class="n">width</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span>
                              <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">a</span><span class="p">(</span><span class="n">href</span><span class="o">=</span><span class="s2">&quot;https://github.com/kkaivola/LBD_FTD_Structural_variants&quot;</span><span class="p">,</span> 
                                     <span class="s2">&quot;Want to visualize structural variants with regulatory regions, gnomAD variants and more? Get BigBed files for Ensembl/UCSC genome browser from here!&quot;</span><span class="p">),</span>
                              <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot; &quot;</span><span class="p">),</span>
                              <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;Number displayed on structural variants is the allele frequency&quot;</span><span class="p">),</span>
                              <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;Click plot to visualize changes after updating genomic coordinates, using zoom, gene search or changing phenotype&quot;</span><span class="p">),</span>
                              <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">br</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s2">&quot;line-height: 10px&quot;</span><span class="p">),</span>
                              <span class="n">radioButtons</span><span class="p">(</span><span class="n">inputId</span> <span class="o">=</span> <span class="s2">&quot;pheno&quot;</span><span class="p">,</span> <span class="n">label</span> <span class="o">=</span> <span class="s2">&quot;Phenotype&quot;</span><span class="p">,</span> <span class="n">choiceValues</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="s2">&quot;LBD&quot;</span><span class="p">,</span> <span class="s2">&quot;FTD&quot;</span><span class="p">),</span> <span class="n">choiceNames</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="s2">&quot;LBD&quot;</span><span class="p">,</span> <span class="s2">&quot;FTD/ALS&quot;</span><span class="p">),</span> <span class="n">inline</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),</span>
                              <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;Gene search&quot;</span><span class="p">,</span> <span class="n">style</span> <span class="o">=</span> <span class="s2">&quot;font-size:14px;&quot;</span><span class="p">),</span>
                              <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot; &quot;</span><span class="p">),</span>
                              <span class="n">textInput</span><span class="o">.</span><span class="n">typeahead</span><span class="p">(</span><span class="nb">id</span><span class="o">=</span><span class="s2">&quot;gene_search&quot;</span><span class="p">,</span> <span class="n">placeholder</span> <span class="o">=</span> <span class="s2">&quot;Gene name, e.g. GBA&quot;</span><span class="p">,</span> <span class="n">local</span><span class="o">=</span><span class="n">data</span><span class="o">.</span><span class="n">frame</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="n">c</span><span class="p">(</span><span class="n">gene_info</span><span class="err">$</span><span class="n">external_gene_name</span><span class="p">)),</span> <span class="n">valueKey</span> <span class="o">=</span> <span class="s2">&quot;name&quot;</span><span class="p">,</span> <span class="n">tokens</span> <span class="o">=</span>  <span class="n">c</span><span class="p">(</span><span class="mi">1</span><span class="p">:</span><span class="n">length</span><span class="p">(</span><span class="n">gene_info</span><span class="err">$</span><span class="n">external_gene_name</span><span class="p">)),</span> <span class="n">template</span> <span class="o">=</span> <span class="n">HTML</span><span class="p">(</span><span class="s2">&quot;&lt;p style=&#39;font-size:12px;&#39;class=&#39;repo-name&#39;&gt;{{name}}&lt;/p&gt;&quot;</span><span class="p">)),</span>
                              <span class="n">pickerInput</span><span class="p">(</span><span class="n">inputId</span> <span class="o">=</span> <span class="s2">&quot;chrom&quot;</span><span class="p">,</span> <span class="n">label</span> <span class="o">=</span> <span class="s2">&quot;Chromosome&quot;</span><span class="p">,</span> <span class="n">choices</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="s2">&quot;chr1&quot;</span><span class="p">,</span><span class="s2">&quot;chr2&quot;</span><span class="p">,</span><span class="s2">&quot;chr3&quot;</span><span class="p">,</span><span class="s2">&quot;chr4&quot;</span><span class="p">,</span><span class="s2">&quot;chr5&quot;</span><span class="p">,</span><span class="s2">&quot;chr6&quot;</span><span class="p">,</span><span class="s2">&quot;chr7&quot;</span><span class="p">,</span><span class="s2">&quot;chr8&quot;</span><span class="p">,</span><span class="s2">&quot;chr9&quot;</span><span class="p">,</span><span class="s2">&quot;chr10&quot;</span><span class="p">,</span><span class="s2">&quot;chr11&quot;</span><span class="p">,</span><span class="s2">&quot;chr12&quot;</span><span class="p">,</span><span class="s2">&quot;chr13&quot;</span><span class="p">,</span><span class="s2">&quot;chr14&quot;</span><span class="p">,</span><span class="s2">&quot;chr15&quot;</span><span class="p">,</span><span class="s2">&quot;chr16&quot;</span><span class="p">,</span><span class="s2">&quot;chr17&quot;</span><span class="p">,</span><span class="s2">&quot;chr18&quot;</span><span class="p">,</span><span class="s2">&quot;chr19&quot;</span><span class="p">,</span><span class="s2">&quot;chr20&quot;</span><span class="p">,</span><span class="s2">&quot;chr21&quot;</span><span class="p">,</span><span class="s2">&quot;chr22&quot;</span><span class="p">,</span><span class="s2">&quot;chrX&quot;</span><span class="p">),</span> <span class="n">multiple</span> <span class="o">=</span> <span class="n">FALSE</span><span class="p">),</span>
                              <span class="n">numericRangeInput</span><span class="p">(</span><span class="n">inputId</span> <span class="o">=</span> <span class="s2">&quot;genomic_range&quot;</span><span class="p">,</span> <span class="n">label</span> <span class="o">=</span> <span class="s2">&quot;Genomic range in bp (hg38)&quot;</span><span class="p">,</span> <span class="n">value</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="mi">155234452</span><span class="p">,</span><span class="mi">155244699</span><span class="p">)),</span>
                              <span class="n">shiny</span><span class="p">::</span><span class="n">actionButton</span><span class="p">(</span><span class="n">inputId</span> <span class="o">=</span> <span class="s2">&quot;zoom_out_3&quot;</span><span class="p">,</span> <span class="n">label</span> <span class="o">=</span> <span class="s2">&quot;Zoom out x3&quot;</span><span class="p">),</span>
                              <span class="n">shiny</span><span class="p">::</span><span class="n">actionButton</span><span class="p">(</span><span class="n">inputId</span> <span class="o">=</span> <span class="s2">&quot;zoom_in_3&quot;</span><span class="p">,</span> <span class="n">label</span> <span class="o">=</span> <span class="s2">&quot;Zoom in x3&quot;</span><span class="p">),</span>
                              <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">br</span><span class="p">(),</span>
                              <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot; &quot;</span><span class="p">),</span>
                              <span class="n">shiny</span><span class="p">::</span><span class="n">actionButton</span><span class="p">(</span><span class="n">inputId</span> <span class="o">=</span> <span class="s2">&quot;plot&quot;</span><span class="p">,</span> <span class="n">label</span> <span class="o">=</span> <span class="s2">&quot;Plot&quot;</span><span class="p">,</span> <span class="n">icon</span> <span class="o">=</span> <span class="n">icon</span><span class="p">(</span><span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;play&quot;</span><span class="p">,</span> <span class="n">lib</span> <span class="o">=</span> <span class="s2">&quot;font-awesome&quot;</span><span class="p">),</span> <span class="n">style</span><span class="o">=</span><span class="s1">&#39;font-size:20px&#39;</span><span class="p">),</span>
                              <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">br</span><span class="p">(),</span>
                              <span class="n">imageOutput</span><span class="p">(</span><span class="n">outputId</span> <span class="o">=</span> <span class="s2">&quot;color_legend2&quot;</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="s2">&quot;100%&quot;</span><span class="p">)</span>
                            <span class="p">),</span>
                            <span class="n">mainPanel</span><span class="p">(</span>
                              <span class="n">imageOutput</span><span class="p">(</span><span class="n">outputId</span> <span class="o">=</span> <span class="s2">&quot;img3&quot;</span><span class="p">)</span>
                            <span class="p">)</span>
                          <span class="p">)</span>
                          
                 <span class="p">),</span>
                 <span class="n">tabPanel</span><span class="p">(</span><span class="s2">&quot;Genome-wide data frame&quot;</span><span class="p">,</span> <span class="n">icon</span> <span class="o">=</span> <span class="n">icon</span><span class="p">(</span><span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;table-list&quot;</span><span class="p">,</span> <span class="n">lib</span> <span class="o">=</span> <span class="s2">&quot;font-awesome&quot;</span><span class="p">),</span>
                          <span class="n">sidebarLayout</span><span class="p">(</span>
                            <span class="n">sidebarPanel</span><span class="p">(</span>
                              <span class="n">width</span> <span class="o">=</span> <span class="mi">3</span><span class="p">,</span>
                              <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">a</span><span class="p">(</span><span class="n">href</span><span class="o">=</span><span class="s2">&quot;https://github.com/kkaivola/LBD_FTD_Structural_variants&quot;</span><span class="p">,</span> 
                                     <span class="s2">&quot;Download the data frames from here&quot;</span><span class="p">),</span>
                              <span class="n">radioButtons</span><span class="p">(</span><span class="n">inputId</span> <span class="o">=</span> <span class="s2">&quot;pheno_gw&quot;</span><span class="p">,</span> <span class="n">label</span> <span class="o">=</span> <span class="s2">&quot;Phenotype&quot;</span><span class="p">,</span> <span class="n">choiceValues</span> <span class="o">=</span>  <span class="n">c</span><span class="p">(</span><span class="s2">&quot;LBD_gw&quot;</span><span class="p">,</span> <span class="s2">&quot;FTD_gw&quot;</span><span class="p">),</span> <span class="n">choiceNames</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="s2">&quot;LBD&quot;</span><span class="p">,</span> <span class="s2">&quot;FTD/ALS&quot;</span><span class="p">)),</span>
                              <span class="n">filter_data_ui</span><span class="p">(</span><span class="s2">&quot;filtering_gw&quot;</span><span class="p">)</span>
                            <span class="p">),</span>
                            <span class="n">mainPanel</span><span class="p">(</span>
                              <span class="n">DT</span><span class="p">::</span><span class="n">dataTableOutput</span><span class="p">(</span><span class="n">outputId</span> <span class="o">=</span> <span class="s2">&quot;table_gw&quot;</span><span class="p">),</span>
                            <span class="p">)</span>
                          <span class="p">)</span>
                 <span class="p">),</span>
                 <span class="n">tabPanel</span><span class="p">(</span><span class="s2">&quot;Info&quot;</span><span class="p">,</span> <span class="n">icon</span> <span class="o">=</span> <span class="n">icon</span><span class="p">(</span><span class="s2">&quot;info&quot;</span><span class="p">),</span>
                        <span class="n">mainPanel</span><span class="p">(</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">h1</span><span class="p">(</span><span class="s2">&quot;Structural variants in non-Alzheimer&#39;s dementias app&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">h2</span><span class="p">(</span><span class="s2">&quot;Background&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;Structural variants cause most variation in genomes but they have been rarely studied due to technical difficulties. This app presents data from Lewy body dementia (LBD) and frontotemporal dementia-amyotrophic lateral sclerosis (FTD/ALS) patients and neurologically healthy controls.&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;For more details please see our manuscript.&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">h2</span><span class="p">(</span><span class="s2">&quot;Structural variant calling and filtering&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;Structural variant calling was performed with GATK-SV, which uses five structural variant detection algorithms and machine learning to produce consensus variant calls and genotype filtering. The data presented in this app is a high-quality subset of structural variants that was created by further filtering the GATK-SV output with the following criteria:&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;1. 1 </span><span class="si">% F</span><span class="s2">DR&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;2. Variant filter type PASS, MULTIALLELIC, UNRESOLVED&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;3. GQ &lt; 300 set to missing&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;4. Genotyping rate &gt; 95 %&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;5. HWE mid-p adjusted p-value &gt; 0.000001&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;The final LBD dataset consisted of 2,355 LBD vs. 3,700 controls with 150,752 structural variants&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;The final FTD/ALS dataset consisted of 2,307 LBD vs. 3,677 controls with 158,991 structural variants&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">h2</span><span class="p">(</span><span class="s2">&quot;Contact information&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">h4</span><span class="p">(</span><span class="s2">&quot;Karri Kaivola (Application questions)&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;karri.kaivola@nih.gov / karri.kaivola@helsinki.fi&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;Neurodegenerative Diseases Research Unit&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;National Institute of Neurological Disorders and Stroke&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;National Institutes of Health&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;Bethesda, Maryland 20892&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;United States&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">h4</span><span class="p">(</span><span class="s2">&quot;Sonja W. Scholz (PI, LBD)&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;sonja.scholz@nih.gov&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;Neurodegenerative Diseases Research Unit&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;National Institute of Neurological Disorders and Stroke&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;National Institutes of Health&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;Bethesda, Maryland 20892&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;United States&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">h4</span><span class="p">(</span><span class="s2">&quot;Bryan J. Traynor (PI, FTD/ALS)&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;bryan.traynor@nih.gov&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;Neuromuscular Diseases Research Section&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;Laboratory of Neurogenetics&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;National Institute on Aging&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;National Institutes of Health&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;Bethesda, Maryland 20892&quot;</span><span class="p">),</span>
                          <span class="n">shiny</span><span class="p">::</span><span class="n">tags</span><span class="err">$</span><span class="n">p</span><span class="p">(</span><span class="s2">&quot;United States&quot;</span><span class="p">)</span>
                        <span class="p">)</span>  
                          <span class="p">)</span>
<span class="p">)</span>


<span class="c1">#Define server</span>
<span class="n">server</span> <span class="o">&lt;-</span> <span class="n">function</span><span class="p">(</span><span class="nb">input</span><span class="p">,</span> <span class="n">output</span><span class="p">,</span> <span class="n">session</span><span class="p">)</span> <span class="p">{</span>
  
  <span class="c1">##Test</span>
  <span class="n">data_gw</span> <span class="o">&lt;-</span> <span class="n">reactive</span><span class="p">({</span>
    <span class="n">get</span><span class="p">(</span><span class="nb">input</span><span class="err">$</span><span class="n">pheno_gw</span><span class="p">)</span>
  <span class="p">})</span>
  
  <span class="n">vars_gw</span> <span class="o">&lt;-</span> <span class="n">reactive</span><span class="p">({</span>
    <span class="k">if</span> <span class="p">(</span><span class="n">identical</span><span class="p">(</span><span class="nb">input</span><span class="err">$</span><span class="n">pheno_gw</span><span class="p">,</span> <span class="s2">&quot;LBD_gw&quot;</span><span class="p">))</span> <span class="p">{</span>
      <span class="n">colnames</span><span class="p">(</span><span class="n">LBD_gw</span><span class="p">)[</span><span class="mi">2</span><span class="p">:</span><span class="mi">12</span><span class="p">]</span>
    <span class="p">}</span> <span class="k">else</span> <span class="p">{</span>
      <span class="n">colnames</span><span class="p">(</span><span class="n">FTD_gw</span><span class="p">)[</span><span class="mi">2</span><span class="p">:</span><span class="mi">12</span><span class="p">]</span>
    <span class="p">}</span>
  <span class="p">})</span>
  
  <span class="n">res_filter_gw</span> <span class="o">&lt;-</span> <span class="n">filter_data_server</span><span class="p">(</span>
    <span class="nb">id</span> <span class="o">=</span> <span class="s2">&quot;filtering_gw&quot;</span><span class="p">,</span>
    <span class="n">data</span> <span class="o">=</span> <span class="n">data_gw</span><span class="p">,</span>
    <span class="n">name</span> <span class="o">=</span> <span class="n">reactive</span><span class="p">(</span><span class="nb">input</span><span class="err">$</span><span class="n">pheno_gw</span><span class="p">),</span>
    <span class="n">widget_num</span> <span class="o">=</span> <span class="s2">&quot;range&quot;</span><span class="p">,</span>
    <span class="n">widget_date</span> <span class="o">=</span> <span class="s2">&quot;slider&quot;</span><span class="p">,</span>
    <span class="n">widget_char</span> <span class="o">=</span> <span class="s2">&quot;picker&quot;</span><span class="p">,</span>
    <span class="n">label_na</span> <span class="o">=</span> <span class="s2">&quot;Missing&quot;</span><span class="p">,</span>
    <span class="n">drop_ids</span> <span class="o">=</span> <span class="n">FALSE</span><span class="p">,</span>
    <span class="nb">vars</span> <span class="o">=</span> <span class="n">vars_gw</span>
  <span class="p">)</span>
  
  <span class="n">output</span><span class="err">$</span><span class="n">table_gw</span> <span class="o">&lt;-</span> <span class="n">DT</span><span class="p">::</span><span class="n">renderDT</span><span class="p">({</span>
    <span class="n">res_filter_gw</span><span class="err">$</span><span class="n">filtered</span><span class="p">()</span>
  <span class="p">},</span> <span class="n">options</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">pageLength</span> <span class="o">=</span> <span class="mi">50</span><span class="p">))</span>

  <span class="c1">##Outputs of SV color legends for SV plots    </span>
  <span class="n">output</span><span class="err">$</span><span class="n">color_legend</span> <span class="o">&lt;-</span> <span class="n">renderImage</span><span class="p">({</span>
    <span class="nb">list</span><span class="p">(</span><span class="n">src</span><span class="o">=</span><span class="n">file</span><span class="o">.</span><span class="n">path</span><span class="p">(</span><span class="s2">&quot;color_code_legend.png&quot;</span><span class="p">))</span>
  <span class="p">},</span> <span class="n">deleteFile</span> <span class="o">=</span> <span class="n">FALSE</span><span class="p">)</span>
  
  <span class="n">output</span><span class="err">$</span><span class="n">color_legend2</span> <span class="o">&lt;-</span> <span class="n">renderImage</span><span class="p">({</span>
    <span class="nb">list</span><span class="p">(</span><span class="n">src</span><span class="o">=</span><span class="n">file</span><span class="o">.</span><span class="n">path</span><span class="p">(</span><span class="s2">&quot;color_code_legend.png&quot;</span><span class="p">))</span>
  <span class="p">},</span> <span class="n">deleteFile</span> <span class="o">=</span> <span class="n">FALSE</span><span class="p">)</span>
  
  <span class="c1">##Output NDD gene pre-rendered plots</span>
  <span class="n">output</span><span class="err">$</span><span class="n">img2</span> <span class="o">&lt;-</span> <span class="n">renderImage</span><span class="p">({</span>
    <span class="nb">list</span><span class="p">(</span>
      <span class="n">src</span> <span class="o">=</span> <span class="n">file</span><span class="o">.</span><span class="n">path</span><span class="p">(</span><span class="s2">&quot;NIH_SV_images/&quot;</span><span class="p">,</span><span class="nb">input</span><span class="err">$</span><span class="n">pheno2</span><span class="p">,</span><span class="n">paste0</span><span class="p">(</span><span class="nb">input</span><span class="err">$</span><span class="n">gene2</span><span class="p">,</span><span class="s2">&quot;_&quot;</span><span class="p">,</span><span class="nb">input</span><span class="err">$</span><span class="n">pheno2</span><span class="p">,</span><span class="s2">&quot;_candidate_gene_overlap.png&quot;</span><span class="p">)))</span>
  <span class="p">},</span> <span class="n">deleteFile</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span>
  
  <span class="c1">##Output info text</span>
  <span class="n">output</span><span class="err">$</span><span class="n">SV_info2</span> <span class="o">&lt;-</span> <span class="n">renderText</span><span class="p">({</span>
    <span class="s2">&quot;Allele frequency presented on structural variants </span><span class="se">\n</span><span class="s2">&quot;</span>
  <span class="p">})</span>
  
  <span class="c1">##NDD gene data frame filtering and output</span>
  <span class="n">data</span> <span class="o">&lt;-</span> <span class="n">reactive</span><span class="p">({</span>
    <span class="n">get</span><span class="p">(</span><span class="nb">input</span><span class="err">$</span><span class="n">dataset</span><span class="p">)</span>
  <span class="p">})</span>
  
  
  <span class="nb">vars</span> <span class="o">&lt;-</span> <span class="n">reactive</span><span class="p">({</span>
    <span class="k">if</span> <span class="p">(</span><span class="n">identical</span><span class="p">(</span><span class="nb">input</span><span class="err">$</span><span class="n">dataset</span><span class="p">,</span> <span class="s2">&quot;LBD&quot;</span><span class="p">))</span> <span class="p">{</span>
      <span class="n">colnames</span><span class="p">(</span><span class="n">LBD</span><span class="p">)[</span><span class="n">c</span><span class="p">(</span><span class="mi">1</span><span class="p">:</span><span class="mi">3</span><span class="p">,</span><span class="mi">6</span><span class="p">:</span><span class="mi">22</span><span class="p">)]</span>
    <span class="p">}</span> <span class="k">else</span> <span class="p">{</span>
      <span class="n">colnames</span><span class="p">(</span><span class="n">FTD</span><span class="p">)[</span><span class="n">c</span><span class="p">(</span><span class="mi">1</span><span class="p">:</span><span class="mi">3</span><span class="p">,</span><span class="mi">6</span><span class="p">:</span><span class="mi">22</span><span class="p">)]</span>
    <span class="p">}</span>
  <span class="p">})</span>
  
  <span class="n">res_filter</span> <span class="o">&lt;-</span> <span class="n">filter_data_server</span><span class="p">(</span>
    <span class="nb">id</span> <span class="o">=</span> <span class="s2">&quot;filtering&quot;</span><span class="p">,</span>
    <span class="n">data</span> <span class="o">=</span> <span class="n">data</span><span class="p">,</span>
    <span class="n">name</span> <span class="o">=</span> <span class="n">reactive</span><span class="p">(</span><span class="nb">input</span><span class="err">$</span><span class="n">dataset</span><span class="p">),</span>
    <span class="n">widget_num</span> <span class="o">=</span> <span class="s2">&quot;range&quot;</span><span class="p">,</span>
    <span class="n">widget_date</span> <span class="o">=</span> <span class="s2">&quot;slider&quot;</span><span class="p">,</span>
    <span class="n">widget_char</span> <span class="o">=</span> <span class="s2">&quot;picker&quot;</span><span class="p">,</span>
    <span class="n">label_na</span> <span class="o">=</span> <span class="s2">&quot;Missing&quot;</span><span class="p">,</span>
    <span class="n">drop_ids</span> <span class="o">=</span> <span class="n">FALSE</span><span class="p">,</span>
    <span class="nb">vars</span> <span class="o">=</span> <span class="nb">vars</span>
  <span class="p">)</span>
  
  <span class="n">output</span><span class="err">$</span><span class="n">table</span> <span class="o">&lt;-</span> <span class="n">DT</span><span class="p">::</span><span class="n">renderDT</span><span class="p">({</span>
    <span class="n">res_filter</span><span class="err">$</span><span class="n">filtered</span><span class="p">()</span>
  <span class="p">},</span> <span class="n">options</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">pageLength</span> <span class="o">=</span> <span class="mi">50</span><span class="p">))</span>
  
  <span class="c1">##Genome-wide plotting</span>
  <span class="c1">###Input from gene search box updates chromosome and genomic coordinates</span>
  <span class="n">observe</span><span class="p">({</span>
    <span class="n">gene_name</span> <span class="o">&lt;-</span> <span class="nb">input</span><span class="err">$</span><span class="n">gene_search</span>
    <span class="n">updatePickerInput</span><span class="p">(</span>
      <span class="n">session</span> <span class="o">=</span> <span class="n">session</span><span class="p">,</span>
      <span class="n">inputId</span> <span class="o">=</span> <span class="s2">&quot;chrom&quot;</span><span class="p">,</span>
      <span class="n">selected</span> <span class="o">=</span> <span class="n">gene_info</span><span class="err">$</span><span class="n">chromosome_name</span><span class="p">[</span><span class="n">gene_info</span><span class="err">$</span><span class="n">external_gene_name</span> <span class="o">==</span> <span class="n">gene_name</span><span class="p">]</span>
    <span class="p">)</span>
    <span class="n">updateNumericRangeInput</span><span class="p">(</span>
      <span class="n">session</span> <span class="o">=</span> <span class="n">session</span><span class="p">,</span>
      <span class="n">inputId</span> <span class="o">=</span> <span class="s2">&quot;genomic_range&quot;</span><span class="p">,</span>
      <span class="n">value</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="n">gene_info</span><span class="err">$</span><span class="n">start_position</span><span class="p">[</span><span class="n">gene_info</span><span class="err">$</span><span class="n">external_gene_name</span> <span class="o">==</span> <span class="n">gene_name</span><span class="p">],</span> <span class="n">gene_info</span><span class="err">$</span><span class="n">end_position</span><span class="p">[</span><span class="n">gene_info</span><span class="err">$</span><span class="n">external_gene_name</span> <span class="o">==</span> <span class="n">gene_name</span><span class="p">]),</span>
    <span class="p">)</span>
  <span class="p">})</span> <span class="o">%&gt;%</span> <span class="n">bindEvent</span><span class="p">(</span><span class="nb">input</span><span class="err">$</span><span class="n">gene_search</span><span class="p">,</span> <span class="n">ignoreInit</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">)</span>
  
  <span class="c1">###Input from zoom buttons updates genomic coordinates</span>
  <span class="n">observe</span><span class="p">({</span>
    <span class="n">updateNumericRangeInput</span><span class="p">(</span>
      <span class="n">session</span> <span class="o">=</span> <span class="n">session</span><span class="p">,</span>
      <span class="n">inputId</span> <span class="o">=</span> <span class="s2">&quot;genomic_range&quot;</span><span class="p">,</span>
      <span class="n">value</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="nb">round</span><span class="p">(</span><span class="nb">input</span><span class="err">$</span><span class="n">genomic_range</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">-</span><span class="p">((</span><span class="nb">input</span><span class="err">$</span><span class="n">genomic_range</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="o">-</span><span class="nb">input</span><span class="err">$</span><span class="n">genomic_range</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span><span class="o">*</span><span class="mf">1.5</span><span class="p">)),</span> <span class="nb">round</span><span class="p">(</span><span class="nb">input</span><span class="err">$</span><span class="n">genomic_range</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="o">+</span><span class="p">((</span><span class="nb">input</span><span class="err">$</span><span class="n">genomic_range</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="o">-</span><span class="nb">input</span><span class="err">$</span><span class="n">genomic_range</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span><span class="o">*</span><span class="mf">1.5</span><span class="p">)))</span>
    <span class="p">)</span>
  <span class="p">})</span> <span class="o">%&gt;%</span> <span class="n">bindEvent</span><span class="p">(</span><span class="nb">input</span><span class="err">$</span><span class="n">zoom_out_3</span><span class="p">,</span> <span class="n">ignoreInit</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">)</span>
  
  <span class="n">observe</span><span class="p">({</span>
    <span class="n">updateNumericRangeInput</span><span class="p">(</span>
      <span class="n">session</span> <span class="o">=</span> <span class="n">session</span><span class="p">,</span>
      <span class="n">inputId</span> <span class="o">=</span> <span class="s2">&quot;genomic_range&quot;</span><span class="p">,</span>
      <span class="n">value</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="nb">round</span><span class="p">(</span><span class="nb">input</span><span class="err">$</span><span class="n">genomic_range</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">+</span><span class="p">((</span><span class="nb">input</span><span class="err">$</span><span class="n">genomic_range</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="o">-</span><span class="nb">input</span><span class="err">$</span><span class="n">genomic_range</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span><span class="o">/</span><span class="mf">1.5</span><span class="p">)),</span> <span class="nb">round</span><span class="p">(</span><span class="nb">input</span><span class="err">$</span><span class="n">genomic_range</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="o">-</span><span class="p">((</span><span class="nb">input</span><span class="err">$</span><span class="n">genomic_range</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="o">-</span><span class="nb">input</span><span class="err">$</span><span class="n">genomic_range</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span><span class="o">/</span><span class="mf">1.5</span><span class="p">)))</span>
    <span class="p">)</span>
  <span class="p">})</span> <span class="o">%&gt;%</span> <span class="n">bindEvent</span><span class="p">(</span><span class="nb">input</span><span class="err">$</span><span class="n">zoom_in_3</span><span class="p">,</span> <span class="n">ignoreInit</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">)</span>
  
  <span class="c1">###Plot user-defined genomic region</span>
  <span class="n">output</span><span class="err">$</span><span class="n">img3</span> <span class="o">&lt;-</span> <span class="n">renderPlot</span><span class="p">(</span><span class="n">width</span> <span class="o">=</span> <span class="s2">&quot;auto&quot;</span><span class="p">,</span> <span class="n">height</span> <span class="o">=</span> <span class="s2">&quot;auto&quot;</span><span class="p">,</span> <span class="p">{</span>
    <span class="c1">#Set variables based on genomic range input</span>
    <span class="nb">chr</span> <span class="o">&lt;-</span> <span class="nb">input</span><span class="err">$</span><span class="n">chrom</span>
    <span class="n">start</span> <span class="o">&lt;-</span> <span class="nb">input</span><span class="err">$</span><span class="n">genomic_range</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
    <span class="n">stop</span> <span class="o">&lt;-</span> <span class="nb">input</span><span class="err">$</span><span class="n">genomic_range</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span>
    <span class="c1">#Select df to plot</span>
    <span class="k">if</span><span class="p">(</span><span class="nb">input</span><span class="err">$</span><span class="n">pheno</span> <span class="o">==</span> <span class="s2">&quot;LBD&quot;</span><span class="p">)</span> <span class="p">{</span>
      <span class="n">df_cases</span> <span class="o">&lt;-</span> <span class="n">LBD_cases</span>
      <span class="n">df_controls</span> <span class="o">&lt;-</span> <span class="n">LBD_controls</span>
    <span class="p">}</span> <span class="k">else</span> <span class="p">{</span>
      <span class="n">df_cases</span> <span class="o">&lt;-</span> <span class="n">FTD_cases</span>
      <span class="n">df_controls</span> <span class="o">&lt;-</span> <span class="n">FTD_controls</span>
    <span class="p">}</span>
    <span class="c1">##Find overlaps (3 different criteria) with input region and dfs and extract IDs for filtering</span>
    <span class="n">tmp_cases1</span> <span class="o">&lt;-</span> <span class="n">df_cases</span> <span class="o">%&gt;%</span>
      <span class="nb">filter</span><span class="p">(</span><span class="n">CHROM</span> <span class="o">==</span> <span class="nb">chr</span><span class="p">)</span> <span class="o">%&gt;%</span>
      <span class="nb">filter</span><span class="p">(</span><span class="n">END</span> <span class="o">&gt;=</span> <span class="n">start</span> <span class="o">&amp;</span> <span class="n">END</span> <span class="o">&lt;=</span> <span class="n">stop</span><span class="p">)</span>
    <span class="n">tmp_cases2</span> <span class="o">&lt;-</span> <span class="n">df_cases</span> <span class="o">%&gt;%</span>
      <span class="nb">filter</span><span class="p">(</span><span class="n">CHROM</span> <span class="o">==</span> <span class="nb">chr</span><span class="p">)</span> <span class="o">%&gt;%</span>
      <span class="nb">filter</span><span class="p">(</span><span class="n">POS</span> <span class="o">&gt;=</span> <span class="n">start</span> <span class="o">&amp;</span> <span class="n">POS</span> <span class="o">&lt;=</span> <span class="n">stop</span><span class="p">)</span>
    <span class="n">tmp_cases3</span> <span class="o">&lt;-</span> <span class="n">df_cases</span> <span class="o">%&gt;%</span>
      <span class="nb">filter</span><span class="p">(</span><span class="n">CHROM</span> <span class="o">==</span> <span class="nb">chr</span><span class="p">)</span> <span class="o">%&gt;%</span>
      <span class="nb">filter</span><span class="p">(</span><span class="n">POS</span> <span class="o">&lt;=</span> <span class="n">start</span> <span class="o">&amp;</span> <span class="n">END</span> <span class="o">&gt;=</span><span class="n">stop</span><span class="p">)</span>
    <span class="n">tmp_IDs_cases</span> <span class="o">&lt;-</span> <span class="n">unique</span><span class="p">(</span><span class="n">c</span><span class="p">(</span><span class="n">tmp_cases1</span><span class="err">$</span><span class="n">ID</span><span class="p">,</span> <span class="n">tmp_cases2</span><span class="err">$</span><span class="n">ID</span><span class="p">,</span> <span class="n">tmp_cases3</span><span class="err">$</span><span class="n">ID</span><span class="p">))</span>
    <span class="n">tmp_cases</span> <span class="o">&lt;-</span> <span class="n">df_cases</span> <span class="o">%&gt;%</span>
      <span class="nb">filter</span><span class="p">(</span><span class="n">ID</span> <span class="o">%</span><span class="k">in</span>% tmp_IDs_cases)
    
    <span class="n">tmp_controls1</span> <span class="o">&lt;-</span> <span class="n">df_controls</span> <span class="o">%&gt;%</span>
      <span class="nb">filter</span><span class="p">(</span><span class="n">CHROM</span> <span class="o">==</span> <span class="nb">chr</span><span class="p">)</span> <span class="o">%&gt;%</span>
      <span class="nb">filter</span><span class="p">(</span><span class="n">END</span> <span class="o">&gt;=</span> <span class="n">start</span> <span class="o">&amp;</span> <span class="n">END</span> <span class="o">&lt;=</span> <span class="n">stop</span><span class="p">)</span>
    <span class="n">tmp_controls2</span> <span class="o">&lt;-</span> <span class="n">df_controls</span> <span class="o">%&gt;%</span>
      <span class="nb">filter</span><span class="p">(</span><span class="n">CHROM</span> <span class="o">==</span> <span class="nb">chr</span><span class="p">)</span> <span class="o">%&gt;%</span>
      <span class="nb">filter</span><span class="p">(</span><span class="n">POS</span> <span class="o">&gt;=</span> <span class="n">start</span> <span class="o">&amp;</span> <span class="n">POS</span> <span class="o">&lt;=</span> <span class="n">stop</span><span class="p">)</span>
    <span class="n">tmp_controls3</span> <span class="o">&lt;-</span> <span class="n">df_controls</span> <span class="o">%&gt;%</span>
      <span class="nb">filter</span><span class="p">(</span><span class="n">CHROM</span> <span class="o">==</span> <span class="nb">chr</span><span class="p">)</span> <span class="o">%&gt;%</span>
      <span class="nb">filter</span><span class="p">(</span><span class="n">POS</span> <span class="o">&lt;=</span> <span class="n">start</span> <span class="o">&amp;</span> <span class="n">END</span> <span class="o">&gt;=</span><span class="n">stop</span><span class="p">)</span>
    <span class="n">tmp_IDs_controls</span> <span class="o">&lt;-</span> <span class="n">unique</span><span class="p">(</span><span class="n">c</span><span class="p">(</span><span class="n">tmp_controls1</span><span class="err">$</span><span class="n">ID</span><span class="p">,</span> <span class="n">tmp_controls2</span><span class="err">$</span><span class="n">ID</span><span class="p">,</span> <span class="n">tmp_controls3</span><span class="err">$</span><span class="n">ID</span><span class="p">))</span>
    <span class="n">tmp_controls</span> <span class="o">&lt;-</span> <span class="n">df_controls</span> <span class="o">%&gt;%</span>
      <span class="nb">filter</span><span class="p">(</span><span class="n">ID</span> <span class="o">%</span><span class="k">in</span>% tmp_IDs_controls)
    
    <span class="c1">##So that text is shown in plot: if SV POS &lt; start, make POS = start.</span>
    <span class="n">tmp_cases</span><span class="err">$</span><span class="n">POS</span><span class="p">[</span><span class="n">tmp_cases</span><span class="err">$</span><span class="n">POS</span> <span class="o">&lt;</span> <span class="n">start</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">start</span>
    <span class="n">tmp_controls</span><span class="err">$</span><span class="n">POS</span><span class="p">[</span><span class="n">tmp_controls</span><span class="err">$</span><span class="n">POS</span> <span class="o">&lt;</span> <span class="n">start</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">start</span>
    <span class="c1">##So that text is shown in plot: if SV END &gt; stop, make END = stop.</span>
    <span class="n">tmp_cases</span><span class="err">$</span><span class="n">END</span><span class="p">[</span><span class="n">tmp_cases</span><span class="err">$</span><span class="n">END</span> <span class="o">&gt;</span>  <span class="n">stop</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">stop</span>
    <span class="n">tmp_controls</span><span class="err">$</span><span class="n">END</span><span class="p">[</span><span class="n">tmp_controls</span><span class="err">$</span><span class="n">END</span> <span class="o">&gt;</span> <span class="n">stop</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">stop</span>
    <span class="c1">##Make tracks and plot</span>
    <span class="k">if</span> <span class="p">((</span><span class="n">nrow</span><span class="p">(</span><span class="n">tmp_cases</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">nrow</span><span class="p">(</span><span class="n">tmp_controls</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">))</span> <span class="p">{</span>
      <span class="n">gtrack</span> <span class="o">&lt;-</span> <span class="n">GenomeAxisTrack</span><span class="p">(</span><span class="n">fontsize</span><span class="o">=</span><span class="mi">12</span><span class="p">)</span>
      <span class="n">itrack</span> <span class="o">&lt;-</span> <span class="n">IdeogramTrack</span><span class="p">(</span><span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="nb">chr</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
      <span class="n">biomTrack</span> <span class="o">&lt;-</span> <span class="n">BiomartGeneRegionTrack</span><span class="p">(</span><span class="n">transcriptAnnotation</span> <span class="o">=</span> <span class="s2">&quot;symbol&quot;</span><span class="p">,</span> <span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span><span class="n">chromosome</span> <span class="o">=</span> <span class="nb">chr</span><span class="p">,</span> <span class="n">start</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;Gene&quot;</span><span class="p">,</span> <span class="nb">filter</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">transcript_is_canonical</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),</span> <span class="n">biomart</span> <span class="o">=</span> <span class="n">bm</span><span class="p">,</span> <span class="n">just</span><span class="o">.</span><span class="n">group</span><span class="o">=</span><span class="s2">&quot;below&quot;</span><span class="p">,</span> <span class="n">protein_coding</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr3</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr5</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">)</span>
      <span class="n">atrack_cases</span> <span class="o">&lt;-</span> <span class="n">AnnotationTrack</span><span class="p">(</span><span class="n">start</span><span class="o">=</span><span class="n">tmp_cases</span><span class="err">$</span><span class="n">POS</span><span class="p">,</span> <span class="n">end</span><span class="o">=</span><span class="n">tmp_cases</span><span class="err">$</span><span class="n">END</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="nb">chr</span><span class="p">,</span> <span class="n">genome</span><span class="o">=</span><span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;Cases&quot;</span><span class="p">,</span> <span class="n">feature</span> <span class="o">=</span> <span class="n">tmp_cases</span><span class="err">$</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="nb">id</span><span class="o">=</span><span class="n">tmp_cases</span><span class="err">$</span><span class="n">AF_cases</span><span class="p">,</span> <span class="n">showFeatureId</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">item</span><span class="o">=</span><span class="s2">&quot;black&quot;</span><span class="p">,</span> <span class="n">col</span><span class="o">.</span><span class="n">line</span><span class="o">=</span><span class="s2">&quot;white&quot;</span><span class="p">,</span> <span class="nb">min</span><span class="o">.</span><span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
      <span class="n">atrack_controls</span> <span class="o">&lt;-</span> <span class="n">AnnotationTrack</span><span class="p">(</span><span class="n">start</span><span class="o">=</span><span class="n">tmp_controls</span><span class="err">$</span><span class="n">POS</span><span class="p">,</span> <span class="n">end</span><span class="o">=</span><span class="n">tmp_controls</span><span class="err">$</span><span class="n">END</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="nb">chr</span><span class="p">,</span> <span class="n">genome</span><span class="o">=</span><span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;Controls&quot;</span><span class="p">,</span> <span class="n">feature</span> <span class="o">=</span> <span class="n">tmp_controls</span><span class="err">$</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="nb">id</span><span class="o">=</span><span class="n">tmp_controls</span><span class="err">$</span><span class="n">AF_controls</span><span class="p">,</span> <span class="n">showFeatureId</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">item</span><span class="o">=</span><span class="s2">&quot;black&quot;</span><span class="p">,</span> <span class="nb">min</span><span class="o">.</span><span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
      <span class="n">plotTracks</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span><span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack_cases</span><span class="p">,</span> <span class="n">atrack_controls</span><span class="p">,</span> <span class="n">biomTrack</span><span class="p">),</span> <span class="kn">from</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;#808080&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">group</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">BND</span><span class="o">=</span><span class="s2">&quot;#377246&quot;</span><span class="p">,</span> <span class="n">CPX</span><span class="o">=</span><span class="s2">&quot;#86E496&quot;</span><span class="p">,</span> <span class="n">DEL</span><span class="o">=</span><span class="s2">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="n">DUP</span><span class="o">=</span><span class="s2">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="n">INS</span><span class="o">=</span><span class="s2">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="n">INS_ME</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_ALU</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_LINE1</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_SVA</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INV</span><span class="o">=</span><span class="s2">&quot;#F69A57&quot;</span><span class="p">)</span>
    <span class="p">}</span> <span class="k">else</span> <span class="k">if</span> <span class="p">((</span><span class="n">nrow</span><span class="p">(</span><span class="n">tmp_cases</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">nrow</span><span class="p">(</span><span class="n">tmp_controls</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">))</span> <span class="p">{</span>
      <span class="n">gtrack</span> <span class="o">&lt;-</span> <span class="n">GenomeAxisTrack</span><span class="p">(</span><span class="n">fontsize</span><span class="o">=</span><span class="mi">12</span><span class="p">)</span>
      <span class="n">itrack</span> <span class="o">&lt;-</span> <span class="n">IdeogramTrack</span><span class="p">(</span><span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="nb">chr</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
      <span class="n">biomTrack</span> <span class="o">&lt;-</span> <span class="n">BiomartGeneRegionTrack</span><span class="p">(</span><span class="n">transcriptAnnotation</span> <span class="o">=</span> <span class="s2">&quot;symbol&quot;</span><span class="p">,</span> <span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span><span class="n">chromosome</span> <span class="o">=</span> <span class="nb">chr</span><span class="p">,</span> <span class="n">start</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;Gene&quot;</span><span class="p">,</span> <span class="nb">filter</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">transcript_is_canonical</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),</span> <span class="n">biomart</span> <span class="o">=</span> <span class="n">bm</span><span class="p">,</span> <span class="n">just</span><span class="o">.</span><span class="n">group</span><span class="o">=</span><span class="s2">&quot;below&quot;</span><span class="p">,</span> <span class="n">protein_coding</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr3</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr5</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">)</span>
      <span class="n">atrack_cases</span> <span class="o">&lt;-</span> <span class="n">AnnotationTrack</span><span class="p">(</span><span class="n">start</span><span class="o">=</span><span class="n">tmp_cases</span><span class="err">$</span><span class="n">POS</span><span class="p">,</span> <span class="n">end</span><span class="o">=</span><span class="n">tmp_cases</span><span class="err">$</span><span class="n">END</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="nb">chr</span><span class="p">,</span> <span class="n">genome</span><span class="o">=</span><span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;Cases&quot;</span><span class="p">,</span> <span class="n">feature</span> <span class="o">=</span> <span class="n">tmp_cases</span><span class="err">$</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="nb">id</span><span class="o">=</span><span class="n">tmp_cases</span><span class="err">$</span><span class="n">AF_cases</span><span class="p">,</span> <span class="n">showFeatureId</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">item</span><span class="o">=</span><span class="s2">&quot;black&quot;</span><span class="p">,</span> <span class="nb">min</span><span class="o">.</span><span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
      <span class="n">plotTracks</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span><span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack_cases</span><span class="p">,</span> <span class="n">biomTrack</span><span class="p">),</span> <span class="kn">from</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;#808080&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">group</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">BND</span><span class="o">=</span><span class="s2">&quot;#377246&quot;</span><span class="p">,</span> <span class="n">CPX</span><span class="o">=</span><span class="s2">&quot;#86E496&quot;</span><span class="p">,</span> <span class="n">DEL</span><span class="o">=</span><span class="s2">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="n">DUP</span><span class="o">=</span><span class="s2">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="n">INS</span><span class="o">=</span><span class="s2">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="n">INS_ME</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_ALU</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_LINE1</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_SVA</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INV</span><span class="o">=</span><span class="s2">&quot;#F69A57&quot;</span><span class="p">)</span>
    <span class="p">}</span> <span class="k">else</span> <span class="k">if</span> <span class="p">((</span><span class="n">nrow</span><span class="p">(</span><span class="n">tmp_cases</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">nrow</span><span class="p">(</span><span class="n">tmp_controls</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">))</span> <span class="p">{</span>
      <span class="n">gtrack</span> <span class="o">&lt;-</span> <span class="n">GenomeAxisTrack</span><span class="p">(</span><span class="n">showTitle</span><span class="o">=</span><span class="n">TRUE</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;No SVs&quot;</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;#808080&quot;</span><span class="p">,</span> <span class="n">rotation</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
      <span class="n">plotTracks</span><span class="p">(</span><span class="n">gtrack</span><span class="p">,</span> <span class="kn">from</span> <span class="o">=</span> <span class="mi">100</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="mi">110</span><span class="p">,</span> <span class="n">title</span><span class="o">.</span><span class="n">width</span> <span class="o">=</span> <span class="mi">5</span><span class="p">)</span>
    <span class="p">}</span> <span class="k">else</span> <span class="p">{</span>
      <span class="n">gtrack</span> <span class="o">&lt;-</span> <span class="n">GenomeAxisTrack</span><span class="p">(</span><span class="n">fontsize</span><span class="o">=</span><span class="mi">12</span><span class="p">)</span>
      <span class="n">itrack</span> <span class="o">&lt;-</span> <span class="n">IdeogramTrack</span><span class="p">(</span><span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="nb">chr</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
      <span class="n">biomTrack</span> <span class="o">&lt;-</span> <span class="n">BiomartGeneRegionTrack</span><span class="p">(</span><span class="n">transcriptAnnotation</span> <span class="o">=</span> <span class="s2">&quot;symbol&quot;</span><span class="p">,</span> <span class="n">genome</span> <span class="o">=</span> <span class="s2">&quot;hg38&quot;</span><span class="p">,</span><span class="n">chromosome</span> <span class="o">=</span> <span class="nb">chr</span><span class="p">,</span> <span class="n">start</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;Gene&quot;</span><span class="p">,</span> <span class="nb">filter</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">transcript_is_canonical</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),</span> <span class="n">biomart</span> <span class="o">=</span> <span class="n">bm</span><span class="p">,</span> <span class="n">just</span><span class="o">.</span><span class="n">group</span><span class="o">=</span><span class="s2">&quot;below&quot;</span><span class="p">,</span> <span class="n">protein_coding</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr3</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">utr5</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="s2">&quot;gray&quot;</span><span class="p">)</span>
      <span class="n">atrack_controls</span> <span class="o">&lt;-</span> <span class="n">AnnotationTrack</span><span class="p">(</span><span class="n">start</span><span class="o">=</span><span class="n">tmp_controls</span><span class="err">$</span><span class="n">POS</span><span class="p">,</span> <span class="n">end</span><span class="o">=</span><span class="n">tmp_controls</span><span class="err">$</span><span class="n">END</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="nb">chr</span><span class="p">,</span> <span class="n">genome</span><span class="o">=</span><span class="s2">&quot;hg38&quot;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;Controls&quot;</span><span class="p">,</span> <span class="n">feature</span> <span class="o">=</span> <span class="n">tmp_controls</span><span class="err">$</span><span class="n">SVTYPE</span><span class="p">,</span> <span class="nb">id</span><span class="o">=</span><span class="n">tmp_controls</span><span class="err">$</span><span class="n">AF_controls</span><span class="p">,</span> <span class="n">showFeatureId</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">item</span><span class="o">=</span><span class="s2">&quot;black&quot;</span><span class="p">,</span> <span class="nb">min</span><span class="o">.</span><span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
      <span class="n">plotTracks</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span><span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack_controls</span><span class="p">,</span> <span class="n">biomTrack</span><span class="p">),</span> <span class="kn">from</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="n">stop</span><span class="p">,</span> <span class="n">fontcolor</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;#808080&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">group</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">.</span><span class="n">title</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">BND</span><span class="o">=</span><span class="s2">&quot;#377246&quot;</span><span class="p">,</span> <span class="n">CPX</span><span class="o">=</span><span class="s2">&quot;#86E496&quot;</span><span class="p">,</span> <span class="n">DEL</span><span class="o">=</span><span class="s2">&quot;#D6402D&quot;</span><span class="p">,</span> <span class="n">DUP</span><span class="o">=</span><span class="s2">&quot;#5282AF&quot;</span><span class="p">,</span> <span class="n">INS</span><span class="o">=</span><span class="s2">&quot;#D579E1&quot;</span><span class="p">,</span> <span class="n">INS_ME</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_ALU</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_LINE1</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INS_ME_SVA</span><span class="o">=</span><span class="s2">&quot;#F6AAFF&quot;</span><span class="p">,</span> <span class="n">INV</span><span class="o">=</span><span class="s2">&quot;#F69A57&quot;</span><span class="p">)</span>
    <span class="p">}</span>
  <span class="p">})</span> <span class="o">%&gt;%</span> <span class="n">bindEvent</span><span class="p">(</span><span class="nb">input</span><span class="err">$</span><span class="n">plot</span><span class="p">,</span> <span class="n">ignoreInit</span> <span class="o">=</span> <span class="n">FALSE</span><span class="p">,</span> <span class="n">ignoreNULL</span> <span class="o">=</span> <span class="n">FALSE</span><span class="p">)</span>
<span class="p">}</span>

<span class="c1">#Launch shiny app</span>
<span class="n">shinyApp</span><span class="p">(</span><span class="n">ui</span><span class="p">,</span> <span class="n">server</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Miscellaneous">Miscellaneous<a class="anchor-link" href="#Miscellaneous">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Manhattan-plots">Manhattan plots<a class="anchor-link" href="#Manhattan-plots">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Check number of SVs in GWAS result files to determine Bonferroni correction threshold</span>

<span class="nb">echo</span> <span class="s2">&quot;Number of SVs in LBD analysis:&quot;</span>
grep -v <span class="s1">&#39;CHROM&#39;</span> <span class="nv">$PATH5</span>/Analysis.GLM.hg38_minGQ300/LBD/LBD.controls.UNRELATED.SNVindels.SV-1perc-fdr-minGQ300.hg38.hwe1e-6.maf001cases.exclTelomereFlank100kb.exclCentromereFlank2.5Mb.exclVDJ.QUAL500.autosomesOnly.withVarFilterInfo.PASS.UNRESOLVED.QUAL500.TIDY.txt <span class="p">|</span> wc -l
<span class="nb">echo</span> <span class="s2">&quot;Number of SVs in FTD analysis:&quot;</span>
grep -v <span class="s1">&#39;CHROM&#39;</span> <span class="nv">$PATH5</span>/Analysis.GLM.hg38_minGQ300/FTD/FTD.controls.UNRELATED.SNVindels.SV-1perc-fdr-minGQ300.hg38.hwe1e-6.maf001cases.exclTelomereFlank100kb.exclCentromereFlank2.5Mb.exclVDJ.QUAL500.autosomesOnly.withVarFilterInfo.PASS.UNRESOLVED.QUAL500.TIDY.txt <span class="p">|</span> wc -l
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1">#Bonferroni corrected thresholds are:</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;LBD:&quot;</span><span class="p">,</span> <span class="mf">0.05</span><span class="o">/</span><span class="mi">4889</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;FTD:&quot;</span><span class="p">,</span> <span class="mf">0.05</span><span class="o">/</span><span class="mi">4699</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Lewy-body-dementias-plot">Lewy body dementias plot<a class="anchor-link" href="#Lewy-body-dementias-plot">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%</span><span class="k">load_ext</span> rpy2.ipython
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="nf">subset</span><span class="p">(</span><span class="n">gwas_data</span><span class="p">,</span> <span class="n">ID</span> <span class="o">%in%</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;LBD_DUP_chr4_3651&quot;</span><span class="p">,</span> <span class="s">&quot;LBD_DEL_chr1_11039&quot;</span><span class="p">,</span> <span class="s">&quot;LBD_DUP_chr21_888&quot;</span><span class="p">))</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">#Code modified from https://danielroelfs.com/blog/how-i-create-manhattan-plots-using-ggplot/ </span>

<span class="c1">#Plot LBD manhattan plot</span>

<span class="c1">##Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggtext</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggrepel</span><span class="p">)</span>

<span class="c1">##Import data</span>
<span class="n">LBD</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;$PATH5/Analysis.GLM.hg38_minGQ300/LBD/LBD.controls.UNRELATED.SNVindels.SV-1perc-fdr-minGQ300.hg38.hwe1e-6.maf001cases.exclTelomereFlank100kb.exclCentromereFlank2.5Mb.exclVDJ.QUAL500.autosomesOnly.withVarFilterInfo.PASS.UNRESOLVED.QUAL500.TIDY.txt&quot;</span><span class="p">)</span>

<span class="c1">##Set significance threshold</span>
<span class="n">sig</span> <span class="o">&lt;-</span> <span class="m">0.05</span><span class="o">/</span><span class="m">4889</span>

<span class="c1">##Rename data to suit downstream code</span>
<span class="n">gwas_data</span> <span class="o">&lt;-</span> <span class="n">LBD</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">gwas_data</span><span class="p">)[</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="m">12</span><span class="p">)]</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span> <span class="s">&quot;bp&quot;</span><span class="p">,</span> <span class="s">&quot;p&quot;</span><span class="p">)</span>

<span class="c1">##Downstream code</span>
<span class="n">data_cum</span> <span class="o">&lt;-</span> <span class="n">gwas_data</span> <span class="o">%&gt;%</span> 
  <span class="nf">group_by</span><span class="p">(</span><span class="n">chr</span><span class="p">)</span> <span class="o">%&gt;%</span> 
  <span class="nf">summarise</span><span class="p">(</span><span class="n">max_bp</span> <span class="o">=</span> <span class="nf">max</span><span class="p">(</span><span class="n">bp</span><span class="p">))</span> <span class="o">%&gt;%</span> 
  <span class="nf">mutate</span><span class="p">(</span><span class="n">bp_add</span> <span class="o">=</span> <span class="nf">lag</span><span class="p">(</span><span class="nf">cumsum</span><span class="p">(</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">max_bp</span><span class="p">)),</span> <span class="n">default</span> <span class="o">=</span> <span class="m">0</span><span class="p">))</span> <span class="o">%&gt;%</span> 
  <span class="nf">select</span><span class="p">(</span><span class="n">chr</span><span class="p">,</span> <span class="n">bp_add</span><span class="p">)</span>


<span class="n">gwas_data</span> <span class="o">&lt;-</span> <span class="n">gwas_data</span> <span class="o">%&gt;%</span> 
  <span class="nf">inner_join</span><span class="p">(</span><span class="n">data_cum</span><span class="p">,</span> <span class="n">by</span> <span class="o">=</span> <span class="s">&quot;chr&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span> 
  <span class="nf">mutate</span><span class="p">(</span><span class="n">bp_cum</span> <span class="o">=</span> <span class="n">bp</span> <span class="o">+</span> <span class="n">bp_add</span><span class="p">)</span>


<span class="n">axis_set</span> <span class="o">&lt;-</span> <span class="n">gwas_data</span> <span class="o">%&gt;%</span> 
  <span class="nf">group_by</span><span class="p">(</span><span class="n">chr</span><span class="p">)</span> <span class="o">%&gt;%</span> 
  <span class="nf">summarize</span><span class="p">(</span><span class="n">center</span> <span class="o">=</span> <span class="nf">mean</span><span class="p">(</span><span class="n">bp_cum</span><span class="p">))</span>

<span class="n">ylim</span> <span class="o">&lt;-</span> <span class="n">gwas_data</span> <span class="o">%&gt;%</span> 
  <span class="nf">filter</span><span class="p">(</span><span class="n">p</span> <span class="o">==</span> <span class="nf">min</span><span class="p">(</span><span class="n">p</span><span class="p">))</span> <span class="o">%&gt;%</span> 
  <span class="nf">mutate</span><span class="p">(</span><span class="n">ylim</span> <span class="o">=</span> <span class="nf">abs</span><span class="p">(</span><span class="nf">floor</span><span class="p">(</span><span class="nf">log10</span><span class="p">(</span><span class="n">p</span><span class="p">)))</span> <span class="o">+</span> <span class="m">2</span><span class="p">)</span> <span class="o">%&gt;%</span> 
  <span class="nf">pull</span><span class="p">(</span><span class="n">ylim</span><span class="p">)</span>

<span class="n">highlight_df</span> <span class="o">&lt;-</span> <span class="n">gwas_data</span> <span class="o">%&gt;%</span> 
             <span class="nf">filter</span><span class="p">(</span><span class="n">p</span><span class="o">&lt;=</span><span class="m">0.000012</span><span class="p">)</span>


<span class="n">manhplot</span> <span class="o">&lt;-</span> <span class="nf">ggplot</span><span class="p">(</span><span class="n">gwas_data</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">bp_cum</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="o">-</span><span class="nf">log10</span><span class="p">(</span><span class="n">p</span><span class="p">),</span> 
                                  <span class="n">color</span> <span class="o">=</span> <span class="nf">as_factor</span><span class="p">(</span><span class="n">chr</span><span class="p">)))</span> <span class="o">+</span>
  <span class="nf">geom_hline</span><span class="p">(</span><span class="n">yintercept</span> <span class="o">=</span> <span class="o">-</span><span class="nf">log10</span><span class="p">(</span><span class="n">sig</span><span class="p">),</span> <span class="n">color</span> <span class="o">=</span> <span class="s">&quot;grey40&quot;</span><span class="p">,</span> <span class="n">linetype</span> <span class="o">=</span> <span class="s">&quot;dashed&quot;</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">alpha</span> <span class="o">=</span> <span class="m">0.75</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">highlight_df</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">bp_cum</span><span class="p">,</span><span class="n">y</span><span class="o">=-</span><span class="nf">log10</span><span class="p">(</span><span class="n">p</span><span class="p">)),</span> <span class="n">color</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;gray&#39;</span><span class="p">,</span><span class="s">&#39;red&#39;</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">geom_text</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">label</span><span class="o">=</span><span class="nf">ifelse</span><span class="p">(</span><span class="n">p</span><span class="o">&lt;</span><span class="n">sig</span><span class="p">,</span><span class="nf">as.character</span><span class="p">(</span><span class="n">ID</span><span class="p">),</span><span class="s">&#39;&#39;</span><span class="p">)),</span> <span class="n">color</span><span class="o">=</span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="n">hjust</span><span class="o">=</span><span class="m">0</span><span class="p">,</span><span class="n">vjust</span><span class="o">=</span><span class="m">-2</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">scale_x_continuous</span><span class="p">(</span><span class="n">label</span> <span class="o">=</span> <span class="n">axis_set</span><span class="o">$</span><span class="n">chr</span><span class="p">,</span> <span class="n">breaks</span> <span class="o">=</span> <span class="n">axis_set</span><span class="o">$</span><span class="n">center</span><span class="p">,</span> <span class="n">expand</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0.01</span><span class="p">,</span><span class="m">0.01</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">scale_y_continuous</span><span class="p">(</span><span class="n">expand</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0</span><span class="p">),</span> <span class="n">limits</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span> <span class="n">ylim</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span> <span class="o">=</span> <span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;#276FBF&quot;</span><span class="p">,</span> <span class="s">&quot;#183059&quot;</span><span class="p">),</span> <span class="nf">unique</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">axis_set</span><span class="o">$</span><span class="n">chr</span><span class="p">))))</span> <span class="o">+</span>
  <span class="nf">scale_size_continuous</span><span class="p">(</span><span class="n">range</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0.5</span><span class="p">,</span><span class="m">3</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">labs</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="kc">NULL</span><span class="p">,</span> 
       <span class="n">y</span> <span class="o">=</span> <span class="s">&quot;-log&lt;sub&gt;10&lt;/sub&gt;(p)&quot;</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">theme_minimal</span><span class="p">()</span> <span class="o">+</span>
  <span class="nf">theme</span><span class="p">(</span> 
    <span class="n">legend.position</span> <span class="o">=</span> <span class="s">&quot;none&quot;</span><span class="p">,</span>
    <span class="n">panel.grid.major.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
    <span class="n">panel.grid.minor.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
    <span class="n">axis.title.y</span> <span class="o">=</span> <span class="nf">element_markdown</span><span class="p">(),</span>
    <span class="n">axis.line</span> <span class="o">=</span> <span class="nf">element_line</span><span class="p">(</span><span class="n">colour</span> <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">,</span> <span class="n">linetype</span><span class="o">=</span><span class="m">1</span><span class="p">)</span>
  <span class="p">)</span>


<span class="n">manhplot</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">plot</span> <span class="o">=</span> <span class="n">manhplot</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">10</span><span class="p">,</span> <span class="n">height</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">,</span> <span class="n">filename</span> <span class="o">=</span> <span class="s">&quot;LBD_1percfdr_GQ300_TIDY_manhattan.pdf&quot;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Frontotemporal-dementia/amyotrophic-lateral-sclerosis-plot">Frontotemporal dementia/amyotrophic lateral sclerosis plot<a class="anchor-link" href="#Frontotemporal-dementia/amyotrophic-lateral-sclerosis-plot">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">#Code modified from https://danielroelfs.com/blog/how-i-create-manhattan-plots-using-ggplot/ </span>
<span class="nf">options</span><span class="p">(</span><span class="n">repr.plot.width</span><span class="o">=</span><span class="m">6</span><span class="p">)</span>
<span class="c1">#Plot LBD manhattan plot</span>

<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggtext</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="n">LBD</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;$PATH5/Analysis.GLM.hg38_minGQ300/FTD/FTD.controls.UNRELATED.SNVindels.SV-1perc-fdr-minGQ300.hg38.hwe1e-6.maf001cases.exclTelomereFlank100kb.exclCentromereFlank2.5Mb.exclVDJ.QUAL500.autosomesOnly.withVarFilterInfo.PASS.UNRESOLVED.QUAL500.TIDY.txt&quot;</span><span class="p">)</span>

<span class="n">sig</span> <span class="o">&lt;-</span> <span class="m">0.05</span><span class="o">/</span><span class="m">4699</span>

<span class="n">gwas_data</span> <span class="o">&lt;-</span> <span class="n">LBD</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">gwas_data</span><span class="p">)[</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="m">12</span><span class="p">)]</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span> <span class="s">&quot;bp&quot;</span><span class="p">,</span> <span class="s">&quot;p&quot;</span><span class="p">)</span>


<span class="n">data_cum</span> <span class="o">&lt;-</span> <span class="n">gwas_data</span> <span class="o">%&gt;%</span> 
  <span class="nf">group_by</span><span class="p">(</span><span class="n">chr</span><span class="p">)</span> <span class="o">%&gt;%</span> 
  <span class="nf">summarise</span><span class="p">(</span><span class="n">max_bp</span> <span class="o">=</span> <span class="nf">max</span><span class="p">(</span><span class="n">bp</span><span class="p">))</span> <span class="o">%&gt;%</span> 
  <span class="nf">mutate</span><span class="p">(</span><span class="n">bp_add</span> <span class="o">=</span> <span class="nf">lag</span><span class="p">(</span><span class="nf">cumsum</span><span class="p">(</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">max_bp</span><span class="p">)),</span> <span class="n">default</span> <span class="o">=</span> <span class="m">0</span><span class="p">))</span> <span class="o">%&gt;%</span> 
  <span class="nf">select</span><span class="p">(</span><span class="n">chr</span><span class="p">,</span> <span class="n">bp_add</span><span class="p">)</span>


<span class="n">gwas_data</span> <span class="o">&lt;-</span> <span class="n">gwas_data</span> <span class="o">%&gt;%</span> 
  <span class="nf">inner_join</span><span class="p">(</span><span class="n">data_cum</span><span class="p">,</span> <span class="n">by</span> <span class="o">=</span> <span class="s">&quot;chr&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span> 
  <span class="nf">mutate</span><span class="p">(</span><span class="n">bp_cum</span> <span class="o">=</span> <span class="n">bp</span> <span class="o">+</span> <span class="n">bp_add</span><span class="p">)</span>


<span class="n">axis_set</span> <span class="o">&lt;-</span> <span class="n">gwas_data</span> <span class="o">%&gt;%</span> 
  <span class="nf">group_by</span><span class="p">(</span><span class="n">chr</span><span class="p">)</span> <span class="o">%&gt;%</span> 
  <span class="nf">summarize</span><span class="p">(</span><span class="n">center</span> <span class="o">=</span> <span class="nf">mean</span><span class="p">(</span><span class="n">bp_cum</span><span class="p">))</span>

<span class="n">ylim</span> <span class="o">&lt;-</span> <span class="n">gwas_data</span> <span class="o">%&gt;%</span> 
  <span class="nf">filter</span><span class="p">(</span><span class="n">p</span> <span class="o">==</span> <span class="nf">min</span><span class="p">(</span><span class="n">p</span><span class="p">))</span> <span class="o">%&gt;%</span> 
  <span class="nf">mutate</span><span class="p">(</span><span class="n">ylim</span> <span class="o">=</span> <span class="nf">abs</span><span class="p">(</span><span class="nf">floor</span><span class="p">(</span><span class="nf">log10</span><span class="p">(</span><span class="n">p</span><span class="p">)))</span> <span class="o">+</span> <span class="m">2</span><span class="p">)</span> <span class="o">%&gt;%</span> 
  <span class="nf">pull</span><span class="p">(</span><span class="n">ylim</span><span class="p">)</span>



<span class="n">highlight_df</span> <span class="o">&lt;-</span> <span class="n">gwas_data</span> <span class="o">%&gt;%</span> 
             <span class="nf">filter</span><span class="p">(</span><span class="n">p</span><span class="o">&lt;=</span><span class="m">0.000012</span><span class="p">)</span>


<span class="n">manhplot</span> <span class="o">&lt;-</span> <span class="nf">ggplot</span><span class="p">(</span><span class="n">gwas_data</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">bp_cum</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="o">-</span><span class="nf">log10</span><span class="p">(</span><span class="n">p</span><span class="p">),</span> 
                                  <span class="n">color</span> <span class="o">=</span> <span class="nf">as_factor</span><span class="p">(</span><span class="n">chr</span><span class="p">)))</span> <span class="o">+</span>
  <span class="nf">geom_hline</span><span class="p">(</span><span class="n">yintercept</span> <span class="o">=</span> <span class="o">-</span><span class="nf">log10</span><span class="p">(</span><span class="n">sig</span><span class="p">),</span> <span class="n">color</span> <span class="o">=</span> <span class="s">&quot;grey40&quot;</span><span class="p">,</span> <span class="n">linetype</span> <span class="o">=</span> <span class="s">&quot;dashed&quot;</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">alpha</span> <span class="o">=</span> <span class="m">0.75</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">highlight_df</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">bp_cum</span><span class="p">,</span><span class="n">y</span><span class="o">=-</span><span class="nf">log10</span><span class="p">(</span><span class="n">p</span><span class="p">)),</span> <span class="n">color</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;red&#39;</span><span class="p">,</span><span class="s">&#39;red&#39;</span><span class="p">,</span><span class="s">&#39;gray&#39;</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">geom_text</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">label</span><span class="o">=</span><span class="nf">ifelse</span><span class="p">(</span><span class="n">p</span><span class="o">&lt;</span><span class="n">sig</span><span class="p">,</span><span class="nf">as.character</span><span class="p">(</span><span class="n">ID</span><span class="p">),</span><span class="s">&#39;&#39;</span><span class="p">)),</span> <span class="n">color</span><span class="o">=</span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="n">hjust</span><span class="o">=</span><span class="m">0.8</span><span class="p">,</span><span class="n">vjust</span><span class="o">=</span><span class="m">-2</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">scale_x_continuous</span><span class="p">(</span><span class="n">label</span> <span class="o">=</span> <span class="n">axis_set</span><span class="o">$</span><span class="n">chr</span><span class="p">,</span> <span class="n">breaks</span> <span class="o">=</span> <span class="n">axis_set</span><span class="o">$</span><span class="n">center</span><span class="p">,</span> <span class="n">expand</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0.01</span><span class="p">,</span><span class="m">0.01</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">scale_y_continuous</span><span class="p">(</span><span class="n">expand</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0</span><span class="p">),</span> <span class="n">limits</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span> <span class="n">ylim</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span> <span class="o">=</span> <span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;#276FBF&quot;</span><span class="p">,</span> <span class="s">&quot;#183059&quot;</span><span class="p">),</span> <span class="nf">unique</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">axis_set</span><span class="o">$</span><span class="n">chr</span><span class="p">))))</span> <span class="o">+</span>
  <span class="nf">scale_size_continuous</span><span class="p">(</span><span class="n">range</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0.5</span><span class="p">,</span><span class="m">3</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">labs</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="kc">NULL</span><span class="p">,</span> 
       <span class="n">y</span> <span class="o">=</span> <span class="s">&quot;-log&lt;sub&gt;10&lt;/sub&gt;(p)&quot;</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">theme_minimal</span><span class="p">()</span> <span class="o">+</span>
  <span class="nf">theme</span><span class="p">(</span> 
    <span class="n">legend.position</span> <span class="o">=</span> <span class="s">&quot;none&quot;</span><span class="p">,</span>
    <span class="n">panel.grid.major.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
    <span class="n">panel.grid.minor.x</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
    <span class="n">axis.title.y</span> <span class="o">=</span> <span class="nf">element_markdown</span><span class="p">(),</span>
    <span class="n">axis.line</span> <span class="o">=</span> <span class="nf">element_line</span><span class="p">(</span><span class="n">colour</span> <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">,</span> <span class="n">linetype</span><span class="o">=</span><span class="m">1</span><span class="p">)</span>
  <span class="p">)</span>


<span class="n">manhplot</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">plot</span> <span class="o">=</span> <span class="n">manhplot</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">10</span><span class="p">,</span> <span class="n">height</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">,</span> <span class="n">filename</span> <span class="o">=</span> <span class="s">&quot;FTD_1percfdr_GQ300_TIDY_manhattan.pdf&quot;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="95-%-condifence-intervals-for-genome-wide-association-study-hits">95 % condifence intervals for genome-wide association study hits<a class="anchor-link" href="#95-%-condifence-intervals-for-genome-wide-association-study-hits">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>95% CIs were not included in PLINK output, but they can be calculated with:<br>
lower = BETA - 1.96<em>SE<br>
upper = BETA + 1.96</em>SE</p>
<p>Since we want CIs for OR, we do e^lower and e^upper.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Extract SV ID, Beta and SE from GWAS top variants (also P and OR value to ease reporting) and calculate CIs</span>

awk <span class="s1">&#39;{print $3, $14, $15, $16, $17,2.7182818^($15-1.96*$16),2.7182818^($15+1.96*$16)}&#39;</span>  <span class="nv">$PATH5</span>/Analysis.GLM.hg38_minGQ300/FTD/FTD.controls.UNRELATED.SNVindels.SV-1perc-fdr-minGQ300.hg38.hwe1e-6.maf001cases.exclTelomereFlank100kb.exclCentromereFlank2.5Mb.exclVDJ.QUAL500.autosomesOnly.withVarFilterInfo.PASS.UNRESOLVED.QUAL500.FINAL.TIDY.txt &gt; <span class="nv">$PATH1</span>/NIH_FALL_2021_SVs_1percFDR/FTD_GWAS_hit_95CIs.txt
awk <span class="s1">&#39;{print $3, $14, $15, $16, $17,2.7182818^($15-1.96*$16),2.7182818^($15+1.96*$16)}&#39;</span>  <span class="nv">$PATH5</span>/Analysis.GLM.hg38_minGQ300/LBD/LBD.controls.UNRELATED.SNVindels.SV-1perc-fdr-minGQ300.hg38.hwe1e-6.maf001cases.exclTelomereFlank100kb.exclCentromereFlank2.5Mb.exclVDJ.QUAL500.autosomesOnly.withVarFilterInfo.PASS.UNRESOLVED.QUAL500.FINAL.TIDY.txt &gt; <span class="nv">$PATH1</span>/NIH_FALL_2021_SVs_1percFDR/LBD_all_GWAS_hit_95CIs.txt
awk <span class="s1">&#39;{print $3, $14, $15, $16, $17,2.7182818^($15-1.96*$16),2.7182818^($15+1.96*$16)}&#39;</span>  <span class="nv">$PATH5</span>/Analysis.GLM.hg38_minGQ300/LBD.path.controls1/LBDpath.controls.UNRELATED.SNVindels.SV-1perc-fdr-minGQ300.hg38.hwe1e-6.maf001cases.exclTelomereFlank100kb.exclCentromereFlank2.5Mb.exclVDJ.QUAL500.autosomesOnly.withVarFilterInfo.PASS.UNRESOLVED.QUAL500.FINAL.TIDY.txt &gt; <span class="nv">$PATH1</span>/NIH_FALL_2021_SVs_1percFDR/LBD_path_GWAS_hit_95CIs.txt
awk <span class="s1">&#39;{print $3, $14, $15, $16, $17,2.7182818^($15-1.96*$16),2.7182818^($15+1.96*$16)}&#39;</span>  <span class="nv">$PATH5</span>/Analysis.GLM.hg38_minGQ300/LBD.clin.controls2/LBDclin.controls.UNRELATED.SNVindels.SV-1perc-fdr-minGQ300.hg38.hwe1e-6.maf001cases.exclTelomereFlank100kb.exclCentromereFlank2.5Mb.exclVDJ.QUAL500.autosomesOnly.withVarFilterInfo.PASS.UNRESOLVED.QUAL500.FINAL.TIDY.txt &gt; <span class="nv">$PATH1</span>/NIH_FALL_2021_SVs_1percFDR/LBD_clin_GWAS_hit_95CIs.txt
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Grep top hits from the right file</span>
<span class="nb">echo</span> <span class="s2">&quot;LBD path&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;ID OR BETA SE P 95%CI_lower 95%CI_upper&quot;</span>
grep -we <span class="s1">&#39;LBD_DUP_chr7_2098&#39;</span> -we <span class="s1">&#39;LBD_DEL_chr18_4975&#39;</span> <span class="nv">$PATH1</span>/NIH_FALL_2021_SVs_1percFDR/LBD_path_GWAS_hit_95CIs.txt

<span class="nb">echo</span> -e <span class="s2">&quot;\n LBD clin&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;ID OR BETA SE P 95%CI_lower 95%CI_upper&quot;</span>
grep -we <span class="s1">&#39;LBD_DEL_chr21_374&#39;</span> -we <span class="s1">&#39;LBD_DEL_chr8_2722&#39;</span> <span class="nv">$PATH1</span>/NIH_FALL_2021_SVs_1percFDR/LBD_clin_GWAS_hit_95CIs.txt

<span class="nb">echo</span> -e <span class="s2">&quot;\n LBD all&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;ID OR BETA SE P 95%CI_lower 95%CI_upper&quot;</span>
grep -we <span class="s1">&#39;LBD_DUP_chr7_2098&#39;</span> -we <span class="s1">&#39;LBD_DEL_chr12_9172&#39;</span> -e <span class="s1">&#39;LBD_DEL_chr18_4975&#39;</span> -e <span class="s1">&#39;LBD_DEL_chr21_374&#39;</span> -e <span class="s1">&#39;LBD_DEL_chr8_2722&#39;</span> <span class="nv">$PATH1</span>/NIH_FALL_2021_SVs_1percFDR/LBD_all_GWAS_hit_95CIs.txt

<span class="nb">echo</span> -e <span class="s2">&quot;\n FTD all&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;ID OR BETA SE P 95%CI_lower 95%CI_upper&quot;</span>
grep -we <span class="s1">&#39;FTD_BND_chr9_581&#39;</span> -we <span class="s1">&#39;FTD_DEL_chr21_304&#39;</span> -we <span class="s1">&#39;FTD_CPX_chr17_77&#39;</span> <span class="nv">$PATH1</span>/NIH_FALL_2021_SVs_1percFDR/FTD_GWAS_hit_95CIs.txt
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="TPCN1-plots">TPCN1 plots<a class="anchor-link" href="#TPCN1-plots">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Locus-zoom">Locus zoom<a class="anchor-link" href="#Locus-zoom">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH3</span>/
mkdir TPCN1_locuszoom_prep
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Locuszoom takes LD file only in a specific format. Variant ID is chr:pos_ref/alt Data wrangle to that format</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/TPCN1_locuszoom_prep
module load plink/2.3-alpha

<span class="c1">##Extract variants from region of interest from clean(ish) files</span>
plink2 <span class="se">\</span>
--bfile <span class="nv">$PATH5</span>/LBD_minGQ300/SNVindels.SV.LBD.UNRELATED.merged.chr12 <span class="se">\</span>
--keep-allele-order <span class="se">\</span>
--chr <span class="m">12</span> <span class="se">\</span>
--from-bp <span class="m">113045316</span> <span class="se">\</span>
--to-bp <span class="m">113445316</span> <span class="se">\</span>
--make-bed <span class="se">\</span>
--out tmp_TPCN1_locus

<span class="c1">##Make file with new IDs</span>
awk <span class="s1">&#39;{print $1&quot;:&quot;$4&quot;_&quot;$6&quot;/&quot;$5,$2}&#39;</span> tmp_TPCN1_locus.bim &gt; tmp_TPCN1_locus_SV_newnames.txt

<span class="c1">##Update IDs</span>
plink2 --bfile tmp_TPCN1_locus --update-name tmp_TPCN1_locus_SV_newnames.txt <span class="m">1</span> <span class="m">2</span> --make-bed --out tmp_TPCN1_locus_newnames
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Calculate LD and format according to locuszooms instructions: R2</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/TPCN1_locuszoom_prep
module load plink/1.9

plink <span class="se">\</span>
--r2 <span class="se">\</span>
--bfile tmp_TPCN1_locus_newnames <span class="se">\</span>
--ld-window <span class="m">499999</span> <span class="se">\</span>
--ld-window-kb <span class="m">500</span> <span class="se">\</span>
--ld-window-r2 <span class="m">0</span>.0 <span class="se">\</span>
--ld-snp <span class="s2">&quot;12:113245316_N/&lt;DEL&gt;&quot;</span>

cat plink.ld <span class="p">|</span> tail -n+2 <span class="p">|</span> sed <span class="s1">&#39;s/^[[:space:]]*//g&#39;</span> <span class="p">|</span> sed <span class="s1">&#39;s/[[:space:]]*$//g&#39;</span> <span class="p">|</span> tr -s <span class="s1">&#39; &#39;</span> <span class="s1">&#39;\t&#39;</span> <span class="p">|</span> sort -k4,4 -k5,5n <span class="p">|</span> bgzip &gt; plink.ld.tab.gz <span class="o">&amp;&amp;</span> tabix -s4 -b5 -e5 plink.ld.tab.gz
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Calculate LD and format according to locuszooms instructions: D&#39;</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/TPCN1_locuszoom_prep
module load plink/1.9
module load samtools

plink <span class="se">\</span>
--r2 dprime <span class="se">\</span>
--bfile tmp_TPCN1_locus_newnames <span class="se">\</span>
--ld-window <span class="m">499999</span> <span class="se">\</span>
--ld-window-kb <span class="m">500</span> <span class="se">\</span>
--ld-window-r2 <span class="m">0</span>.0 <span class="se">\</span>
--ld-snp <span class="s2">&quot;12:113245316_N/&lt;DEL&gt;&quot;</span> <span class="se">\</span>
--out dprime_ld

<span class="c1">##Format to make locuszoom compatible, add step to cut R2 column</span>

cat dprime_ld.ld <span class="p">|</span> tail -n+2 <span class="p">|</span> sed <span class="s1">&#39;s/^[[:space:]]*//g&#39;</span> <span class="p">|</span> sed <span class="s1">&#39;s/[[:space:]]*$//g&#39;</span> <span class="p">|</span> tr -s <span class="s1">&#39; &#39;</span> <span class="s1">&#39;\t&#39;</span> <span class="p">|</span> sort -k4,4 -k5,5n <span class="p">|</span> cut -f1,2,3,4,5,6,8 <span class="p">|</span> bgzip &gt; dprime_ld.ld.tab.gz <span class="o">&amp;&amp;</span> tabix -s4 -b5 -e5 dprime_ld.ld.tab.gz
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">##Run glm again for this region only to get result file</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/TPCN1_locuszoom_prep
module load plink/2.3-alpha

plink2 <span class="se">\</span>
--bfile tmp_TPCN1_locus_newnames <span class="se">\</span>
--covar <span class="nv">$PATH5</span>/COVARIATES.SNVindels.SV.freeze9.LBD.controls.UNRELATED_minGQ300.txt <span class="se">\</span>
--covar-name GENDER,CONSENSUS_AGE,PC1,PC3,PC4,PC7,PC8 <span class="se">\</span>
--covar-variance-standardize <span class="se">\</span>
--glm hide-covar firth-fallback <span class="nv">cols</span><span class="o">=</span>+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err <span class="se">\</span>
--out tmp_TPCN1_region_glmrerun <span class="se">\</span>
--pheno <span class="nv">$PATH5</span>/COVARIATES.SNVindels.SV.freeze9.LBD.controls.UNRELATED_minGQ300.txt <span class="se">\</span>
--pheno-name AFFECTION_STATUS
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Beta-beta-plots">Beta-beta plots<a class="anchor-link" href="#Beta-beta-plots">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Beta-beta-plot-of-Belenguez-Alzheimer's-disease-TPCN1-and-our-TPCN1-loci">Beta-beta plot of Belenguez Alzheimer's disease TPCN1 and our TPCN1 loci<a class="anchor-link" href="#Beta-beta-plot-of-Belenguez-Alzheimer's-disease-TPCN1-and-our-TPCN1-loci">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/beta_beta_plots

<span class="c1">##Download Belenguez harmonized summary statistics</span>
wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST90027001-GCST90028000/GCST90027158/harmonised/35379992-GCST90027158-MONDO_0004975.h.tsv.gz
gunzip  <span class="m">35379992</span>-GCST90027158-MONDO_0004975.h.tsv.gz
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/beta_beta_plots

<span class="c1">#Download Chia et al harmonized summary statistics</span>
wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST90001001-GCST90002000/GCST90001390/harmonised/33589841-GCST90001390-EFO_0006792.h.tsv.gz
gunzip <span class="m">33589841</span>-GCST90001390-EFO_0006792.h.tsv.gz
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Extract analyzed SNPS/indels from the TPCN1 LD block (region from locuszoom plot) from LBD summary stats</span>
<span class="nb">cd</span> <span class="nv">$PATH2</span>/beta_beta_plots

<span class="c1">##Extract TPCN1 haplotype region</span>
awk <span class="s1">&#39;($3==&quot;hm_chrom&quot; || ($3==12 &amp;&amp; $4&gt;113150000 &amp;&amp; $4&lt;113350000))&#39;</span> <span class="m">33589841</span>-GCST90001390-EFO_0006792.h.tsv &gt; TPCN1_analyzed_snps_indels_glm.txt

<span class="c1">##Get IDs</span>
awk <span class="s1">&#39;(NR&gt;1) {print $1}&#39;</span> TPCN1_analyzed_snps_indels_glm.txt &gt; TPCN1_analyzed_snps_indels.txt
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Grep TPCN1 haplotype SNPs from harmonized Belenguez summary stats</span>
<span class="nb">cd</span> <span class="nv">$PATH2</span>/beta_beta_plots

head -n1 <span class="m">35379992</span>-GCST90027158-MONDO_0004975.h.tsv &gt; <span class="m">35379992</span>-GCST90027158-MONDO_0004975.h._TPCN1_SNPs.tsv
grep -Ff TPCN1_analyzed_snps_indels.txt <span class="m">35379992</span>-GCST90027158-MONDO_0004975.h.tsv &gt;&gt; <span class="m">35379992</span>-GCST90027158-MONDO_0004975.h._TPCN1_SNPs.tsv

<span class="c1">##Check how many were found</span>
<span class="nb">echo</span> <span class="s2">&quot;number of searched SNPs:&quot;</span>
wc -l TPCN1_analyzed_snps_indels.txt
<span class="nb">echo</span> <span class="s2">&quot;Number of found SNPs&quot;</span>
wc -l <span class="m">35379992</span>-GCST90027158-MONDO_0004975.h._TPCN1_SNPs.tsv
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%</span><span class="k">load_ext</span> rpy2.ipython
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">#####beta-beta plot TPCN1 us vs belenguez</span>
<span class="c1">##Set wd</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;$PATH2/beta_beta_plots&quot;</span><span class="p">)</span>

<span class="c1">##Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">scales</span><span class="p">)</span>

<span class="c1">##Import data</span>
<span class="n">our</span> <span class="o">&lt;-</span> <span class="nf">read.delim</span><span class="p">(</span><span class="s">&quot;TPCN1_analyzed_snps_indels_glm.txt&quot;</span><span class="p">)</span>
<span class="n">belenguez</span> <span class="o">&lt;-</span> <span class="nf">read.delim</span><span class="p">(</span><span class="s">&quot;35379992-GCST90027158-MONDO_0004975.h._TPCN1_SNPs.tsv&quot;</span><span class="p">)</span>

<span class="c1">##Merge dfs by SNP ID</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">our</span><span class="p">,</span> <span class="n">belenguez</span><span class="p">,</span> <span class="n">by</span><span class="o">=</span><span class="s">&quot;hm_variant_id&quot;</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="s">&quot;union&quot;</span><span class="p">,</span> <span class="n">suffixes</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;_LBD&quot;</span><span class="p">,</span> <span class="s">&quot;_AD&quot;</span><span class="p">))</span>

<span class="nf">print</span><span class="p">(</span><span class="nf">head</span><span class="p">(</span><span class="n">df</span><span class="p">))</span>
<span class="c1">##Harmonize allele for which effect was tested tested between our and Belenguez data, if all effect alleles are not the same</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">hm_effect_allele_LBD</span> <span class="o">==</span> <span class="n">df</span><span class="o">$</span><span class="n">hm_effect_allele_AD</span><span class="p">))</span>

<span class="c1">##Create variable that shows only suggestive p value variants</span>
<span class="n">df</span><span class="o">$</span><span class="n">suggestive</span> <span class="o">&lt;-</span> <span class="nf">ifelse</span><span class="p">((</span><span class="n">df</span><span class="o">$</span><span class="n">p_value_LBD</span> <span class="o">&lt;</span> <span class="m">0.00001</span> <span class="o">|</span> <span class="n">df</span><span class="o">$</span><span class="n">p_value_AD</span> <span class="o">&lt;</span> <span class="m">0.00001</span><span class="p">),</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="kc">FALSE</span><span class="p">)</span>

<span class="c1">##Subset only variants that show suggestive association to either trait</span>
<span class="n">df2</span> <span class="o">&lt;-</span> <span class="n">df</span> <span class="o">%&gt;%</span>
  <span class="nf">filter</span><span class="p">(</span><span class="n">p_value_LBD</span> <span class="o">&lt;</span> <span class="m">0.005</span> <span class="o">|</span> <span class="n">p_value_AD</span> <span class="o">&lt;</span> <span class="m">0.005</span><span class="p">)</span>

<span class="n">df3</span> <span class="o">&lt;-</span> <span class="n">df</span> <span class="o">%&gt;%</span>
  <span class="nf">filter</span><span class="p">(</span><span class="n">p_value_LBD</span> <span class="o">&lt;</span> <span class="m">0.00001</span> <span class="o">|</span> <span class="n">p_value_AD</span> <span class="o">&lt;</span><span class="m">0.00001</span><span class="p">)</span>

<span class="c1">##Correlation tests</span>
<span class="n">cor1</span> <span class="o">&lt;-</span> <span class="nf">cor.test</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">hm_beta_LBD</span><span class="p">,</span> <span class="n">df</span><span class="o">$</span><span class="n">hm_beta_AD</span><span class="p">,</span> 
         <span class="n">method</span> <span class="o">=</span> <span class="s">&quot;pearson&quot;</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">cor1</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">cor1</span><span class="o">$</span><span class="n">p.value</span><span class="p">)</span>

<span class="n">cor2</span> <span class="o">&lt;-</span> <span class="nf">cor.test</span><span class="p">(</span><span class="n">df2</span><span class="o">$</span><span class="n">hm_beta_LBD</span><span class="p">,</span> <span class="n">df2</span><span class="o">$</span><span class="n">hm_beta_AD</span><span class="p">,</span> 
         <span class="n">method</span> <span class="o">=</span> <span class="s">&quot;pearson&quot;</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">cor2</span><span class="p">)</span>

<span class="n">cor3</span> <span class="o">&lt;-</span> <span class="nf">cor.test</span><span class="p">(</span><span class="n">df3</span><span class="o">$</span><span class="n">hm_beta_LBD</span><span class="p">,</span> <span class="n">df3</span><span class="o">$</span><span class="n">hm_beta_AD</span><span class="p">,</span> 
         <span class="n">method</span> <span class="o">=</span> <span class="s">&quot;pearson&quot;</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">cor3</span><span class="p">)</span>

<span class="c1">##Fancier plot (modifier Cornelis&#39;s code)</span>
<span class="c1">###Plot unfiltered</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">hm_beta_AD</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">hm_beta_LBD</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="n">suggestive</span><span class="p">))</span> <span class="o">+</span> 
  <span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#39;black&#39;</span><span class="p">,</span> <span class="s">&#39;red&#39;</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">theme_bw</span><span class="p">()</span> <span class="o">+</span> 
  <span class="nf">theme</span><span class="p">(</span><span class="n">legend.position</span><span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">geom_smooth</span><span class="p">(</span><span class="n">se</span> <span class="o">=</span> <span class="bp">T</span><span class="p">,</span> <span class="n">method</span> <span class="o">=</span> <span class="n">lm</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span> <span class="o">=</span> <span class="m">0</span><span class="p">,</span> <span class="n">linetype</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">geom_hline</span><span class="p">(</span><span class="n">yintercept</span> <span class="o">=</span> <span class="m">0</span><span class="p">,</span> <span class="n">linetype</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;TPCN1 locus beta in AD&quot;</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;TPCN1 locus beta in LBD&quot;</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">ylim</span><span class="p">(</span><span class="m">-0.4</span><span class="p">,</span><span class="m">0.4</span><span class="p">))</span>
      
<span class="c1">#if want to add annotation annotate(&quot;text&quot;, size = 3, x=-0.1, y=0.35, label= paste(&quot;cor =&quot;, round(cor1$estimate, 2), &quot;\n&quot;, &quot;p = &quot;, scientific(cor1$p.value, 2))))</span>

<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span> <span class="o">=</span> <span class="s">&quot;beta_beta_plot_TPCN1_LD_region.pdf&quot;</span><span class="p">,</span> <span class="n">plot</span> <span class="o">=</span> <span class="nf">last_plot</span><span class="p">(),</span> <span class="n">device</span> <span class="o">=</span> <span class="s">&quot;pdf&quot;</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">3</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="m">4.4</span><span class="p">,</span> <span class="n">units</span> <span class="o">=</span> <span class="s">&quot;in&quot;</span> <span class="p">)</span>


<span class="c1">###Plot p value filtered</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">df2</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">hm_beta_AD</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">hm_beta_LBD</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="n">suggestive</span><span class="p">))</span> <span class="o">+</span> 
  <span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#39;black&#39;</span><span class="p">,</span> <span class="s">&#39;red&#39;</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">theme_bw</span><span class="p">()</span> <span class="o">+</span> 
  <span class="nf">theme</span><span class="p">(</span><span class="n">legend.position</span><span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">geom_smooth</span><span class="p">(</span><span class="n">se</span> <span class="o">=</span> <span class="bp">T</span><span class="p">,</span> <span class="n">method</span> <span class="o">=</span> <span class="n">lm</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span> <span class="o">=</span> <span class="m">0</span><span class="p">,</span> <span class="n">linetype</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">geom_hline</span><span class="p">(</span><span class="n">yintercept</span> <span class="o">=</span> <span class="m">0</span><span class="p">,</span> <span class="n">linetype</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;beta in Belenguez et. al. GWAS&quot;</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;beta in this study&quot;</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">annotate</span><span class="p">(</span><span class="s">&quot;text&quot;</span><span class="p">,</span> <span class="n">size</span> <span class="o">=</span> <span class="m">8</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="m">-0.1</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="m">0.3</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span> <span class="nf">paste</span><span class="p">(</span><span class="s">&quot;cor =&quot;</span><span class="p">,</span> <span class="nf">round</span><span class="p">(</span><span class="n">cor2</span><span class="o">$</span><span class="n">estimate</span><span class="p">,</span> <span class="m">2</span><span class="p">),</span> <span class="s">&quot;,&quot;</span><span class="p">,</span> <span class="s">&quot;p = &quot;</span><span class="p">,</span> <span class="nf">scientific</span><span class="p">(</span><span class="n">cor2</span><span class="o">$</span><span class="n">p.value</span><span class="p">,</span> <span class="m">2</span><span class="p">))))</span>

<span class="c1">###Plot p value filtered2</span>
<span class="nf">ggplot</span><span class="p">(</span><span class="n">df3</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">hm_beta_AD</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">beta_LBD</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="n">suggestive</span><span class="p">))</span> <span class="o">+</span> 
  <span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#39;black&#39;</span><span class="p">,</span> <span class="s">&#39;red&#39;</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">theme_bw</span><span class="p">()</span> <span class="o">+</span> 
  <span class="nf">theme</span><span class="p">(</span><span class="n">legend.position</span><span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">geom_smooth</span><span class="p">(</span><span class="n">se</span> <span class="o">=</span> <span class="bp">T</span><span class="p">,</span> <span class="n">method</span> <span class="o">=</span> <span class="n">lm</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span> <span class="o">=</span> <span class="m">0</span><span class="p">,</span> <span class="n">linetype</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">geom_hline</span><span class="p">(</span><span class="n">yintercept</span> <span class="o">=</span> <span class="m">0</span><span class="p">,</span> <span class="n">linetype</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;beta in Belenguez et. al. GWAS&quot;</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;beta in this study&quot;</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">annotate</span><span class="p">(</span><span class="s">&quot;text&quot;</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="m">-0.05</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="m">0.5</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span> <span class="nf">paste</span><span class="p">(</span><span class="s">&quot;cor =&quot;</span><span class="p">,</span> <span class="nf">round</span><span class="p">(</span><span class="n">cor3</span><span class="o">$</span><span class="n">estimate</span><span class="p">,</span> <span class="m">2</span><span class="p">),</span> <span class="s">&quot;,&quot;</span><span class="p">,</span> <span class="s">&quot;p = &quot;</span><span class="p">,</span> <span class="nf">scientific</span><span class="p">(</span><span class="n">cor3</span><span class="o">$</span><span class="n">p.value</span><span class="p">,</span> <span class="m">2</span><span class="p">)))</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Beta-beta-plot-of-Lewy-body-dementias-and-Parkinsons'-disease">Beta-beta plot of Lewy body dementias and Parkinsons' disease<a class="anchor-link" href="#Beta-beta-plot-of-Lewy-body-dementias-and-Parkinsons'-disease">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Copy Nalls et al PD symmary stats (without 23andme) from Ruth</span>
<span class="nb">cd</span> /<span class="nv">$PATH2</span>/beta_beta_plots
cp <span class="nv">$PATH9</span>/GWAS_LBD.v2/FIGURES_PUBLICATION/AdditionalInfo/LBDoverlap.AD.PD.GWAS/nallsEtAl2019_excluding23andMe_allVariants.TIDY.hg38.txt .

<span class="c1">#Extract only TPCN1 LD block</span>
awk <span class="s1">&#39;(NR==1 || ($1 == 12 &amp;&amp; $10 &gt; 113150000 &amp;&amp; $10 &lt; 113350000))&#39;</span> nallsEtAl2019_excluding23andMe_allVariants.TIDY.hg38.txt &gt; nallsEtAl2019_excluding23andMe_TPCN1_LD_block.TIDY.hg38.txt
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">#####beta-beta plot TPCN1 us vs nalls</span>
<span class="c1">#####Talked with Raph, he suggested either defining the locus or using +-1Mb of the top variant. He also said he often includes only variants that are suggestive in either data set</span>
<span class="c1">#####We have defined the region with locuszoom, so use that region. </span>
<span class="c1">##Set wd</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;$PATH2/beta_beta_plots&quot;</span><span class="p">)</span>

<span class="c1">##Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">scales</span><span class="p">)</span>

<span class="c1">#Import files</span>
<span class="n">LBD</span> <span class="o">&lt;-</span> <span class="nf">read.delim</span><span class="p">(</span><span class="s">&quot;TPCN1_analyzed_snps_indels_glm.txt&quot;</span><span class="p">)</span>
<span class="n">PD</span> <span class="o">&lt;-</span> <span class="nf">read.delim</span><span class="p">(</span><span class="s">&quot;nallsEtAl2019_excluding23andMe_TPCN1_LD_block.TIDY.hg38.txt&quot;</span><span class="p">)</span>

<span class="c1">#Merge</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">LBD</span><span class="p">,</span> <span class="n">PD</span><span class="p">,</span> <span class="n">by.x</span> <span class="o">=</span> <span class="s">&quot;hm_pos&quot;</span><span class="p">,</span> <span class="n">by.y</span><span class="o">=</span><span class="s">&quot;POS_hg38&quot;</span><span class="p">,</span> <span class="n">how</span> <span class="o">=</span> <span class="n">union</span><span class="p">)</span>

<span class="c1">#check effect allele matching</span>

<span class="nf">print</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">hm_effect_allele</span> <span class="o">==</span> <span class="n">df</span><span class="o">$</span><span class="n">EffectAllele</span><span class="p">))</span>

<span class="c1">#Exclude those with different effect allele since only 6 of them</span>
<span class="n">match</span> <span class="o">&lt;-</span> <span class="n">df</span><span class="o">$</span><span class="n">hm_effect_allele</span> <span class="o">==</span> <span class="n">df</span><span class="o">$</span><span class="n">EffectAllele</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="n">df</span><span class="p">[</span><span class="n">match</span><span class="p">,]</span>

<span class="c1">#check effect allele matching v2</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">hm_effect_allele</span> <span class="o">==</span> <span class="n">df</span><span class="o">$</span><span class="n">EffectAllele</span><span class="p">))</span>

<span class="c1">#calculate correlation</span>

<span class="n">cor1</span> <span class="o">&lt;-</span> <span class="nf">cor.test</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">hm_beta</span><span class="p">,</span> <span class="n">df</span><span class="o">$</span><span class="n">PD.beta</span><span class="p">,</span> <span class="n">method</span> <span class="o">=</span> <span class="s">&quot;pearson&quot;</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">cor1</span><span class="p">)</span>

<span class="c1">#Plot</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">PD.beta</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">hm_beta</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">geom_point</span><span class="p">()</span> <span class="o">+</span>
        <span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#39;black&#39;</span><span class="p">,</span> <span class="s">&#39;red&#39;</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">theme_bw</span><span class="p">()</span> <span class="o">+</span> 
        <span class="nf">theme</span><span class="p">(</span><span class="n">legend.position</span><span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">geom_smooth</span><span class="p">(</span><span class="n">se</span> <span class="o">=</span> <span class="bp">T</span><span class="p">,</span> <span class="n">method</span> <span class="o">=</span> <span class="n">lm</span><span class="p">)</span> <span class="o">+</span> 
        <span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span> <span class="o">=</span> <span class="m">0</span><span class="p">,</span> <span class="n">linetype</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span> <span class="o">+</span> 
        <span class="nf">geom_hline</span><span class="p">(</span><span class="n">yintercept</span> <span class="o">=</span> <span class="m">0</span><span class="p">,</span> <span class="n">linetype</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;TPCN1 locus beta in PD&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;TPCN1 locus beta in LBD&quot;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">ylim</span><span class="p">(</span><span class="m">-0.4</span><span class="p">,</span><span class="m">0.4</span><span class="p">))</span>

<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span> <span class="o">=</span> <span class="s">&quot;beta_beta_plot_TPCN1_LD_region_PD.pdf&quot;</span><span class="p">,</span> <span class="n">plot</span> <span class="o">=</span> <span class="nf">last_plot</span><span class="p">(),</span> <span class="n">device</span> <span class="o">=</span> <span class="s">&quot;pdf&quot;</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">3</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="m">4.4</span><span class="p">,</span> <span class="n">units</span> <span class="o">=</span> <span class="s">&quot;in&quot;</span> <span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Cohort-demographics">Cohort demographics<a class="anchor-link" href="#Cohort-demographics">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%</span><span class="k">load_ext</span> rpy2.ipython
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">###LBD GATK-SV cohort</span>
<span class="c1">#Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">gmodels</span><span class="p">)</span>

<span class="c1">#Import files</span>
<span class="n">LBD_analyzed_samples</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;$PATH2/clean_high_quality_vcfs/tmp_LBD_analyzed_samples.txt&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="n">LBD_covars</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;$PATH5/COVARIATES.SNVindels.SV.freeze9.LBD.controls.UNRELATED_minGQ300.txt&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>

<span class="c1">#Subset covar file to include only analyzed samples</span>
<span class="n">LBD_covars</span> <span class="o">&lt;-</span> <span class="n">LBD_covars</span> <span class="o">%&gt;%</span>
    <span class="nf">filter</span><span class="p">(</span><span class="n">IID</span> <span class="o">%in%</span> <span class="n">LBD_analyzed_samples</span><span class="o">$</span><span class="n">V1</span><span class="p">)</span>

<span class="c1">#Get sex proportions per case-control group</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">CrossTable</span><span class="p">(</span><span class="n">LBD_covars</span><span class="o">$</span><span class="n">AFFECTION_STATUS</span><span class="p">,</span><span class="n">LBD_covars</span><span class="o">$</span><span class="n">SEX</span><span class="p">,</span> <span class="n">prop.r</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> <span class="n">prop.c</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span> <span class="n">prop.t</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span> <span class="n">prop.chisq</span><span class="o">=</span><span class="bp">F</span><span class="p">))</span>

<span class="c1">#Get age summaries per phenotype</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">summary</span><span class="p">(</span><span class="nf">subset</span><span class="p">(</span><span class="n">LBD_covars</span><span class="p">,</span> <span class="n">AFFECTION_STATUS</span> <span class="o">==</span> <span class="m">1</span><span class="p">,</span> <span class="n">CONSENSUS_AGE</span><span class="p">)))</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">summary</span><span class="p">(</span><span class="nf">subset</span><span class="p">(</span><span class="n">LBD_covars</span><span class="p">,</span> <span class="n">AFFECTION_STATUS</span> <span class="o">==</span> <span class="m">2</span><span class="p">,</span> <span class="n">CONSENSUS_AGE</span><span class="p">)))</span>

<span class="c1">#Get clin vs path dg summaries</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">LBD_covars</span><span class="o">$</span><span class="n">AFFECTION_STATUS</span><span class="p">,</span> <span class="n">LBD_covars</span><span class="o">$</span><span class="n">DIAGNOSIS_TYPE</span><span class="p">,</span> <span class="n">useNA</span><span class="o">=</span><span class="s">&quot;ifany&quot;</span><span class="p">))</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">###FTD GATK-SV cohort</span>
<span class="c1">#Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">gmodels</span><span class="p">)</span>

<span class="c1">#Import files</span>
<span class="n">FTD_analyzed_samples</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;$PATH2/clean_high_quality_vcfs/tmp_FTD_analyzed_samples.txt&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="n">FTD_covars</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;$PATH5/COVARIATES.SNVindels.SV.freeze9.FTD.controls.UNRELATED_minGQ300.txt&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>

<span class="c1">#Subset covar file to include only analyzed samples</span>
<span class="n">FTD_covars</span> <span class="o">&lt;-</span> <span class="n">FTD_covars</span> <span class="o">%&gt;%</span>
    <span class="nf">filter</span><span class="p">(</span><span class="n">IID</span> <span class="o">%in%</span> <span class="n">FTD_analyzed_samples</span><span class="o">$</span><span class="n">V1</span><span class="p">)</span>

<span class="c1">#Get sex proportions per case-control group</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">CrossTable</span><span class="p">(</span><span class="n">FTD_covars</span><span class="o">$</span><span class="n">AFFECTION_STATUS</span><span class="p">,</span><span class="n">FTD_covars</span><span class="o">$</span><span class="n">SEX</span><span class="p">,</span> <span class="n">prop.r</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> <span class="n">prop.c</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span> <span class="n">prop.t</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span> <span class="n">prop.chisq</span><span class="o">=</span><span class="bp">F</span><span class="p">))</span>

<span class="c1">#Get age summaries per phenotype</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">summary</span><span class="p">(</span><span class="nf">subset</span><span class="p">(</span><span class="n">FTD_covars</span><span class="p">,</span> <span class="n">AFFECTION_STATUS</span> <span class="o">==</span> <span class="m">1</span><span class="p">,</span> <span class="n">CONSENSUS_AGE</span><span class="p">)))</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">summary</span><span class="p">(</span><span class="nf">subset</span><span class="p">(</span><span class="n">FTD_covars</span><span class="p">,</span> <span class="n">AFFECTION_STATUS</span> <span class="o">==</span> <span class="m">2</span><span class="p">,</span> <span class="n">CONSENSUS_AGE</span><span class="p">)))</span>

<span class="c1">#Get clin vs path dg summaries</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">FTD_covars</span><span class="o">$</span><span class="n">AFFECTION_STATUS</span><span class="p">,</span> <span class="n">FTD_covars</span><span class="o">$</span><span class="n">DIAGNOSIS_TYPE</span><span class="p">,</span> <span class="n">useNA</span><span class="o">=</span><span class="s">&quot;ifany&quot;</span><span class="p">))</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">###LBD replication (BBSeq and Demseq2) cohort</span>
<span class="c1">#Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">gmodels</span><span class="p">)</span>

<span class="c1">#Import files</span>
<span class="n">LBD_analyzed_samples</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;$PATH3/replication_TPCN1_BBSeq_Demseq2/tmp_analyzed_samples.txt&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="n">LBD_age_sex</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;$PATH3/replication_TPCN1_BBSeq_Demseq2/BBSeq_Demseq2_covariates.txt&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="n">LBD_pheno</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;pheno_LBD_cases.txt&quot;</span><span class="p">)</span>

<span class="c1">#Subset covar file to include only analyzed samples</span>
<span class="n">LBD_pheno</span> <span class="o">&lt;-</span> <span class="n">LBD_pheno</span> <span class="o">%&gt;%</span>
    <span class="nf">filter</span><span class="p">(</span><span class="n">IID</span> <span class="o">%in%</span> <span class="n">LBD_analyzed_samples</span><span class="o">$</span><span class="n">V1</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">dim</span><span class="p">(</span><span class="n">LBD_pheno</span><span class="p">))</span>
<span class="n">LBD_age_sex</span> <span class="o">&lt;-</span> <span class="n">LBD_age_sex</span> <span class="o">%&gt;%</span>
    <span class="nf">filter</span><span class="p">(</span><span class="n">IID</span> <span class="o">%in%</span> <span class="n">LBD_analyzed_samples</span><span class="o">$</span><span class="n">V1</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">dim</span><span class="p">(</span><span class="n">LBD_age_sex</span><span class="p">))</span>

<span class="c1">#Merge LBD_pheno and LBD covar since ~23 samples have wrong phenotype in covar file</span>
<span class="n">df</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">LBD_pheno</span><span class="p">,</span> <span class="n">LBD_age_sex</span><span class="p">,</span> <span class="n">by</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;IID&quot;</span><span class="p">,</span><span class="s">&quot;FID&quot;</span><span class="p">))</span>

<span class="c1">#Get sex proportions per case-control group</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">CrossTable</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">LBD</span><span class="p">,</span><span class="n">df</span><span class="o">$</span><span class="n">SEX</span><span class="p">,</span> <span class="n">prop.r</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> <span class="n">prop.c</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span> <span class="n">prop.t</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span> <span class="n">prop.chisq</span><span class="o">=</span><span class="bp">F</span><span class="p">))</span>

<span class="c1">#Get age summaries per phenotype</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">summary</span><span class="p">(</span><span class="nf">subset</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">LBD</span> <span class="o">==</span> <span class="m">1</span><span class="p">,</span> <span class="n">Age</span><span class="p">)))</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">summary</span><span class="p">(</span><span class="nf">subset</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">LBD</span> <span class="o">==</span> <span class="m">2</span><span class="p">,</span> <span class="n">Age</span><span class="p">)))</span>

<span class="c1">#Get clin vs path dg summaries</span>
<span class="c1">#print(table(df$LBD, df$DIAGNOSIS_TYPE, useNA=&quot;ifany&quot;))</span>

<span class="nf">head</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Get numbers of path vs clin cases in LBD replication cohort</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/replication_TPCN1_BBSeq_Demseq2
grep -wFf tmp_analyzed_samples.txt pheno_LBD_cases.txt <span class="p">|</span> awk <span class="s1">&#39;($3==2 &amp;&amp; $1 ~/BBS/) {print $1}&#39;</span> <span class="p">|</span> wc -l
grep -wFf tmp_analyzed_samples.txt pheno_LBD_cases.txt <span class="p">|</span> awk <span class="s1">&#39;($3==2 &amp;&amp; $1 ~/RES/) {print $1}&#39;</span> &gt; tmp_analyzed_demseq_cases.txt

grep -wFf tmp_analyzed_demseq_cases.txt <span class="nv">$PATH10</span>/LBD.FTD/Phenotype.BBS_DementiaSeq_v2_harmonized.txt <span class="p">|</span> grep -i <span class="s1">&#39;Pathological&#39;</span> <span class="p">|</span> wc -l
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>All BBSeq samples are neuropathological, so the pathological samples are 58+352/555 = 73.5%</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Bigbed-files-for-UCSC/Ensembl">Bigbed files for UCSC/Ensembl<a class="anchor-link" href="#Bigbed-files-for-UCSC/Ensembl">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Make bigbed files for LBD cases, LBD controls, FTD cases, FTD controls</span>
<span class="nb">cd</span> <span class="nv">$PATH3</span>/ucsc_bigbed_track
module load bcftools
module load ucsc

<span class="c1">##Fetch chrom sizes (UCSC binary)</span>
fetchChromSizes hg38 &gt; hg38.chrom.sizes

<span class="c1">##Autosql file is same for all bigbeds, file template from UCSC</span>
cat fieldnames.as

<span class="c1">##Create bigbeds</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
  <span class="k">for</span> STATUS <span class="k">in</span> cases controls<span class="p">;</span> <span class="k">do</span>
        bcftools query -f <span class="s1">&#39;%CHROM\t%POS\t%END\t&quot;INFO&quot;\t%QUAL\t&quot;.&quot;\t%POS\t%END\t&quot;RGB&quot;\t%AF\t%SVLEN\t%SVTYPE\t%ID\n&#39;</span> <span class="nv">$PATH2</span>/clean_high_quality_vcfs/<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz &gt; tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>_analyzed.bed <span class="c1">#Extract info per variant</span>
        sed -i <span class="s1">&#39;s/&quot;//g&#39;</span> tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>_analyzed.bed <span class="c1">#Remove quote marks</span>
        awk <span class="s1">&#39;($10 &gt; 0)&#39;</span> tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>_analyzed.bed &gt; tmp.txt <span class="c1">#Filter out variants with AF = 0</span>
        awk <span class="s1">&#39;{if($12==&quot;INS&quot;) {$3=$2 ; $8=$2; print} else{print $0}}&#39;</span> tmp.txt &gt; tmp2.txt <span class="c1">#If SVTYPE is INS, make END=POS</span>
        awk <span class="s1">&#39;{if($12==&quot;INS&quot;) {$11=1; print} else{print $0}}&#39;</span> tmp2.txt &gt; tmp3.txt <span class="c1">#For INS, set SVLEN = 1</span>
        awk <span class="s1">&#39;{if($11&lt;0) {$11=1; print} else{print $0}}&#39;</span> tmp3.txt &gt; tmp4.txt <span class="c1">#For those SVs (BNDs, CTX) where SVLEN is under -1 (undetermined), set SVLEN to 1</span>
        awk <span class="s1">&#39;{if($12 == &quot;DUP&quot;) {$9=&quot;0,0,255&quot;; print} else{print $0}}&#39;</span> tmp4.txt &gt; tmp5.txt <span class="c1">#Add color code</span>
        awk <span class="s1">&#39;{if($12 == &quot;DEL&quot;) {$9=&quot;255,0,0&quot;; print} else{print $0}}&#39;</span> tmp5.txt &gt; tmp6.txt <span class="c1">#Add color code</span>
        awk <span class="s1">&#39;{if($12 == &quot;INS&quot;) {$9=&quot;255,153,51&quot;; print} else{print $0}}&#39;</span> tmp6.txt &gt; tmp7.txt <span class="c1">#Add color code</span>
        awk <span class="s1">&#39;{if($12 == &quot;INV&quot;) {$9=&quot;102,0,204&quot;; print} else{print $0}}&#39;</span> tmp7.txt &gt; tmp8.txt <span class="c1">#Add color code</span>
        awk <span class="s1">&#39;{if($12 == &quot;INV&quot;) {$9=&quot;102,0,204&quot;; print} else{print $0}}&#39;</span> tmp8.txt &gt; tmp9.txt <span class="c1">#Add color code</span>
        awk <span class="s1">&#39;{if($9 == &quot;RGB&quot;) {$9=&quot;128,128,128&quot;; print} else{print $0}}&#39;</span> tmp9.txt &gt; tmp10.txt <span class="c1">#Add color code</span>
        awk <span class="s1">&#39;{printf &quot;%.2f%\n&quot;,$10*100}&#39;</span> tmp10.txt &gt; tmp11.txt <span class="c1">#Make INFO field as &quot;SVTYPE_AF(rounded % two decimals)_QUAL&quot;</span>
        paste tmp10.txt tmp11.txt &gt; tmp12.txt
        awk <span class="s1">&#39;$4=$12&quot;_AF=&quot;$14&quot;_QUAL=&quot;$5&#39;</span> tmp12.txt &gt; tmp13.txt
        awk <span class="s1">&#39;$2=$2-1&#39;</span> tmp13.txt &gt; tmp14.txt <span class="c1">#Make 0-based bed by substracting 1 from POS</span>
        cut -d <span class="s2">&quot; &quot;</span> -f1-13 tmp14.txt &gt; <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>_analyzed.bed
        sort -k1,1 -k2,2n <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>_analyzed.bed -o <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>_analyzed.bed <span class="c1">#Sort bed file</span>
        bedToBigBed  -as<span class="o">=</span>fieldnames.as -type<span class="o">=</span>bed9+4 <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>_analyzed.bed hg38.chrom.sizes <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>_analyzed.bb <span class="c1">#Make bed into bigbed file</span>
    <span class="k">done</span>
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Number-of-tested-filtered-clean-minor-allele-frequecy-&lt;1%-and-&gt;=1%-structural-variants-tested-against-Nanopore">Number of tested filtered clean minor allele frequecy &lt;1% and &gt;=1% structural variants tested against Nanopore<a class="anchor-link" href="#Number-of-tested-filtered-clean-minor-allele-frequecy-&lt;1%-and-&gt;=1%-structural-variants-tested-against-Nanopore">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1">#Make a list of all SR SVs tested against nanopore</span>
<span class="nb">cd</span> <span class="nv">$PATH2</span>/nanopore_validation/SR_vcfs/split_SR_vcfs/truvari_bench
module load bcftools

<span class="nv">SAMPLES</span><span class="o">=</span><span class="s2">&quot;RES01827 RES01897 RES02284 UMARY-5077 RES08227 UMARY-1571 UMARY-634 RES01859 RES01877 UMARY-1544 RES00032 RES00064 RES00275 UMARY-4542 RES00071 RES00048 RES05717 RES05720 RES05722 RES05724&quot;</span>

<span class="c1">##Over MAF 1%</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> SVTYPE <span class="k">in</span> DEL DUP INS INV<span class="p">;</span> <span class="k">do</span>
        bcftools query -f <span class="s1">&#39;%ID\n&#39;</span> truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">SVTYPE</span><span class="si">}</span>_over_0.01/tp-call.vcf &gt;&gt; tmp_IDs_over0.01.txt
        bcftools query -f <span class="s1">&#39;%ID\n&#39;</span> truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">SVTYPE</span><span class="si">}</span>_over_0.01/fp.vcf &gt;&gt; tmp_IDs_over0.01.txt
    <span class="k">done</span>
<span class="k">done</span>

<span class="nb">echo</span> <span class="s2">&quot;Unique SVs over MAF 1%&quot;</span>
sort tmp_IDs_over0.01.txt <span class="p">|</span> uniq <span class="p">|</span> wc -l


<span class="c1">##under MAF 1%</span>
<span class="k">for</span> SAMPLE <span class="k">in</span> <span class="nv">$SAMPLES</span><span class="p">;</span> <span class="k">do</span>
    <span class="k">for</span> SVTYPE <span class="k">in</span> DEL DUP INS INV<span class="p">;</span> <span class="k">do</span>
        bcftools query -f <span class="s1">&#39;%ID\n&#39;</span> truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">SVTYPE</span><span class="si">}</span>_under_0.01/tp-call.vcf &gt;&gt; tmp_IDs_under0.01.txt
        bcftools query -f <span class="s1">&#39;%ID\n&#39;</span> truvari_bench_<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_<span class="si">${</span><span class="nv">SVTYPE</span><span class="si">}</span>_under_0.01/fp.vcf &gt;&gt; tmp_IDs_under0.01.txt
    <span class="k">done</span>
<span class="k">done</span>

<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;Unique SVs under MAF 1%&quot;</span>
sort tmp_IDs_under0.01.txt <span class="p">|</span> uniq <span class="p">|</span> wc -l
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Generate-spreadsheets-for-high-quality-structural-variants">Generate spreadsheets for high-quality structural variants<a class="anchor-link" href="#Generate-spreadsheets-for-high-quality-structural-variants">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span>%%bash
<span class="nb">cd</span> <span class="nv">$PATH2</span>/clean_high_quality_vcfs
module load bcftools

<span class="c1">#Make header line and extract information from cleaned vcfs</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span> <span class="k">do</span>
    <span class="nb">echo</span> <span class="s2">&quot;ID CHROM POS END SVTYPE SVLEN QUAL AF ALGORITHMS EVIDENCE&quot;</span> &gt; tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.txt
    bcftools query -f <span class="s1">&#39;%ID %CHROM %POS %END %SVTYPE %SVLEN %QUAL %AF %ALGORITHMS %EVIDENCE\n&#39;</span> <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz &gt;&gt; tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.txt
<span class="k">done</span>

<span class="c1">#Extract case-control allele frequencies from case-control vcfs. If variant does not exit in that file, its AF in that cohort was 0</span>
<span class="k">for</span> PHENO <span class="k">in</span> LBD FTD<span class="p">;</span><span class="k">do</span>
    <span class="k">for</span> STATUS <span class="k">in</span> cases controls<span class="p">;</span> <span class="k">do</span>
        <span class="nb">echo</span> <span class="s2">&quot;ID AF&quot;</span> &gt; tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>_AFs.txt
        bcftools query -f <span class="s2">&quot;%ID %AF\n&quot;</span> <span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>_high_quality_subset_clean_GQ300missing_all_chr.vcf.gz &gt;&gt; tmp_<span class="si">${</span><span class="nv">PHENO</span><span class="si">}</span>_<span class="si">${</span><span class="nv">STATUS</span><span class="si">}</span>_AFs.txt
    <span class="k">done</span>
<span class="k">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%</span><span class="k">load_ext</span> rpy2.ipython
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">####LBD</span>
<span class="c1">#Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>

<span class="c1">#Import files</span>
<span class="n">LBD</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;$PATH2/clean_high_quality_vcfs/tmp_LBD_high_quality_subset_clean_GQ300missing_all_chr.txt&quot;</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">LBD</span><span class="p">)</span>
<span class="n">LBD_cases_AF</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;$PATH2/clean_high_quality_vcfs/tmp_LBD_cases_AFs.txt&quot;</span><span class="p">)</span>
<span class="n">LBD_controls_AF</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;$PATH2/clean_high_quality_vcfs/tmp_LBD_controls_AFs.txt&quot;</span><span class="p">)</span>

<span class="c1">#Check df dimensions</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">dim</span><span class="p">(</span><span class="n">LBD</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">dim</span><span class="p">(</span><span class="n">LBD_cases_AF</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">dim</span><span class="p">(</span><span class="n">LBD_controls_AF</span><span class="p">))</span>

<span class="c1">#Merge case-control allele frequencies: include all SVs from both dfs</span>
<span class="n">LBD_AFs</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">LBD_cases_AF</span><span class="p">,</span><span class="n">LBD_controls_AF</span><span class="p">,</span> <span class="n">by</span> <span class="o">=</span> <span class="s">&quot;ID&quot;</span><span class="p">,</span> <span class="n">all</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">suffixes</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;_cases&quot;</span><span class="p">,</span> <span class="s">&quot;_controls&quot;</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">dim</span><span class="p">(</span><span class="n">LBD_AFs</span><span class="p">))</span>

<span class="c1">#Change NA AF values to 0</span>
<span class="n">LBD_AFs</span><span class="o">$</span><span class="n">AF_cases</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">LBD_AFs</span><span class="o">$</span><span class="n">AF_cases</span><span class="p">)]</span> <span class="o">&lt;-</span> <span class="m">0</span>
<span class="n">LBD_AFs</span><span class="o">$</span><span class="n">AF_controls</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">LBD_AFs</span><span class="o">$</span><span class="n">AF_controls</span><span class="p">)]</span> <span class="o">&lt;-</span> <span class="m">0</span>

<span class="c1">#Merge AF data with other data</span>
<span class="n">LBD</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">LBD</span><span class="p">,</span> <span class="n">LBD_AFs</span><span class="p">,</span> <span class="n">by</span><span class="o">=</span><span class="s">&quot;ID&quot;</span><span class="p">)</span>

<span class="c1">#Reorganize column order, check file output and dimensions</span>
<span class="n">LBD</span> <span class="o">&lt;-</span> <span class="n">LBD</span><span class="p">[,</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="m">3</span><span class="p">,</span><span class="m">4</span><span class="p">,</span><span class="m">5</span><span class="p">,</span><span class="m">6</span><span class="p">,</span><span class="m">7</span><span class="p">,</span><span class="m">8</span><span class="p">,</span><span class="m">11</span><span class="p">,</span><span class="m">12</span><span class="p">,</span><span class="m">9</span><span class="p">,</span><span class="m">10</span><span class="p">)]</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">head</span><span class="p">(</span><span class="n">LBD</span><span class="p">))</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">LBD</span><span class="p">)</span>

<span class="c1">#Save as an excel file</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">LBD</span><span class="p">,</span> <span class="s">&quot;LBD_high_quality_subset_clean_GQ300missing_all_chr.txt&quot;</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">quote</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="c1">####FTD</span>
<span class="c1">#Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="s">&quot;xlsx&quot;</span><span class="p">)</span>

<span class="c1">#Import files</span>
<span class="n">FTD</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;$PATH2/clean_high_quality_vcfs/tmp_FTD_high_quality_subset_clean_GQ300missing_all_chr.txt&quot;</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">FTD</span><span class="p">)</span>
<span class="n">FTD_cases_AF</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;$PATH2/clean_high_quality_vcfs/tmp_FTD_cases_AFs.txt&quot;</span><span class="p">)</span>
<span class="n">FTD_controls_AF</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="s">&quot;$PATH2/clean_high_quality_vcfs/tmp_FTD_controls_AFs.txt&quot;</span><span class="p">)</span>

<span class="c1">#Check df dimensions</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">dim</span><span class="p">(</span><span class="n">FTD</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">dim</span><span class="p">(</span><span class="n">FTD_cases_AF</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">dim</span><span class="p">(</span><span class="n">FTD_controls_AF</span><span class="p">))</span>

<span class="c1">#Merge case-control allele frequencies: include all SVs from both dfs</span>
<span class="n">FTD_AFs</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">FTD_cases_AF</span><span class="p">,</span><span class="n">FTD_controls_AF</span><span class="p">,</span> <span class="n">by</span> <span class="o">=</span> <span class="s">&quot;ID&quot;</span><span class="p">,</span> <span class="n">all</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">suffixes</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;_cases&quot;</span><span class="p">,</span> <span class="s">&quot;_controls&quot;</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">dim</span><span class="p">(</span><span class="n">FTD_AFs</span><span class="p">))</span>

<span class="c1">#Change NA AF values to 0</span>
<span class="n">FTD_AFs</span><span class="o">$</span><span class="n">AF_cases</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">FTD_AFs</span><span class="o">$</span><span class="n">AF_cases</span><span class="p">)]</span> <span class="o">&lt;-</span> <span class="m">0</span>
<span class="n">FTD_AFs</span><span class="o">$</span><span class="n">AF_controls</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">FTD_AFs</span><span class="o">$</span><span class="n">AF_controls</span><span class="p">)]</span> <span class="o">&lt;-</span> <span class="m">0</span>

<span class="c1">#Merge AF data with other data</span>
<span class="n">FTD</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">FTD</span><span class="p">,</span> <span class="n">FTD_AFs</span><span class="p">,</span> <span class="n">by</span><span class="o">=</span><span class="s">&quot;ID&quot;</span><span class="p">)</span>

<span class="c1">#Reorganize column order, check file output and dimensions</span>
<span class="n">FTD</span> <span class="o">&lt;-</span> <span class="n">FTD</span><span class="p">[,</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="m">3</span><span class="p">,</span><span class="m">4</span><span class="p">,</span><span class="m">5</span><span class="p">,</span><span class="m">6</span><span class="p">,</span><span class="m">7</span><span class="p">,</span><span class="m">8</span><span class="p">,</span><span class="m">11</span><span class="p">,</span><span class="m">12</span><span class="p">,</span><span class="m">9</span><span class="p">,</span><span class="m">10</span><span class="p">)]</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">head</span><span class="p">(</span><span class="n">FTD</span><span class="p">))</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">FTD</span><span class="p">)</span>

<span class="c1">#Save as an excel file</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">FTD</span><span class="p">,</span> <span class="s">&quot;FTD_high_quality_subset_clean_GQ300missing_all_chr.txt&quot;</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">quote</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
    </div>
  </div>
</body>

 


</html>
